{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Manipulation of Trans Proteomic Pipeline (TPP) PeptideProphet peptide validation results from Comet database searched*Prochlorococus MED4* LC-MS/MS data using python.\n",
    "\n",
    "Starting with: \n",
    "\n",
    "- PeptideProphet output (.xlxs and .csv) of PTM-optimized database searches >90% probability\n",
    "\n",
    "Goal:\n",
    "\n",
    "- Files with stripped (no PTMs or tryptic ends) peptide lists and\n",
    "- Columns with #'s of each modification in every sequence\n",
    "- Column with stripped peptide lengths (# amino acids)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For technical duplicates, I exported PeptideProphet results as both Excel files and as CSVs into my ETNP 2017 git repo:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP\n"
     ]
    }
   ],
   "source": [
    "cd /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " RAL4_MED2_combine_Comet25_AA_NAAF.csv\r\n",
      " RAL4_MED2_combine_Comet2.5Xcorr_proteins.txt\r\n",
      " RAL4_MED2_combine_Comet3_AA_NAAF.csv\r\n",
      " RAL4_MED2_combine_Comet3Xcorr_proteins.txt\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_Comet.csv\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_Comet_stripped.csv\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_Comet_stripped_peptides_2.5XCorr.txt\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_Comet_stripped_peptides.txt\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_Comet_stripped_work.ods\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_Comet_unfiltered.csv\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_Comet.xlsx\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_PepProp90.csv\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_PepProp90_stripped.csv\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_PepProp90_stripped_peptides\r\n",
      " RAL4_MED2_trypsin_1_PTMopt_PepProp90.xlsx\r\n",
      " RAL4_MED2_trypsin_2_PTMopt_Comet.csv\r\n",
      " RAL4_MED2_trypsin_2_PTMopt_Comet_stripped.csv\r\n",
      " RAL4_MED2_trypsin_2_PTMopt_Comet_stripped_peptides_2.5XCorr.txt\r\n",
      " RAL4_MED2_trypsin_2_PTMopt_Comet_stripped_peptides.txt\r\n",
      " RAL4_MED2_trypsin_2_PTMopt_Comet_stripped_work.ods\r\n",
      " RAL4_MED2_trypsin_2_PTMopt_Comet.xlsx\r\n",
      " RAL4_MED2_trypsin_2_PTMopt_PepProp90.csv\r\n",
      " RAL4_MED2_trypsin_2_PTMopt_PepProp90_stripped.csv\r\n",
      " RAL4_MED2_trypsin_2_PTMopt_PepProp90_stripped_peptides\r\n",
      " RAL4_MED2_trypsin_2_PTMopt_PepProp90.xlsx\r\n",
      " RAL4_MED2_trypsin_combine_PTMopt_Comet_3XCorr_noDECOY.csv\r\n",
      " RAL4_MED2_trypsin_combine_PTMopt_Comet_3XCorr_noDECOY.txt\r\n",
      " RAL4_MED2_trypsin_combine_PTMopt_Comet_stripped__nonredundant_peptides_2.5XCorr.txt\r\n",
      " RAL4_MED2_trypsin_combine_PTMopt_Comet_stripped_peptides_2.5XCorr.txt\r\n",
      " RAL4_MED2_trypsin_combine_PTMopt_Comet_stripped_peptides_3XCorr.txt\r\n",
      " RAL4_MED2_trypsin_combine_PTMopt_Comet_stripped_peptides.txt\r\n",
      " RAL95_MED2_trypsin_combine_Comet25_NAAF_totals.csv\r\n",
      " RAL95_MED2_trypsin_combine_Comet25_totals.csv\r\n",
      " RAL95_MED2_trypsin_combine_Comet3_NAAF_totals.csv\r\n",
      " RAL95_MED2_trypsin_combine_Comet3_totals.csv\r\n",
      " RAL95_MED2_trypsin_combine_cometpmm25_totals.csv\r\n",
      " RAL95_MED2_trypsin_combine_cometpmm3_totals.csv\r\n",
      " RAL95_MED2_trypsin_combine_Comet_totals.csv\r\n",
      " ral_95_med2_trypsin_combine_quant_concat_pepxml.csv\r\n",
      "'ral_95_med2_trypsin_combine_quant.pep 2b.xls'\r\n",
      " ral_95_med2_trypsin_combine_quant.pep.sort.csv\r\n",
      " ral_95_med2_trypsin_combine_quant.pep.srt.xls\r\n",
      " ral_95_med2_trypsin_combine_quant.pep.xls\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.0.5'"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#import pandas library for working with tabular data\n",
    "import os\n",
    "os.getcwd()\n",
    "import pandas as pd\n",
    "#import regular expresson (regex)\n",
    "import re\n",
    "#check pandas version\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20421\n",
      "20179\n",
      "40600\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>probability</th>\n",
       "      <th>spectrum</th>\n",
       "      <th>expect</th>\n",
       "      <th>ions</th>\n",
       "      <th>peptide</th>\n",
       "      <th>protein</th>\n",
       "      <th>calc_neutral_pep_mass</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.0</td>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.04846.04846.2</td>\n",
       "      <td>2.520000e-04</td>\n",
       "      <td>18-Jul</td>\n",
       "      <td>K.VGAATETEM[147.04]K.Y</td>\n",
       "      <td>PMM0452,PMM1436</td>\n",
       "      <td>1051.4856</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.05313.05313.3</td>\n",
       "      <td>6.030000e-04</td>\n",
       "      <td>Aug-48</td>\n",
       "      <td>K.VEAHPIPEHPRPR.R</td>\n",
       "      <td>PMM0760</td>\n",
       "      <td>1533.8164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1.0</td>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.05371.05371.3</td>\n",
       "      <td>3.790000e-07</td>\n",
       "      <td>17/48</td>\n",
       "      <td>K.FHSAEVDSETDHR.V</td>\n",
       "      <td>PMM0613</td>\n",
       "      <td>1528.6542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1.0</td>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.05377.05377.2</td>\n",
       "      <td>3.590000e-07</td>\n",
       "      <td>16-Sep</td>\n",
       "      <td>K.YC[160.03]DDAINKR.E</td>\n",
       "      <td>PMM0574</td>\n",
       "      <td>1153.5186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1.0</td>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.05457.05457.4</td>\n",
       "      <td>6.070000e-03</td>\n",
       "      <td>Jul-90</td>\n",
       "      <td>R.HGGGAFSGKDPTKVDR.S</td>\n",
       "      <td>PMM0311</td>\n",
       "      <td>1627.8067</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   probability                                     spectrum        expect  \\\n",
       "0          1.0  022016_RAL4_95_MED2_trypsin_1.04846.04846.2  2.520000e-04   \n",
       "1          1.0  022016_RAL4_95_MED2_trypsin_1.05313.05313.3  6.030000e-04   \n",
       "2          1.0  022016_RAL4_95_MED2_trypsin_1.05371.05371.3  3.790000e-07   \n",
       "3          1.0  022016_RAL4_95_MED2_trypsin_1.05377.05377.2  3.590000e-07   \n",
       "4          1.0  022016_RAL4_95_MED2_trypsin_1.05457.05457.4  6.070000e-03   \n",
       "\n",
       "     ions                 peptide          protein  calc_neutral_pep_mass  \n",
       "0  18-Jul  K.VGAATETEM[147.04]K.Y  PMM0452,PMM1436              1051.4856  \n",
       "1  Aug-48       K.VEAHPIPEHPRPR.R          PMM0760              1533.8164  \n",
       "2   17/48       K.FHSAEVDSETDHR.V          PMM0613              1528.6542  \n",
       "3  16-Sep   K.YC[160.03]DDAINKR.E          PMM0574              1153.5186  \n",
       "4  Jul-90    R.HGGGAFSGKDPTKVDR.S          PMM0311              1627.8067  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#read the CSV into a datadrame we name 'comet' using the pandas read_csv function\n",
    "peppro1 = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_1_PTMopt_PepProp90.csv\")\n",
    "peppro2 = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_2_PTMopt_PepProp90.csv\")\n",
    "\n",
    "print(len(peppro1))\n",
    "print(len(peppro2))\n",
    "\n",
    "#print(peppro1.columns)\n",
    "\n",
    "frames = [peppro1, peppro2]\n",
    "\n",
    "# concatenate dataframes\n",
    "peppro = pd.concat(frames, sort=False)\n",
    "print(len(peppro))\n",
    "\n",
    "#look at the dataframe\n",
    "peppro.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The peptide column has the residues before and after the tryptic terminii as well as masses of modified residues (e.g., 160.03 Da for carbamidomethylated cysteine). We want to make new columns with all that information and make a column with only the 'stripped' peptide sequence that's just amino acids - this we can then align against other sequences, for example.\n",
    "\n",
    "Modified residues were allowed for:\n",
    "\n",
    "    fixed carbamodimethylation of cysteine 57.021464 C\n",
    "    varialbe oxidation of methionine: 15.9949 M\n",
    "    variable deamidation of asparagine, glumatine: 0.984016 NQ\n",
    "    variable iron cation on lysine: 54.010565 K\n",
    "    variable methylation of lysine and arginine: 14.015650 KR\n",
    "\n",
    "We'll then write this manipulated dataframe to a new file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>probability</th>\n",
       "      <th>spectrum</th>\n",
       "      <th>expect</th>\n",
       "      <th>ions</th>\n",
       "      <th>peptide</th>\n",
       "      <th>protein</th>\n",
       "      <th>calc_neutral_pep_mass</th>\n",
       "      <th>L terminus</th>\n",
       "      <th>R terminus</th>\n",
       "      <th>c-carb</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>k-iron</th>\n",
       "      <th>k-meth</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>stripped length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.0</td>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.04846.04846.2</td>\n",
       "      <td>2.520000e-04</td>\n",
       "      <td>18-Jul</td>\n",
       "      <td>K.VGAATETEM[147.04]K.Y</td>\n",
       "      <td>PMM0452,PMM1436</td>\n",
       "      <td>1051.4856</td>\n",
       "      <td>K</td>\n",
       "      <td>Y</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>VGAATETEMK</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.05313.05313.3</td>\n",
       "      <td>6.030000e-04</td>\n",
       "      <td>Aug-48</td>\n",
       "      <td>K.VEAHPIPEHPRPR.R</td>\n",
       "      <td>PMM0760</td>\n",
       "      <td>1533.8164</td>\n",
       "      <td>K</td>\n",
       "      <td>R</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>VEAHPIPEHPRPR</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1.0</td>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.05371.05371.3</td>\n",
       "      <td>3.790000e-07</td>\n",
       "      <td>17/48</td>\n",
       "      <td>K.FHSAEVDSETDHR.V</td>\n",
       "      <td>PMM0613</td>\n",
       "      <td>1528.6542</td>\n",
       "      <td>K</td>\n",
       "      <td>V</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>FHSAEVDSETDHR</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1.0</td>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.05377.05377.2</td>\n",
       "      <td>3.590000e-07</td>\n",
       "      <td>16-Sep</td>\n",
       "      <td>K.YC[160.03]DDAINKR.E</td>\n",
       "      <td>PMM0574</td>\n",
       "      <td>1153.5186</td>\n",
       "      <td>K</td>\n",
       "      <td>E</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>YCDDAINKR</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1.0</td>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.05457.05457.4</td>\n",
       "      <td>6.070000e-03</td>\n",
       "      <td>Jul-90</td>\n",
       "      <td>R.HGGGAFSGKDPTKVDR.S</td>\n",
       "      <td>PMM0311</td>\n",
       "      <td>1627.8067</td>\n",
       "      <td>R</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>HGGGAFSGKDPTKVDR</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   probability                                     spectrum        expect  \\\n",
       "0          1.0  022016_RAL4_95_MED2_trypsin_1.04846.04846.2  2.520000e-04   \n",
       "1          1.0  022016_RAL4_95_MED2_trypsin_1.05313.05313.3  6.030000e-04   \n",
       "2          1.0  022016_RAL4_95_MED2_trypsin_1.05371.05371.3  3.790000e-07   \n",
       "3          1.0  022016_RAL4_95_MED2_trypsin_1.05377.05377.2  3.590000e-07   \n",
       "4          1.0  022016_RAL4_95_MED2_trypsin_1.05457.05457.4  6.070000e-03   \n",
       "\n",
       "     ions                 peptide          protein  calc_neutral_pep_mass  \\\n",
       "0  18-Jul  K.VGAATETEM[147.04]K.Y  PMM0452,PMM1436              1051.4856   \n",
       "1  Aug-48       K.VEAHPIPEHPRPR.R          PMM0760              1533.8164   \n",
       "2   17/48       K.FHSAEVDSETDHR.V          PMM0613              1528.6542   \n",
       "3  16-Sep   K.YC[160.03]DDAINKR.E          PMM0574              1153.5186   \n",
       "4  Jul-90    R.HGGGAFSGKDPTKVDR.S          PMM0311              1627.8067   \n",
       "\n",
       "  L terminus R terminus  c-carb  m-oxid  n-deam  q-deam  k-iron  k-meth  \\\n",
       "0          K          Y       0       1       0       0       0       0   \n",
       "1          K          R       0       0       0       0       0       0   \n",
       "2          K          V       0       0       0       0       0       0   \n",
       "3          K          E       1       0       0       0       0       0   \n",
       "4          R          S       0       0       0       0       0       0   \n",
       "\n",
       "   r-meth  stripped peptide  stripped length  \n",
       "0       0        VGAATETEMK               10  \n",
       "1       0     VEAHPIPEHPRPR               13  \n",
       "2       0     FHSAEVDSETDHR               13  \n",
       "3       0         YCDDAINKR                9  \n",
       "4       0  HGGGAFSGKDPTKVDR               16  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use str.strip with indexing by str[0] to add a column with the peptide's left terminus\n",
    "peppro['L terminus'] = peppro['peptide'].astype(str).str[0]\n",
    "\n",
    "# use str.strip with indexing by str[-1] to add a column with the peptide's left terminus\n",
    "peppro['R terminus'] = peppro['peptide'].str.strip().str[-1]\n",
    "\n",
    "# use a count function to enumerate the # of ccarbamidomethylated C's in each peptide\n",
    "peppro['c-carb'] = peppro['peptide'].str.count(\"160.03\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "peppro['m-oxid'] = peppro['peptide'].str.count(\"147.04\")\n",
    "\n",
    "# use a count function to enumerate the # of deamidated N's in each peptide\n",
    "peppro['n-deam'] = peppro['peptide'].str.count(\"115.03\")\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "peppro['q-deam'] = peppro['peptide'].str.count(\"129.04\")\n",
    "\n",
    "# use a count function to enumerate the # of iron adducted K's in each peptide\n",
    "peppro['k-iron'] = peppro['peptide'].str.count(\"182.11\")\n",
    "\n",
    "# use a count function to enumerate the # of methylated K's in each peptide\n",
    "peppro['k-meth'] = peppro['peptide'].str.count(\"142.11\")\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "peppro['r-meth'] = peppro['peptide'].str.count(\"170.12\")\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "peppro['stripped peptide'] = peppro['peptide'].str[2:].str[:-2].str.replace(r\"\\[.*\\]\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "peppro['stripped length'] = peppro['stripped peptide'].apply(len)\n",
    "\n",
    "# write modified dataframe to new txt file, same name + 'stripped'\n",
    "peppro.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_combine_PTMopt_PepProp90_stripped.csv\")\n",
    "\n",
    "# check out the results\n",
    "peppro.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# redundant peppro peptides >2.5 XCorr 38886\n",
      "# nonredundant peppro peptides >2.5 XCOrr 12285\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>VGAATETEMK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>VEAHPIPEHPRPR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FHSAEVDSETDHR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>YCDDAINKR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HGGGAFSGKDPTKVDR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   stripped peptide\n",
       "0        VGAATETEMK\n",
       "1     VEAHPIPEHPRPR\n",
       "2     FHSAEVDSETDHR\n",
       "3         YCDDAINKR\n",
       "4  HGGGAFSGKDPTKVDR"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# keep only peptides  >95% probability\n",
    "# need to convert Xcorr column from strings to numeric so we can use loc\n",
    "peppro['probability'] = pd.to_numeric(peppro['probability'])\n",
    "\n",
    "peppro95 = peppro.loc[peppro['probability'] >= 0.95]\n",
    "\n",
    "# keep only peptide column \n",
    "pep95 = peppro95[[\"stripped peptide\"]]\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "pep95.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_combine_PTMopt_PepPro_stripped_peptides_95.txt\", header=False, index=False)\n",
    "\n",
    "# removing redundancy\n",
    "pep95dedup = pd.DataFrame.drop_duplicates(pep95)\n",
    "\n",
    "print(\"# redundant peppro peptides >2.5 XCorr\", len(pep95))\n",
    "print(\"# nonredundant peppro peptides >2.5 XCOrr\", len(pep95dedup))\n",
    "\n",
    "pep95.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
