
### Directory containing all data files and analyses pertaining to the following manuscript:


### *Protein cycling in the eastern tropical North Pacific oxygen deficient zone: a de novo-discovery peptidomic approach*

Megan E. Duffy<sup>1</sup>, Jacquelyn A. Neibauer<sup>1</sup>, Jamee Adams<sup>1</sup>, Clara A. Fuchsman<sup>2</sup>, and Richard G. Keil<sup>1</sup>

<sup>1</sup>School of Oceanography, University of Washington, Seattle, WA

<sup>2</sup>Horn Point Laboratory, University of Maryland Center for Environmental Science, Cambridge, MD

- Corresponding author: duffyme@uw.edu

What it is:

Broadly, this work is part of a larger effort to illuminate details of carbon cycling. How does organic matter break down, particularly in dymanic regions like an oxygen deficient zone?  Here we use a new approach to environmental protein sequencing, de novo-assisted sequencing, to better detect the protineacous organic matter, both living and degraded, made by organisms in this dynamic environment. Because the amino acid sequences of peptides can be specific to the organism that made them, as well as to the function of the protein, sequencing represents a powerful mining of potential information about community function. Furthermore, proteins are important components of the total organic matter pool. Therefore, details of their production, degradation, recycling, and preservation matter critically to how carbon and nitrogen cycle throughout the ocean over short and long time scales. 

These data are from a January 2017 cruise on the R/V Sikuliaq to the eastern tropical North Pacific. We extracted protein from suspended material (collected with McLane *in situ* pumps) and from sinking material (collected with sediment trap-incubator systems). 

This repository also contains data and analyses from a preliminary culture study to evaluate the usefulness and bias of de novo-assisted peptide sequencing. The MS/MS data is from extracted protein of a *Prochloroccus marinus* strain MED4 collected on a Thermo QExactive HF by Rachel Lundeen. 


![cone sediment trap on boat](https://github.com/MeganEDuffy/2017-etnp/blob/master/images/sikuliaq2017-cone-trap.jpeg)

ABSTRACT

eptides and proteins were identified using a novel de novo-discovery approach in suspended and sinking organic particles from the eastern tropical North Pacific and in a culture of a dominant autotroph from the region, the cyanobacterium Prochlorococcus. De novo peptide sequencing, where the sequence of amino acids is determined directly from mass spectra rather than from comparison to theoretical spectra from a selected sequence database, was found to be a useful tool for discovery of peptides present in a sample but not initially included in the search database. Iterative de novo-informed database search results suggested the presence of fungal peptides and proteins in deep sinking particles, consistent with growing evidence that fungi play an important role in degradation of sinking material in the ocean. The de novo-discovery approach also allowed the tracking of modified autotrophic cyanobacterial peptides to the deep sea, where they contributed 0.63% of the phylum-level identifiable peptide pool in a bathymetric sediment trap sample. Overall, the amino acid composition of the peptides in the sinking material showed little change with depth, consistent with earlier observations of bulk organic matter and/or amino acid composition during the early stages of degradation. However, we identified an abundance of modified amino acids in sinking and suspended particles, including high levels of deamidation, suggesting that partial degradation of protein could potentially fuel observed anammox and contribute to observed pool of refractory organic nitrogen. We also observe methylation of arginine, which has previously been shown to slow degradation of peptides in seawater. Our results demonstrate several examples how de novo-discovery allows for a deeper evaluation of proteins and peptides in environmental systems undergoing degradation.
