{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Manipulation of Comet database search results of ETNP 2017 P2 samples LC-MS/MS data using python.\n",
    "\n",
    "Starting with:\n",
    "\n",
    "    Comet search results (.csv) of PTM-optimized database searches\n",
    "    Includes precursor intensities and protein descriptions\n",
    "\n",
    "Goal:\n",
    "\n",
    "    Files with stripped (no PTMs) peptide lists and\n",
    "    Columns with #'s of each modification in every sequence\n",
    "    Column with stripped peptide lengths (# amino acids)\n",
    "    \n",
    "### To use for a different file:\n",
    "\n",
    "#### 1. Change the input file name in *IN 4*\n",
    "#### 2. Use 'find + replace' (Esc + F) to replace the running # (e.g., 233) for another\n",
    "#### 3. Update the NAAF factor calculated in *IN 6* into *IN 7*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We don't have technical duplicates here, sadly, unlike the MED4 Pro samples. I exported Comet search results CSVs into my ETNP 2017 git repo:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/TPP-PTMopt/ETNP-SKQ17-TPP-PTMopt-hyroxylation\n"
     ]
    }
   ],
   "source": [
    "cd /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/TPP-PTMopt/ETNP-SKQ17-TPP-PTMopt-hyroxylation/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.0.5'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# LIBRARIES\n",
    "#import pandas library for working with tabular data\n",
    "import os\n",
    "os.getcwd()\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "from scipy.stats import kde\n",
    "#import regular expresson (regex)\n",
    "import re\n",
    "#check pandas version\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20170410_JA4_01_quant_interact.pep.xls\r\n",
      "20170410_JA4_01_quant_interact.pep.xlsx\r\n",
      "20170531_ETNP_JA14_01_PTMopt_quant_interact.pep.xls\r\n",
      "20181003_4-19_965m_top_PTMopt_quant_interact.pep.xls\r\n",
      "20181004_4-19_265m_topP_278_90min_PTMopt_quant_interact.pep.xls\r\n",
      "20181214_378_etnp2017_100m_trap_PTMopt_quant_interact.pep.xls\r\n",
      "ETNP-SKQ17-231-100m-0.3-JA2_PTMopt_Comet3_AA_NAAF.csv\r\n",
      "ETNP-SKQ17-231-100m-0.3-JA2_PTMopt_Comet3_totals.csv\r\n",
      "ETNP-SKQ17-231-100m-0.3-JA2_PTMopt_Comet50_totals.csv\r\n",
      "ETNP-SKQ17-231-100m-0.3-JA2_PTMopt_Comet50_ulfiltered.csv\r\n",
      "JA2_PTMopt_interact.pep.xls\r\n",
      "JA2_PTMopt_interact_quant_nopro.pep.csv\r\n",
      "JA2_PTMopt_interact_quant_nopro.pep.ods\r\n",
      "JA2_PTMopt_interact_quant_nopro.pep.xlsx\r\n",
      "JA2_PTMopt_interact_quant.pep.csv\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 231: 100 m McLane pump filtered on 0.3 um GF-75"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# redundant Comet peptides in combined dataframe 16916\n",
      "# nonredundant Comet peptides in combined dataframe 16916\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>spectrum</th>\n",
       "      <th>xcorr</th>\n",
       "      <th>deltacn</th>\n",
       "      <th>expect</th>\n",
       "      <th>ions</th>\n",
       "      <th>peptide</th>\n",
       "      <th>protein</th>\n",
       "      <th>calc_neutral_pep_mass</th>\n",
       "      <th>precursor_intensity</th>\n",
       "      <th>protein_descr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20170410_JA4_01.33094.33094.3</td>\n",
       "      <td>5.437</td>\n",
       "      <td>1.000</td>\n",
       "      <td>1.030000e-07</td>\n",
       "      <td>30/88</td>\n",
       "      <td>R.AIQQQIENPLAQQILSGELVPGK.V</td>\n",
       "      <td>gi|54036848|sp|P63284.1|CLPB_ECOLI</td>\n",
       "      <td>2473.3540</td>\n",
       "      <td>29704700.0</td>\n",
       "      <td>RecName: Full=Chaperone protein ClpB; AltName:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20170410_JA4_01.33121.33121.2</td>\n",
       "      <td>5.391</td>\n",
       "      <td>1.000</td>\n",
       "      <td>1.490000e-08</td>\n",
       "      <td>26/44</td>\n",
       "      <td>R.AIQQQIENPLAQQILSGELVPGK.V</td>\n",
       "      <td>gi|54036848|sp|P63284.1|CLPB_ECOLI</td>\n",
       "      <td>2473.3540</td>\n",
       "      <td>2342260.0</td>\n",
       "      <td>RecName: Full=Chaperone protein ClpB; AltName:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20170410_JA4_01.30594.30594.3</td>\n",
       "      <td>5.229</td>\n",
       "      <td>0.553</td>\n",
       "      <td>1.370000e-11</td>\n",
       "      <td>34/80</td>\n",
       "      <td>R.TQVESGNVQWDIVDVLPHEAR.V</td>\n",
       "      <td>ETNP_120m_PROKKA_307121,ETNP_120m_particle_PRO...</td>\n",
       "      <td>2391.1819</td>\n",
       "      <td>4257740.0</td>\n",
       "      <td>Bacterial extracellular solute-binding protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20170410_JA4_01.23249.23249.3</td>\n",
       "      <td>5.213</td>\n",
       "      <td>0.583</td>\n",
       "      <td>7.600000e-14</td>\n",
       "      <td>30/92</td>\n",
       "      <td>K.AYGDTFTGGKATFVNYNGGLGEVR.T</td>\n",
       "      <td>ETNP_120m_PROKKA_307121,ETNP_120m_particle_PRO...</td>\n",
       "      <td>2493.1925</td>\n",
       "      <td>3257690.0</td>\n",
       "      <td>Bacterial extracellular solute-binding protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20170410_JA4_01.12092.12092.3</td>\n",
       "      <td>4.856</td>\n",
       "      <td>0.345</td>\n",
       "      <td>1.890000e+00</td>\n",
       "      <td>25/48</td>\n",
       "      <td>R.KEINSADDLKGLK.M</td>\n",
       "      <td>ETNP_120m_PROKKA_234900,ETNP_120m_PROKKA_26254...</td>\n",
       "      <td>1429.7777</td>\n",
       "      <td>396663.0</td>\n",
       "      <td>Alpha-keto acid-binding periplasmic protein Ta...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        spectrum  xcorr  deltacn        expect   ions  \\\n",
       "0  20170410_JA4_01.33094.33094.3  5.437    1.000  1.030000e-07  30/88   \n",
       "1  20170410_JA4_01.33121.33121.2  5.391    1.000  1.490000e-08  26/44   \n",
       "2  20170410_JA4_01.30594.30594.3  5.229    0.553  1.370000e-11  34/80   \n",
       "3  20170410_JA4_01.23249.23249.3  5.213    0.583  7.600000e-14  30/92   \n",
       "4  20170410_JA4_01.12092.12092.3  4.856    0.345  1.890000e+00  25/48   \n",
       "\n",
       "                        peptide  \\\n",
       "0   R.AIQQQIENPLAQQILSGELVPGK.V   \n",
       "1   R.AIQQQIENPLAQQILSGELVPGK.V   \n",
       "2     R.TQVESGNVQWDIVDVLPHEAR.V   \n",
       "3  K.AYGDTFTGGKATFVNYNGGLGEVR.T   \n",
       "4             R.KEINSADDLKGLK.M   \n",
       "\n",
       "                                             protein  calc_neutral_pep_mass  \\\n",
       "0                 gi|54036848|sp|P63284.1|CLPB_ECOLI              2473.3540   \n",
       "1                 gi|54036848|sp|P63284.1|CLPB_ECOLI              2473.3540   \n",
       "2  ETNP_120m_PROKKA_307121,ETNP_120m_particle_PRO...              2391.1819   \n",
       "3  ETNP_120m_PROKKA_307121,ETNP_120m_particle_PRO...              2493.1925   \n",
       "4  ETNP_120m_PROKKA_234900,ETNP_120m_PROKKA_26254...              1429.7777   \n",
       "\n",
       "   precursor_intensity                                      protein_descr  \n",
       "0           29704700.0  RecName: Full=Chaperone protein ClpB; AltName:...  \n",
       "1            2342260.0  RecName: Full=Chaperone protein ClpB; AltName:...  \n",
       "2            4257740.0     Bacterial extracellular solute-binding protein  \n",
       "3            3257690.0     Bacterial extracellular solute-binding protein  \n",
       "4             396663.0  Alpha-keto acid-binding periplasmic protein Ta...  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read the CSV into a dataframe using the pandas read_csv function\n",
    "#cometdup233 = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/TPP-PTMopt/ETNP-SKQ17-TPP-PTMopt-hyroxylation/JA2_PTMopt_interact_quant_nopro.pep.csv\", index_col='spectrum')\n",
    "\n",
    "cometdup233 = pd.read_excel(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/TPP-PTMopt/ETNP-SKQ17-TPP-PTMopt-hyroxylation/20170410_JA4_01_quant_interact.pep.xlsx\")\n",
    "\n",
    "# remove redundant rows\n",
    "comet233 = pd.DataFrame.drop_duplicates(cometdup233)\n",
    "\n",
    "\n",
    "print(\"# redundant Comet peptides in combined dataframe\", len(cometdup233))\n",
    "print(\"# nonredundant Comet peptides in combined dataframe\", len(comet233))\n",
    "\n",
    "#look at the dataframe\n",
    "comet233.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/millieginty/anaconda3/lib/python3.8/site-packages/pandas/core/ops/array_ops.py:253: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
      "  res_values = method(rvalues)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>spectrum</th>\n",
       "      <th>xcorr</th>\n",
       "      <th>deltacn</th>\n",
       "      <th>expect</th>\n",
       "      <th>ions</th>\n",
       "      <th>peptide</th>\n",
       "      <th>protein</th>\n",
       "      <th>calc_neutral_pep_mass</th>\n",
       "      <th>precursor_intensity</th>\n",
       "      <th>protein_descr</th>\n",
       "      <th>...</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>k-hydr</th>\n",
       "      <th>p-hydr</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>stripped length</th>\n",
       "      <th>NAAF num.</th>\n",
       "      <th>ptm-total</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20170410_JA4_01.33094.33094.3</td>\n",
       "      <td>5.437</td>\n",
       "      <td>1.000</td>\n",
       "      <td>1.030000e-07</td>\n",
       "      <td>30/88</td>\n",
       "      <td>R.AIQQQIENPLAQQILSGELVPGK.V</td>\n",
       "      <td>gi|54036848|sp|P63284.1|CLPB_ECOLI</td>\n",
       "      <td>2473.3540</td>\n",
       "      <td>29704700.0</td>\n",
       "      <td>RecName: Full=Chaperone protein ClpB; AltName:...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>R.AIQQQIENPLAQQILSGELVPGK.V</td>\n",
       "      <td>27</td>\n",
       "      <td>1.100174e+06</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20170410_JA4_01.33121.33121.2</td>\n",
       "      <td>5.391</td>\n",
       "      <td>1.000</td>\n",
       "      <td>1.490000e-08</td>\n",
       "      <td>26/44</td>\n",
       "      <td>R.AIQQQIENPLAQQILSGELVPGK.V</td>\n",
       "      <td>gi|54036848|sp|P63284.1|CLPB_ECOLI</td>\n",
       "      <td>2473.3540</td>\n",
       "      <td>2342260.0</td>\n",
       "      <td>RecName: Full=Chaperone protein ClpB; AltName:...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>R.AIQQQIENPLAQQILSGELVPGK.V</td>\n",
       "      <td>27</td>\n",
       "      <td>8.675037e+04</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20170410_JA4_01.30594.30594.3</td>\n",
       "      <td>5.229</td>\n",
       "      <td>0.553</td>\n",
       "      <td>1.370000e-11</td>\n",
       "      <td>34/80</td>\n",
       "      <td>R.TQVESGNVQWDIVDVLPHEAR.V</td>\n",
       "      <td>ETNP_120m_PROKKA_307121,ETNP_120m_particle_PRO...</td>\n",
       "      <td>2391.1819</td>\n",
       "      <td>4257740.0</td>\n",
       "      <td>Bacterial extracellular solute-binding protein</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>R.TQVESGNVQWDIVDVLPHEAR.V</td>\n",
       "      <td>25</td>\n",
       "      <td>1.703096e+05</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20170410_JA4_01.23249.23249.3</td>\n",
       "      <td>5.213</td>\n",
       "      <td>0.583</td>\n",
       "      <td>7.600000e-14</td>\n",
       "      <td>30/92</td>\n",
       "      <td>K.AYGDTFTGGKATFVNYNGGLGEVR.T</td>\n",
       "      <td>ETNP_120m_PROKKA_307121,ETNP_120m_particle_PRO...</td>\n",
       "      <td>2493.1925</td>\n",
       "      <td>3257690.0</td>\n",
       "      <td>Bacterial extracellular solute-binding protein</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>K.AYGDTFTGGKATFVNYNGGLGEVR.T</td>\n",
       "      <td>28</td>\n",
       "      <td>1.163461e+05</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20170410_JA4_01.12092.12092.3</td>\n",
       "      <td>4.856</td>\n",
       "      <td>0.345</td>\n",
       "      <td>1.890000e+00</td>\n",
       "      <td>25/48</td>\n",
       "      <td>R.KEINSADDLKGLK.M</td>\n",
       "      <td>ETNP_120m_PROKKA_234900,ETNP_120m_PROKKA_26254...</td>\n",
       "      <td>1429.7777</td>\n",
       "      <td>396663.0</td>\n",
       "      <td>Alpha-keto acid-binding periplasmic protein Ta...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>R.KEINSADDLKGLK.M</td>\n",
       "      <td>17</td>\n",
       "      <td>2.333312e+04</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 43 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        spectrum  xcorr  deltacn        expect   ions  \\\n",
       "0  20170410_JA4_01.33094.33094.3  5.437    1.000  1.030000e-07  30/88   \n",
       "1  20170410_JA4_01.33121.33121.2  5.391    1.000  1.490000e-08  26/44   \n",
       "2  20170410_JA4_01.30594.30594.3  5.229    0.553  1.370000e-11  34/80   \n",
       "3  20170410_JA4_01.23249.23249.3  5.213    0.583  7.600000e-14  30/92   \n",
       "4  20170410_JA4_01.12092.12092.3  4.856    0.345  1.890000e+00  25/48   \n",
       "\n",
       "                        peptide  \\\n",
       "0   R.AIQQQIENPLAQQILSGELVPGK.V   \n",
       "1   R.AIQQQIENPLAQQILSGELVPGK.V   \n",
       "2     R.TQVESGNVQWDIVDVLPHEAR.V   \n",
       "3  K.AYGDTFTGGKATFVNYNGGLGEVR.T   \n",
       "4             R.KEINSADDLKGLK.M   \n",
       "\n",
       "                                             protein  calc_neutral_pep_mass  \\\n",
       "0                 gi|54036848|sp|P63284.1|CLPB_ECOLI              2473.3540   \n",
       "1                 gi|54036848|sp|P63284.1|CLPB_ECOLI              2473.3540   \n",
       "2  ETNP_120m_PROKKA_307121,ETNP_120m_particle_PRO...              2391.1819   \n",
       "3  ETNP_120m_PROKKA_307121,ETNP_120m_particle_PRO...              2493.1925   \n",
       "4  ETNP_120m_PROKKA_234900,ETNP_120m_PROKKA_26254...              1429.7777   \n",
       "\n",
       "   precursor_intensity                                      protein_descr  \\\n",
       "0           29704700.0  RecName: Full=Chaperone protein ClpB; AltName:...   \n",
       "1            2342260.0  RecName: Full=Chaperone protein ClpB; AltName:...   \n",
       "2            4257740.0     Bacterial extracellular solute-binding protein   \n",
       "3            3257690.0     Bacterial extracellular solute-binding protein   \n",
       "4             396663.0  Alpha-keto acid-binding periplasmic protein Ta...   \n",
       "\n",
       "   ... m-oxid n-deam  q-deam  k-hydr  p-hydr  r-meth  \\\n",
       "0  ...      0      0       0       0       0       0   \n",
       "1  ...      0      0       0       0       0       0   \n",
       "2  ...      0      0       0       0       0       0   \n",
       "3  ...      0      0       0       0       0       0   \n",
       "4  ...      0      0       0       0       0       0   \n",
       "\n",
       "               stripped peptide  stripped length     NAAF num.  ptm-total  \n",
       "0   R.AIQQQIENPLAQQILSGELVPGK.V               27  1.100174e+06          0  \n",
       "1   R.AIQQQIENPLAQQILSGELVPGK.V               27  8.675037e+04          0  \n",
       "2     R.TQVESGNVQWDIVDVLPHEAR.V               25  1.703096e+05          0  \n",
       "3  K.AYGDTFTGGKATFVNYNGGLGEVR.T               28  1.163461e+05          0  \n",
       "4             R.KEINSADDLKGLK.M               17  2.333312e+04          0  \n",
       "\n",
       "[5 rows x 43 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get rid of rows where the xcorr is unavailable (usually 3 or so)\n",
    "comet233 = comet233[comet233.xcorr != '[unavailable]']\n",
    "\n",
    "# use str.strip with indexing by str[0] to add a column with the peptide's left terminus\n",
    "comet233['L terminus'] = comet233['peptide'].astype(str).str[0]\n",
    "\n",
    "# use str.strip with indexing by str[-1] to add a column with the peptide's left terminus\n",
    "comet233['R terminus'] = comet233['peptide'].str.strip().str[-1]\n",
    "\n",
    "# use a count function to enumerate the # of A's (alanines) in each peptide\n",
    "comet233['A'] = comet233['peptide'].str.count(\"A\")\n",
    "\n",
    "# use a count function to enumerate the # of C's (cysteines) in each peptide\n",
    "comet233['C'] = comet233['peptide'].str.count(\"C\")\n",
    "\n",
    "# use a count function to enumerate the # of D's (aspartic acids) in each peptide\n",
    "comet233['D'] = comet233['peptide'].str.count(\"D\")\n",
    "\n",
    "# use a count function to enumerate the # of E's (glutamic acids) in each peptide\n",
    "comet233['E'] = comet233['peptide'].str.count(\"E\")\n",
    "\n",
    "# use a count function to enumerate the # of F's (phenylalanines) in each peptide\n",
    "comet233['F'] = comet233['peptide'].str.count(\"F\")\n",
    "\n",
    "# use a count function to enumerate the # of G's (glycines) in each peptide\n",
    "comet233['G'] = comet233['peptide'].str.count(\"G\")\n",
    "\n",
    "# use a count function to enumerate the # of H's (histidines) in each peptide\n",
    "comet233['H'] = comet233['peptide'].str.count(\"H\")\n",
    "\n",
    "# use a count function to enumerate the # of I's (isoleucines) in each peptide\n",
    "# in comet233 output, there will be no isoleucines (they're lumped in with leucines)\n",
    "comet233['I'] = comet233['peptide'].str.count(\"I\")\n",
    "\n",
    "# use a count function to enumerate the # of K's (lysines) in each peptide\n",
    "comet233['K'] = comet233['peptide'].str.count(\"K\")\n",
    "\n",
    "# use a count function to enumerate the # of L's (leucines) in each peptide\n",
    "# also these include the isoleucines\n",
    "comet233['L'] = comet233['peptide'].str.count(\"L\")\n",
    "\n",
    "# use a count function to enumerate the # of M's (methionines) in each peptide\n",
    "comet233['M'] = comet233['peptide'].str.count(\"M\")\n",
    "\n",
    "# use a count function to enumerate the # of N's (asparagines) in each peptide\n",
    "comet233['N'] = comet233['peptide'].str.count(\"N\")\n",
    "\n",
    "# use a count function to enumerate the # of P's ([prolines]) in each peptide\n",
    "comet233['P'] = comet233['peptide'].str.count(\"P\")\n",
    "\n",
    "# use a count function to enumerate the # of Q's (glutamines) in each peptide\n",
    "comet233['Q'] = comet233['peptide'].str.count(\"Q\")\n",
    "\n",
    "# use a count function to enumerate the # of R's (arginines) in each peptide\n",
    "comet233['R'] = comet233['peptide'].str.count(\"R\")\n",
    "\n",
    "# use a count function to enumerate the # of S's (serines) in each peptide\n",
    "comet233['S'] = comet233['peptide'].str.count(\"S\")\n",
    "\n",
    "# use a count function to enumerate the # of T's (threonines) in each peptide\n",
    "comet233['T'] = comet233['peptide'].str.count(\"T\")\n",
    "\n",
    "# use a count function to enumerate the # of V's (valines) in each peptide\n",
    "comet233['V'] = comet233['peptide'].str.count(\"V\")\n",
    "\n",
    "# use a count function to enumerate the # of W's (tryptophans) in each peptide\n",
    "comet233['W'] = comet233['peptide'].str.count(\"W\")\n",
    "\n",
    "# use a count function to enumerate the # of Y's (tyrosines) in each peptide\n",
    "comet233['Y'] = comet233['peptide'].str.count(\"Y\")\n",
    "\n",
    "# use a count function to enumerate the # of carbamidomethylated C's in each peptide\n",
    "comet233['c-carb'] = comet233['peptide'].str.count(\"160.03\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "comet233['m-oxid'] = comet233['peptide'].str.count(\"147.04\")\n",
    "\n",
    "# use a count function to enumerate the # of deamidated N's in each peptide\n",
    "comet233['n-deam'] = comet233['peptide'].str.count(\"115.03\")\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "comet233['q-deam'] = comet233['peptide'].str.count(\"129.04\")\n",
    "\n",
    "# use a count function to enumerate the # of hydroxylated K's in each peptide\n",
    "comet233['k-hydr'] = comet233['peptide'].str.count(\"144.09\")\n",
    "\n",
    "# use a count function to enumerate the # of hydroxylated P's in each peptide\n",
    "comet233['p-hydr'] = comet233['peptide'].str.count(\"131.05\")\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "comet233['r-meth'] = comet233['peptide'].str.count(\"170.12\")\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "comet233['stripped peptide'] = comet233['peptide'].str.replace(r\"\\(.*\\)\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "comet233['stripped length'] = comet233['stripped peptide'].apply(len)\n",
    "\n",
    "comet233['NAAF num.'] = comet233['precursor_intensity'] / comet233['stripped length']\n",
    "\n",
    "# total the number of modifications in sequence\n",
    "comet233['ptm-total'] = comet233['c-carb'] + comet233['m-oxid'] + comet233['n-deam'] + comet233['q-deam'] + comet233['k-hydr'] + comet233['p-hydr'] + comet233['r-meth']\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "comet233.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/TPP-PTMopt/ETNP-SKQ17-TPP-PTMopt-hyroxylation/ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_Comet50_ulfiltered.csv\")\n",
    "\n",
    "# check out the results\n",
    "comet233.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Calculating the false discovery rate (% FDR)\n",
    "\n",
    "### Filtering PSMs > a selected XCorr value and exporting peptides"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# real Comet PSMs 9612\n",
      "# decoy Comet PSMs 7304\n",
      "False discovery rate =  75.98834789846026\n"
     ]
    }
   ],
   "source": [
    "# Let's separate out the decoy hits from the good ones\n",
    "\n",
    "comet233pmm = comet233[~comet233['protein'].str.contains(\"DECOY\")]\n",
    "comet233dec = comet233[comet233['protein'].str.contains(\"DECOY\")]\n",
    "\n",
    "# how many PSM that are only PMM (proteins in the database)?\n",
    "\n",
    "print(\"# real Comet PSMs\", len(comet233pmm))\n",
    "\n",
    "# compared to how many PSMs containing decoys?\n",
    "\n",
    "print(\"# decoy Comet PSMs\", len(comet233dec))\n",
    "\n",
    "# calculate the bulk FDR (all PSMs so let's not beat ourselves up)\n",
    "\n",
    "r = len(comet233pmm)\n",
    "d = len(comet233dec)\n",
    "\n",
    "FDR = d/r*100\n",
    "\n",
    "print(\"False discovery rate = \", FDR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# real Comet PSMs 84\n",
      "# decoy Comet PSMs 2\n",
      "False discovery rate =  2.3255813953488373\n"
     ]
    }
   ],
   "source": [
    "# keep only peptides  >3 XCorr\n",
    "# need to convert Xcorr column from strings to numeric so we can use loc\n",
    "comet233['xcorr'] = pd.to_numeric(comet233['xcorr'])\n",
    "\n",
    "comet233_3 = comet233.loc[comet233['xcorr'] >= 3]\n",
    "\n",
    "# What's the FDR?\n",
    "\n",
    "# Let's separate out the decoy hits from the good ones\n",
    "\n",
    "comet233pmm3 = comet233_3[~comet233_3['protein'].str.contains(\"DECOY\")]\n",
    "comet233dec3 = comet233_3[comet233_3['protein'].str.contains(\"DECOY\")]\n",
    "\n",
    "# how many PSM that are only PMM (proteins in the database)?\n",
    "\n",
    "print(\"# real Comet PSMs\", len(comet233pmm3))\n",
    "\n",
    "# compared to how many PSMs containing decoys?\n",
    "\n",
    "print(\"# decoy Comet PSMs\", len(comet233dec3))\n",
    "\n",
    "# calculate the FDR \n",
    "\n",
    "r = len(comet233pmm3)\n",
    "d = len(comet233dec3)\n",
    "\n",
    "FDR = d/(d+r)*100\n",
    "\n",
    "print(\"False discovery rate = \", FDR)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## NAAF correction and exporting files with AA and PTM totals:\n",
    "\n",
    "### Only for DECOY-filtered Comet PSMs > XCorr 3.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>K</th>\n",
       "      <th>I</th>\n",
       "      <th>...</th>\n",
       "      <th>Total area</th>\n",
       "      <th>Total length</th>\n",
       "      <th>% C w/ carb</th>\n",
       "      <th>% M w/ oxid</th>\n",
       "      <th>% N w/ deam</th>\n",
       "      <th>% Q w/ deam</th>\n",
       "      <th>% K w/ hydr</th>\n",
       "      <th>% P w/ hydr</th>\n",
       "      <th>% R w/ meth</th>\n",
       "      <th>NAAF denom.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>195</td>\n",
       "      <td>3</td>\n",
       "      <td>87</td>\n",
       "      <td>115</td>\n",
       "      <td>25</td>\n",
       "      <td>166</td>\n",
       "      <td>10</td>\n",
       "      <td>84</td>\n",
       "      <td>99</td>\n",
       "      <td>84</td>\n",
       "      <td>...</td>\n",
       "      <td>718754825.0</td>\n",
       "      <td>1960</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.083333</td>\n",
       "      <td>0.084112</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>366711.645408</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                A  C   D    E   F    G   H   I   K   I  ...   Total area  \\\n",
       "sample total  195  3  87  115  25  166  10  84  99  84  ...  718754825.0   \n",
       "\n",
       "              Total length  % C w/ carb  % M w/ oxid  % N w/ deam  \\\n",
       "sample total          1960          1.0          0.2     0.083333   \n",
       "\n",
       "              % Q w/ deam  % K w/ hydr  % P w/ hydr  % R w/ meth  \\\n",
       "sample total     0.084112          0.0          0.0          0.0   \n",
       "\n",
       "                NAAF denom.  \n",
       "sample total  366711.645408  \n",
       "\n",
       "[1 rows x 38 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of certain columns in the modified\n",
    "# peptide dataframe above (for >XCorr 3)\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'A': comet233pmm3['A'].sum(),\n",
    "        'C': comet233pmm3['C'].sum(),\n",
    "        'D': comet233pmm3['D'].sum(),\n",
    "        'E': comet233pmm3['E'].sum(),\n",
    "        'F': comet233pmm3['F'].sum(),\n",
    "        'G': comet233pmm3['G'].sum(),\n",
    "        'H': comet233pmm3['H'].sum(),\n",
    "        'I': comet233pmm3['I'].sum(),\n",
    "        'K': comet233pmm3['K'].sum(),\n",
    "        'I': comet233pmm3['I'].sum(),\n",
    "        'L': comet233pmm3['L'].sum(),\n",
    "        'M': comet233pmm3['M'].sum(),\n",
    "        'N': comet233pmm3['N'].sum(),\n",
    "        'P': comet233pmm3['P'].sum(),\n",
    "        'Q': comet233pmm3['Q'].sum(),\n",
    "        'R': comet233pmm3['R'].sum(),\n",
    "        'S': comet233pmm3['S'].sum(),\n",
    "        'T': comet233pmm3['T'].sum(),\n",
    "        'V': comet233pmm3['V'].sum(),\n",
    "        'W': comet233pmm3['W'].sum(),\n",
    "        'Y': comet233pmm3['Y'].sum(),\n",
    "        'c-carb': comet233pmm3['c-carb'].sum(),\n",
    "        'm-oxid': comet233pmm3['m-oxid'].sum(),\n",
    "        'n-deam': comet233pmm3['n-deam'].sum(),\n",
    "        'q-deam': comet233pmm3['q-deam'].sum(),\n",
    "        'k-hydr': comet233pmm3['k-hydr'].sum(),\n",
    "        'p-hydr': comet233pmm3['p-hydr'].sum(),\n",
    "        'r-meth': comet233pmm3['r-meth'].sum(),\n",
    "        'Total area': comet233pmm3['precursor_intensity'].sum(),\n",
    "        'Total length': comet233pmm3['stripped length'].sum()\n",
    "       }\n",
    "\n",
    "totalcomet233pmm3 = pd.DataFrame(data, columns=['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'I', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', 'm-oxid', 'n-deam', 'q-deam', 'k-hydr', 'p-hydr', 'r-meth', 'Total area', 'Total length'], index=index)\n",
    "\n",
    "# calculate percentage of C's with carb (should be 1.0)\n",
    "totalcomet233pmm3['% C w/ carb'] = totalcomet233pmm3['c-carb'] / totalcomet233pmm3['C'] \n",
    "\n",
    "# calculate percentage of M's that are oxidized\n",
    "totalcomet233pmm3['% M w/ oxid'] = totalcomet233pmm3['m-oxid'] / totalcomet233pmm3['M'] \n",
    "\n",
    "# calculate percentage of N's that are deamidated\n",
    "totalcomet233pmm3['% N w/ deam'] = totalcomet233pmm3['n-deam'] / totalcomet233pmm3['N'] \n",
    "\n",
    "# calculate percentage of Q's that are deamidated\n",
    "totalcomet233pmm3['% Q w/ deam'] = totalcomet233pmm3['q-deam'] / totalcomet233pmm3['Q'] \n",
    "\n",
    "# calculate percentage of K's that are hydroxylated\n",
    "totalcomet233pmm3['% K w/ hydr'] = totalcomet233pmm3['k-hydr'] / totalcomet233pmm3['K'] \n",
    "\n",
    "# calculate percentage of P's that are hydroxylated\n",
    "totalcomet233pmm3['% P w/ hydr'] = totalcomet233pmm3['p-hydr'] / totalcomet233pmm3['P'] \n",
    "\n",
    "# calculate percentage of R's that are methylated\n",
    "totalcomet233pmm3['% R w/ meth'] = totalcomet233pmm3['r-meth'] / totalcomet233pmm3['R'] \n",
    "\n",
    "# calculate NAAF denominator for all peptides in dataset i\n",
    "totalcomet233pmm3['NAAF denom.'] = totalcomet233pmm3['Total area'] / totalcomet233pmm3['Total length']\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "totalcomet233pmm3.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/TPP-PTMopt/ETNP-SKQ17-TPP-PTMopt-hyroxylation/ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_Comet3_totals.csv\")\n",
    "\n",
    "totalcomet233pmm3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-9-430b81bee5df>:7: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  comet233pmm3['NAAF factor'] = (comet233pmm3['NAAF num.'])/NAAF3\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>NAAF factor</th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>K</th>\n",
       "      <th>...</th>\n",
       "      <th>V-NAAF3</th>\n",
       "      <th>W-NAAF3</th>\n",
       "      <th>Y-NAAF3</th>\n",
       "      <th>ccarb-NAAF3</th>\n",
       "      <th>moxid-NAAF3</th>\n",
       "      <th>ndeam-NAAF3</th>\n",
       "      <th>qdeam-NAAF3</th>\n",
       "      <th>khydr-NAAF3</th>\n",
       "      <th>phydr-NAAF3</th>\n",
       "      <th>rmeth-NAAF3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R.AIQQQIENPLAQQILSGELVPGK.V</td>\n",
       "      <td>3.000107</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>6.000213</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>R.AIQQQIENPLAQQILSGELVPGK.V</td>\n",
       "      <td>0.236563</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.473126</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>R.TQVESGNVQWDIVDVLPHEAR.V</td>\n",
       "      <td>0.464424</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>2.322119</td>\n",
       "      <td>0.464424</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>K.AYGDTFTGGKATFVNYNGGLGEVR.T</td>\n",
       "      <td>0.317269</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0.634537</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.634537</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>R.KEINSADDLKGLK.M</td>\n",
       "      <td>0.063628</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 56 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               stripped peptide  NAAF factor  A  C  D  E  F  G  H  K  ...  \\\n",
       "0   R.AIQQQIENPLAQQILSGELVPGK.V     3.000107  2  0  0  2  0  2  0  1  ...   \n",
       "1   R.AIQQQIENPLAQQILSGELVPGK.V     0.236563  2  0  0  2  0  2  0  1  ...   \n",
       "2     R.TQVESGNVQWDIVDVLPHEAR.V     0.464424  1  0  2  2  0  1  1  0  ...   \n",
       "3  K.AYGDTFTGGKATFVNYNGGLGEVR.T     0.317269  2  0  1  1  2  6  0  2  ...   \n",
       "4             R.KEINSADDLKGLK.M     0.063628  1  0  2  1  0  1  0  3  ...   \n",
       "\n",
       "    V-NAAF3   W-NAAF3   Y-NAAF3  ccarb-NAAF3  moxid-NAAF3  ndeam-NAAF3  \\\n",
       "0  6.000213  0.000000  0.000000          0.0          0.0          0.0   \n",
       "1  0.473126  0.000000  0.000000          0.0          0.0          0.0   \n",
       "2  2.322119  0.464424  0.000000          0.0          0.0          0.0   \n",
       "3  0.634537  0.000000  0.634537          0.0          0.0          0.0   \n",
       "4  0.000000  0.000000  0.000000          0.0          0.0          0.0   \n",
       "\n",
       "   qdeam-NAAF3  khydr-NAAF3  phydr-NAAF3  rmeth-NAAF3  \n",
       "0          0.0          0.0          0.0          0.0  \n",
       "1          0.0          0.0          0.0          0.0  \n",
       "2          0.0          0.0          0.0          0.0  \n",
       "3          0.0          0.0          0.0          0.0  \n",
       "4          0.0          0.0          0.0          0.0  \n",
       "\n",
       "[5 rows x 56 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use the calculated NAAF factor (in totalcomet233pmm3 dataframe, above) to caluclate the NAAF \n",
    "# NAAF: normalized normalized area abundance factor\n",
    "\n",
    "NAAF3 = 366711.645408\n",
    "\n",
    "# use NAAF >XCorr 3 to get NAAF\n",
    "comet233pmm3['NAAF factor'] = (comet233pmm3['NAAF num.'])/NAAF3\n",
    "\n",
    "# make a dataframe that contains only what we need: sequences, AAs, PTMs\n",
    "comet233pmm3_AA = comet233pmm3[['stripped peptide', 'NAAF factor', 'A', 'C', 'D', 'E', 'F', 'G', 'H', 'K', 'I', 'L', \\\n",
    "                                'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', 'm-oxid', 'n-deam', \\\n",
    "                                'q-deam', 'k-hydr', 'p-hydr', 'r-meth']].copy()\n",
    "\n",
    "# multiply the NAAF3 factor by the AAs to normalize its abundance by peak area and peptide length\n",
    "\n",
    "comet233pmm3_AA['A-NAAF3'] = comet233pmm3_AA['A'] * comet233pmm3['NAAF factor']\n",
    "comet233pmm3_AA['C-NAAF3'] = comet233pmm3_AA['C'] * comet233pmm3['NAAF factor']\n",
    "comet233pmm3_AA['D-NAAF3'] = comet233pmm3_AA['D'] * comet233pmm3['NAAF factor']\n",
    "comet233pmm3_AA['E-NAAF3'] = comet233pmm3_AA['E'] * comet233pmm3['NAAF factor']\n",
    "comet233pmm3_AA['F-NAAF3'] = comet233pmm3_AA['F'] * comet233pmm3['NAAF factor']\n",
    "comet233pmm3_AA['G-NAAF3'] = comet233pmm3_AA['G'] * comet233pmm3['NAAF factor']\n",
    "comet233pmm3_AA['H-NAAF3'] = comet233pmm3_AA['H'] * comet233pmm3['NAAF factor']\n",
    "comet233pmm3_AA['K-NAAF3'] = comet233pmm3_AA['K'] * comet233pmm3['NAAF factor']\n",
    "comet233pmm3_AA['I-NAAF3'] = comet233pmm3_AA['I'] * comet233pmm3['NAAF factor']\n",
    "comet233pmm3_AA['L-NAAF3'] = comet233pmm3_AA['L'] * comet233pmm3['NAAF factor']\n",
    "comet233pmm3_AA['M-NAAF3'] = comet233pmm3_AA['M'] * comet233pmm3['NAAF factor']\n",
    "comet233pmm3_AA['N-NAAF3'] = comet233pmm3_AA['N'] * comet233pmm3['NAAF factor']\n",
    "comet233pmm3_AA['P-NAAF3'] = comet233pmm3_AA['P'] * comet233pmm3['NAAF factor']\n",
    "comet233pmm3_AA['Q-NAAF3'] = comet233pmm3_AA['Q'] * comet233pmm3['NAAF factor']\n",
    "comet233pmm3_AA['R-NAAF3'] = comet233pmm3_AA['R'] * comet233pmm3['NAAF factor']\n",
    "comet233pmm3_AA['S-NAAF3'] = comet233pmm3_AA['S'] * comet233pmm3['NAAF factor']\n",
    "comet233pmm3_AA['T-NAAF3'] = comet233pmm3_AA['T'] * comet233pmm3['NAAF factor']\n",
    "comet233pmm3_AA['V-NAAF3'] = comet233pmm3_AA['V'] * comet233pmm3['NAAF factor']\n",
    "comet233pmm3_AA['W-NAAF3'] = comet233pmm3_AA['W'] * comet233pmm3['NAAF factor']\n",
    "comet233pmm3_AA['Y-NAAF3'] = comet233pmm3_AA['Y'] * comet233pmm3['NAAF factor']\n",
    "\n",
    "# multiply the NAAF3 factor by the PTMs normalize its abundance by peak area and peptide length\n",
    "\n",
    "comet233pmm3_AA['ccarb-NAAF3'] = comet233pmm3_AA['c-carb'] * comet233pmm3_AA['NAAF factor']\n",
    "comet233pmm3_AA['moxid-NAAF3'] = comet233pmm3_AA['m-oxid'] * comet233pmm3_AA['NAAF factor']\n",
    "comet233pmm3_AA['ndeam-NAAF3'] = comet233pmm3_AA['n-deam'] * comet233pmm3_AA['NAAF factor']\n",
    "comet233pmm3_AA['qdeam-NAAF3'] = comet233pmm3_AA['q-deam'] * comet233pmm3_AA['NAAF factor']\n",
    "comet233pmm3_AA['khydr-NAAF3'] = comet233pmm3_AA['k-hydr'] * comet233pmm3_AA['NAAF factor']\n",
    "comet233pmm3_AA['phydr-NAAF3'] = comet233pmm3_AA['p-hydr'] * comet233pmm3_AA['NAAF factor']\n",
    "comet233pmm3_AA['rmeth-NAAF3'] = comet233pmm3_AA['r-meth'] * comet233pmm3_AA['NAAF factor']\n",
    "\n",
    "# write the dataframe to a new csv\n",
    "comet233pmm3_AA.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/TPP-PTMopt/ETNP-SKQ17-TPP-PTMopt-hyroxylation//ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_Comet3_AA_NAAF.csv\")\n",
    "\n",
    "comet233pmm3_AA.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Visualizing the results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# take only AA totals and transpose for easier bar plotting in matplotlib\n",
    "\n",
    "comet231aatot = totalcomet231pmm3[['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y']].copy().T\n",
    "\n",
    "# take only PTMs and transpose for easier bar plotting in matplotlib\n",
    "\n",
    "#comet231ptmtot = totalcomet231pmm3[['% C w/ carb.', '% M w/ oxid', '% N w/ deam', '% Q w/ deam', '% K w/ hydr', '% P w/ hydr', '% R w/ meth']].copy().T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# bar plot of residue totals\n",
    "\n",
    "x_labels = ['sample total']\n",
    "\n",
    "ax = comet231aatot.plot(y=['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y'], kind=\"bar\", title = '100 m suspended')\n",
    "plt.xticks(rotation=0)\n",
    "ax.get_legend().remove()\n",
    "ax.set_xticklabels(x_labels)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# bar plot of residue totals\n",
    "# there is no isoleucine (I) in Peaks data, which is why L is really big and I is 0\n",
    "\n",
    "my_colors = [(x/10.0, x/20.0, 0.75) for x in range(len(peaksaatot))] # <-- Quick gradient example along the Red/Green dimensions.\n",
    "\n",
    "ax = peaksaatot.plot(y=['sample total'], kind=\"bar\", color = 'green', title = '100 m suspended')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# bar plot of relative modifications\n",
    "\n",
    "ax = totalpeaks.plot(y=['% C w/ carb.', '% M w/ oxid', '% N w/ deam', '% Q w/ deam', '% K w/ hydr', '% P w/ hydr', '% K w/ meth', '% R w/ meth'], kind=\"bar\", title = '100 m suspended')\n",
    "ax.set_xticklabels([])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# bar plot of relative mods\n",
    "\n",
    "\n",
    "ax = peaksreltot.plot(y=['sample total'], kind=\"bar\", title = '100 m suspended')\n",
    "plt.xticks(rotation=45)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# making evenly spaced bins for the ALC data based on the min and max, called above\n",
    "bins = [50, 55, 60, 65, 70, 75, 80, 85, 90, 95, 100]\n",
    "labels = ['50-55', '55-60', '60-65', '65-70', '70-75', '75-80', '80-85', '85-90', '90-95', '95-100']\n",
    "\n",
    "# use pandas cut function to do the binning itself\n",
    "peaks['binned'] = pd.cut(peaks['ALC (%)'], bins=bins, labels=labels)\n",
    "\n",
    "# bar plots of binned PTM data\n",
    "\n",
    "index = ['50-55', '55-60', '60-65', '65-70', '70-75', '75-80', '80-85', '85-90', '90-95', '95-100']\n",
    "data = {'Total PTMs': [peaks.groupby('binned')['ptm-total'].sum()['50-55'], peaks.groupby('binned')['ptm-total'].sum()['55-60'], peaks.groupby('binned')['ptm-total'].sum()['60-65'], peaks.groupby('binned')['ptm-total'].sum()['65-70'], peaks.groupby('binned')['ptm-total'].sum()['70-75'], peaks.groupby('binned')['ptm-total'].sum()['75-80'], peaks.groupby('binned')['ptm-total'].sum()['80-85'], peaks.groupby('binned')['ptm-total'].sum()['85-90'], peaks.groupby('binned')['ptm-total'].sum()['90-95'], peaks.groupby('binned')['ptm-total'].sum()['95-100']],\n",
    "        'Cys carb.': [peaks.groupby('binned')['c-carb'].sum()['50-55'], peaks.groupby('binned')['c-carb'].sum()['55-60'], peaks.groupby('binned')['c-carb'].sum()['60-65'], peaks.groupby('binned')['c-carb'].sum()['65-70'], peaks.groupby('binned')['c-carb'].sum()['70-75'], peaks.groupby('binned')['c-carb'].sum()['75-80'], peaks.groupby('binned')['c-carb'].sum()['80-85'], peaks.groupby('binned')['c-carb'].sum()['85-90'], peaks.groupby('binned')['c-carb'].sum()['90-95'], peaks.groupby('binned')['c-carb'].sum()['95-100']],\n",
    "        'Met oxi.': [peaks.groupby('binned')['m-oxid'].sum()['50-55'], peaks.groupby('binned')['m-oxid'].sum()['55-60'], peaks.groupby('binned')['m-oxid'].sum()['60-65'], peaks.groupby('binned')['m-oxid'].sum()['65-70'], peaks.groupby('binned')['m-oxid'].sum()['70-75'], peaks.groupby('binned')['m-oxid'].sum()['75-80'], peaks.groupby('binned')['m-oxid'].sum()['80-85'], peaks.groupby('binned')['m-oxid'].sum()['85-90'], peaks.groupby('binned')['m-oxid'].sum()['90-95'], peaks.groupby('binned')['m-oxid'].sum()['95-100']],\n",
    "        'Asp deam.': [peaks.groupby('binned')['n-deam'].sum()['50-55'], peaks.groupby('binned')['n-deam'].sum()['55-60'], peaks.groupby('binned')['n-deam'].sum()['60-65'], peaks.groupby('binned')['n-deam'].sum()['65-70'], peaks.groupby('binned')['n-deam'].sum()['70-75'], peaks.groupby('binned')['n-deam'].sum()['75-80'], peaks.groupby('binned')['n-deam'].sum()['80-85'], peaks.groupby('binned')['n-deam'].sum()['85-90'], peaks.groupby('binned')['n-deam'].sum()['90-95'], peaks.groupby('binned')['n-deam'].sum()['95-100']],\n",
    "        'Glut deam.': [peaks.groupby('binned')['q-deam'].sum()['50-55'], peaks.groupby('binned')['q-deam'].sum()['55-60'], peaks.groupby('binned')['q-deam'].sum()['60-65'], peaks.groupby('binned')['q-deam'].sum()['65-70'], peaks.groupby('binned')['q-deam'].sum()['70-75'], peaks.groupby('binned')['q-deam'].sum()['75-80'], peaks.groupby('binned')['q-deam'].sum()['80-85'], peaks.groupby('binned')['q-deam'].sum()['85-90'], peaks.groupby('binned')['q-deam'].sum()['90-95'], peaks.groupby('binned')['q-deam'].sum()['95-100']],\n",
    "        'Lys hydr': [peaks.groupby('binned')['k-hydr'].sum()['50-55'], peaks.groupby('binned')['k-hydr'].sum()['55-60'], peaks.groupby('binned')['k-hydr'].sum()['60-65'], peaks.groupby('binned')['k-hydr'].sum()['65-70'], peaks.groupby('binned')['k-hydr'].sum()['70-75'], peaks.groupby('binned')['k-hydr'].sum()['75-80'], peaks.groupby('binned')['k-hydr'].sum()['80-85'], peaks.groupby('binned')['k-hydr'].sum()['85-90'], peaks.groupby('binned')['k-hydr'].sum()['90-95'], peaks.groupby('binned')['k-hydr'].sum()['95-100']],\n",
    "        'Pro hydr': [peaks.groupby('binned')['p-hydr'].sum()['50-55'], peaks.groupby('binned')['p-hydr'].sum()['55-60'], peaks.groupby('binned')['p-hydr'].sum()['60-65'], peaks.groupby('binned')['p-hydr'].sum()['65-70'], peaks.groupby('binned')['p-hydr'].sum()['70-75'], peaks.groupby('binned')['p-hydr'].sum()['75-80'], peaks.groupby('binned')['p-hydr'].sum()['80-85'], peaks.groupby('binned')['p-hydr'].sum()['85-90'], peaks.groupby('binned')['p-hydr'].sum()['90-95'], peaks.groupby('binned')['p-hydr'].sum()['95-100']],\n",
    "        'Lys meth.': [peaks.groupby('binned')['k-meth'].sum()['50-55'], peaks.groupby('binned')['k-meth'].sum()['55-60'], peaks.groupby('binned')['k-meth'].sum()['60-65'], peaks.groupby('binned')['k-meth'].sum()['65-70'], peaks.groupby('binned')['k-meth'].sum()['70-75'], peaks.groupby('binned')['k-meth'].sum()['75-80'], peaks.groupby('binned')['k-meth'].sum()['80-85'], peaks.groupby('binned')['k-meth'].sum()['85-90'], peaks.groupby('binned')['k-meth'].sum()['90-95'], peaks.groupby('binned')['k-meth'].sum()['95-100']],\n",
    "        'Arg meth.': [peaks.groupby('binned')['r-meth'].sum()['50-55'], peaks.groupby('binned')['r-meth'].sum()['55-60'], peaks.groupby('binned')['r-meth'].sum()['60-65'], peaks.groupby('binned')['r-meth'].sum()['65-70'], peaks.groupby('binned')['r-meth'].sum()['70-75'], peaks.groupby('binned')['r-meth'].sum()['75-80'], peaks.groupby('binned')['r-meth'].sum()['80-85'], peaks.groupby('binned')['r-meth'].sum()['85-90'], peaks.groupby('binned')['r-meth'].sum()['90-95'], peaks.groupby('binned')['r-meth'].sum()['95-100']]\n",
    "        }\n",
    "\n",
    "peaksbin = pd.DataFrame(data, columns=['Total PTMs','Cys carb.','Met oxi.','Asp deam.', 'Glut deam.', 'Lys hydr', 'Pro hydr', 'Lys meth.', 'Arg meth.'], index=index)\n",
    "\n",
    "# write the peaks bin ptm dataframe to a csv:\n",
    "peaksbin.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_ptm.csv\")\n",
    "\n",
    "ax1 = peaksbin.plot.bar(y='Total PTMs', rot=45)\n",
    "ax1.set_title('Total PTMs')\n",
    "\n",
    "ax2 = peaksbin.plot.bar(y='Cys carb.', rot=45)\n",
    "ax2.set_title('Cysteine carbamidomethylation')\n",
    "\n",
    "ax3 = peaksbin.plot.bar(y='Met oxi.', rot=45)\n",
    "ax3.set_title('Methionine oxidation')\n",
    "\n",
    "ax4 = peaksbin.plot.bar(y='Asp deam.', rot=45)\n",
    "ax4.set_title('Asparagine deamidation')\n",
    "\n",
    "ax5 = peaksbin.plot.bar(y='Glut deam.', rot=45)\n",
    "ax5.set_title('Glutamine deamidation')\n",
    "\n",
    "ax6 = peaksbin.plot.bar(y='Lys hydr', rot=45)\n",
    "ax6.set_title('Lysine hydroxylation')\n",
    "\n",
    "ax7 = peaksbin.plot.bar(y='Pro hydr', rot=45)\n",
    "ax7.set_title('Proline hydroxylation')\n",
    "\n",
    "ax8 = peaksbin.plot.bar(y='Lys meth.', rot=45)\n",
    "ax8.set_title('Lysine methylation')\n",
    "\n",
    "ax9 = peaksbin.plot.bar(y='Arg meth.', rot=45)\n",
    "ax9.set_title('Arginine methylation')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exporting txt files of stripped peptides at confidence cutoffs:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# keep only peptide column >50% ALC\n",
    "pep = peaks[[\"stripped peptide\"]]\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "pep.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.txt\", header=False, index=False)\n",
    "\n",
    "# made the text file into a FASTA \n",
    "\n",
    "!awk '{print \">\"NR\"\\n\"$0}' /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.txt > /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas\n",
    "\n",
    "# look\n",
    "\n",
    "print(\"# of DN peptide >50% ALC\", len(pep))\n",
    "pep.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# keep only peptides  >80% ALC\n",
    "peaks80 = peaks.loc[peaks['ALC (%)'] >= 80]\n",
    "\n",
    "# see how many rows and double check\n",
    "# peaks80.head(-10)\n",
    "\n",
    "# keep only peptide column \n",
    "pep80 = peaks80[[\"stripped peptide\"]]\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "pep80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN80_stripped_peptides.txt\", header=False, index=False)\n",
    "\n",
    "# made the text file into a FASTA \n",
    "\n",
    "!awk '{print \">\"NR\"\\n\"$0}' /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN80_stripped_peptides.txt > /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN80_stripped_peptides.fas\n",
    "\n",
    "print(\"# of DN peptide >80% ALC\", len(pep80))\n",
    "pep80.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Using BioPython to query peptide sequences\n",
    "\n",
    "I installed the BioPython package using `pip install biopython`. All instructions and information [here](https://www.tutorialspoint.com/biopython/index.htm). \n",
    "\n",
    "GitHub project: https://github.com/biopython/biopython\n",
    "\n",
    "I'm relying on the ProtParam module to parse sequences for relative AA composition, instability, secondary structure, instability, and hydrophobicity. You can read more about that module and the studies the indecies are derived from here:\n",
    "\n",
    "https://biopython.org/wiki/ProtParam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# Bio.SeqIO is the standard Sequence Input/Output interface for BioPython 1.43 and later\n",
    "# Bio.SeqIO provides a simple uniform interface to input and output assorted sequence file formats.\n",
    "# (including multiple sequence alignments), but will only deal with sequences as SeqRecord objects\n",
    "\n",
    "# for accepted file formats see https://biopython.org/wiki/SeqIO\n",
    "\n",
    "from Bio import SeqIO\n",
    "#for seq_record in SeqIO.parse(\"/home/millieginty/Documents/git-repos/2017-etnp/data/MED4/MED2_tryp_1raw_db_peptides_nmod.fasta\", \"fasta\"):\n",
    "    #print(seq_record.id)\n",
    "    #print(repr(seq_record.seq))\n",
    "    #print(len(seq_record))\n",
    "    \n",
    "# I commented the print functions out so the output doesn't take up too much space. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# seeing what the ProtParam module can do with a single protein sequence:\n",
    "\n",
    "from Bio.SeqUtils.ProtParam import ProteinAnalysis\n",
    "\n",
    "test_seq = \"MAEGEITTFTALTEKFNLPPGNYKKPKLLYCSNGGHFLRILPDGTVDGTRDRSDQHIQLQLSAESVGEVYIKSTETGQYLAMDTSGLLYGSQTPSEECLFLERLEENHYNTYTSKKHAEKNWFVGLKKNGSCKRGPRTHYGQKAILFLPLPV\"\n",
    "\n",
    "analysed_seq = ProteinAnalysis(test_seq)\n",
    "print(\"molecular weight of seq =\", analysed_seq.molecular_weight())\n",
    "\n",
    "# calculates the aromaticity value of a protein according to Lobry & Gautier (1994, Nucleic Acids Res., 22, 3174-3180). \n",
    "# it's simply the relative frequency of Phe+Trp+Tyr.\n",
    "\n",
    "analysed_seq.aromaticity()\n",
    "print(\"aromaticity of seq =\", analysed_seq.aromaticity())\n",
    "\n",
    "# secondary_structure_fraction:\n",
    "# this methods returns a list of the fraction of amino acids which tend to be in helix, turn or sheet. \n",
    "# AAs in helix: V, I, Y, F, W, L\n",
    "# AAs in turn: N, P, G, S\n",
    "# AAs in sheet: E, M, A, L\n",
    "# the returned list contains 3 values: [Helix, Turn, Sheet]\n",
    "\n",
    "analysed_seq.secondary_structure_fraction()\n",
    "print(\"frac in H T S =\", analysed_seq.secondary_structure_fraction())\n",
    "\n",
    "# the instability index, an implementation of the method of Guruprasad et al. (1990, Protein Engineering, 4, 155-161).\n",
    "# this method tests a protein for stability. \n",
    "# any value above 40 means the protein is unstable (=has a short half life)\n",
    "# NOT SURE WHAT THIS MEANS FOR PEPTIDES, BUT WE COULD DO THIS FOR PROTEINS\n",
    "\n",
    "analysed_seq.instability_index()\n",
    "print(\"instability =\", analysed_seq.instability_index())\n",
    "\n",
    "# count_amino_acids will do just that, and get_amino_acids_percent will return %'s for each AA across the sequence. \n",
    "analysed_seq.get_amino_acids_percent()\n",
    "\n",
    "# taking the returned dictionary and converting to a dataframe\n",
    "\n",
    "aadict = analysed_seq.get_amino_acids_percent()\n",
    "aadf = pd.DataFrame(list(aadict.items()),columns = ['residue','% occurance']) \n",
    "\n",
    "aadf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# use SeqIO and a loop to apply count_amino_acids to each sequence in the file\n",
    "# aatot will give us the total number of each residue in the entire sample output\n",
    "\n",
    "import collections\n",
    "from Bio import SeqIO\n",
    "from Bio.SeqUtils.ProtParam import ProteinAnalysis\n",
    "\n",
    "all_aas = collections.defaultdict(int)\n",
    "for record in SeqIO.parse(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas\", \"fasta\"):\n",
    "    x = ProteinAnalysis(str(record.seq))\n",
    "    #print(record.id, x.count_amino_acids())\n",
    "    for aa, count in x.count_amino_acids().items():\n",
    "        all_aas[aa] += count        \n",
    "        \n",
    "# made a dataframe for amino acid total counts        \n",
    "data = (all_aas)\n",
    "aatot = pd.DataFrame(data, index = ['sample sequence total'])\n",
    "aatot.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pandas import Series, DataFrame\n",
    "\n",
    "with open('/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas') as fasta_file:  # Will close handle cleanly\n",
    "    identifiers = []\n",
    "    lengths = []\n",
    "    for seq_record in SeqIO.parse(fasta_file, 'fasta'):  # (generator)\n",
    "        identifiers.append(seq_record.id)\n",
    "        lengths.append(len(seq_record.seq))\n",
    "        \n",
    "        \n",
    "#converting lists to pandas Series    \n",
    "s1 = Series(identifiers, name='ID')\n",
    "s2 = Series(lengths, name='length')\n",
    "\n",
    "#Gathering Series into a pandas DataFrame and rename index as ID column\n",
    "idseq = DataFrame(dict(ID=s1, length=s2)).set_index(['ID'])\n",
    "\n",
    "idseq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pandas import Series, DataFrame\n",
    "\n",
    "\n",
    "\n",
    "with open('/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas) as fasta_file:  # Will close handle cleanly\n",
    "    identifiers = []\n",
    "    lengths = []\n",
    "    aa = []\n",
    "    for seq_record in SeqIO.parse(fasta_file, 'fasta'):  # (generator)\n",
    "        identifiers.append(seq_record.id)\n",
    "        lengths.append(len(seq_record.seq))\n",
    "        aa.count_amino_acids(seq_record.seq)\n",
    "        \n",
    "        \n",
    "#converting lists to pandas Series    \n",
    "s1 = Series(identifiers, name='ID')\n",
    "s2 = Series(lengths, name='length')\n",
    "s3 = Series(aa, name='AAs')\n",
    "\n",
    "#Gathering Series into a pandas DataFrame and rename index as ID column\n",
    "idseq = DataFrame(dict(ID=s1, length=s2, AAs=s3)).set_index(['ID'])\n",
    "\n",
    "idseq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio.SeqUtils import ProtParam\n",
    "\n",
    "handle = open(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas\") \n",
    "for record in SeqIO.parse(handle, \"fasta\"): \n",
    "    seq = str(record.seq)\n",
    "    X = ProtParam.ProteinAnalysis(seq)\n",
    "    print(X.count_amino_acids()) \n",
    "    #print X.get_amino_acids_percent() \n",
    "    #print X.molecular_weight() \n",
    "    #print X.aromaticity() \n",
    "    #print X.instability_index() \n",
    "    #print X.flexibility() \n",
    "    #print X.isoelectric_point() \n",
    "    #print X.secondary_structure_fraction()\n",
    "    \n",
    "# made a data series from the count_amino_acids function\n",
    "# aacount = {X.count_amino_acids()}\n",
    "\n",
    "# made a pandas dataframe from the series generated above\n",
    "# aacount = pd.DataFrame(list(data.items()),columns = ['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y']) \n",
    "\n",
    "aacount = pd.DataFrame(X.count_amino_acids(), index=[0])\n",
    "\n",
    "# look at new dataframe\n",
    "\n",
    "# aacount.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
