{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Manipulation of Trans Proteomic Pipeline (TPP) Comet database search results of *Prochlorococus MED4* LC-MS/MS data using python.\n",
    "\n",
    "Starting with: \n",
    "\n",
    "- Comet output (.xlxs and .csv) of PTM-optimized database searches, sorted by XCorr (descending) and run through XInteract to extract precursor intensities and protein descriptions mapped from the search database.\n",
    "\n",
    "Ending with:\n",
    "\n",
    "- Files with stripped (no PTMs or tryptic ends) peptide lists and\n",
    "- Columns with #'s of each modification in every sequence\n",
    "- Column with stripped peptide lengths (# amino acids)\n",
    "- Histogram of sequence lengths\n",
    "- Bar plots of PTM occurance\n",
    "\n",
    "### To use:\n",
    "\n",
    "#### 1. Change the input file name in *IN 4*\n",
    "#### 2. Change output file name in *IN 6*, *IN 7*, *IN 8*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For technical duplicates, I exported Comet search results as both Excel files and as CSVs into my ETNP 2017 git repo:\n",
    "\n",
    "Also, when running through XInteract in the TPP, I combined the duplicate injections into a single PepXML file which I exported as an xls file and converted to a csv."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP\n"
     ]
    }
   ],
   "source": [
    "cd /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RAL4_MED2_trypsin_1_PTMopt_Comet.csv\r\n",
      "RAL4_MED2_trypsin_1_PTMopt_Comet_stripped.csv\r\n",
      "RAL4_MED2_trypsin_1_PTMopt_Comet_stripped_peptides_2.5XCorr.txt\r\n",
      "RAL4_MED2_trypsin_1_PTMopt_Comet_stripped_peptides.txt\r\n",
      "RAL4_MED2_trypsin_1_PTMopt_Comet_stripped_work.ods\r\n",
      "RAL4_MED2_trypsin_1_PTMopt_Comet.xlsx\r\n",
      "RAL4_MED2_trypsin_1_PTMopt_PepProp90.csv\r\n",
      "RAL4_MED2_trypsin_1_PTMopt_PepProp90_stripped.csv\r\n",
      "RAL4_MED2_trypsin_1_PTMopt_PepProp90_stripped_peptides\r\n",
      "RAL4_MED2_trypsin_1_PTMopt_PepProp90.xlsx\r\n",
      "RAL4_MED2_trypsin_2_PTMopt_Comet.csv\r\n",
      "RAL4_MED2_trypsin_2_PTMopt_Comet_stripped.csv\r\n",
      "RAL4_MED2_trypsin_2_PTMopt_Comet_stripped_peptides_2.5XCorr.txt\r\n",
      "RAL4_MED2_trypsin_2_PTMopt_Comet_stripped_peptides.txt\r\n",
      "RAL4_MED2_trypsin_2_PTMopt_Comet_stripped_work.ods\r\n",
      "RAL4_MED2_trypsin_2_PTMopt_Comet.xlsx\r\n",
      "RAL4_MED2_trypsin_2_PTMopt_PepProp90.csv\r\n",
      "RAL4_MED2_trypsin_2_PTMopt_PepProp90_stripped.csv\r\n",
      "RAL4_MED2_trypsin_2_PTMopt_PepProp90_stripped_peptides\r\n",
      "RAL4_MED2_trypsin_2_PTMopt_PepProp90.xlsx\r\n",
      "RAL4_MED2_trypsin_combine_PTMopt_Comet_stripped__nonredundant_peptides_2.5XCorr.txt\r\n",
      "RAL4_MED2_trypsin_combine_PTMopt_Comet_stripped_peptides_2.5XCorr.txt\r\n",
      "RAL4_MED2_trypsin_combine_PTMopt_Comet_stripped_peptides.txt\r\n",
      "RAL95_MED2_trypsin_combine_PTMopt_DN50_nonredundant_stripped_peptides.txt\r\n",
      "RAL95_MED2_trypsin_combine_PTMopt_DN80_nonredundant_stripped_peptides.txt\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.0.5'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# LIBRARIES\n",
    "#import pandas library for working with tabular data\n",
    "import os\n",
    "os.getcwd()\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "#from matplotlib import pyplot\n",
    "import seaborn as sns\n",
    "from scipy.stats import kde\n",
    "#import regular expresson (regex)\n",
    "import re\n",
    "#check pandas version\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# redundant Comet peptides in combined dataframe 118797\n",
      "# nonredundant Comet peptides in combined dataframe 118797\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>spectrum</th>\n",
       "      <th>xcorr</th>\n",
       "      <th>deltacn</th>\n",
       "      <th>expect</th>\n",
       "      <th>ions</th>\n",
       "      <th>peptide</th>\n",
       "      <th>protein</th>\n",
       "      <th>calc_neutral_pep_mass</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.41211.41211.4</td>\n",
       "      <td>8.768</td>\n",
       "      <td>1</td>\n",
       "      <td>1.42E-09</td>\n",
       "      <td>22/156</td>\n",
       "      <td>K.LFADENHLSPAVTAIQIEDIDAEQFRK.N</td>\n",
       "      <td>PMM0035</td>\n",
       "      <td>3069.5407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.32793.32793.4</td>\n",
       "      <td>8.578</td>\n",
       "      <td>1</td>\n",
       "      <td>9.31E-10</td>\n",
       "      <td>18/204</td>\n",
       "      <td>K.LAIDDSSIN[115.03]LDQVDYIN[115.03]AHGTSTTANDK...</td>\n",
       "      <td>PMM1609</td>\n",
       "      <td>3735.7599</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.50751.50751.4</td>\n",
       "      <td>8.469</td>\n",
       "      <td>0.646</td>\n",
       "      <td>3.43E-14</td>\n",
       "      <td>24/156</td>\n",
       "      <td>R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D</td>\n",
       "      <td>PMM1436</td>\n",
       "      <td>2831.4409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.31065.31065.3</td>\n",
       "      <td>8.061</td>\n",
       "      <td>0.561</td>\n",
       "      <td>2.18E-11</td>\n",
       "      <td>17/92</td>\n",
       "      <td>K.IDIEDGGIVTIASHDGAAAEEAQK.I</td>\n",
       "      <td>PMM1191</td>\n",
       "      <td>2409.166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.50278.50278.4</td>\n",
       "      <td>7.949</td>\n",
       "      <td>0.584</td>\n",
       "      <td>3.02E-09</td>\n",
       "      <td>23/156</td>\n",
       "      <td>R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D</td>\n",
       "      <td>PMM1436</td>\n",
       "      <td>2831.4409</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      spectrum  xcorr deltacn    expect  \\\n",
       "0  022016_RAL4_95_MED2_trypsin_1.41211.41211.4  8.768       1  1.42E-09   \n",
       "1  022016_RAL4_95_MED2_trypsin_1.32793.32793.4  8.578       1  9.31E-10   \n",
       "2  022016_RAL4_95_MED2_trypsin_1.50751.50751.4  8.469   0.646  3.43E-14   \n",
       "3  022016_RAL4_95_MED2_trypsin_1.31065.31065.3  8.061   0.561  2.18E-11   \n",
       "4  022016_RAL4_95_MED2_trypsin_1.50278.50278.4  7.949   0.584  3.02E-09   \n",
       "\n",
       "     ions                                            peptide  protein  \\\n",
       "0  22/156                    K.LFADENHLSPAVTAIQIEDIDAEQFRK.N  PMM0035   \n",
       "1  18/204  K.LAIDDSSIN[115.03]LDQVDYIN[115.03]AHGTSTTANDK...  PMM1609   \n",
       "2  24/156    R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D  PMM1436   \n",
       "3   17/92                       K.IDIEDGGIVTIASHDGAAAEEAQK.I  PMM1191   \n",
       "4  23/156    R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D  PMM1436   \n",
       "\n",
       "  calc_neutral_pep_mass  \n",
       "0             3069.5407  \n",
       "1             3735.7599  \n",
       "2             2831.4409  \n",
       "3              2409.166  \n",
       "4             2831.4409  "
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#read the CSVs of each replicate into a datadrame we name 'comet' using the pandas read_csv function\n",
    "comet1 = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_1_PTMopt_Comet.csv\")\n",
    "comet2 = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_2_PTMopt_Comet.csv\")\n",
    "\n",
    "frames = [comet1, comet2]\n",
    "\n",
    "# concatenate dataframes\n",
    "cometdup = pd.concat(frames, sort=False)\n",
    "\n",
    "# remove redundant rows\n",
    "# actually, this won't work since the file name will be different and spectrum #. We'll deduplicate later but I'll leave this for now. \n",
    "comet = pd.DataFrame.drop_duplicates(cometdup)\n",
    "\n",
    "print(\"# redundant Comet peptides in combined dataframe\", len(cometdup))\n",
    "print(\"# nonredundant Comet peptides in combined dataframe\", len(comet))\n",
    "\n",
    "#look at the dataframe\n",
    "comet.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The peptide column has the residues before and after the tryptic terminii as well as masses of modified residues (e.g., 160.03 Da for carbamidomethylated cysteine). We want to make new columns with all that information and make a column with only the 'stripped' peptide sequence that's just amino acids - this we can then align against other sequences, for example. \n",
    "\n",
    "Modified residues were allowed for:\n",
    "\n",
    "- fixed carbamodimethylation of cysteine 57.021464 C\n",
    "- varialbe oxidation of methionine: 15.9949 M\n",
    "- variable deamidation of asparagine, glumatine: 0.984016 NQ\n",
    "- variable iron cation on lysine: 54.010565 K\n",
    "- variable methylation of lysine and arginine: 14.015650 KR\n",
    "\n",
    "We'll then write this manipulated dataframe to a new file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>spectrum</th>\n",
       "      <th>xcorr</th>\n",
       "      <th>deltacn</th>\n",
       "      <th>expect</th>\n",
       "      <th>ions</th>\n",
       "      <th>peptide</th>\n",
       "      <th>protein</th>\n",
       "      <th>calc_neutral_pep_mass</th>\n",
       "      <th>L terminus</th>\n",
       "      <th>R terminus</th>\n",
       "      <th>...</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>k-iron</th>\n",
       "      <th>k-meth</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>stripped length</th>\n",
       "      <th>ptm-total</th>\n",
       "      <th>ptm-norm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.41211.41211.4</td>\n",
       "      <td>8.768</td>\n",
       "      <td>1</td>\n",
       "      <td>1.42E-09</td>\n",
       "      <td>22/156</td>\n",
       "      <td>K.LFADENHLSPAVTAIQIEDIDAEQFRK.N</td>\n",
       "      <td>PMM0035</td>\n",
       "      <td>3069.5407</td>\n",
       "      <td>K</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>LFADENHLSPAVTAIQIEDIDAEQFRK</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.32793.32793.4</td>\n",
       "      <td>8.578</td>\n",
       "      <td>1</td>\n",
       "      <td>9.31E-10</td>\n",
       "      <td>18/204</td>\n",
       "      <td>K.LAIDDSSIN[115.03]LDQVDYIN[115.03]AHGTSTTANDK...</td>\n",
       "      <td>PMM1609</td>\n",
       "      <td>3735.7599</td>\n",
       "      <td>K</td>\n",
       "      <td>S</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>LAIDDSSINAHGTSTTANDKNETSAIK</td>\n",
       "      <td>27</td>\n",
       "      <td>2</td>\n",
       "      <td>0.074074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.50751.50751.4</td>\n",
       "      <td>8.469</td>\n",
       "      <td>0.646</td>\n",
       "      <td>3.43E-14</td>\n",
       "      <td>24/156</td>\n",
       "      <td>R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D</td>\n",
       "      <td>PMM1436</td>\n",
       "      <td>2831.4409</td>\n",
       "      <td>R</td>\n",
       "      <td>D</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SGLQNAASIAGMIVADLPEKK</td>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "      <td>0.095238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.31065.31065.3</td>\n",
       "      <td>8.061</td>\n",
       "      <td>0.561</td>\n",
       "      <td>2.18E-11</td>\n",
       "      <td>17/92</td>\n",
       "      <td>K.IDIEDGGIVTIASHDGAAAEEAQK.I</td>\n",
       "      <td>PMM1191</td>\n",
       "      <td>2409.166</td>\n",
       "      <td>K</td>\n",
       "      <td>I</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>IDIEDGGIVTIASHDGAAAEEAQK</td>\n",
       "      <td>24</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.50278.50278.4</td>\n",
       "      <td>7.949</td>\n",
       "      <td>0.584</td>\n",
       "      <td>3.02E-09</td>\n",
       "      <td>23/156</td>\n",
       "      <td>R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D</td>\n",
       "      <td>PMM1436</td>\n",
       "      <td>2831.4409</td>\n",
       "      <td>R</td>\n",
       "      <td>D</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SGLQNAASIAGMIVADLPEKK</td>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "      <td>0.095238</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      spectrum  xcorr deltacn    expect  \\\n",
       "0  022016_RAL4_95_MED2_trypsin_1.41211.41211.4  8.768       1  1.42E-09   \n",
       "1  022016_RAL4_95_MED2_trypsin_1.32793.32793.4  8.578       1  9.31E-10   \n",
       "2  022016_RAL4_95_MED2_trypsin_1.50751.50751.4  8.469   0.646  3.43E-14   \n",
       "3  022016_RAL4_95_MED2_trypsin_1.31065.31065.3  8.061   0.561  2.18E-11   \n",
       "4  022016_RAL4_95_MED2_trypsin_1.50278.50278.4  7.949   0.584  3.02E-09   \n",
       "\n",
       "     ions                                            peptide  protein  \\\n",
       "0  22/156                    K.LFADENHLSPAVTAIQIEDIDAEQFRK.N  PMM0035   \n",
       "1  18/204  K.LAIDDSSIN[115.03]LDQVDYIN[115.03]AHGTSTTANDK...  PMM1609   \n",
       "2  24/156    R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D  PMM1436   \n",
       "3   17/92                       K.IDIEDGGIVTIASHDGAAAEEAQK.I  PMM1191   \n",
       "4  23/156    R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D  PMM1436   \n",
       "\n",
       "  calc_neutral_pep_mass L terminus R terminus  ...  m-oxid  n-deam  q-deam  \\\n",
       "0             3069.5407          K          N  ...       0       0       0   \n",
       "1             3735.7599          K          S  ...       0       2       0   \n",
       "2             2831.4409          R          D  ...       1       0       0   \n",
       "3              2409.166          K          I  ...       0       0       0   \n",
       "4             2831.4409          R          D  ...       1       0       0   \n",
       "\n",
       "   k-iron  k-meth  r-meth             stripped peptide  stripped length  \\\n",
       "0       0       0       0  LFADENHLSPAVTAIQIEDIDAEQFRK               27   \n",
       "1       0       0       0  LAIDDSSINAHGTSTTANDKNETSAIK               27   \n",
       "2       0       0       0        SGLQNAASIAGMIVADLPEKK               21   \n",
       "3       0       0       0     IDIEDGGIVTIASHDGAAAEEAQK               24   \n",
       "4       0       0       0        SGLQNAASIAGMIVADLPEKK               21   \n",
       "\n",
       "   ptm-total  ptm-norm  \n",
       "0          0  0.000000  \n",
       "1          2  0.074074  \n",
       "2          2  0.095238  \n",
       "3          0  0.000000  \n",
       "4          2  0.095238  \n",
       "\n",
       "[5 rows x 41 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get rid of rows where the xcorr is unavailable (usually 3 or so)\n",
    "comet = comet[comet.xcorr != '[unavailable]']\n",
    "\n",
    "# use str.strip with indexing by str[0] to add a column with the peptide's left terminus\n",
    "comet['L terminus'] = comet['peptide'].astype(str).str[0]\n",
    "\n",
    "# use str.strip with indexing by str[-1] to add a column with the peptide's left terminus\n",
    "comet['R terminus'] = comet['peptide'].str.strip().str[-1]\n",
    "\n",
    "# use a count function to enumerate the # of A's (alanines) in each peptide\n",
    "comet['A'] = comet['peptide'].str.count(\"A\")\n",
    "\n",
    "# use a count function to enumerate the # of C's (cysteines) in each peptide\n",
    "comet['C'] = comet['peptide'].str.count(\"C\")\n",
    "\n",
    "# use a count function to enumerate the # of D's (aspartic acids) in each peptide\n",
    "comet['D'] = comet['peptide'].str.count(\"D\")\n",
    "\n",
    "# use a count function to enumerate the # of E's (glutamic acids) in each peptide\n",
    "comet['E'] = comet['peptide'].str.count(\"E\")\n",
    "\n",
    "# use a count function to enumerate the # of F's (phenylalanines) in each peptide\n",
    "comet['F'] = comet['peptide'].str.count(\"F\")\n",
    "\n",
    "# use a count function to enumerate the # of G's (glycines) in each peptide\n",
    "comet['G'] = comet['peptide'].str.count(\"G\")\n",
    "\n",
    "# use a count function to enumerate the # of H's (histidines) in each peptide\n",
    "comet['H'] = comet['peptide'].str.count(\"H\")\n",
    "\n",
    "# use a count function to enumerate the # of I's (isoleucines) in each peptide\n",
    "# in comet output, there will be no isoleucines (they're lumped in with leucines)\n",
    "comet['I'] = comet['peptide'].str.count(\"I\")\n",
    "\n",
    "# use a count function to enumerate the # of K's (lysines) in each peptide\n",
    "comet['K'] = comet['peptide'].str.count(\"K\")\n",
    "\n",
    "# use a count function to enumerate the # of L's (leucines) in each peptide\n",
    "comet['L'] = comet['peptide'].str.count(\"L\")\n",
    "\n",
    "# use a count function to enumerate the # of M's (methionines) in each peptide\n",
    "comet['M'] = comet['peptide'].str.count(\"M\")\n",
    "\n",
    "# use a count function to enumerate the # of N's (asparagines) in each peptide\n",
    "comet['N'] = comet['peptide'].str.count(\"N\")\n",
    "\n",
    "# use a count function to enumerate the # of P's ([prolines]) in each peptide\n",
    "comet['P'] = comet['peptide'].str.count(\"P\")\n",
    "\n",
    "# use a count function to enumerate the # of Q's (glutamines) in each peptide\n",
    "comet['Q'] = comet['peptide'].str.count(\"Q\")\n",
    "\n",
    "# use a count function to enumerate the # of R's (arginines) in each peptide\n",
    "comet['R'] = comet['peptide'].str.count(\"R\")\n",
    "\n",
    "# use a count function to enumerate the # of S's (serines) in each peptide\n",
    "comet['S'] = comet['peptide'].str.count(\"S\")\n",
    "\n",
    "# use a count function to enumerate the # of T's (threonines) in each peptide\n",
    "comet['T'] = comet['peptide'].str.count(\"T\")\n",
    "\n",
    "# use a count function to enumerate the # of V's (valines) in each peptide\n",
    "comet['V'] = comet['peptide'].str.count(\"V\")\n",
    "\n",
    "# use a count function to enumerate the # of W's (tryptophans) in each peptide\n",
    "comet['W'] = comet['peptide'].str.count(\"W\")\n",
    "\n",
    "# use a count function to enumerate the # of Y's (tyrosines) in each peptide\n",
    "comet['Y'] = comet['peptide'].str.count(\"Y\")\n",
    "\n",
    "# use a count function to enumerate the # of ccarbamidomethylated C's in each peptide\n",
    "comet['c-carb'] = comet['peptide'].str.count(\"160.03\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "comet['m-oxid'] = comet['peptide'].str.count(\"147.04\")\n",
    "\n",
    "# use a count function to enumerate the # of deamidated N's in each peptide\n",
    "comet['n-deam'] = comet['peptide'].str.count(\"115.03\")\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "comet['q-deam'] = comet['peptide'].str.count(\"129.04\")\n",
    "\n",
    "# use a count function to enumerate the # of iron adducted K's in each peptide\n",
    "comet['k-iron'] = comet['peptide'].str.count(\"182.11\")\n",
    "\n",
    "# use a count function to enumerate the # of methylated K's in each peptide\n",
    "comet['k-meth'] = comet['peptide'].str.count(\"142.11\")\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "comet['r-meth'] = comet['peptide'].str.count(\"170.12\")\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "comet['stripped peptide'] = comet['peptide'].str[2:].str[:-2].str.replace(r\"\\[.*\\]\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "comet['stripped length'] = comet['stripped peptide'].apply(len)\n",
    "\n",
    "# total the number of modifications in sequence\n",
    "comet['ptm-total'] = comet['c-carb'] + comet['m-oxid'] + comet['n-deam'] + comet['q-deam'] + comet['k-iron'] + comet['k-meth'] + comet['r-meth']\n",
    "\n",
    "# total the number of modifications in sequence relative to # of AAs\n",
    "comet['ptm-norm'] = comet['ptm-total'] / comet['stripped length']\n",
    "\n",
    "# write modified dataframe to new txt file, same name + 'stripped'\n",
    "comet.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_1_PTMopt_Comet_stripped.csv\")\n",
    "\n",
    "\n",
    "# check out the results\n",
    "comet.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# redundant Comet peptides 115289\n",
      "Max Xcorr:  8.78\n",
      "Min Xcorr:  0.001\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LFADENHLSPAVTAIQIEDIDAEQFRK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LAIDDSSINAHGTSTTANDKNETSAIK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SGLQNAASIAGMIVADLPEKK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IDIEDGGIVTIASHDGAAAEEAQK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SGLQNAASIAGMIVADLPEKK</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              stripped peptide\n",
       "0  LFADENHLSPAVTAIQIEDIDAEQFRK\n",
       "1  LAIDDSSINAHGTSTTANDKNETSAIK\n",
       "2        SGLQNAASIAGMIVADLPEKK\n",
       "3     IDIEDGGIVTIASHDGAAAEEAQK\n",
       "4        SGLQNAASIAGMIVADLPEKK"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# keep only stripped peptide column\n",
    "pep = comet[[\"stripped peptide\"]]\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "pep.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_combine_PTMopt_Comet_stripped_peptides.txt\", header=False, index=False)\n",
    "\n",
    "\n",
    "# look\n",
    "\n",
    "print(\"# redundant Comet peptides\", len(pep))\n",
    "print(\"Max Xcorr: \", comet['xcorr'].max())\n",
    "print(\"Min Xcorr: \", comet['xcorr'].min())\n",
    "\n",
    "pep.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Calculating the false discovery rate (% FDR)\n",
    "\n",
    "### Filtering PSMs > a selected XCorr value and exporting peptides"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# real Comet PSMs 80626\n",
      "# decoy Comet PSMs 34663\n",
      "False discovery rate =  42.99233497879096\n"
     ]
    }
   ],
   "source": [
    "# Let's separate out the decoy hits from the good ones\n",
    "\n",
    "cometpmm = comet[~comet['protein'].str.contains(\"DECOY\")]\n",
    "cometdec = comet[comet['protein'].str.contains(\"DECOY\")]\n",
    "\n",
    "# how many PSM that are only PMM (proteins in the database)?\n",
    "\n",
    "print(\"# real Comet PSMs\", len(cometpmm))\n",
    "\n",
    "# compared to how many PSMs containing decoys?\n",
    "\n",
    "print(\"# decoy Comet PSMs\", len(cometdec))\n",
    "\n",
    "# calculate the bulk FDR (all PSMs so let's not beat ourselves up)\n",
    "\n",
    "r = len(cometpmm)\n",
    "d = len(cometdec)\n",
    "\n",
    "FDR = d/r*100\n",
    "\n",
    "print(\"False discovery rate = \", FDR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# real Comet PSMs 35928\n",
      "# decoy Comet PSMs 1273\n",
      "False discovery rate =  3.4219510228219674\n"
     ]
    }
   ],
   "source": [
    "# keep only peptides  >2.5 XCorr\n",
    "# need to convert Xcorr column from strings to numeric so we can use loc\n",
    "comet['xcorr'] = pd.to_numeric(comet['xcorr'])\n",
    "\n",
    "comet25 = comet.loc[comet['xcorr'] >= 2.5]\n",
    "\n",
    "# What's the FDR?\n",
    "\n",
    "# Let's separate out the decoy hits from the good ones\n",
    "\n",
    "cometpmm25 = comet25[~comet25['protein'].str.contains(\"DECOY\")]\n",
    "cometdec25 = comet25[comet25['protein'].str.contains(\"DECOY\")]\n",
    "\n",
    "# how many PSM that are only PMM (proteins in the database)?\n",
    "\n",
    "print(\"# real Comet PSMs\", len(cometpmm25))\n",
    "\n",
    "# compared to how many PSMs containing decoys?\n",
    "\n",
    "print(\"# decoy Comet PSMs\", len(cometdec25))\n",
    "\n",
    "# calculate the FDR \n",
    "\n",
    "r = len(cometpmm25)\n",
    "d = len(cometdec25)\n",
    "\n",
    "FDR = d/(d+r)*100\n",
    "\n",
    "print(\"False discovery rate = \", FDR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# real Comet PSMs 26923\n",
      "# decoy Comet PSMs 258\n",
      "False discovery rate =  0.9491924506088812\n"
     ]
    }
   ],
   "source": [
    "# keep only peptides  >3 XCorr\n",
    "# need to convert Xcorr column from strings to numeric so we can use loc\n",
    "comet['xcorr'] = pd.to_numeric(comet['xcorr'])\n",
    "\n",
    "comet3 = comet.loc[comet['xcorr'] >= 3]\n",
    "\n",
    "# What's the FDR?\n",
    "\n",
    "# Let's separate out the decoy hits from the good ones\n",
    "\n",
    "cometpmm3 = comet3[~comet3['protein'].str.contains(\"DECOY\")]\n",
    "cometdec3 = comet3[comet3['protein'].str.contains(\"DECOY\")]\n",
    "\n",
    "# how many PSM that are only PMM (proteins in the database)?\n",
    "\n",
    "print(\"# real Comet PSMs\", len(cometpmm3))\n",
    "\n",
    "# compared to how many PSMs containing decoys?\n",
    "\n",
    "print(\"# decoy Comet PSMs\", len(cometdec3))\n",
    "\n",
    "# calculate the FDR \n",
    "\n",
    "r = len(cometpmm3)\n",
    "d = len(cometdec3)\n",
    "\n",
    "FDR = d/(d+r)*100\n",
    "\n",
    "print(\"False discovery rate = \", FDR)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exporting peptides from a XCorr > 2.5 and XCorr > 3 thresholds:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# redundant Comet peptides >2.5 XCorr 35928\n",
      "# nonredundant Comet peptides >2.5 XCOrr 12280\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LFADENHLSPAVTAIQIEDIDAEQFRK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LAIDDSSINAHGTSTTANDKNETSAIK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SGLQNAASIAGMIVADLPEKK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IDIEDGGIVTIASHDGAAAEEAQK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SGLQNAASIAGMIVADLPEKK</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              stripped peptide\n",
       "0  LFADENHLSPAVTAIQIEDIDAEQFRK\n",
       "1  LAIDDSSINAHGTSTTANDKNETSAIK\n",
       "2        SGLQNAASIAGMIVADLPEKK\n",
       "3     IDIEDGGIVTIASHDGAAAEEAQK\n",
       "4        SGLQNAASIAGMIVADLPEKK"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# keep only peptides  >2.5 XCorr\n",
    "# need to convert Xcorr column from strings to numeric so we can use loc\n",
    "comet['xcorr'] = pd.to_numeric(comet['xcorr'])\n",
    "\n",
    "comet25 = comet.loc[comet['xcorr'] >= 2.5]\n",
    "\n",
    "# Let's separate out the decoy hits from the good ones\n",
    "\n",
    "cometpmm25 = comet25[~comet25['protein'].str.contains(\"DECOY\")]\n",
    "cometdec25 = comet25[comet25['protein'].str.contains(\"DECOY\")]\n",
    "\n",
    "\n",
    "# keep only peptide column \n",
    "pep25 = cometpmm25[[\"stripped peptide\"]]\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "pep25.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_combine_PTMopt_Comet_stripped_peptides_2.5XCorr.txt\", header=False, index=False)\n",
    "\n",
    "# removing redundancy\n",
    "pep25dedup = pd.DataFrame.drop_duplicates(pep25)\n",
    "\n",
    "print(\"# redundant Comet peptides >2.5 XCorr\", len(pep25))\n",
    "print(\"# nonredundant Comet peptides >2.5 XCOrr\", len(pep25dedup))\n",
    "\n",
    "pep25.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# redundant Comet peptides >3 XCorr 26923\n",
      "# nonredundant Comet peptides >3 XCOrr 9213\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LFADENHLSPAVTAIQIEDIDAEQFRK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LAIDDSSINAHGTSTTANDKNETSAIK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SGLQNAASIAGMIVADLPEKK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IDIEDGGIVTIASHDGAAAEEAQK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SGLQNAASIAGMIVADLPEKK</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              stripped peptide\n",
       "0  LFADENHLSPAVTAIQIEDIDAEQFRK\n",
       "1  LAIDDSSINAHGTSTTANDKNETSAIK\n",
       "2        SGLQNAASIAGMIVADLPEKK\n",
       "3     IDIEDGGIVTIASHDGAAAEEAQK\n",
       "4        SGLQNAASIAGMIVADLPEKK"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# keep only peptides  >3 XCorr\n",
    "# need to convert Xcorr column from strings to numeric so we can use loc\n",
    "comet['xcorr'] = pd.to_numeric(comet['xcorr'])\n",
    "\n",
    "comet3 = comet.loc[comet['xcorr'] >= 3]\n",
    "\n",
    "# Let's separate out the decoy hits from the good ones\n",
    "\n",
    "cometpmm3 = comet3[~comet3['protein'].str.contains(\"DECOY\")]\n",
    "cometdec3 = comet3[comet3['protein'].str.contains(\"DECOY\")]\n",
    "\n",
    "# export the whole table for Comet XCorr > 3\n",
    "cometpmm3.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_combine_PTMopt_Comet_3XCorr_noDECOY.csv\")\n",
    "\n",
    "# keep only peptide column \n",
    "pep3 = cometpmm3[[\"stripped peptide\"]]\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "pep3.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_combine_PTMopt_Comet_stripped_peptides_3XCorr.txt\", header=False, index=False)\n",
    "\n",
    "# removing redundancy\n",
    "pep3dedup = pd.DataFrame.drop_duplicates(pep3)\n",
    "\n",
    "print(\"# redundant Comet peptides >3 XCorr\", len(pep3))\n",
    "print(\"# nonredundant Comet peptides >3 XCOrr\", len(pep3dedup))\n",
    "\n",
    "pep3.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Exporting files with PTM totals:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>K</th>\n",
       "      <th>L</th>\n",
       "      <th>...</th>\n",
       "      <th>k-iron</th>\n",
       "      <th>k-meth</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>% C w/ carb.</th>\n",
       "      <th>% M w/ oxid</th>\n",
       "      <th>% N w/ deam</th>\n",
       "      <th>% Q w/ deam</th>\n",
       "      <th>% K w/ iron</th>\n",
       "      <th>% K w/ meth</th>\n",
       "      <th>% R w/ meth</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>44906</td>\n",
       "      <td>5271</td>\n",
       "      <td>39296</td>\n",
       "      <td>49238</td>\n",
       "      <td>22760</td>\n",
       "      <td>45755</td>\n",
       "      <td>8080</td>\n",
       "      <td>52085</td>\n",
       "      <td>58223</td>\n",
       "      <td>63935</td>\n",
       "      <td>...</td>\n",
       "      <td>1784</td>\n",
       "      <td>1106</td>\n",
       "      <td>470</td>\n",
       "      <td>0.926769</td>\n",
       "      <td>0.308879</td>\n",
       "      <td>0.106737</td>\n",
       "      <td>0.10592</td>\n",
       "      <td>0.030641</td>\n",
       "      <td>0.018996</td>\n",
       "      <td>0.016276</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  A     C      D      E      F      G     H      I      K  \\\n",
       "sample total  44906  5271  39296  49238  22760  45755  8080  52085  58223   \n",
       "\n",
       "                  L  ...  k-iron  k-meth  r-meth  % C w/ carb.  % M w/ oxid  \\\n",
       "sample total  63935  ...    1784    1106     470      0.926769     0.308879   \n",
       "\n",
       "              % N w/ deam  % Q w/ deam  % K w/ iron  % K w/ meth  % R w/ meth  \n",
       "sample total     0.106737      0.10592     0.030641     0.018996     0.016276  \n",
       "\n",
       "[1 rows x 34 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of certain columns in the stripped peptide dataframe above\n",
    "# choosing the XCorr > 2.5 filtered results\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'A': comet25['A'].sum(),\n",
    "        'C': comet25['C'].sum(),\n",
    "        'D': comet25['D'].sum(),\n",
    "        'E': comet25['E'].sum(),\n",
    "        'F': comet25['F'].sum(),\n",
    "        'G': comet25['G'].sum(),\n",
    "        'H': comet25['H'].sum(),\n",
    "        'I': comet25['I'].sum(),\n",
    "        'K': comet25['K'].sum(),\n",
    "        'L': comet25['L'].sum(),\n",
    "        'M': comet25['M'].sum(),\n",
    "        'N': comet25['N'].sum(),\n",
    "        'P': comet25['P'].sum(),\n",
    "        'Q': comet25['Q'].sum(),\n",
    "        'R': comet25['R'].sum(),\n",
    "        'S': comet25['S'].sum(),\n",
    "        'T': comet25['T'].sum(),\n",
    "        'V': comet25['V'].sum(),\n",
    "        'W': comet25['W'].sum(),\n",
    "        'Y': comet25['Y'].sum(),\n",
    "        'c-carb': comet25['c-carb'].sum(),\n",
    "        'm-oxid': comet25['m-oxid'].sum(),\n",
    "        'n-deam': comet25['n-deam'].sum(),\n",
    "        'q-deam': comet25['q-deam'].sum(),\n",
    "        'k-iron': comet25['k-iron'].sum(),\n",
    "        'k-meth': comet25['k-meth'].sum(),\n",
    "        'r-meth': comet25['r-meth'].sum()\n",
    "       }\n",
    "\n",
    "totalcomet25 = pd.DataFrame(data, columns=['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', 'm-oxid', 'n-deam', 'q-deam', 'k-iron', 'k-meth', 'r-meth' ], index=index)\n",
    "\n",
    "# calculate percentage of C's with carb (should be 1.0)\n",
    "totalcomet25['% C w/ carb.'] = totalcomet25['c-carb'] / totalcomet25['C'] \n",
    "\n",
    "# calculate percentage of M's that are oxidized\n",
    "totalcomet25['% M w/ oxid'] = totalcomet25['m-oxid'] / totalcomet25['M'] \n",
    "\n",
    "# calculate percentage of N's that are deamidated\n",
    "totalcomet25['% N w/ deam'] = totalcomet25['n-deam'] / totalcomet25['N'] \n",
    "\n",
    "# calculate percentage of Q's that are deamidated\n",
    "totalcomet25['% Q w/ deam'] = totalcomet25['q-deam'] / totalcomet25['Q'] \n",
    "\n",
    "# calculate percentage of K's that are hydroxylated\n",
    "totalcomet25['% K w/ iron'] = totalcomet25['k-iron'] / totalcomet25['K'] \n",
    "\n",
    "# calculate percentage of K's that are methylated\n",
    "totalcomet25['% K w/ meth'] = totalcomet25['k-meth'] / totalcomet25['K'] \n",
    "\n",
    "# calculate percentage of R's that are methylated\n",
    "totalcomet25['% R w/ meth'] = totalcomet25['r-meth'] / totalcomet25['R'] \n",
    "\n",
    "# write modified dataframe to new txt file, same name + 'stripped'\n",
    "totalcomet25.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL95_MED2_trypsin_combine_Comet25_totals.csv\")\n",
    "\n",
    "totalcomet25.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>K</th>\n",
       "      <th>L</th>\n",
       "      <th>...</th>\n",
       "      <th>k-iron</th>\n",
       "      <th>k-meth</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>% C w/ carb.</th>\n",
       "      <th>% M w/ oxid</th>\n",
       "      <th>% N w/ deam</th>\n",
       "      <th>% Q w/ deam</th>\n",
       "      <th>% K w/ iron</th>\n",
       "      <th>% K w/ meth</th>\n",
       "      <th>% R w/ meth</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>34766</td>\n",
       "      <td>3876</td>\n",
       "      <td>29841</td>\n",
       "      <td>37360</td>\n",
       "      <td>16417</td>\n",
       "      <td>34864</td>\n",
       "      <td>6466</td>\n",
       "      <td>38307</td>\n",
       "      <td>42186</td>\n",
       "      <td>47856</td>\n",
       "      <td>...</td>\n",
       "      <td>472</td>\n",
       "      <td>247</td>\n",
       "      <td>108</td>\n",
       "      <td>0.928535</td>\n",
       "      <td>0.293919</td>\n",
       "      <td>0.072883</td>\n",
       "      <td>0.07465</td>\n",
       "      <td>0.011189</td>\n",
       "      <td>0.005855</td>\n",
       "      <td>0.005027</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  A     C      D      E      F      G     H      I      K  \\\n",
       "sample total  34766  3876  29841  37360  16417  34864  6466  38307  42186   \n",
       "\n",
       "                  L  ...  k-iron  k-meth  r-meth  % C w/ carb.  % M w/ oxid  \\\n",
       "sample total  47856  ...     472     247     108      0.928535     0.293919   \n",
       "\n",
       "              % N w/ deam  % Q w/ deam  % K w/ iron  % K w/ meth  % R w/ meth  \n",
       "sample total     0.072883      0.07465     0.011189     0.005855     0.005027  \n",
       "\n",
       "[1 rows x 34 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of certain columns in the stripped peptide dataframe above\n",
    "# choosing the XCorr > 3 filtered results\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'A': comet3['A'].sum(),\n",
    "        'C': comet3['C'].sum(),\n",
    "        'D': comet3['D'].sum(),\n",
    "        'E': comet3['E'].sum(),\n",
    "        'F': comet3['F'].sum(),\n",
    "        'G': comet3['G'].sum(),\n",
    "        'H': comet3['H'].sum(),\n",
    "        'I': comet3['I'].sum(),\n",
    "        'K': comet3['K'].sum(),\n",
    "        'L': comet3['L'].sum(),\n",
    "        'M': comet3['M'].sum(),\n",
    "        'N': comet3['N'].sum(),\n",
    "        'P': comet3['P'].sum(),\n",
    "        'Q': comet3['Q'].sum(),\n",
    "        'R': comet3['R'].sum(),\n",
    "        'S': comet3['S'].sum(),\n",
    "        'T': comet3['T'].sum(),\n",
    "        'V': comet3['V'].sum(),\n",
    "        'W': comet3['W'].sum(),\n",
    "        'Y': comet3['Y'].sum(),\n",
    "        'c-carb': comet3['c-carb'].sum(),\n",
    "        'm-oxid': comet3['m-oxid'].sum(),\n",
    "        'n-deam': comet3['n-deam'].sum(),\n",
    "        'q-deam': comet3['q-deam'].sum(),\n",
    "        'k-iron': comet3['k-iron'].sum(),\n",
    "        'k-meth': comet3['k-meth'].sum(),\n",
    "        'r-meth': comet3['r-meth'].sum()\n",
    "       }\n",
    "\n",
    "totalcomet3 = pd.DataFrame(data, columns=['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', 'm-oxid', 'n-deam', 'q-deam', 'k-iron', 'k-meth', 'r-meth' ], index=index)\n",
    "\n",
    "# calculate percentage of C's with carb (should be 1.0)\n",
    "totalcomet3['% C w/ carb.'] = totalcomet3['c-carb'] / totalcomet3['C'] \n",
    "\n",
    "# calculate percentage of M's that are oxidized\n",
    "totalcomet3['% M w/ oxid'] = totalcomet3['m-oxid'] / totalcomet3['M'] \n",
    "\n",
    "# calculate percentage of N's that are deamidated\n",
    "totalcomet3['% N w/ deam'] = totalcomet3['n-deam'] / totalcomet3['N'] \n",
    "\n",
    "# calculate percentage of Q's that are deamidated\n",
    "totalcomet3['% Q w/ deam'] = totalcomet3['q-deam'] / totalcomet3['Q'] \n",
    "\n",
    "# calculate percentage of K's that are hydroxylated\n",
    "totalcomet3['% K w/ iron'] = totalcomet3['k-iron'] / totalcomet3['K'] \n",
    "\n",
    "# calculate percentage of K's that are methylated\n",
    "totalcomet3['% K w/ meth'] = totalcomet3['k-meth'] / totalcomet3['K'] \n",
    "\n",
    "# calculate percentage of R's that are methylated\n",
    "totalcomet3['% R w/ meth'] = totalcomet3['r-meth'] / totalcomet3['R'] \n",
    "\n",
    "# write modified dataframe to new txt file, same name + 'stripped'\n",
    "totalcomet3.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL95_MED2_trypsin_combine_Comet3_totals.csv\")\n",
    "\n",
    "totalcomet3.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### All XCorr - Visualizing the results:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# making evenly spaced bins for the Xcorr data based on the min and max, called above\n",
    "bins = [0, 0.5, 1, 1.5, 2, 2.5, 3, 3.5, 4, 4.5, 5, 5.5, 6, 6.5, 7, 7.5, 8, 8.5, 9]\n",
    "labels = ['0-0.5', '0.5-1', '1-1.5', '1.5-2', '2-2.5', '2.5-3', '3-3.5', '3.5-4', '4-4.5', '4.5-5', '5-5.5', '5.5-6', '6-6.5', '6.5-7', '7-7.5', '7.5-8', '8-8.5', '8.5-9']\n",
    "\n",
    "# use pandas cut function to do the binning itself\n",
    "comet['binned'] = pd.cut(comet['xcorr'], bins=bins, labels=labels)\n",
    "\n",
    "# bar plots of binned PTM data\n",
    "\n",
    "index = ['0-0.5', '0.5-1', '1-1.5', '1.5-2', '2-2.5', '2.5-3', '3-3.5', '3.5-4', '4-4.5', '4.5-5', '5-5.5', '5.5-6', '6-6.5', '6.5-7', '7-7.5', '7.5-8', '8-8.5', '8.5-9']\n",
    "data = {'Total PTMs': [comet.groupby('binned')['ptm-total'].sum()['0-0.5'], comet.groupby('binned')['ptm-total'].sum()['0.5-1'], comet.groupby('binned')['ptm-total'].sum()['1-1.5'], comet.groupby('binned')['ptm-total'].sum()['1.5-2'], comet.groupby('binned')['ptm-total'].sum()['2-2.5'], comet.groupby('binned')['ptm-total'].sum()['2.5-3'], comet.groupby('binned')['ptm-total'].sum()['3-3.5'], comet.groupby('binned')['ptm-total'].sum()['3.5-4'], comet.groupby('binned')['ptm-total'].sum()['4-4.5'], comet.groupby('binned')['ptm-total'].sum()['4.5-5'], comet.groupby('binned')['ptm-total'].sum()['5-5.5'], comet.groupby('binned')['ptm-total'].sum()['5.5-6'], comet.groupby('binned')['ptm-total'].sum()['6-6.5'], comet.groupby('binned')['ptm-total'].sum()['6.5-7'], comet.groupby('binned')['ptm-total'].sum()['7-7.5'], comet.groupby('binned')['ptm-total'].sum()['7.5-8'], comet.groupby('binned')['ptm-total'].sum()['8-8.5'], comet.groupby('binned')['ptm-total'].sum()['8.5-9']],\n",
    "        'Cys carb.': [comet.groupby('binned')['c-carb'].sum()['0-0.5'], comet.groupby('binned')['c-carb'].sum()['0.5-1'], comet.groupby('binned')['c-carb'].sum()['1-1.5'], comet.groupby('binned')['c-carb'].sum()['1.5-2'], comet.groupby('binned')['c-carb'].sum()['2-2.5'], comet.groupby('binned')['c-carb'].sum()['2.5-3'], comet.groupby('binned')['c-carb'].sum()['3-3.5'], comet.groupby('binned')['c-carb'].sum()['3.5-4'], comet.groupby('binned')['c-carb'].sum()['4-4.5'], comet.groupby('binned')['c-carb'].sum()['4.5-5'], comet.groupby('binned')['c-carb'].sum()['5-5.5'], comet.groupby('binned')['c-carb'].sum()['5.5-6'], comet.groupby('binned')['c-carb'].sum()['6-6.5'], comet.groupby('binned')['c-carb'].sum()['6.5-7'], comet.groupby('binned')['c-carb'].sum()['7-7.5'], comet.groupby('binned')['c-carb'].sum()['7.5-8'], comet.groupby('binned')['c-carb'].sum()['8-8.5'], comet.groupby('binned')['c-carb'].sum()['8.5-9']],\n",
    "        'Met oxi.': [comet.groupby('binned')['m-oxid'].sum()['0-0.5'], comet.groupby('binned')['m-oxid'].sum()['0.5-1'], comet.groupby('binned')['m-oxid'].sum()['1-1.5'], comet.groupby('binned')['m-oxid'].sum()['1.5-2'], comet.groupby('binned')['m-oxid'].sum()['2-2.5'], comet.groupby('binned')['m-oxid'].sum()['2.5-3'], comet.groupby('binned')['m-oxid'].sum()['3-3.5'], comet.groupby('binned')['m-oxid'].sum()['3.5-4'], comet.groupby('binned')['m-oxid'].sum()['4-4.5'], comet.groupby('binned')['m-oxid'].sum()['4.5-5'], comet.groupby('binned')['m-oxid'].sum()['5-5.5'], comet.groupby('binned')['m-oxid'].sum()['5.5-6'], comet.groupby('binned')['m-oxid'].sum()['6-6.5'], comet.groupby('binned')['m-oxid'].sum()['6.5-7'], comet.groupby('binned')['m-oxid'].sum()['7-7.5'], comet.groupby('binned')['m-oxid'].sum()['7.5-8'], comet.groupby('binned')['m-oxid'].sum()['8-8.5'], comet.groupby('binned')['m-oxid'].sum()['8.5-9']],\n",
    "        'Asp deam.': [comet.groupby('binned')['n-deam'].sum()['0-0.5'], comet.groupby('binned')['n-deam'].sum()['0.5-1'], comet.groupby('binned')['n-deam'].sum()['1-1.5'], comet.groupby('binned')['n-deam'].sum()['1.5-2'], comet.groupby('binned')['n-deam'].sum()['2-2.5'], comet.groupby('binned')['n-deam'].sum()['2.5-3'], comet.groupby('binned')['n-deam'].sum()['3-3.5'], comet.groupby('binned')['n-deam'].sum()['3.5-4'], comet.groupby('binned')['n-deam'].sum()['4-4.5'], comet.groupby('binned')['n-deam'].sum()['4.5-5'], comet.groupby('binned')['n-deam'].sum()['5-5.5'], comet.groupby('binned')['n-deam'].sum()['5.5-6'], comet.groupby('binned')['n-deam'].sum()['6-6.5'], comet.groupby('binned')['n-deam'].sum()['6.5-7'], comet.groupby('binned')['n-deam'].sum()['7-7.5'], comet.groupby('binned')['n-deam'].sum()['7.5-8'], comet.groupby('binned')['n-deam'].sum()['8-8.5'], comet.groupby('binned')['n-deam'].sum()['8.5-9']],\n",
    "        'Glut deam.': [comet.groupby('binned')['q-deam'].sum()['0-0.5'], comet.groupby('binned')['q-deam'].sum()['0.5-1'], comet.groupby('binned')['q-deam'].sum()['1-1.5'], comet.groupby('binned')['q-deam'].sum()['1.5-2'], comet.groupby('binned')['q-deam'].sum()['2-2.5'], comet.groupby('binned')['q-deam'].sum()['2.5-3'], comet.groupby('binned')['q-deam'].sum()['3-3.5'], comet.groupby('binned')['q-deam'].sum()['3.5-4'], comet.groupby('binned')['q-deam'].sum()['4-4.5'], comet.groupby('binned')['q-deam'].sum()['4.5-5'], comet.groupby('binned')['q-deam'].sum()['5-5.5'], comet.groupby('binned')['q-deam'].sum()['5.5-6'], comet.groupby('binned')['q-deam'].sum()['6-6.5'], comet.groupby('binned')['q-deam'].sum()['6.5-7'], comet.groupby('binned')['q-deam'].sum()['7-7.5'], comet.groupby('binned')['q-deam'].sum()['7.5-8'], comet.groupby('binned')['q-deam'].sum()['8-8.5'], comet.groupby('binned')['q-deam'].sum()['8.5-9']],\n",
    "        'Lys iron': [comet.groupby('binned')['k-iron'].sum()['0-0.5'], comet.groupby('binned')['k-iron'].sum()['0.5-1'], comet.groupby('binned')['k-iron'].sum()['1-1.5'], comet.groupby('binned')['k-iron'].sum()['1.5-2'], comet.groupby('binned')['k-iron'].sum()['2-2.5'], comet.groupby('binned')['k-iron'].sum()['2.5-3'], comet.groupby('binned')['k-iron'].sum()['3-3.5'], comet.groupby('binned')['k-iron'].sum()['3.5-4'], comet.groupby('binned')['k-iron'].sum()['4-4.5'], comet.groupby('binned')['k-iron'].sum()['4.5-5'], comet.groupby('binned')['k-iron'].sum()['5-5.5'], comet.groupby('binned')['k-iron'].sum()['5.5-6'], comet.groupby('binned')['k-iron'].sum()['6-6.5'], comet.groupby('binned')['k-iron'].sum()['6.5-7'], comet.groupby('binned')['k-iron'].sum()['7-7.5'], comet.groupby('binned')['k-iron'].sum()['7.5-8'], comet.groupby('binned')['k-iron'].sum()['8-8.5'], comet.groupby('binned')['k-iron'].sum()['8.5-9']],\n",
    "        'Lys meth.': [comet.groupby('binned')['k-meth'].sum()['0-0.5'], comet.groupby('binned')['k-meth'].sum()['0.5-1'], comet.groupby('binned')['k-meth'].sum()['1-1.5'], comet.groupby('binned')['k-meth'].sum()['1.5-2'], comet.groupby('binned')['k-meth'].sum()['2-2.5'], comet.groupby('binned')['k-meth'].sum()['2.5-3'], comet.groupby('binned')['k-meth'].sum()['3-3.5'], comet.groupby('binned')['k-meth'].sum()['3.5-4'], comet.groupby('binned')['k-meth'].sum()['4-4.5'], comet.groupby('binned')['k-meth'].sum()['4.5-5'], comet.groupby('binned')['k-meth'].sum()['5-5.5'], comet.groupby('binned')['k-meth'].sum()['5.5-6'], comet.groupby('binned')['k-meth'].sum()['6-6.5'], comet.groupby('binned')['k-meth'].sum()['6.5-7'], comet.groupby('binned')['k-meth'].sum()['7-7.5'], comet.groupby('binned')['k-meth'].sum()['7.5-8'], comet.groupby('binned')['k-meth'].sum()['8-8.5'], comet.groupby('binned')['k-meth'].sum()['8.5-9']],\n",
    "        'Arg meth.': [comet.groupby('binned')['r-meth'].sum()['0-0.5'], comet.groupby('binned')['r-meth'].sum()['0.5-1'], comet.groupby('binned')['r-meth'].sum()['1-1.5'], comet.groupby('binned')['r-meth'].sum()['1.5-2'], comet.groupby('binned')['r-meth'].sum()['2-2.5'], comet.groupby('binned')['r-meth'].sum()['2.5-3'], comet.groupby('binned')['r-meth'].sum()['3-3.5'], comet.groupby('binned')['r-meth'].sum()['3.5-4'], comet.groupby('binned')['r-meth'].sum()['4-4.5'], comet.groupby('binned')['r-meth'].sum()['4.5-5'], comet.groupby('binned')['r-meth'].sum()['5-5.5'], comet.groupby('binned')['r-meth'].sum()['5.5-6'], comet.groupby('binned')['r-meth'].sum()['6-6.5'], comet.groupby('binned')['r-meth'].sum()['6.5-7'], comet.groupby('binned')['r-meth'].sum()['7-7.5'], comet.groupby('binned')['r-meth'].sum()['7.5-8'], comet.groupby('binned')['r-meth'].sum()['8-8.5'], comet.groupby('binned')['r-meth'].sum()['8.5-9']]\n",
    "        }\n",
    "\n",
    "cometbin = pd.DataFrame(data, columns=['Total PTMs','Cys carb.','Met oxi.','Asp deam.', 'Glut deam.', 'Lys iron', 'Lys meth.', 'Arg meth.'], index=index)\n",
    "\n",
    "ax1 = cometbin.plot.bar(y='Total PTMs', rot=45)\n",
    "ax1.set_title('Total PTMs')\n",
    "\n",
    "ax2 = cometbin.plot.bar(y='Cys carb.', rot=45)\n",
    "ax2.set_title('Cysteine carbamidomethylation')\n",
    "\n",
    "ax3 = cometbin.plot.bar(y='Met oxi.', rot=45)\n",
    "ax3.set_title('Methionine oxidation')\n",
    "\n",
    "ax4 = cometbin.plot.bar(y='Asp deam.', rot=45)\n",
    "ax4.set_title('Asparagine deamidation')\n",
    "\n",
    "ax5 = cometbin.plot.bar(y='Glut deam.', rot=45)\n",
    "ax5.set_title('Glutamine deamidation')\n",
    "\n",
    "ax6 = cometbin.plot.bar(y='Lys iron', rot=45)\n",
    "ax6.set_title('Lysine iron adduct')\n",
    "\n",
    "ax7 = cometbin.plot.bar(y='Lys meth.', rot=45)\n",
    "ax7.set_title('Lysine methylation')\n",
    "\n",
    "ax7 = cometbin.plot.bar(y='Arg meth.', rot=45)\n",
    "ax7.set_title('Arginine methylation')\n",
    "\n",
    "#plt.savefig('/home/millieginty/Documents/git-repos/2017-etnp/analyses/pronovo-2020/pronovo-2020-ptm/MED4_trypsin1_Comet_PTMopt.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# histogram of stripped peptide lengths\n",
    "\n",
    "#plt.rcdefaults()\n",
    "#fig, ax = plt.subplots()\n",
    "\n",
    "comet.plot(y='stripped length', kind = 'hist', bins = 20, title = 'Peptide length')\n",
    "plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# histogram of total peptide amounts\n",
    "\n",
    "#plt.rcdefaults()\n",
    "#fig, ax = plt.subplots()\n",
    "\n",
    "comet.plot(y='ptm-total', kind = 'hist', bins = 20, title = 'PTMs/sequence')\n",
    "plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from matplotlib import pyplot\n",
    "# density plot of xcorr vs length (idea from https://python-graph-gallery.com/85-density-plot-with-matplotlib/)\n",
    "\n",
    "# read in data\n",
    "x = comet['xcorr']\n",
    "y = comet['stripped length']\n",
    " \n",
    "# evaluate a gaussian kernel density estimation (KDE) on a regular grid of nbins x nbins over data extents\n",
    "nbins=300\n",
    "k = kde.gaussian_kde([x,y])\n",
    "xi, yi = np.mgrid[x.min():x.max():nbins*1j, y.min():y.max():nbins*1j]\n",
    "zi = k(np.vstack([xi.flatten(), yi.flatten()]))\n",
    " \n",
    "# make the plot\n",
    "plt.pcolormesh(xi, yi, zi.reshape(xi.shape))\n",
    "plt.colorbar()\n",
    "plt.show() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# plot with density of xcorr vs length and xcorr vs total ptms\n",
    "plt.figure()\n",
    "\n",
    "# read in data\n",
    "x = comet['xcorr']\n",
    "y = comet['stripped length']\n",
    "\n",
    "a = comet['xcorr']\n",
    "b = comet['ptm-total']\n",
    "\n",
    "# evaluate a gaussian kernel density estimation (KDE) on a regular grid of nbins x nbins over data extents\n",
    "nbins=300\n",
    "k = kde.gaussian_kde([x,y])\n",
    "xi, yi = np.mgrid[x.min():x.max():nbins*1j, y.min():y.max():nbins*1j]\n",
    "zi = k(np.vstack([xi.flatten(), yi.flatten()]))\n",
    "\n",
    "c = kde.gaussian_kde([a,b])\n",
    "ai, bi = np.mgrid[a.min():a.max():nbins*1j, b.min():b.max():nbins*1j]\n",
    "di = k(np.vstack([ai.flatten(), bi.flatten()]))\n",
    "\n",
    "# density plot of length vs xcorr\n",
    "plt.subplot(221)\n",
    "plt.pcolormesh(xi, yi, zi.reshape(xi.shape))\n",
    "plt.title('logit')\n",
    "plt.colorbar()\n",
    "\n",
    "# density plot of xcorr vs ptms\n",
    "plt.subplot(222)\n",
    "plt.pcolormesh(ai, bi, di.reshape(ai.shape))\n",
    "plt.title('Combined MED4 Comet XCorr vs total PTMs')\n",
    "plt.colorbar()\n",
    "\n",
    "\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### XCorr > 2.5 - Visualizing the results:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# making evenly spaced bins for the Xcorr data based on the min and max, called above\n",
    "bins = [2.5, 3, 3.5, 4, 4.5, 5, 5.5, 6, 6.5, 7, 7.5, 8, 8.5, 9]\n",
    "labels = ['2.5-3', '3-3.5', '3.5-4', '4-4.5', '4.5-5', '5-5.5', '5.5-6', '6-6.5', '6.5-7', '7-7.5', '7.5-8', '8-8.5', '8.5-9']\n",
    "\n",
    "# use pandas cut function to do the binning itself\n",
    "comet25['binned'] = pd.cut(comet25['xcorr'], bins=bins, labels=labels)\n",
    "\n",
    "# bar plots of binned PTM data\n",
    "\n",
    "index = ['2.5-3', '3-3.5', '3.5-4', '4-4.5', '4.5-5', '5-5.5', '5.5-6', '6-6.5', '6.5-7', '7-7.5', '7.5-8', '8-8.5', '8.5-9']\n",
    "data = {'Total PTMs': [comet25.groupby('binned')['ptm-total'].sum()['2.5-3'], comet25.groupby('binned')['ptm-total'].sum()['3-3.5'], comet25.groupby('binned')['ptm-total'].sum()['3.5-4'], comet25.groupby('binned')['ptm-total'].sum()['4-4.5'], comet25.groupby('binned')['ptm-total'].sum()['4.5-5'], comet25.groupby('binned')['ptm-total'].sum()['5-5.5'], comet25.groupby('binned')['ptm-total'].sum()['5.5-6'], comet25.groupby('binned')['ptm-total'].sum()['6-6.5'], comet25.groupby('binned')['ptm-total'].sum()['6.5-7'], comet25.groupby('binned')['ptm-total'].sum()['7-7.5'], comet25.groupby('binned')['ptm-total'].sum()['7.5-8'], comet25.groupby('binned')['ptm-total'].sum()['8-8.5'], comet25.groupby('binned')['ptm-total'].sum()['8.5-9']],\n",
    "        'Cys carb.': [comet25.groupby('binned')['c-carb'].sum()['2.5-3'], comet25.groupby('binned')['c-carb'].sum()['3-3.5'], comet25.groupby('binned')['c-carb'].sum()['3.5-4'], comet25.groupby('binned')['c-carb'].sum()['4-4.5'], comet25.groupby('binned')['c-carb'].sum()['4.5-5'], comet25.groupby('binned')['c-carb'].sum()['5-5.5'], comet25.groupby('binned')['c-carb'].sum()['5.5-6'], comet25.groupby('binned')['c-carb'].sum()['6-6.5'], comet25.groupby('binned')['c-carb'].sum()['6.5-7'], comet25.groupby('binned')['c-carb'].sum()['7-7.5'], comet25.groupby('binned')['c-carb'].sum()['7.5-8'], comet25.groupby('binned')['c-carb'].sum()['8-8.5'], comet25.groupby('binned')['c-carb'].sum()['8.5-9']],\n",
    "        'Met oxi.': [comet25.groupby('binned')['m-oxid'].sum()['2.5-3'], comet25.groupby('binned')['m-oxid'].sum()['3-3.5'], comet25.groupby('binned')['m-oxid'].sum()['3.5-4'], comet25.groupby('binned')['m-oxid'].sum()['4-4.5'], comet25.groupby('binned')['m-oxid'].sum()['4.5-5'], comet25.groupby('binned')['m-oxid'].sum()['5-5.5'], comet25.groupby('binned')['m-oxid'].sum()['5.5-6'], comet25.groupby('binned')['m-oxid'].sum()['6-6.5'], comet25.groupby('binned')['m-oxid'].sum()['6.5-7'], comet25.groupby('binned')['m-oxid'].sum()['7-7.5'], comet25.groupby('binned')['m-oxid'].sum()['7.5-8'], comet25.groupby('binned')['m-oxid'].sum()['8-8.5'], comet25.groupby('binned')['m-oxid'].sum()['8.5-9']],\n",
    "        'Asp deam.': [comet25.groupby('binned')['n-deam'].sum()['2.5-3'], comet25.groupby('binned')['n-deam'].sum()['3-3.5'], comet25.groupby('binned')['n-deam'].sum()['3.5-4'], comet25.groupby('binned')['n-deam'].sum()['4-4.5'], comet25.groupby('binned')['n-deam'].sum()['4.5-5'], comet25.groupby('binned')['n-deam'].sum()['5-5.5'], comet25.groupby('binned')['n-deam'].sum()['5.5-6'], comet25.groupby('binned')['n-deam'].sum()['6-6.5'], comet25.groupby('binned')['n-deam'].sum()['6.5-7'], comet25.groupby('binned')['n-deam'].sum()['7-7.5'], comet25.groupby('binned')['n-deam'].sum()['7.5-8'], comet25.groupby('binned')['n-deam'].sum()['8-8.5'], comet25.groupby('binned')['n-deam'].sum()['8.5-9']],\n",
    "        'Glut deam.': [comet25.groupby('binned')['q-deam'].sum()['2.5-3'], comet25.groupby('binned')['q-deam'].sum()['3-3.5'], comet25.groupby('binned')['q-deam'].sum()['3.5-4'], comet25.groupby('binned')['q-deam'].sum()['4-4.5'], comet25.groupby('binned')['q-deam'].sum()['4.5-5'], comet25.groupby('binned')['q-deam'].sum()['5-5.5'], comet25.groupby('binned')['q-deam'].sum()['5.5-6'], comet25.groupby('binned')['q-deam'].sum()['6-6.5'], comet25.groupby('binned')['q-deam'].sum()['6.5-7'], comet25.groupby('binned')['q-deam'].sum()['7-7.5'], comet25.groupby('binned')['q-deam'].sum()['7.5-8'], comet25.groupby('binned')['q-deam'].sum()['8-8.5'], comet25.groupby('binned')['q-deam'].sum()['8.5-9']],\n",
    "        'Lys iron': [comet25.groupby('binned')['k-iron'].sum()['2.5-3'], comet25.groupby('binned')['k-iron'].sum()['3-3.5'], comet25.groupby('binned')['k-iron'].sum()['3.5-4'], comet25.groupby('binned')['k-iron'].sum()['4-4.5'], comet25.groupby('binned')['k-iron'].sum()['4.5-5'], comet25.groupby('binned')['k-iron'].sum()['5-5.5'], comet25.groupby('binned')['k-iron'].sum()['5.5-6'], comet25.groupby('binned')['k-iron'].sum()['6-6.5'], comet25.groupby('binned')['k-iron'].sum()['6.5-7'], comet25.groupby('binned')['k-iron'].sum()['7-7.5'], comet25.groupby('binned')['k-iron'].sum()['7.5-8'], comet25.groupby('binned')['k-iron'].sum()['8-8.5'], comet25.groupby('binned')['k-iron'].sum()['8.5-9']],\n",
    "        'Lys meth.': [comet25.groupby('binned')['k-meth'].sum()['2.5-3'], comet25.groupby('binned')['k-meth'].sum()['3-3.5'], comet25.groupby('binned')['k-meth'].sum()['3.5-4'], comet25.groupby('binned')['k-meth'].sum()['4-4.5'], comet25.groupby('binned')['k-meth'].sum()['4.5-5'], comet25.groupby('binned')['k-meth'].sum()['5-5.5'], comet25.groupby('binned')['k-meth'].sum()['5.5-6'], comet25.groupby('binned')['k-meth'].sum()['6-6.5'], comet25.groupby('binned')['k-meth'].sum()['6.5-7'], comet25.groupby('binned')['k-meth'].sum()['7-7.5'], comet25.groupby('binned')['k-meth'].sum()['7.5-8'], comet25.groupby('binned')['k-meth'].sum()['8-8.5'], comet25.groupby('binned')['k-meth'].sum()['8.5-9']],\n",
    "        'Arg meth.': [comet25.groupby('binned')['r-meth'].sum()['2.5-3'], comet25.groupby('binned')['r-meth'].sum()['3-3.5'], comet25.groupby('binned')['r-meth'].sum()['3.5-4'], comet25.groupby('binned')['r-meth'].sum()['4-4.5'], comet25.groupby('binned')['r-meth'].sum()['4.5-5'], comet25.groupby('binned')['r-meth'].sum()['5-5.5'], comet25.groupby('binned')['r-meth'].sum()['5.5-6'], comet25.groupby('binned')['r-meth'].sum()['6-6.5'], comet25.groupby('binned')['r-meth'].sum()['6.5-7'], comet25.groupby('binned')['r-meth'].sum()['7-7.5'], comet25.groupby('binned')['r-meth'].sum()['7.5-8'], comet25.groupby('binned')['r-meth'].sum()['8-8.5'], comet25.groupby('binned')['r-meth'].sum()['8.5-9']]\n",
    "        }\n",
    "\n",
    "comet25bin = pd.DataFrame(data, columns=['Total PTMs','Cys carb.','Met oxi.','Asp deam.', 'Glut deam.', 'Lys iron', 'Lys meth.', 'Arg meth.'], index=index)\n",
    "\n",
    "ax1 = comet25bin.plot.bar(y='Total PTMs', rot=45)\n",
    "ax1.set_title('Total PTMs')\n",
    "\n",
    "ax2 = comet25bin.plot.bar(y='Cys carb.', rot=45)\n",
    "ax2.set_title('Cysteine carbamidomethylation')\n",
    "\n",
    "ax3 = comet25bin.plot.bar(y='Met oxi.', rot=45)\n",
    "ax3.set_title('Methionine oxidation')\n",
    "\n",
    "ax4 = comet25bin.plot.bar(y='Asp deam.', rot=45)\n",
    "ax4.set_title('Asparagine deamidation')\n",
    "\n",
    "ax5 = comet25bin.plot.bar(y='Glut deam.', rot=45)\n",
    "ax5.set_title('Glutamine deamidation')\n",
    "\n",
    "ax6 = comet25bin.plot.bar(y='Lys iron', rot=45)\n",
    "ax6.set_title('Lysine iron adduct')\n",
    "\n",
    "ax7 = comet25bin.plot.bar(y='Lys meth.', rot=45)\n",
    "ax7.set_title('Lysine methylation')\n",
    "\n",
    "ax7 = comet25bin.plot.bar(y='Arg meth.', rot=45)\n",
    "ax7.set_title('Arginine methylation')\n",
    "\n",
    "#plt.savefig('/home/millieginty/Documents/git-repos/2017-etnp/analyses/pronovo-2020/pronovo-2020-ptm/MED4_trypsin1_comet25_PTMopt.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# histogram of stripped peptide lengths\n",
    "\n",
    "#plt.rcdefaults()\n",
    "#fig, ax = plt.subplots()\n",
    "\n",
    "comet25.plot(y='stripped length', kind = 'hist', bins = 20, title = 'Peptide length')\n",
    "plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# histogram of total peptide amounts\n",
    "\n",
    "#plt.rcdefaults()\n",
    "#fig, ax = plt.subplots()\n",
    "\n",
    "comet25.plot(y='ptm-total', kind = 'hist', bins = 20, title = 'PTMs/sequence')\n",
    "plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from matplotlib import pyplot\n",
    "# density plot of xcorr vs length (idea from https://python-graph-gallery.com/85-density-plot-with-matplotlib/)\n",
    "\n",
    "# read in data\n",
    "x = comet25['xcorr']\n",
    "y = comet25['stripped length']\n",
    " \n",
    "# evaluate a gaussian kernel density estimation (KDE) on a regular grid of nbins x nbins over data extents\n",
    "nbins=300\n",
    "k = kde.gaussian_kde([x,y])\n",
    "xi, yi = np.mgrid[x.min():x.max():nbins*1j, y.min():y.max():nbins*1j]\n",
    "zi = k(np.vstack([xi.flatten(), yi.flatten()]))\n",
    " \n",
    "# make the plot\n",
    "plt.pcolormesh(xi, yi, zi.reshape(xi.shape))\n",
    "plt.colorbar()\n",
    "plt.show() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# now we have the stripped peptide csvs and txt files in the same data dir:\n",
    "!ls /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
