{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Manipulation of Comet database search results of ETNP 2017 P2 samples LC-MS/MS data using python.\n",
    "\n",
    "Starting with:\n",
    "\n",
    "    Comet search results (.csv) of PTM-optimized database searches\n",
    "    These were all searched with 15 ppm precursor tolerance and 0.5 ppm fragement ion tolerance\n",
    "    Search database included marine fungi and labrinthulomyces discovered using de novo peptide sequencing\n",
    "    XInteract file includes precursor intensities and protein descriptions\n",
    "\n",
    "Goal:\n",
    "\n",
    "    Files with stripped (no PTMs) peptide lists and\n",
    "    Columns with #'s of each modification in every sequence\n",
    "    Column with stripped peptide lengths (# amino acids)\n",
    "    \n",
    "### To use for a different file:\n",
    "\n",
    "#### 1. Change the input file name in *IN 4*\n",
    "#### 2. Use 'find + replace' (Esc + F) to replace the running # (e.g., 233) for another\n",
    "#### 3. Update the NAAF factor calculated in *IN 6* into *IN 7*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We don't have technical duplicates here, sadly, unlike the MED4 Pro samples. I exported Comet search results after running through XInteract and saving as `.xlsx` files (pandas really doesn't like to read the `xls` verions of the XInteract output becase there are so many characters in the `protein` column for these samples) into my ETNP 2017 git repo:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/TPP-PTMopt/ETNP-SKQ17-TPP-PTM-opt_15ppm_FUNGI\n"
     ]
    }
   ],
   "source": [
    "cd /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/TPP-PTMopt/ETNP-SKQ17-TPP-PTM-opt_15ppm_FUNGI/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.0.5'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# LIBRARIES\n",
    "#import pandas library for working with tabular data\n",
    "import os\n",
    "os.getcwd()\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "from scipy.stats import kde\n",
    "#import regular expresson (regex)\n",
    "import re\n",
    "#check pandas version\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "JA14_15ppm_quant.xlsx  JA4_15ppm_quant.xlsx        trap_265m_15ppm_quant.xlsx\r\n",
      "JA2_15ppm_quant.xlsx   trap_100m_15ppm_quant.xlsx  trap_965m_15ppm_quant.xlsx\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 231: 100 m McLane pump filtered on 0.3 um GF-75"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# redundant Comet peptides in combined dataframe 19263\n",
      "# nonredundant Comet peptides in combined dataframe 19263\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>spectrum</th>\n",
       "      <th>xcorr</th>\n",
       "      <th>deltacn</th>\n",
       "      <th>expect</th>\n",
       "      <th>ions</th>\n",
       "      <th>peptide</th>\n",
       "      <th>calc_neutral_pep_mass</th>\n",
       "      <th>precursor_intensity</th>\n",
       "      <th>protein</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20170410_JA2_01.34842.34842.3</td>\n",
       "      <td>5.552</td>\n",
       "      <td>0.520</td>\n",
       "      <td>4.130000e-09</td>\n",
       "      <td>31/88</td>\n",
       "      <td>R.AIQQQIENPLAQQILSGELVPGK.V</td>\n",
       "      <td>2473.3540</td>\n",
       "      <td>19969800.0</td>\n",
       "      <td>gi|54036848|sp|P63284.1|CLPB_ECOLI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20170410_JA2_01.23179.23179.2</td>\n",
       "      <td>4.989</td>\n",
       "      <td>0.657</td>\n",
       "      <td>2.350000e-14</td>\n",
       "      <td>29/42</td>\n",
       "      <td>K.IVVGGPYSSVSDAASSLDSSQK.S</td>\n",
       "      <td>2153.0488</td>\n",
       "      <td>4575480.0</td>\n",
       "      <td>ETNP_120m_PROKKA_24357,ETNP_100m_PROKKA_52245,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20170410_JA2_01.34897.34897.2</td>\n",
       "      <td>4.888</td>\n",
       "      <td>0.623</td>\n",
       "      <td>1.260000e-09</td>\n",
       "      <td>25/44</td>\n",
       "      <td>R.AIQQQIENPLAQQILSGELVPGK.V</td>\n",
       "      <td>2473.3540</td>\n",
       "      <td>632913.0</td>\n",
       "      <td>gi|54036848|sp|P63284.1|CLPB_ECOLI</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20170410_JA2_01.23135.23135.3</td>\n",
       "      <td>4.853</td>\n",
       "      <td>0.600</td>\n",
       "      <td>6.590000e-11</td>\n",
       "      <td>29/84</td>\n",
       "      <td>K.IVVGGPYSSVSDAASSLDSSQK.S</td>\n",
       "      <td>2153.0488</td>\n",
       "      <td>6257440.0</td>\n",
       "      <td>ETNP_120m_PROKKA_24357,ETNP_100m_PROKKA_52245,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20170410_JA2_01.28172.28172.2</td>\n",
       "      <td>4.826</td>\n",
       "      <td>0.657</td>\n",
       "      <td>2.910000e-14</td>\n",
       "      <td>29/42</td>\n",
       "      <td>K.IVVGGPYSSVSDAASVLDGSQK.S</td>\n",
       "      <td>2135.0746</td>\n",
       "      <td>3597780.0</td>\n",
       "      <td>ETNP_120m_PROKKA_126325,ETNP_100m_PROKKA_02317...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        spectrum  xcorr  deltacn        expect   ions  \\\n",
       "0  20170410_JA2_01.34842.34842.3  5.552    0.520  4.130000e-09  31/88   \n",
       "1  20170410_JA2_01.23179.23179.2  4.989    0.657  2.350000e-14  29/42   \n",
       "2  20170410_JA2_01.34897.34897.2  4.888    0.623  1.260000e-09  25/44   \n",
       "3  20170410_JA2_01.23135.23135.3  4.853    0.600  6.590000e-11  29/84   \n",
       "4  20170410_JA2_01.28172.28172.2  4.826    0.657  2.910000e-14  29/42   \n",
       "\n",
       "                       peptide  calc_neutral_pep_mass  precursor_intensity  \\\n",
       "0  R.AIQQQIENPLAQQILSGELVPGK.V              2473.3540           19969800.0   \n",
       "1   K.IVVGGPYSSVSDAASSLDSSQK.S              2153.0488            4575480.0   \n",
       "2  R.AIQQQIENPLAQQILSGELVPGK.V              2473.3540             632913.0   \n",
       "3   K.IVVGGPYSSVSDAASSLDSSQK.S              2153.0488            6257440.0   \n",
       "4   K.IVVGGPYSSVSDAASVLDGSQK.S              2135.0746            3597780.0   \n",
       "\n",
       "                                             protein  \n",
       "0                 gi|54036848|sp|P63284.1|CLPB_ECOLI  \n",
       "1  ETNP_120m_PROKKA_24357,ETNP_100m_PROKKA_52245,...  \n",
       "2                 gi|54036848|sp|P63284.1|CLPB_ECOLI  \n",
       "3  ETNP_120m_PROKKA_24357,ETNP_100m_PROKKA_52245,...  \n",
       "4  ETNP_120m_PROKKA_126325,ETNP_100m_PROKKA_02317...  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read the CSV into a dataframe using the pandas read_excel function\n",
    "#cometdup231 = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/TPP-PTMopt/ETNP-SKQ17-TPP-PTMopt-hyroxylation/JA2_PTMopt_interact_quant_nopro.pep.csv\", index_col='spectrum')\n",
    "\n",
    "cometdup231 = pd.read_excel(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/TPP-PTMopt/ETNP-SKQ17-TPP-PTM-opt_15ppm_FUNGI/JA2_15ppm_quant.xlsx\")\n",
    "\n",
    "# remove redundant rows\n",
    "comet231 = pd.DataFrame.drop_duplicates(cometdup231)\n",
    "\n",
    "print(\"# redundant Comet peptides in combined dataframe\", len(cometdup231))\n",
    "print(\"# nonredundant Comet peptides in combined dataframe\", len(comet231))\n",
    "\n",
    "#look at the dataframe\n",
    "comet231.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/millieginty/anaconda3/lib/python3.8/site-packages/pandas/core/ops/array_ops.py:253: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison\n",
      "  res_values = method(rvalues)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>spectrum</th>\n",
       "      <th>xcorr</th>\n",
       "      <th>deltacn</th>\n",
       "      <th>expect</th>\n",
       "      <th>ions</th>\n",
       "      <th>peptide</th>\n",
       "      <th>calc_neutral_pep_mass</th>\n",
       "      <th>precursor_intensity</th>\n",
       "      <th>protein</th>\n",
       "      <th>L terminus</th>\n",
       "      <th>...</th>\n",
       "      <th>c-carb</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>k-hydr</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>stripped length</th>\n",
       "      <th>NAAF num.</th>\n",
       "      <th>ptm-total</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20170410_JA2_01.34842.34842.3</td>\n",
       "      <td>5.552</td>\n",
       "      <td>0.520</td>\n",
       "      <td>4.130000e-09</td>\n",
       "      <td>31/88</td>\n",
       "      <td>R.AIQQQIENPLAQQILSGELVPGK.V</td>\n",
       "      <td>2473.3540</td>\n",
       "      <td>19969800.0</td>\n",
       "      <td>gi|54036848|sp|P63284.1|CLPB_ECOLI</td>\n",
       "      <td>R</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>R.AIQQQIENPLAQQILSGELVPGK.V</td>\n",
       "      <td>27</td>\n",
       "      <td>739622.222222</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20170410_JA2_01.23179.23179.2</td>\n",
       "      <td>4.989</td>\n",
       "      <td>0.657</td>\n",
       "      <td>2.350000e-14</td>\n",
       "      <td>29/42</td>\n",
       "      <td>K.IVVGGPYSSVSDAASSLDSSQK.S</td>\n",
       "      <td>2153.0488</td>\n",
       "      <td>4575480.0</td>\n",
       "      <td>ETNP_120m_PROKKA_24357,ETNP_100m_PROKKA_52245,...</td>\n",
       "      <td>K</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>K.IVVGGPYSSVSDAASSLDSSQK.S</td>\n",
       "      <td>26</td>\n",
       "      <td>175980.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20170410_JA2_01.34897.34897.2</td>\n",
       "      <td>4.888</td>\n",
       "      <td>0.623</td>\n",
       "      <td>1.260000e-09</td>\n",
       "      <td>25/44</td>\n",
       "      <td>R.AIQQQIENPLAQQILSGELVPGK.V</td>\n",
       "      <td>2473.3540</td>\n",
       "      <td>632913.0</td>\n",
       "      <td>gi|54036848|sp|P63284.1|CLPB_ECOLI</td>\n",
       "      <td>R</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>R.AIQQQIENPLAQQILSGELVPGK.V</td>\n",
       "      <td>27</td>\n",
       "      <td>23441.222222</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20170410_JA2_01.23135.23135.3</td>\n",
       "      <td>4.853</td>\n",
       "      <td>0.600</td>\n",
       "      <td>6.590000e-11</td>\n",
       "      <td>29/84</td>\n",
       "      <td>K.IVVGGPYSSVSDAASSLDSSQK.S</td>\n",
       "      <td>2153.0488</td>\n",
       "      <td>6257440.0</td>\n",
       "      <td>ETNP_120m_PROKKA_24357,ETNP_100m_PROKKA_52245,...</td>\n",
       "      <td>K</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>K.IVVGGPYSSVSDAASSLDSSQK.S</td>\n",
       "      <td>26</td>\n",
       "      <td>240670.769231</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20170410_JA2_01.28172.28172.2</td>\n",
       "      <td>4.826</td>\n",
       "      <td>0.657</td>\n",
       "      <td>2.910000e-14</td>\n",
       "      <td>29/42</td>\n",
       "      <td>K.IVVGGPYSSVSDAASVLDGSQK.S</td>\n",
       "      <td>2135.0746</td>\n",
       "      <td>3597780.0</td>\n",
       "      <td>ETNP_120m_PROKKA_126325,ETNP_100m_PROKKA_02317...</td>\n",
       "      <td>K</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>K.IVVGGPYSSVSDAASVLDGSQK.S</td>\n",
       "      <td>26</td>\n",
       "      <td>138376.153846</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        spectrum  xcorr  deltacn        expect   ions  \\\n",
       "0  20170410_JA2_01.34842.34842.3  5.552    0.520  4.130000e-09  31/88   \n",
       "1  20170410_JA2_01.23179.23179.2  4.989    0.657  2.350000e-14  29/42   \n",
       "2  20170410_JA2_01.34897.34897.2  4.888    0.623  1.260000e-09  25/44   \n",
       "3  20170410_JA2_01.23135.23135.3  4.853    0.600  6.590000e-11  29/84   \n",
       "4  20170410_JA2_01.28172.28172.2  4.826    0.657  2.910000e-14  29/42   \n",
       "\n",
       "                       peptide  calc_neutral_pep_mass  precursor_intensity  \\\n",
       "0  R.AIQQQIENPLAQQILSGELVPGK.V              2473.3540           19969800.0   \n",
       "1   K.IVVGGPYSSVSDAASSLDSSQK.S              2153.0488            4575480.0   \n",
       "2  R.AIQQQIENPLAQQILSGELVPGK.V              2473.3540             632913.0   \n",
       "3   K.IVVGGPYSSVSDAASSLDSSQK.S              2153.0488            6257440.0   \n",
       "4   K.IVVGGPYSSVSDAASVLDGSQK.S              2135.0746            3597780.0   \n",
       "\n",
       "                                             protein L terminus  ... c-carb  \\\n",
       "0                 gi|54036848|sp|P63284.1|CLPB_ECOLI          R  ...      0   \n",
       "1  ETNP_120m_PROKKA_24357,ETNP_100m_PROKKA_52245,...          K  ...      0   \n",
       "2                 gi|54036848|sp|P63284.1|CLPB_ECOLI          R  ...      0   \n",
       "3  ETNP_120m_PROKKA_24357,ETNP_100m_PROKKA_52245,...          K  ...      0   \n",
       "4  ETNP_120m_PROKKA_126325,ETNP_100m_PROKKA_02317...          K  ...      0   \n",
       "\n",
       "   m-oxid  n-deam  q-deam  k-hydr  r-meth             stripped peptide  \\\n",
       "0       0       0       0       0       0  R.AIQQQIENPLAQQILSGELVPGK.V   \n",
       "1       0       0       0       0       0   K.IVVGGPYSSVSDAASSLDSSQK.S   \n",
       "2       0       0       0       0       0  R.AIQQQIENPLAQQILSGELVPGK.V   \n",
       "3       0       0       0       0       0   K.IVVGGPYSSVSDAASSLDSSQK.S   \n",
       "4       0       0       0       0       0   K.IVVGGPYSSVSDAASVLDGSQK.S   \n",
       "\n",
       "   stripped length      NAAF num.  ptm-total  \n",
       "0               27  739622.222222          0  \n",
       "1               26  175980.000000          0  \n",
       "2               27   23441.222222          0  \n",
       "3               26  240670.769231          0  \n",
       "4               26  138376.153846          0  \n",
       "\n",
       "[5 rows x 41 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get rid of rows where the xcorr is unavailable (usually 3 or so)\n",
    "comet231 = comet231[comet231.xcorr != '[unavailable]']\n",
    "\n",
    "# use str.strip with indexing by str[0] to add a column with the peptide's left terminus\n",
    "comet231['L terminus'] = comet231['peptide'].astype(str).str[0]\n",
    "\n",
    "# use str.strip with indexing by str[-1] to add a column with the peptide's left terminus\n",
    "comet231['R terminus'] = comet231['peptide'].str.strip().str[-1]\n",
    "\n",
    "# use a count function to enumerate the # of A's (alanines) in each peptide\n",
    "comet231['A'] = comet231['peptide'].str.count(\"A\")\n",
    "\n",
    "# use a count function to enumerate the # of C's (cysteines) in each peptide\n",
    "comet231['C'] = comet231['peptide'].str.count(\"C\")\n",
    "\n",
    "# use a count function to enumerate the # of D's (aspartic acids) in each peptide\n",
    "comet231['D'] = comet231['peptide'].str.count(\"D\")\n",
    "\n",
    "# use a count function to enumerate the # of E's (glutamic acids) in each peptide\n",
    "comet231['E'] = comet231['peptide'].str.count(\"E\")\n",
    "\n",
    "# use a count function to enumerate the # of F's (phenylalanines) in each peptide\n",
    "comet231['F'] = comet231['peptide'].str.count(\"F\")\n",
    "\n",
    "# use a count function to enumerate the # of G's (glycines) in each peptide\n",
    "comet231['G'] = comet231['peptide'].str.count(\"G\")\n",
    "\n",
    "# use a count function to enumerate the # of H's (histidines) in each peptide\n",
    "comet231['H'] = comet231['peptide'].str.count(\"H\")\n",
    "\n",
    "# use a count function to enumerate the # of I's (isoleucines) in each peptide\n",
    "# in comet231 output, there will be no isoleucines (they're lumped in with leucines)\n",
    "comet231['I'] = comet231['peptide'].str.count(\"I\")\n",
    "\n",
    "# use a count function to enumerate the # of K's (lysines) in each peptide\n",
    "comet231['K'] = comet231['peptide'].str.count(\"K\")\n",
    "\n",
    "# use a count function to enumerate the # of L's (leucines) in each peptide\n",
    "# also these include the isoleucines\n",
    "comet231['L'] = comet231['peptide'].str.count(\"L\")\n",
    "\n",
    "# use a count function to enumerate the # of M's (methionines) in each peptide\n",
    "comet231['M'] = comet231['peptide'].str.count(\"M\")\n",
    "\n",
    "# use a count function to enumerate the # of N's (asparagines) in each peptide\n",
    "comet231['N'] = comet231['peptide'].str.count(\"N\")\n",
    "\n",
    "# use a count function to enumerate the # of P's ([prolines]) in each peptide\n",
    "comet231['P'] = comet231['peptide'].str.count(\"P\")\n",
    "\n",
    "# use a count function to enumerate the # of Q's (glutamines) in each peptide\n",
    "comet231['Q'] = comet231['peptide'].str.count(\"Q\")\n",
    "\n",
    "# use a count function to enumerate the # of R's (arginines) in each peptide\n",
    "comet231['R'] = comet231['peptide'].str.count(\"R\")\n",
    "\n",
    "# use a count function to enumerate the # of S's (serines) in each peptide\n",
    "comet231['S'] = comet231['peptide'].str.count(\"S\")\n",
    "\n",
    "# use a count function to enumerate the # of T's (threonines) in each peptide\n",
    "comet231['T'] = comet231['peptide'].str.count(\"T\")\n",
    "\n",
    "# use a count function to enumerate the # of V's (valines) in each peptide\n",
    "comet231['V'] = comet231['peptide'].str.count(\"V\")\n",
    "\n",
    "# use a count function to enumerate the # of W's (tryptophans) in each peptide\n",
    "comet231['W'] = comet231['peptide'].str.count(\"W\")\n",
    "\n",
    "# use a count function to enumerate the # of Y's (tyrosines) in each peptide\n",
    "comet231['Y'] = comet231['peptide'].str.count(\"Y\")\n",
    "\n",
    "# use a count function to enumerate the # of carbamidomethylated C's in each peptide\n",
    "comet231['c-carb'] = comet231['peptide'].str.count(\"160.03\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "comet231['m-oxid'] = comet231['peptide'].str.count(\"147.04\")\n",
    "\n",
    "# use a count function to enumerate the # of deamidated N's in each peptide\n",
    "comet231['n-deam'] = comet231['peptide'].str.count(\"115.03\")\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "comet231['q-deam'] = comet231['peptide'].str.count(\"129.04\")\n",
    "\n",
    "# use a count function to enumerate the # of hydroxylated K's in each peptide\n",
    "comet231['k-hydr'] = comet231['peptide'].str.count(\"144.09\")\n",
    "\n",
    "# use a count function to enumerate the # of hydroxylated P's in each peptide\n",
    "# I removed P hydroxyation in final searches because there were so few\n",
    "#comet231['p-hydr'] = comet231['peptide'].str.count(\"131.05\")\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "comet231['r-meth'] = comet231['peptide'].str.count(\"170.12\")\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "comet231['stripped peptide'] = comet231['peptide'].str.replace(r\"\\(.*\\)\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "comet231['stripped length'] = comet231['stripped peptide'].apply(len)\n",
    "\n",
    "comet231['NAAF num.'] = comet231['precursor_intensity'] / comet231['stripped length']\n",
    "\n",
    "# total the number of modifications in sequence\n",
    "comet231['ptm-total'] = comet231['c-carb'] + comet231['m-oxid'] + comet231['n-deam'] + comet231['q-deam'] + comet231['k-hydr'] + comet231['r-meth']\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "comet231.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/TPP-PTMopt/ETNP-SKQ17-TPP-PTM-opt_15ppm_FUNGI/ETNP-SKQ17-231-100m-0.3-JA2_PTMopt_Comet15_fungidb.csv\")\n",
    "\n",
    "# check out the results\n",
    "comet231.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Calculating the false discovery rate (% FDR)\n",
    "\n",
    "### Filtering PSMs > a selected XCorr value and exporting peptides"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# real Comet PSMs 11445\n",
      "# decoy Comet PSMs 7818\n",
      "False discovery rate =  68.30930537352555\n"
     ]
    }
   ],
   "source": [
    "# Let's separate out the decoy hits from the good ones\n",
    "\n",
    "comet231pmm = comet231[~comet231['protein'].str.contains(\"DECOY\")]\n",
    "comet231dec = comet231[comet231['protein'].str.contains(\"DECOY\")]\n",
    "\n",
    "# how many PSM that are only PMM (proteins in the database)?\n",
    "\n",
    "print(\"# real Comet PSMs\", len(comet231pmm))\n",
    "\n",
    "# compared to how many PSMs containing decoys?\n",
    "\n",
    "print(\"# decoy Comet PSMs\", len(comet231dec))\n",
    "\n",
    "# calculate the bulk FDR (all PSMs so let's not beat ourselves up)\n",
    "\n",
    "r = len(comet231pmm)\n",
    "d = len(comet231dec)\n",
    "\n",
    "FDR = d/r*100\n",
    "\n",
    "print(\"False discovery rate = \", FDR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# real Comet PSMs 167\n",
      "# decoy Comet PSMs 2\n",
      "False discovery rate =  1.183431952662722\n"
     ]
    }
   ],
   "source": [
    "# keep only peptides  >3 XCorr\n",
    "# need to convert Xcorr column from strings to numeric so we can use loc\n",
    "# note that pmm here just means 'non decoy', it's a vestige from the accession prefixes from the MED proteins\n",
    "comet231['xcorr'] = pd.to_numeric(comet231['xcorr'])\n",
    "\n",
    "comet231_3 = comet231.loc[comet231['xcorr'] >= 2.9]\n",
    "\n",
    "# What's the FDR?\n",
    "\n",
    "# Let's separate out the decoy hits from the good ones\n",
    "\n",
    "comet231pmm3 = comet231_3[~comet231_3['protein'].str.contains(\"DECOY\")]\n",
    "comet231dec3 = comet231_3[comet231_3['protein'].str.contains(\"DECOY\")]\n",
    "\n",
    "# how many PSM that are only PMM (proteins in the database)?\n",
    "\n",
    "print(\"# real Comet PSMs\", len(comet231pmm3))\n",
    "\n",
    "# compared to how many PSMs containing decoys?\n",
    "\n",
    "print(\"# decoy Comet PSMs\", len(comet231dec3))\n",
    "\n",
    "# calculate the FDR \n",
    "\n",
    "r = len(comet231pmm3)\n",
    "d = len(comet231dec3)\n",
    "\n",
    "FDR = d/(d+r)*100\n",
    "\n",
    "print(\"False discovery rate = \", FDR)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## NAAF correction and exporting files with AA and PTM totals:\n",
    "\n",
    "### Only for DECOY-filtered Comet PSMs > XCorr 3.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>K</th>\n",
       "      <th>L</th>\n",
       "      <th>...</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>Total area</th>\n",
       "      <th>Total length</th>\n",
       "      <th>% C w/ carb</th>\n",
       "      <th>% M w/ oxid</th>\n",
       "      <th>% N w/ deam</th>\n",
       "      <th>% Q w/ deam</th>\n",
       "      <th>% K w/ hydr</th>\n",
       "      <th>% R w/ meth</th>\n",
       "      <th>NAAF denom.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>415</td>\n",
       "      <td>7</td>\n",
       "      <td>217</td>\n",
       "      <td>196</td>\n",
       "      <td>78</td>\n",
       "      <td>309</td>\n",
       "      <td>9</td>\n",
       "      <td>154</td>\n",
       "      <td>201</td>\n",
       "      <td>207</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>913932978.0</td>\n",
       "      <td>3864</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.516129</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.049587</td>\n",
       "      <td>0.014925</td>\n",
       "      <td>0.0</td>\n",
       "      <td>236525.097826</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                A  C    D    E   F    G  H    I    K    L  ...  r-meth  \\\n",
       "sample total  415  7  217  196  78  309  9  154  201  207  ...       0   \n",
       "\n",
       "               Total area  Total length  % C w/ carb  % M w/ oxid  \\\n",
       "sample total  913932978.0          3864          1.0     0.516129   \n",
       "\n",
       "              % N w/ deam  % Q w/ deam  % K w/ hydr  % R w/ meth  \\\n",
       "sample total          0.0     0.049587     0.014925          0.0   \n",
       "\n",
       "                NAAF denom.  \n",
       "sample total  236525.097826  \n",
       "\n",
       "[1 rows x 35 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of certain columns in the modified\n",
    "# peptide dataframe above (for >XCorr 3)\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'A': comet231pmm3['A'].sum(),\n",
    "        'C': comet231pmm3['C'].sum(),\n",
    "        'D': comet231pmm3['D'].sum(),\n",
    "        'E': comet231pmm3['E'].sum(),\n",
    "        'F': comet231pmm3['F'].sum(),\n",
    "        'G': comet231pmm3['G'].sum(),\n",
    "        'H': comet231pmm3['H'].sum(),\n",
    "        'I': comet231pmm3['I'].sum(),\n",
    "        'K': comet231pmm3['K'].sum(),\n",
    "        'I': comet231pmm3['I'].sum(),\n",
    "        'L': comet231pmm3['L'].sum(),\n",
    "        'M': comet231pmm3['M'].sum(),\n",
    "        'N': comet231pmm3['N'].sum(),\n",
    "        'P': comet231pmm3['P'].sum(),\n",
    "        'Q': comet231pmm3['Q'].sum(),\n",
    "        'R': comet231pmm3['R'].sum(),\n",
    "        'S': comet231pmm3['S'].sum(),\n",
    "        'T': comet231pmm3['T'].sum(),\n",
    "        'V': comet231pmm3['V'].sum(),\n",
    "        'W': comet231pmm3['W'].sum(),\n",
    "        'Y': comet231pmm3['Y'].sum(),\n",
    "        'c-carb': comet231pmm3['c-carb'].sum(),\n",
    "        'm-oxid': comet231pmm3['m-oxid'].sum(),\n",
    "        'n-deam': comet231pmm3['n-deam'].sum(),\n",
    "        'q-deam': comet231pmm3['q-deam'].sum(),\n",
    "        'k-hydr': comet231pmm3['k-hydr'].sum(),\n",
    "        'r-meth': comet231pmm3['r-meth'].sum(),\n",
    "        'Total area': comet231pmm3['precursor_intensity'].sum(),\n",
    "        'Total length': comet231pmm3['stripped length'].sum()\n",
    "       }\n",
    "\n",
    "totalcomet231pmm3 = pd.DataFrame(data, columns=['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', 'm-oxid', 'n-deam', 'q-deam', 'k-hydr', 'r-meth', 'Total area', 'Total length'], index=index)\n",
    "\n",
    "# calculate percentage of C's with carb (should be 1.0)\n",
    "totalcomet231pmm3['% C w/ carb'] = totalcomet231pmm3['c-carb'] / totalcomet231pmm3['C'] \n",
    "\n",
    "# calculate percentage of M's that are oxidized\n",
    "totalcomet231pmm3['% M w/ oxid'] = totalcomet231pmm3['m-oxid'] / totalcomet231pmm3['M'] \n",
    "\n",
    "# calculate percentage of N's that are deamidated\n",
    "totalcomet231pmm3['% N w/ deam'] = totalcomet231pmm3['n-deam'] / totalcomet231pmm3['N'] \n",
    "\n",
    "# calculate percentage of Q's that are deamidated\n",
    "totalcomet231pmm3['% Q w/ deam'] = totalcomet231pmm3['q-deam'] / totalcomet231pmm3['Q'] \n",
    "\n",
    "# calculate percentage of K's that are hydroxylated\n",
    "totalcomet231pmm3['% K w/ hydr'] = totalcomet231pmm3['k-hydr'] / totalcomet231pmm3['K'] \n",
    "\n",
    "# calculate percentage of P's that are hydroxylated\n",
    "#totalcomet231pmm3['% P w/ hydr'] = totalcomet231pmm3['p-hydr'] / totalcomet231pmm3['P'] \n",
    "\n",
    "# calculate percentage of R's that are methylated\n",
    "totalcomet231pmm3['% R w/ meth'] = totalcomet231pmm3['r-meth'] / totalcomet231pmm3['R'] \n",
    "\n",
    "# calculate NAAF denominator for all peptides in dataset i\n",
    "totalcomet231pmm3['NAAF denom.'] = totalcomet231pmm3['Total area'] / totalcomet231pmm3['Total length']\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "totalcomet231pmm3.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/TPP-PTMopt/ETNP-SKQ17-TPP-PTM-opt_15ppm_FUNGI/ETNP-SKQ17-231-100m-0.3-JA2_PTMopt_Comet15_1FDR_totals.csv\")\n",
    "\n",
    "totalcomet231pmm3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-36-ea87c9113921>:7: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  comet231pmm3['NAAF factor'] = (comet231pmm3['NAAF num.'])/NAAF3\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>NAAF factor</th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>K</th>\n",
       "      <th>...</th>\n",
       "      <th>T-NAAF3</th>\n",
       "      <th>V-NAAF3</th>\n",
       "      <th>W-NAAF3</th>\n",
       "      <th>Y-NAAF3</th>\n",
       "      <th>ccarb-NAAF3</th>\n",
       "      <th>moxid-NAAF3</th>\n",
       "      <th>ndeam-NAAF3</th>\n",
       "      <th>qdeam-NAAF3</th>\n",
       "      <th>khydr-NAAF3</th>\n",
       "      <th>rmeth-NAAF3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>R.AIQQQIENPLAQQILSGELVPGK.V</td>\n",
       "      <td>3.127035</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.254070</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>K.IVVGGPYSSVSDAASSLDSSQK.S</td>\n",
       "      <td>0.744023</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.232068</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.744023</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>R.AIQQQIENPLAQQILSGELVPGK.V</td>\n",
       "      <td>0.099107</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.198213</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>K.IVVGGPYSSVSDAASSLDSSQK.S</td>\n",
       "      <td>1.017527</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.052582</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.017527</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>K.IVVGGPYSSVSDAASVLDGSQK.S</td>\n",
       "      <td>0.585038</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.340152</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.585038</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              stripped peptide  NAAF factor  A  C  D  E  F  G  H  K  ...  \\\n",
       "0  R.AIQQQIENPLAQQILSGELVPGK.V     3.127035  2  0  0  2  0  2  0  1  ...   \n",
       "1   K.IVVGGPYSSVSDAASSLDSSQK.S     0.744023  2  0  2  0  0  2  0  2  ...   \n",
       "2  R.AIQQQIENPLAQQILSGELVPGK.V     0.099107  2  0  0  2  0  2  0  1  ...   \n",
       "3   K.IVVGGPYSSVSDAASSLDSSQK.S     1.017527  2  0  2  0  0  2  0  2  ...   \n",
       "4   K.IVVGGPYSSVSDAASVLDGSQK.S     0.585038  2  0  2  0  0  3  0  2  ...   \n",
       "\n",
       "   T-NAAF3   V-NAAF3  W-NAAF3   Y-NAAF3  ccarb-NAAF3  moxid-NAAF3  \\\n",
       "0      0.0  6.254070      0.0  0.000000          0.0          0.0   \n",
       "1      0.0  2.232068      0.0  0.744023          0.0          0.0   \n",
       "2      0.0  0.198213      0.0  0.000000          0.0          0.0   \n",
       "3      0.0  3.052582      0.0  1.017527          0.0          0.0   \n",
       "4      0.0  2.340152      0.0  0.585038          0.0          0.0   \n",
       "\n",
       "   ndeam-NAAF3  qdeam-NAAF3  khydr-NAAF3  rmeth-NAAF3  \n",
       "0          0.0          0.0          0.0          0.0  \n",
       "1          0.0          0.0          0.0          0.0  \n",
       "2          0.0          0.0          0.0          0.0  \n",
       "3          0.0          0.0          0.0          0.0  \n",
       "4          0.0          0.0          0.0          0.0  \n",
       "\n",
       "[5 rows x 54 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use the calculated NAAF factor (in totalcomet231pmm3 dataframe, above) to caluclate the NAAF \n",
    "# NAAF: normalized normalized area abundance factor\n",
    "\n",
    "NAAF3 = 236525.097826\n",
    "\n",
    "# use NAAF >XCorr 3 to get NAAF\n",
    "comet231pmm3['NAAF factor'] = (comet231pmm3['NAAF num.'])/NAAF3\n",
    "\n",
    "# make a dataframe that contains only what we need: sequences, AAs, PTMs\n",
    "comet231pmm3_NAAF = comet231pmm3[['stripped peptide', 'NAAF factor', 'A', 'C', 'D', 'E', 'F', 'G', 'H', 'K', 'I', 'L', \\\n",
    "                                'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', 'm-oxid', 'n-deam', \\\n",
    "                                'q-deam', 'k-hydr', 'r-meth']].copy()\n",
    "\n",
    "# multiply the NAAF3 factor by the AAs to normalize its abundance by peak area and peptide length\n",
    "\n",
    "comet231pmm3_NAAF['A-NAAF3'] = comet231pmm3_NAAF['A'] * comet231pmm3['NAAF factor']\n",
    "comet231pmm3_NAAF['C-NAAF3'] = comet231pmm3_NAAF['C'] * comet231pmm3['NAAF factor']\n",
    "comet231pmm3_NAAF['D-NAAF3'] = comet231pmm3_NAAF['D'] * comet231pmm3['NAAF factor']\n",
    "comet231pmm3_NAAF['E-NAAF3'] = comet231pmm3_NAAF['E'] * comet231pmm3['NAAF factor']\n",
    "comet231pmm3_NAAF['F-NAAF3'] = comet231pmm3_NAAF['F'] * comet231pmm3['NAAF factor']\n",
    "comet231pmm3_NAAF['G-NAAF3'] = comet231pmm3_NAAF['G'] * comet231pmm3['NAAF factor']\n",
    "comet231pmm3_NAAF['H-NAAF3'] = comet231pmm3_NAAF['H'] * comet231pmm3['NAAF factor']\n",
    "comet231pmm3_NAAF['I-NAAF3'] = comet231pmm3_NAAF['I'] * comet231pmm3['NAAF factor']\n",
    "comet231pmm3_NAAF['K-NAAF3'] = comet231pmm3_NAAF['K'] * comet231pmm3['NAAF factor']\n",
    "comet231pmm3_NAAF['L-NAAF3'] = comet231pmm3_NAAF['L'] * comet231pmm3['NAAF factor']\n",
    "comet231pmm3_NAAF['M-NAAF3'] = comet231pmm3_NAAF['M'] * comet231pmm3['NAAF factor']\n",
    "comet231pmm3_NAAF['N-NAAF3'] = comet231pmm3_NAAF['N'] * comet231pmm3['NAAF factor']\n",
    "comet231pmm3_NAAF['P-NAAF3'] = comet231pmm3_NAAF['P'] * comet231pmm3['NAAF factor']\n",
    "comet231pmm3_NAAF['Q-NAAF3'] = comet231pmm3_NAAF['Q'] * comet231pmm3['NAAF factor']\n",
    "comet231pmm3_NAAF['R-NAAF3'] = comet231pmm3_NAAF['R'] * comet231pmm3['NAAF factor']\n",
    "comet231pmm3_NAAF['S-NAAF3'] = comet231pmm3_NAAF['S'] * comet231pmm3['NAAF factor']\n",
    "comet231pmm3_NAAF['T-NAAF3'] = comet231pmm3_NAAF['T'] * comet231pmm3['NAAF factor']\n",
    "comet231pmm3_NAAF['V-NAAF3'] = comet231pmm3_NAAF['V'] * comet231pmm3['NAAF factor']\n",
    "comet231pmm3_NAAF['W-NAAF3'] = comet231pmm3_NAAF['W'] * comet231pmm3['NAAF factor']\n",
    "comet231pmm3_NAAF['Y-NAAF3'] = comet231pmm3_NAAF['Y'] * comet231pmm3['NAAF factor']\n",
    "\n",
    "# multiply the NAAF3 factor by the PTMs normalize its abundance by peak area and peptide length\n",
    "\n",
    "comet231pmm3_NAAF['ccarb-NAAF3'] = comet231pmm3_NAAF['c-carb'] * comet231pmm3_NAAF['NAAF factor']\n",
    "comet231pmm3_NAAF['moxid-NAAF3'] = comet231pmm3_NAAF['m-oxid'] * comet231pmm3_NAAF['NAAF factor']\n",
    "comet231pmm3_NAAF['ndeam-NAAF3'] = comet231pmm3_NAAF['n-deam'] * comet231pmm3_NAAF['NAAF factor']\n",
    "comet231pmm3_NAAF['qdeam-NAAF3'] = comet231pmm3_NAAF['q-deam'] * comet231pmm3_NAAF['NAAF factor']\n",
    "comet231pmm3_NAAF['khydr-NAAF3'] = comet231pmm3_NAAF['k-hydr'] * comet231pmm3_NAAF['NAAF factor']\n",
    "comet231pmm3_NAAF['rmeth-NAAF3'] = comet231pmm3_NAAF['r-meth'] * comet231pmm3_NAAF['NAAF factor']\n",
    "\n",
    "# write the dataframe to a new csv\n",
    "comet231pmm3_NAAF.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/TPP-PTMopt/ETNP-SKQ17-TPP-PTM-opt_15ppm_FUNGI/ETNP-SKQ17-231-100m-0.3-JA2_PTMopt_Comet15_1FDR_NAAF.csv\")\n",
    "\n",
    "comet231pmm3_NAAF.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NAAF</th>\n",
       "      <th>A-NAAF</th>\n",
       "      <th>C-NAAF</th>\n",
       "      <th>D-NAAF</th>\n",
       "      <th>E-NAAF</th>\n",
       "      <th>F-NAAF</th>\n",
       "      <th>G-NAAF</th>\n",
       "      <th>H-NAAF</th>\n",
       "      <th>I-NAAF</th>\n",
       "      <th>K-NAAF</th>\n",
       "      <th>...</th>\n",
       "      <th>Q-deam-NAAF</th>\n",
       "      <th>K-hydr-NAAF</th>\n",
       "      <th>R-meth-NAAF</th>\n",
       "      <th>% C w/ carb. NAAF</th>\n",
       "      <th>% M w/ oxid. NAAF</th>\n",
       "      <th>% N w/ deam. NAAF</th>\n",
       "      <th>% Q w/ deam. NAAF</th>\n",
       "      <th>% K w/ hydr. NAAF</th>\n",
       "      <th>% R w/ meth. NAAF</th>\n",
       "      <th>NAAF check</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>192.809094</td>\n",
       "      <td>417.178448</td>\n",
       "      <td>2.138785</td>\n",
       "      <td>246.305973</td>\n",
       "      <td>252.434195</td>\n",
       "      <td>16.369696</td>\n",
       "      <td>251.610994</td>\n",
       "      <td>5.386227</td>\n",
       "      <td>207.616083</td>\n",
       "      <td>118.783472</td>\n",
       "      <td>...</td>\n",
       "      <td>0.566503</td>\n",
       "      <td>0.086301</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.326099</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.00282</td>\n",
       "      <td>0.000727</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000815</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    NAAF      A-NAAF    C-NAAF      D-NAAF      E-NAAF  \\\n",
       "sample total  192.809094  417.178448  2.138785  246.305973  252.434195   \n",
       "\n",
       "                 F-NAAF      G-NAAF    H-NAAF      I-NAAF      K-NAAF  ...  \\\n",
       "sample total  16.369696  251.610994  5.386227  207.616083  118.783472  ...   \n",
       "\n",
       "              Q-deam-NAAF  K-hydr-NAAF  R-meth-NAAF  % C w/ carb. NAAF  \\\n",
       "sample total     0.566503     0.086301          0.0                1.0   \n",
       "\n",
       "              % M w/ oxid. NAAF  % N w/ deam. NAAF  % Q w/ deam. NAAF  \\\n",
       "sample total           0.326099                0.0            0.00282   \n",
       "\n",
       "              % K w/ hydr. NAAF  % R w/ meth. NAAF  NAAF check  \n",
       "sample total           0.000727                0.0    0.000815  \n",
       "\n",
       "[1 rows x 34 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of NAAF normalized AAs for comet231 Comet results\n",
    "# also contains the sums of the NAAF-corrected PTMs occurances for each affected residue\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'NAAF': comet231pmm3_NAAF['NAAF factor'].sum(),\n",
    "        'A-NAAF': comet231pmm3_NAAF['A-NAAF3'].sum(),\n",
    "        'C-NAAF': comet231pmm3_NAAF['C-NAAF3'].sum(),\n",
    "        'D-NAAF': comet231pmm3_NAAF['D-NAAF3'].sum(),\n",
    "        'E-NAAF': comet231pmm3_NAAF['E-NAAF3'].sum(),\n",
    "        'F-NAAF': comet231pmm3_NAAF['F-NAAF3'].sum(),\n",
    "        'G-NAAF': comet231pmm3_NAAF['G-NAAF3'].sum(),\n",
    "        'H-NAAF': comet231pmm3_NAAF['H-NAAF3'].sum(),\n",
    "        'I-NAAF': comet231pmm3_NAAF['I-NAAF3'].sum(),\n",
    "        'K-NAAF': comet231pmm3_NAAF['K-NAAF3'].sum(),\n",
    "        'L-NAAF': comet231pmm3_NAAF['L-NAAF3'].sum(),\n",
    "        'M-NAAF': comet231pmm3_NAAF['M-NAAF3'].sum(),\n",
    "        'N-NAAF': comet231pmm3_NAAF['N-NAAF3'].sum(),\n",
    "        'P-NAAF': comet231pmm3_NAAF['P-NAAF3'].sum(),\n",
    "        'Q-NAAF': comet231pmm3_NAAF['Q-NAAF3'].sum(),\n",
    "        'R-NAAF': comet231pmm3_NAAF['R-NAAF3'].sum(),\n",
    "        'S-NAAF': comet231pmm3_NAAF['S-NAAF3'].sum(),\n",
    "        'T-NAAF': comet231pmm3_NAAF['T-NAAF3'].sum(),\n",
    "        'V-NAAF': comet231pmm3_NAAF['V-NAAF3'].sum(),\n",
    "        'W-NAAF': comet231pmm3_NAAF['W-NAAF3'].sum(),\n",
    "        'Y-NAAF': comet231pmm3_NAAF['Y-NAAF3'].sum(),\n",
    "        'C-carb-NAAF': comet231pmm3_NAAF['ccarb-NAAF3'].sum(),\n",
    "        'M-oxid-NAAF': comet231pmm3_NAAF['moxid-NAAF3'].sum(),\n",
    "        'N-deam-NAAF': comet231pmm3_NAAF['ndeam-NAAF3'].sum(),\n",
    "        'Q-deam-NAAF': comet231pmm3_NAAF['qdeam-NAAF3'].sum(),\n",
    "        'K-hydr-NAAF': comet231pmm3_NAAF['khydr-NAAF3'].sum(),\n",
    "        'R-meth-NAAF': comet231pmm3_NAAF['rmeth-NAAF3'].sum()\n",
    "       }\n",
    "\n",
    "totalcomet231pmm3_NAAF = pd.DataFrame(data, columns=['NAAF', 'A-NAAF', 'C-NAAF', 'D-NAAF', 'E-NAAF', 'F-NAAF', \\\n",
    "                                                   'G-NAAF', 'H-NAAF', 'I-NAAF','K-NAAF', 'L-NAAF', 'M-NAAF', \\\n",
    "                                                   'N-NAAF', 'P-NAAF', 'Q-NAAF', 'R-NAAF', 'S-NAAF', \\\n",
    "                                                   'T-NAAF', 'V-NAAF', 'W-NAAF', 'Y-NAAF', 'C-carb-NAAF', \\\n",
    "                                                   'M-oxid-NAAF', 'N-deam-NAAF', 'Q-deam-NAAF', 'K-hydr-NAAF',\\\n",
    "                                                   'R-meth-NAAF'], index=index)\n",
    "\n",
    "# calculate the NAAF-corrected % modified C, M, N, Q, K, and Rs\n",
    "\n",
    "\n",
    "totalcomet231pmm3_NAAF['% C w/ carb. NAAF'] = totalcomet231pmm3_NAAF['C-carb-NAAF'] / totalcomet231pmm3_NAAF['C-NAAF']\n",
    "totalcomet231pmm3_NAAF['% M w/ oxid. NAAF'] = totalcomet231pmm3_NAAF['M-oxid-NAAF'] / totalcomet231pmm3_NAAF['M-NAAF']\n",
    "totalcomet231pmm3_NAAF['% N w/ deam. NAAF'] = totalcomet231pmm3_NAAF['N-deam-NAAF'] / totalcomet231pmm3_NAAF['N-NAAF']\n",
    "totalcomet231pmm3_NAAF['% Q w/ deam. NAAF'] = totalcomet231pmm3_NAAF['Q-deam-NAAF'] / totalcomet231pmm3_NAAF['Q-NAAF']\n",
    "totalcomet231pmm3_NAAF['% K w/ hydr. NAAF'] = totalcomet231pmm3_NAAF['K-hydr-NAAF'] / totalcomet231pmm3_NAAF['K-NAAF']\n",
    "totalcomet231pmm3_NAAF['% R w/ meth. NAAF'] = totalcomet231pmm3_NAAF['R-meth-NAAF'] / totalcomet231pmm3_NAAF['R-NAAF']\n",
    "\n",
    "# calculate NAAF summed numerator over denominator (in above cell) for all peptides in dataset i: a check\n",
    "totalcomet231pmm3_NAAF['NAAF check'] = totalcomet231pmm3_NAAF['NAAF'] / 236525.097826\n",
    "\n",
    "# write modified dataframe to new txt file, same name + totals\n",
    "totalcomet231pmm3_NAAF.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/TPP-PTMopt/ETNP-SKQ17-TPP-PTM-opt_15ppm_FUNGI/ETNP-SKQ17-231-100m-0.3-JA2_PTMopt_Comet15_1FDR_NAAF_totals.csv\")\n",
    "\n",
    "totalcomet231pmm3_NAAF.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Visualizing the results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exporting txt files of stripped peptides at confidence cutoffs:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# keep only peptide column >50% ALC\n",
    "pep = peaks[[\"stripped peptide\"]]\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "pep.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.txt\", header=False, index=False)\n",
    "\n",
    "# made the text file into a FASTA \n",
    "\n",
    "!awk '{print \">\"NR\"\\n\"$0}' /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.txt > /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas\n",
    "\n",
    "# look\n",
    "\n",
    "print(\"# of DN peptide >50% ALC\", len(pep))\n",
    "pep.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# keep only peptides  >80% ALC\n",
    "peaks80 = peaks.loc[peaks['ALC (%)'] >= 80]\n",
    "\n",
    "# see how many rows and double check\n",
    "# peaks80.head(-10)\n",
    "\n",
    "# keep only peptide column \n",
    "pep80 = peaks80[[\"stripped peptide\"]]\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "pep80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN80_stripped_peptides.txt\", header=False, index=False)\n",
    "\n",
    "# made the text file into a FASTA \n",
    "\n",
    "!awk '{print \">\"NR\"\\n\"$0}' /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN80_stripped_peptides.txt > /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN80_stripped_peptides.fas\n",
    "\n",
    "print(\"# of DN peptide >80% ALC\", len(pep80))\n",
    "pep80.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Using BioPython to query peptide sequences\n",
    "\n",
    "I installed the BioPython package using `pip install biopython`. All instructions and information [here](https://www.tutorialspoint.com/biopython/index.htm). \n",
    "\n",
    "GitHub project: https://github.com/biopython/biopython\n",
    "\n",
    "I'm relying on the ProtParam module to parse sequences for relative AA composition, instability, secondary structure, instability, and hydrophobicity. You can read more about that module and the studies the indecies are derived from here:\n",
    "\n",
    "https://biopython.org/wiki/ProtParam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# Bio.SeqIO is the standard Sequence Input/Output interface for BioPython 1.43 and later\n",
    "# Bio.SeqIO provides a simple uniform interface to input and output assorted sequence file formats.\n",
    "# (including multiple sequence alignments), but will only deal with sequences as SeqRecord objects\n",
    "\n",
    "# for accepted file formats see https://biopython.org/wiki/SeqIO\n",
    "\n",
    "from Bio import SeqIO\n",
    "#for seq_record in SeqIO.parse(\"/home/millieginty/Documents/git-repos/2017-etnp/data/MED4/MED2_tryp_1raw_db_peptides_nmod.fasta\", \"fasta\"):\n",
    "    #print(seq_record.id)\n",
    "    #print(repr(seq_record.seq))\n",
    "    #print(len(seq_record))\n",
    "    \n",
    "# I commented the print functions out so the output doesn't take up too much space. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# seeing what the ProtParam module can do with a single protein sequence:\n",
    "\n",
    "from Bio.SeqUtils.ProtParam import ProteinAnalysis\n",
    "\n",
    "test_seq = \"MAEGEITTFTALTEKFNLPPGNYKKPKLLYCSNGGHFLRILPDGTVDGTRDRSDQHIQLQLSAESVGEVYIKSTETGQYLAMDTSGLLYGSQTPSEECLFLERLEENHYNTYTSKKHAEKNWFVGLKKNGSCKRGPRTHYGQKAILFLPLPV\"\n",
    "\n",
    "analysed_seq = ProteinAnalysis(test_seq)\n",
    "print(\"molecular weight of seq =\", analysed_seq.molecular_weight())\n",
    "\n",
    "# calculates the aromaticity value of a protein according to Lobry & Gautier (1994, Nucleic Acids Res., 22, 3174-3180). \n",
    "# it's simply the relative frequency of Phe+Trp+Tyr.\n",
    "\n",
    "analysed_seq.aromaticity()\n",
    "print(\"aromaticity of seq =\", analysed_seq.aromaticity())\n",
    "\n",
    "# secondary_structure_fraction:\n",
    "# this methods returns a list of the fraction of amino acids which tend to be in helix, turn or sheet. \n",
    "# AAs in helix: V, I, Y, F, W, L\n",
    "# AAs in turn: N, P, G, S\n",
    "# AAs in sheet: E, M, A, L\n",
    "# the returned list contains 3 values: [Helix, Turn, Sheet]\n",
    "\n",
    "analysed_seq.secondary_structure_fraction()\n",
    "print(\"frac in H T S =\", analysed_seq.secondary_structure_fraction())\n",
    "\n",
    "# the instability index, an implementation of the method of Guruprasad et al. (1990, Protein Engineering, 4, 155-161).\n",
    "# this method tests a protein for stability. \n",
    "# any value above 40 means the protein is unstable (=has a short half life)\n",
    "# NOT SURE WHAT THIS MEANS FOR PEPTIDES, BUT WE COULD DO THIS FOR PROTEINS\n",
    "\n",
    "analysed_seq.instability_index()\n",
    "print(\"instability =\", analysed_seq.instability_index())\n",
    "\n",
    "# count_amino_acids will do just that, and get_amino_acids_percent will return %'s for each AA across the sequence. \n",
    "analysed_seq.get_amino_acids_percent()\n",
    "\n",
    "# taking the returned dictionary and converting to a dataframe\n",
    "\n",
    "aadict = analysed_seq.get_amino_acids_percent()\n",
    "aadf = pd.DataFrame(list(aadict.items()),columns = ['residue','% occurance']) \n",
    "\n",
    "aadf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# use SeqIO and a loop to apply count_amino_acids to each sequence in the file\n",
    "# aatot will give us the total number of each residue in the entire sample output\n",
    "\n",
    "import collections\n",
    "from Bio import SeqIO\n",
    "from Bio.SeqUtils.ProtParam import ProteinAnalysis\n",
    "\n",
    "all_aas = collections.defaultdict(int)\n",
    "for record in SeqIO.parse(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas\", \"fasta\"):\n",
    "    x = ProteinAnalysis(str(record.seq))\n",
    "    #print(record.id, x.count_amino_acids())\n",
    "    for aa, count in x.count_amino_acids().items():\n",
    "        all_aas[aa] += count        \n",
    "        \n",
    "# made a dataframe for amino acid total counts        \n",
    "data = (all_aas)\n",
    "aatot = pd.DataFrame(data, index = ['sample sequence total'])\n",
    "aatot.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pandas import Series, DataFrame\n",
    "\n",
    "with open('/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas') as fasta_file:  # Will close handle cleanly\n",
    "    identifiers = []\n",
    "    lengths = []\n",
    "    for seq_record in SeqIO.parse(fasta_file, 'fasta'):  # (generator)\n",
    "        identifiers.append(seq_record.id)\n",
    "        lengths.append(len(seq_record.seq))\n",
    "        \n",
    "        \n",
    "#converting lists to pandas Series    \n",
    "s1 = Series(identifiers, name='ID')\n",
    "s2 = Series(lengths, name='length')\n",
    "\n",
    "#Gathering Series into a pandas DataFrame and rename index as ID column\n",
    "idseq = DataFrame(dict(ID=s1, length=s2)).set_index(['ID'])\n",
    "\n",
    "idseq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pandas import Series, DataFrame\n",
    "\n",
    "\n",
    "\n",
    "with open('/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas) as fasta_file:  # Will close handle cleanly\n",
    "    identifiers = []\n",
    "    lengths = []\n",
    "    aa = []\n",
    "    for seq_record in SeqIO.parse(fasta_file, 'fasta'):  # (generator)\n",
    "        identifiers.append(seq_record.id)\n",
    "        lengths.append(len(seq_record.seq))\n",
    "        aa.count_amino_acids(seq_record.seq)\n",
    "        \n",
    "        \n",
    "#converting lists to pandas Series    \n",
    "s1 = Series(identifiers, name='ID')\n",
    "s2 = Series(lengths, name='length')\n",
    "s3 = Series(aa, name='AAs')\n",
    "\n",
    "#Gathering Series into a pandas DataFrame and rename index as ID column\n",
    "idseq = DataFrame(dict(ID=s1, length=s2, AAs=s3)).set_index(['ID'])\n",
    "\n",
    "idseq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio.SeqUtils import ProtParam\n",
    "\n",
    "handle = open(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas\") \n",
    "for record in SeqIO.parse(handle, \"fasta\"): \n",
    "    seq = str(record.seq)\n",
    "    X = ProtParam.ProteinAnalysis(seq)\n",
    "    print(X.count_amino_acids()) \n",
    "    #print X.get_amino_acids_percent() \n",
    "    #print X.molecular_weight() \n",
    "    #print X.aromaticity() \n",
    "    #print X.instability_index() \n",
    "    #print X.flexibility() \n",
    "    #print X.isoelectric_point() \n",
    "    #print X.secondary_structure_fraction()\n",
    "    \n",
    "# made a data series from the count_amino_acids function\n",
    "# aacount = {X.count_amino_acids()}\n",
    "\n",
    "# made a pandas dataframe from the series generated above\n",
    "# aacount = pd.DataFrame(list(data.items()),columns = ['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y']) \n",
    "\n",
    "aacount = pd.DataFrame(X.count_amino_acids(), index=[0])\n",
    "\n",
    "# look at new dataframe\n",
    "\n",
    "# aacount.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
