{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Manipulation of Trans Proteomic Pipeline (TPP) Comet database search results of *Prochlorococus MED4* LC-MS/MS data using python.\n",
    "\n",
    "Starting with: \n",
    "\n",
    "- Comet output (.xlxs and .csv) of PTM-optimized database searches\n",
    "\n",
    "Goal:\n",
    "\n",
    "- Files with unique, stripped (no PTMs or tryptic ends) peptide lists\n",
    "- Files with all PTMs "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For technical duplicates, I exported Comet search results as both Excel files and as CSVs into my ETNP 2017 git repo:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP\n"
     ]
    }
   ],
   "source": [
    "cd /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RAL4_MED2_trypsin_1_PTMopt_Comet.csv\r\n",
      "RAL4_MED2_trypsin_1_PTMopt_Comet_stripped.csv\r\n",
      "RAL4_MED2_trypsin_1_PTMopt_Comet.xlsx\r\n",
      "RAL4_MED2_trypsin_1_PTMopt_PepProp90.csv\r\n",
      "RAL4_MED2_trypsin_1_PTMopt_PepProp90.xlsx\r\n",
      "RAL4_MED2_trypsin_2_PTMopt_Comet.csv\r\n",
      "RAL4_MED2_trypsin_2_PTMopt_Comet_stripped.csv\r\n",
      "RAL4_MED2_trypsin_2_PTMopt_Comet.xlsx\r\n",
      "RAL4_MED2_trypsin_2_PTMopt_PepProp90.csv\r\n",
      "RAL4_MED2_trypsin_2_PTMopt_PepProp90.xlsx\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.0.5'"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#import pandas library for working with tabular data\n",
    "import os\n",
    "os.getcwd()\n",
    "import pandas as pd\n",
    "#import regular expresson (regex)\n",
    "import re\n",
    "#check pandas version\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>spectrum</th>\n",
       "      <th>xcorr</th>\n",
       "      <th>deltacn</th>\n",
       "      <th>expect</th>\n",
       "      <th>ions</th>\n",
       "      <th>peptide</th>\n",
       "      <th>protein</th>\n",
       "      <th>calc_neutral_pep_mass</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.41211.41211.4</td>\n",
       "      <td>8.768</td>\n",
       "      <td>1</td>\n",
       "      <td>1.42E-09</td>\n",
       "      <td>22/156</td>\n",
       "      <td>K.LFADENHLSPAVTAIQIEDIDAEQFRK.N</td>\n",
       "      <td>PMM0035</td>\n",
       "      <td>3069.5407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.32793.32793.4</td>\n",
       "      <td>8.578</td>\n",
       "      <td>1</td>\n",
       "      <td>9.31E-10</td>\n",
       "      <td>18/204</td>\n",
       "      <td>K.LAIDDSSIN[115.03]LDQVDYIN[115.03]AHGTSTTANDK...</td>\n",
       "      <td>PMM1609</td>\n",
       "      <td>3735.7599</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.50751.50751.4</td>\n",
       "      <td>8.469</td>\n",
       "      <td>0.646</td>\n",
       "      <td>3.43E-14</td>\n",
       "      <td>24/156</td>\n",
       "      <td>R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D</td>\n",
       "      <td>PMM1436</td>\n",
       "      <td>2831.4409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.31065.31065.3</td>\n",
       "      <td>8.061</td>\n",
       "      <td>0.561</td>\n",
       "      <td>2.18E-11</td>\n",
       "      <td>17/92</td>\n",
       "      <td>K.IDIEDGGIVTIASHDGAAAEEAQK.I</td>\n",
       "      <td>PMM1191</td>\n",
       "      <td>2409.166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.50278.50278.4</td>\n",
       "      <td>7.949</td>\n",
       "      <td>0.584</td>\n",
       "      <td>3.02E-09</td>\n",
       "      <td>23/156</td>\n",
       "      <td>R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D</td>\n",
       "      <td>PMM1436</td>\n",
       "      <td>2831.4409</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      spectrum  xcorr deltacn    expect  \\\n",
       "0  022016_RAL4_95_MED2_trypsin_1.41211.41211.4  8.768       1  1.42E-09   \n",
       "1  022016_RAL4_95_MED2_trypsin_1.32793.32793.4  8.578       1  9.31E-10   \n",
       "2  022016_RAL4_95_MED2_trypsin_1.50751.50751.4  8.469   0.646  3.43E-14   \n",
       "3  022016_RAL4_95_MED2_trypsin_1.31065.31065.3  8.061   0.561  2.18E-11   \n",
       "4  022016_RAL4_95_MED2_trypsin_1.50278.50278.4  7.949   0.584  3.02E-09   \n",
       "\n",
       "     ions                                            peptide  protein  \\\n",
       "0  22/156                    K.LFADENHLSPAVTAIQIEDIDAEQFRK.N  PMM0035   \n",
       "1  18/204  K.LAIDDSSIN[115.03]LDQVDYIN[115.03]AHGTSTTANDK...  PMM1609   \n",
       "2  24/156    R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D  PMM1436   \n",
       "3   17/92                       K.IDIEDGGIVTIASHDGAAAEEAQK.I  PMM1191   \n",
       "4  23/156    R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D  PMM1436   \n",
       "\n",
       "  calc_neutral_pep_mass  \n",
       "0             3069.5407  \n",
       "1             3735.7599  \n",
       "2             2831.4409  \n",
       "3              2409.166  \n",
       "4             2831.4409  "
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#read the CSV into a datadrame we name 'comet' using the pandas read_csv function\n",
    "comet = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_1_PTMopt_Comet.csv\")\n",
    "\n",
    "\n",
    "#look at the dataframe\n",
    "comet.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The peptide column has the residues before and after the tryptic terminii as well as masses of modified residues (e.g., 160.03 Da for carbamidomethylated cysteine). We want to make new columns with all that information and make a column with only the 'stripped' peptide sequence that's just amino acids - this we can then align against other sequences, for example. \n",
    "\n",
    "Modified residues were allowed for:\n",
    "\n",
    "- fixed carbamodimethylation of cysteine 57.021464 C\n",
    "- varialbe oxidation of methionine: 15.9949 M\n",
    "- variable deamidation of asparagine, glumatine: 0.984016 NQ\n",
    "- variable iron cation on lysine: 54.010565 K\n",
    "- variable methylation of lysine and arginine: 14.015650 KR\n",
    "\n",
    "We'll then write this manipulated dataframe to a new file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>spectrum</th>\n",
       "      <th>xcorr</th>\n",
       "      <th>deltacn</th>\n",
       "      <th>expect</th>\n",
       "      <th>ions</th>\n",
       "      <th>peptide</th>\n",
       "      <th>protein</th>\n",
       "      <th>calc_neutral_pep_mass</th>\n",
       "      <th>L terminus</th>\n",
       "      <th>R terminus</th>\n",
       "      <th>c-carb</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>k-iron</th>\n",
       "      <th>k-meth</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>stripped length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.41211.41211.4</td>\n",
       "      <td>8.768</td>\n",
       "      <td>1</td>\n",
       "      <td>1.42E-09</td>\n",
       "      <td>22/156</td>\n",
       "      <td>K.LFADENHLSPAVTAIQIEDIDAEQFRK.N</td>\n",
       "      <td>PMM0035</td>\n",
       "      <td>3069.5407</td>\n",
       "      <td>K</td>\n",
       "      <td>N</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>LFADENHLSPAVTAIQIEDIDAEQFRK</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.32793.32793.4</td>\n",
       "      <td>8.578</td>\n",
       "      <td>1</td>\n",
       "      <td>9.31E-10</td>\n",
       "      <td>18/204</td>\n",
       "      <td>K.LAIDDSSIN[115.03]LDQVDYIN[115.03]AHGTSTTANDK...</td>\n",
       "      <td>PMM1609</td>\n",
       "      <td>3735.7599</td>\n",
       "      <td>K</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>LAIDDSSINAHGTSTTANDKNETSAIK</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.50751.50751.4</td>\n",
       "      <td>8.469</td>\n",
       "      <td>0.646</td>\n",
       "      <td>3.43E-14</td>\n",
       "      <td>24/156</td>\n",
       "      <td>R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D</td>\n",
       "      <td>PMM1436</td>\n",
       "      <td>2831.4409</td>\n",
       "      <td>R</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SGLQNAASIAGMIVADLPEKK</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.31065.31065.3</td>\n",
       "      <td>8.061</td>\n",
       "      <td>0.561</td>\n",
       "      <td>2.18E-11</td>\n",
       "      <td>17/92</td>\n",
       "      <td>K.IDIEDGGIVTIASHDGAAAEEAQK.I</td>\n",
       "      <td>PMM1191</td>\n",
       "      <td>2409.166</td>\n",
       "      <td>K</td>\n",
       "      <td>I</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>IDIEDGGIVTIASHDGAAAEEAQK</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.50278.50278.4</td>\n",
       "      <td>7.949</td>\n",
       "      <td>0.584</td>\n",
       "      <td>3.02E-09</td>\n",
       "      <td>23/156</td>\n",
       "      <td>R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D</td>\n",
       "      <td>PMM1436</td>\n",
       "      <td>2831.4409</td>\n",
       "      <td>R</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SGLQNAASIAGMIVADLPEKK</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      spectrum  xcorr deltacn    expect  \\\n",
       "0  022016_RAL4_95_MED2_trypsin_1.41211.41211.4  8.768       1  1.42E-09   \n",
       "1  022016_RAL4_95_MED2_trypsin_1.32793.32793.4  8.578       1  9.31E-10   \n",
       "2  022016_RAL4_95_MED2_trypsin_1.50751.50751.4  8.469   0.646  3.43E-14   \n",
       "3  022016_RAL4_95_MED2_trypsin_1.31065.31065.3  8.061   0.561  2.18E-11   \n",
       "4  022016_RAL4_95_MED2_trypsin_1.50278.50278.4  7.949   0.584  3.02E-09   \n",
       "\n",
       "     ions                                            peptide  protein  \\\n",
       "0  22/156                    K.LFADENHLSPAVTAIQIEDIDAEQFRK.N  PMM0035   \n",
       "1  18/204  K.LAIDDSSIN[115.03]LDQVDYIN[115.03]AHGTSTTANDK...  PMM1609   \n",
       "2  24/156    R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D  PMM1436   \n",
       "3   17/92                       K.IDIEDGGIVTIASHDGAAAEEAQK.I  PMM1191   \n",
       "4  23/156    R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D  PMM1436   \n",
       "\n",
       "  calc_neutral_pep_mass L terminus R terminus  c-carb  m-oxid  n-deam  q-deam  \\\n",
       "0             3069.5407          K          N       0       0       0       0   \n",
       "1             3735.7599          K          S       0       0       2       0   \n",
       "2             2831.4409          R          D       1       1       0       0   \n",
       "3              2409.166          K          I       0       0       0       0   \n",
       "4             2831.4409          R          D       1       1       0       0   \n",
       "\n",
       "   k-iron  k-meth  r-meth             stripped peptide  stripped length  \n",
       "0       0       0       0  LFADENHLSPAVTAIQIEDIDAEQFRK               31  \n",
       "1       0       0       0  LAIDDSSINAHGTSTTANDKNETSAIK               55  \n",
       "2       0       0       0        SGLQNAASIAGMIVADLPEKK               47  \n",
       "3       0       0       0     IDIEDGGIVTIASHDGAAAEEAQK               28  \n",
       "4       0       0       0        SGLQNAASIAGMIVADLPEKK               47  "
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use str.strip with indexing by str[0] to add a column with the peptide's left terminus\n",
    "comet['L terminus'] = comet['peptide'].astype(str).str[0]\n",
    "\n",
    "# use str.strip with indexing by str[-1] to add a column with the peptide's left terminus\n",
    "comet['R terminus'] = comet['peptide'].str.strip().str[-1]\n",
    "\n",
    "# use a count function to enumerate the # of ccarbamidomethylated C's in each peptide\n",
    "comet['c-carb'] = comet['peptide'].str.count(\"160.03\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "comet['m-oxid'] = comet['peptide'].str.count(\"147.04\")\n",
    "\n",
    "# use a count function to enumerate the # of deamidated N's in each peptide\n",
    "comet['n-deam'] = comet['peptide'].str.count(\"115.03\")\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "comet['q-deam'] = comet['peptide'].str.count(\"129.04\")\n",
    "\n",
    "# use a count function to enumerate the # of iron adducted K's in each peptide\n",
    "comet['k-iron'] = comet['peptide'].str.count(\"182.11\")\n",
    "\n",
    "# use a count function to enumerate the # of methylated K's in each peptide\n",
    "comet['k-meth'] = comet['peptide'].str.count(\"142.11\")\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "comet['r-meth'] = comet['peptide'].str.count(\"170.12\")\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "comet['stripped peptide'] = comet['peptide'].str[2:].str[:-2].str.replace(r\"\\[.*\\]\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "comet['stripped length'] = comet['peptide'].apply(len)\n",
    "\n",
    "# write modified dataframe to new txt file, same name + 'stripped'\n",
    "comet.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_1_PTMopt_Comet_stripped.csv\")\n",
    "\n",
    "\n",
    "# check out the results\n",
    "comet.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now doing the same manipulation for the duplicate MED2_trypsin injection Comet output:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>spectrum</th>\n",
       "      <th>xcorr</th>\n",
       "      <th>deltacn</th>\n",
       "      <th>expect</th>\n",
       "      <th>ions</th>\n",
       "      <th>peptide</th>\n",
       "      <th>protein</th>\n",
       "      <th>calc_neutral_pep_mass</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_2.32518.32518.4</td>\n",
       "      <td>8.78</td>\n",
       "      <td>1</td>\n",
       "      <td>5.07E-12</td>\n",
       "      <td>12/204</td>\n",
       "      <td>K.LAIDDSSINLDQVDYIN[115.03]AHGTSTTANDKNETSAIK.S</td>\n",
       "      <td>PMM1609</td>\n",
       "      <td>3734.7759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_2.50643.50643.4</td>\n",
       "      <td>8.599</td>\n",
       "      <td>0.583</td>\n",
       "      <td>2.14E-12</td>\n",
       "      <td>22/156</td>\n",
       "      <td>R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D</td>\n",
       "      <td>PMM1436</td>\n",
       "      <td>2831.4409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_2.30823.30823.3</td>\n",
       "      <td>8.163</td>\n",
       "      <td>0.532</td>\n",
       "      <td>1.19E-09</td>\n",
       "      <td>18/92</td>\n",
       "      <td>K.IDIEDGGIVTIASHDGAAAEEAQK.I</td>\n",
       "      <td>PMM1191</td>\n",
       "      <td>2409.166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_2.26453.26453.3</td>\n",
       "      <td>7.881</td>\n",
       "      <td>0.5</td>\n",
       "      <td>4.97E-06</td>\n",
       "      <td>21/68</td>\n",
       "      <td>R.DKGSLVLLDAC[160.03]QSLAHQK.L</td>\n",
       "      <td>PMM0070</td>\n",
       "      <td>1982.0255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_2.30811.30811.3</td>\n",
       "      <td>7.779</td>\n",
       "      <td>0.559</td>\n",
       "      <td>8.76E-09</td>\n",
       "      <td>15/92</td>\n",
       "      <td>K.IDIEDGGIVTIASHDGAAAEEAQK.I</td>\n",
       "      <td>PMM1191</td>\n",
       "      <td>2409.166</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      spectrum  xcorr deltacn    expect  \\\n",
       "0  022016_RAL4_95_MED2_trypsin_2.32518.32518.4   8.78       1  5.07E-12   \n",
       "1  022016_RAL4_95_MED2_trypsin_2.50643.50643.4  8.599   0.583  2.14E-12   \n",
       "2  022016_RAL4_95_MED2_trypsin_2.30823.30823.3  8.163   0.532  1.19E-09   \n",
       "3  022016_RAL4_95_MED2_trypsin_2.26453.26453.3  7.881     0.5  4.97E-06   \n",
       "4  022016_RAL4_95_MED2_trypsin_2.30811.30811.3  7.779   0.559  8.76E-09   \n",
       "\n",
       "     ions                                          peptide  protein  \\\n",
       "0  12/204  K.LAIDDSSINLDQVDYIN[115.03]AHGTSTTANDKNETSAIK.S  PMM1609   \n",
       "1  22/156  R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D  PMM1436   \n",
       "2   18/92                     K.IDIEDGGIVTIASHDGAAAEEAQK.I  PMM1191   \n",
       "3   21/68                   R.DKGSLVLLDAC[160.03]QSLAHQK.L  PMM0070   \n",
       "4   15/92                     K.IDIEDGGIVTIASHDGAAAEEAQK.I  PMM1191   \n",
       "\n",
       "  calc_neutral_pep_mass  \n",
       "0             3734.7759  \n",
       "1             2831.4409  \n",
       "2              2409.166  \n",
       "3             1982.0255  \n",
       "4              2409.166  "
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#read the CSV into a datadrame we name 'comet' using the pandas read_csv function\n",
    "comet = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_2_PTMopt_Comet.csv\")\n",
    "\n",
    "\n",
    "#look at the dataframe\n",
    "comet.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>spectrum</th>\n",
       "      <th>xcorr</th>\n",
       "      <th>deltacn</th>\n",
       "      <th>expect</th>\n",
       "      <th>ions</th>\n",
       "      <th>peptide</th>\n",
       "      <th>protein</th>\n",
       "      <th>calc_neutral_pep_mass</th>\n",
       "      <th>L terminus</th>\n",
       "      <th>R terminus</th>\n",
       "      <th>c-carb</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>k-iron</th>\n",
       "      <th>k-meth</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>stripped length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_2.32518.32518.4</td>\n",
       "      <td>8.78</td>\n",
       "      <td>1</td>\n",
       "      <td>5.07E-12</td>\n",
       "      <td>12/204</td>\n",
       "      <td>K.LAIDDSSINLDQVDYIN[115.03]AHGTSTTANDKNETSAIK.S</td>\n",
       "      <td>PMM1609</td>\n",
       "      <td>3734.7759</td>\n",
       "      <td>K</td>\n",
       "      <td>S</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>LAIDDSSINLDQVDYINAHGTSTTANDKNETSAIK</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_2.50643.50643.4</td>\n",
       "      <td>8.599</td>\n",
       "      <td>0.583</td>\n",
       "      <td>2.14E-12</td>\n",
       "      <td>22/156</td>\n",
       "      <td>R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D</td>\n",
       "      <td>PMM1436</td>\n",
       "      <td>2831.4409</td>\n",
       "      <td>R</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SGLQNAASIAGMIVADLPEKK</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_2.30823.30823.3</td>\n",
       "      <td>8.163</td>\n",
       "      <td>0.532</td>\n",
       "      <td>1.19E-09</td>\n",
       "      <td>18/92</td>\n",
       "      <td>K.IDIEDGGIVTIASHDGAAAEEAQK.I</td>\n",
       "      <td>PMM1191</td>\n",
       "      <td>2409.166</td>\n",
       "      <td>K</td>\n",
       "      <td>I</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>IDIEDGGIVTIASHDGAAAEEAQK</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_2.26453.26453.3</td>\n",
       "      <td>7.881</td>\n",
       "      <td>0.5</td>\n",
       "      <td>4.97E-06</td>\n",
       "      <td>21/68</td>\n",
       "      <td>R.DKGSLVLLDAC[160.03]QSLAHQK.L</td>\n",
       "      <td>PMM0070</td>\n",
       "      <td>1982.0255</td>\n",
       "      <td>R</td>\n",
       "      <td>L</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>DKGSLVLLDACQSLAHQK</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_2.30811.30811.3</td>\n",
       "      <td>7.779</td>\n",
       "      <td>0.559</td>\n",
       "      <td>8.76E-09</td>\n",
       "      <td>15/92</td>\n",
       "      <td>K.IDIEDGGIVTIASHDGAAAEEAQK.I</td>\n",
       "      <td>PMM1191</td>\n",
       "      <td>2409.166</td>\n",
       "      <td>K</td>\n",
       "      <td>I</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>IDIEDGGIVTIASHDGAAAEEAQK</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      spectrum  xcorr deltacn    expect  \\\n",
       "0  022016_RAL4_95_MED2_trypsin_2.32518.32518.4   8.78       1  5.07E-12   \n",
       "1  022016_RAL4_95_MED2_trypsin_2.50643.50643.4  8.599   0.583  2.14E-12   \n",
       "2  022016_RAL4_95_MED2_trypsin_2.30823.30823.3  8.163   0.532  1.19E-09   \n",
       "3  022016_RAL4_95_MED2_trypsin_2.26453.26453.3  7.881     0.5  4.97E-06   \n",
       "4  022016_RAL4_95_MED2_trypsin_2.30811.30811.3  7.779   0.559  8.76E-09   \n",
       "\n",
       "     ions                                          peptide  protein  \\\n",
       "0  12/204  K.LAIDDSSINLDQVDYIN[115.03]AHGTSTTANDKNETSAIK.S  PMM1609   \n",
       "1  22/156  R.SGLQNAASIAGM[147.04]VLTTEC[160.03]IVADLPEKK.D  PMM1436   \n",
       "2   18/92                     K.IDIEDGGIVTIASHDGAAAEEAQK.I  PMM1191   \n",
       "3   21/68                   R.DKGSLVLLDAC[160.03]QSLAHQK.L  PMM0070   \n",
       "4   15/92                     K.IDIEDGGIVTIASHDGAAAEEAQK.I  PMM1191   \n",
       "\n",
       "  calc_neutral_pep_mass L terminus R terminus  c-carb  m-oxid  n-deam  q-deam  \\\n",
       "0             3734.7759          K          S       0       0       1       0   \n",
       "1             2831.4409          R          D       1       1       0       0   \n",
       "2              2409.166          K          I       0       0       0       0   \n",
       "3             1982.0255          R          L       1       0       0       0   \n",
       "4              2409.166          K          I       0       0       0       0   \n",
       "\n",
       "   k-iron  k-meth  r-meth                     stripped peptide  \\\n",
       "0       0       0       0  LAIDDSSINLDQVDYINAHGTSTTANDKNETSAIK   \n",
       "1       0       0       0                SGLQNAASIAGMIVADLPEKK   \n",
       "2       0       0       0             IDIEDGGIVTIASHDGAAAEEAQK   \n",
       "3       0       0       0                   DKGSLVLLDACQSLAHQK   \n",
       "4       0       0       0             IDIEDGGIVTIASHDGAAAEEAQK   \n",
       "\n",
       "   stripped length  \n",
       "0               47  \n",
       "1               47  \n",
       "2               28  \n",
       "3               30  \n",
       "4               28  "
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use str.strip with indexing by str[0] to add a column with the peptide's left terminus\n",
    "comet['L terminus'] = comet['peptide'].astype(str).str[0]\n",
    "\n",
    "# use str.strip with indexing by str[-1] to add a column with the peptide's left terminus\n",
    "comet['R terminus'] = comet['peptide'].str.strip().str[-1]\n",
    "\n",
    "# use a count function to enumerate the # of ccarbamidomethylated C's in each peptide\n",
    "comet['c-carb'] = comet['peptide'].str.count(\"160.03\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "comet['m-oxid'] = comet['peptide'].str.count(\"147.04\")\n",
    "\n",
    "# use a count function to enumerate the # of deamidated N's in each peptide\n",
    "comet['n-deam'] = comet['peptide'].str.count(\"115.03\")\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "comet['q-deam'] = comet['peptide'].str.count(\"129.04\")\n",
    "\n",
    "# use a count function to enumerate the # of iron adducted K's in each peptide\n",
    "comet['k-iron'] = comet['peptide'].str.count(\"182.11\")\n",
    "\n",
    "# use a count function to enumerate the # of methylated K's in each peptide\n",
    "comet['k-meth'] = comet['peptide'].str.count(\"142.11\")\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "comet['r-meth'] = comet['peptide'].str.count(\"170.12\")\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "comet['stripped peptide'] = comet['peptide'].str[2:].str[:-2].str.replace(r\"\\[.*\\]\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "comet['stripped length'] = comet['peptide'].apply(len)\n",
    "\n",
    "# write modified dataframe to new txt file, same name + 'stripped'\n",
    "comet.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_2_PTMopt_Comet_stripped.csv\")\n",
    "\n",
    "\n",
    "# check out the results\n",
    "comet.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we should have two stripped peptide csvs in the directory:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RAL4_MED2_trypsin_1_PTMopt_Comet.csv\r\n",
      "RAL4_MED2_trypsin_1_PTMopt_Comet_stripped.csv\r\n",
      "RAL4_MED2_trypsin_1_PTMopt_Comet.xlsx\r\n",
      "RAL4_MED2_trypsin_1_PTMopt_PepProp90.csv\r\n",
      "RAL4_MED2_trypsin_1_PTMopt_PepProp90.xlsx\r\n",
      "RAL4_MED2_trypsin_2_PTMopt_Comet.csv\r\n",
      "RAL4_MED2_trypsin_2_PTMopt_Comet_stripped.csv\r\n",
      "RAL4_MED2_trypsin_2_PTMopt_Comet.xlsx\r\n",
      "RAL4_MED2_trypsin_2_PTMopt_PepProp90.csv\r\n",
      "RAL4_MED2_trypsin_2_PTMopt_PepProp90.xlsx\r\n"
     ]
    }
   ],
   "source": [
    "ls /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
