{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Manipulation of Peaks de novo results of ETNP 2017 P2 samples LC-MS/MS data using python.\n",
    "\n",
    "Starting with:\n",
    "\n",
    "    Peaks de novo results (.csv) of PTM-optimized database searches > ALC 50%\n",
    "\n",
    "Goal:\n",
    "\n",
    "    Files with stripped (no PTMs) peptide lists and\n",
    "    Columns with #'s of each modification in every sequence\n",
    "    Column with stripped peptide lengths (# amino acids)\n",
    "    \n",
    "### To use for a different file:\n",
    "\n",
    "#### 1. Change the input file name in *IN 4*\n",
    "#### 2. Use 'find + replace' (Esc + F) to replace the running # (e.g., 233) for another\n",
    "#### 3. Update the NAAF factor calculated in *IN 6* into *IN 7*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We don't have technical duplicates here, sadly, unlike the MED4 Pro samples. I exported PeaksDN search results CSVs into my ETNP 2017 git repo:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt\n"
     ]
    }
   ],
   "source": [
    "cd /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;34m231\u001b[0m/  \u001b[01;34m233\u001b[0m/  \u001b[01;34m243\u001b[0m/  \u001b[01;34m273\u001b[0m/  \u001b[01;34m278\u001b[0m/  \u001b[01;34m378\u001b[0m/\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ETNP_SKQ17_DENOVO_168_278-265m-trap_DN50_15ppm.csv\r\n",
      "ETNP_SKQ17_DENOVO_168_278-265m-trap_DN50_15ppm.xml\r\n"
     ]
    }
   ],
   "source": [
    "ls 278"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.0.5'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# LIBRARIES\n",
    "#import pandas library for working with tabular data\n",
    "import os\n",
    "os.getcwd()\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "from scipy.stats import kde\n",
    "#import regular expresson (regex)\n",
    "import re\n",
    "#check pandas version\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 278: 265 m +P filtered on 0.3 um GF-75"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# redundant Peaks peptides >50% ALC in dataframe 10488\n",
      "# redundant Peaks peptides >80% ALC in dataframe 1482\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Fraction</th>\n",
       "      <th>Scan</th>\n",
       "      <th>Source File</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Tag Length</th>\n",
       "      <th>ALC (%)</th>\n",
       "      <th>length</th>\n",
       "      <th>m/z</th>\n",
       "      <th>z</th>\n",
       "      <th>RT</th>\n",
       "      <th>Area</th>\n",
       "      <th>Mass</th>\n",
       "      <th>ppm</th>\n",
       "      <th>PTM</th>\n",
       "      <th>local confidence (%)</th>\n",
       "      <th>tag (&gt;=0%)</th>\n",
       "      <th>mode</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>6054</td>\n",
       "      <td>20181004_4-19_265m_topP_278_90min.raw</td>\n",
       "      <td>WLEEDK</td>\n",
       "      <td>6</td>\n",
       "      <td>98</td>\n",
       "      <td>6</td>\n",
       "      <td>410.1956</td>\n",
       "      <td>2</td>\n",
       "      <td>21.80</td>\n",
       "      <td>920000.0</td>\n",
       "      <td>818.3810</td>\n",
       "      <td>-5.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>97 99 100 99 96 97</td>\n",
       "      <td>WLEEDK</td>\n",
       "      <td>CID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>14925</td>\n",
       "      <td>20181004_4-19_265m_topP_278_90min.raw</td>\n",
       "      <td>M(+15.99)TAEELKK</td>\n",
       "      <td>8</td>\n",
       "      <td>98</td>\n",
       "      <td>8</td>\n",
       "      <td>483.2513</td>\n",
       "      <td>2</td>\n",
       "      <td>41.26</td>\n",
       "      <td>21600000.0</td>\n",
       "      <td>964.4899</td>\n",
       "      <td>-2.0</td>\n",
       "      <td>Oxidation (M)</td>\n",
       "      <td>96 98 98 99 99 98 99 97</td>\n",
       "      <td>M(+15.99)TAEELKK</td>\n",
       "      <td>CID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>13192</td>\n",
       "      <td>20181004_4-19_265m_topP_278_90min.raw</td>\n",
       "      <td>SC(+57.02)DWPK</td>\n",
       "      <td>6</td>\n",
       "      <td>97</td>\n",
       "      <td>6</td>\n",
       "      <td>396.6714</td>\n",
       "      <td>2</td>\n",
       "      <td>37.44</td>\n",
       "      <td>4220000.0</td>\n",
       "      <td>791.3272</td>\n",
       "      <td>1.3</td>\n",
       "      <td>Carbamidomethylation</td>\n",
       "      <td>98 99 99 97 96 97</td>\n",
       "      <td>SC(+57.02)DWPK</td>\n",
       "      <td>CID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>16407</td>\n",
       "      <td>20181004_4-19_265m_topP_278_90min.raw</td>\n",
       "      <td>WNQ(+.98)QFLNAAK</td>\n",
       "      <td>10</td>\n",
       "      <td>97</td>\n",
       "      <td>10</td>\n",
       "      <td>610.7994</td>\n",
       "      <td>2</td>\n",
       "      <td>44.52</td>\n",
       "      <td>40600000.0</td>\n",
       "      <td>1219.5986</td>\n",
       "      <td>-11.7</td>\n",
       "      <td>Deamidation (NQ)</td>\n",
       "      <td>95 98 99 96 98 98 96 99 99 98</td>\n",
       "      <td>WNQ(+.98)QFLNAAK</td>\n",
       "      <td>CID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>6599</td>\n",
       "      <td>20181004_4-19_265m_topP_278_90min.raw</td>\n",
       "      <td>SEDVYAK</td>\n",
       "      <td>7</td>\n",
       "      <td>97</td>\n",
       "      <td>7</td>\n",
       "      <td>406.1959</td>\n",
       "      <td>2</td>\n",
       "      <td>22.94</td>\n",
       "      <td>831000.0</td>\n",
       "      <td>810.3759</td>\n",
       "      <td>1.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>95 99 99 96 97 98 98</td>\n",
       "      <td>SEDVYAK</td>\n",
       "      <td>CID</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Fraction   Scan                            Source File           Peptide  \\\n",
       "0         5   6054  20181004_4-19_265m_topP_278_90min.raw            WLEEDK   \n",
       "1         5  14925  20181004_4-19_265m_topP_278_90min.raw  M(+15.99)TAEELKK   \n",
       "2         5  13192  20181004_4-19_265m_topP_278_90min.raw    SC(+57.02)DWPK   \n",
       "3         5  16407  20181004_4-19_265m_topP_278_90min.raw  WNQ(+.98)QFLNAAK   \n",
       "4         5   6599  20181004_4-19_265m_topP_278_90min.raw           SEDVYAK   \n",
       "\n",
       "   Tag Length  ALC (%)  length       m/z  z     RT        Area       Mass  \\\n",
       "0           6       98       6  410.1956  2  21.80    920000.0   818.3810   \n",
       "1           8       98       8  483.2513  2  41.26  21600000.0   964.4899   \n",
       "2           6       97       6  396.6714  2  37.44   4220000.0   791.3272   \n",
       "3          10       97      10  610.7994  2  44.52  40600000.0  1219.5986   \n",
       "4           7       97       7  406.1959  2  22.94    831000.0   810.3759   \n",
       "\n",
       "    ppm                   PTM           local confidence (%)  \\\n",
       "0  -5.3                   NaN             97 99 100 99 96 97   \n",
       "1  -2.0         Oxidation (M)        96 98 98 99 99 98 99 97   \n",
       "2   1.3  Carbamidomethylation              98 99 99 97 96 97   \n",
       "3 -11.7      Deamidation (NQ)  95 98 99 96 98 98 96 99 99 98   \n",
       "4   1.6                   NaN           95 99 99 96 97 98 98   \n",
       "\n",
       "         tag (>=0%) mode  \n",
       "0            WLEEDK  CID  \n",
       "1  M(+15.99)TAEELKK  CID  \n",
       "2    SC(+57.02)DWPK  CID  \n",
       "3  WNQ(+.98)QFLNAAK  CID  \n",
       "4           SEDVYAK  CID  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read the CSV into a dataframe using the pandas read_csv function\n",
    "peaks278_50 = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/278/ETNP_SKQ17_DENOVO_168_278-265m-trap_DN50_15ppm.csv\")\n",
    "\n",
    "# also make a dataframe but keep only peptides >80% ALC\n",
    "peaks278_80 = peaks278_50.loc[peaks278_50['ALC (%)'] >= 80].copy()\n",
    "\n",
    "# how many de novo sequence candidates >50% ALC?\n",
    "print(\"# redundant Peaks peptides >50% ALC in dataframe\", len(peaks278_50))\n",
    "# how many de novo sequence candidates >80ALC?\n",
    "print(\"# redundant Peaks peptides >80% ALC in dataframe\", len(peaks278_80))\n",
    "\n",
    "#look at the dataframe\n",
    "peaks278_50.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Fraction</th>\n",
       "      <th>Scan</th>\n",
       "      <th>Source File</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Tag Length</th>\n",
       "      <th>ALC (%)</th>\n",
       "      <th>length</th>\n",
       "      <th>m/z</th>\n",
       "      <th>z</th>\n",
       "      <th>RT</th>\n",
       "      <th>...</th>\n",
       "      <th>c-carb</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>k-hydr</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>stripped length</th>\n",
       "      <th>ptm-total</th>\n",
       "      <th>NAAF num.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>6054</td>\n",
       "      <td>20181004_4-19_265m_topP_278_90min.raw</td>\n",
       "      <td>WLEEDK</td>\n",
       "      <td>6</td>\n",
       "      <td>98</td>\n",
       "      <td>6</td>\n",
       "      <td>410.1956</td>\n",
       "      <td>2</td>\n",
       "      <td>21.80</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>WLEEDK</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1.533333e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>14925</td>\n",
       "      <td>20181004_4-19_265m_topP_278_90min.raw</td>\n",
       "      <td>M(+15.99)TAEELKK</td>\n",
       "      <td>8</td>\n",
       "      <td>98</td>\n",
       "      <td>8</td>\n",
       "      <td>483.2513</td>\n",
       "      <td>2</td>\n",
       "      <td>41.26</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>MTAEELKK</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>2.700000e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>13192</td>\n",
       "      <td>20181004_4-19_265m_topP_278_90min.raw</td>\n",
       "      <td>SC(+57.02)DWPK</td>\n",
       "      <td>6</td>\n",
       "      <td>97</td>\n",
       "      <td>6</td>\n",
       "      <td>396.6714</td>\n",
       "      <td>2</td>\n",
       "      <td>37.44</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SCDWPK</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>7.033333e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>16407</td>\n",
       "      <td>20181004_4-19_265m_topP_278_90min.raw</td>\n",
       "      <td>WNQ(+.98)QFLNAAK</td>\n",
       "      <td>10</td>\n",
       "      <td>97</td>\n",
       "      <td>10</td>\n",
       "      <td>610.7994</td>\n",
       "      <td>2</td>\n",
       "      <td>44.52</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>WNQQFLNAAK</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>4.060000e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>6599</td>\n",
       "      <td>20181004_4-19_265m_topP_278_90min.raw</td>\n",
       "      <td>SEDVYAK</td>\n",
       "      <td>7</td>\n",
       "      <td>97</td>\n",
       "      <td>7</td>\n",
       "      <td>406.1959</td>\n",
       "      <td>2</td>\n",
       "      <td>22.94</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SEDVYAK</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1.187143e+05</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Fraction   Scan                            Source File           Peptide  \\\n",
       "0         5   6054  20181004_4-19_265m_topP_278_90min.raw            WLEEDK   \n",
       "1         5  14925  20181004_4-19_265m_topP_278_90min.raw  M(+15.99)TAEELKK   \n",
       "2         5  13192  20181004_4-19_265m_topP_278_90min.raw    SC(+57.02)DWPK   \n",
       "3         5  16407  20181004_4-19_265m_topP_278_90min.raw  WNQ(+.98)QFLNAAK   \n",
       "4         5   6599  20181004_4-19_265m_topP_278_90min.raw           SEDVYAK   \n",
       "\n",
       "   Tag Length  ALC (%)  length       m/z  z     RT  ...  c-carb  m-oxid  \\\n",
       "0           6       98       6  410.1956  2  21.80  ...       0       0   \n",
       "1           8       98       8  483.2513  2  41.26  ...       0       1   \n",
       "2           6       97       6  396.6714  2  37.44  ...       1       0   \n",
       "3          10       97      10  610.7994  2  44.52  ...       0       0   \n",
       "4           7       97       7  406.1959  2  22.94  ...       0       0   \n",
       "\n",
       "   n-deam q-deam k-hydr r-meth stripped peptide  stripped length  ptm-total  \\\n",
       "0       0      0      0      0           WLEEDK                6          0   \n",
       "1       0      0      0      0         MTAEELKK                8          1   \n",
       "2       0      0      0      0           SCDWPK                6          1   \n",
       "3       0      1      0      0       WNQQFLNAAK               10          1   \n",
       "4       0      0      0      0          SEDVYAK                7          0   \n",
       "\n",
       "      NAAF num.  \n",
       "0  1.533333e+05  \n",
       "1  2.700000e+06  \n",
       "2  7.033333e+05  \n",
       "3  4.060000e+06  \n",
       "4  1.187143e+05  \n",
       "\n",
       "[5 rows x 47 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use a count function to enumerate the # of A's (alanines) in each peptide\n",
    "peaks278_50['A'] = peaks278_50['Peptide'].str.count(\"A\")\n",
    "\n",
    "# use a count function to enumerate the # of C's (cysteines) in each peptide\n",
    "peaks278_50['C'] = peaks278_50['Peptide'].str.count(\"C\")\n",
    "\n",
    "# use a count function to enumerate the # of D's (aspartic acids) in each peptide\n",
    "peaks278_50['D'] = peaks278_50['Peptide'].str.count(\"D\")\n",
    "\n",
    "# use a count function to enumerate the # of E's (glutamic acids) in each peptide\n",
    "peaks278_50['E'] = peaks278_50['Peptide'].str.count(\"E\")\n",
    "\n",
    "# use a count function to enumerate the # of F's (phenylalanines) in each peptide\n",
    "peaks278_50['F'] = peaks278_50['Peptide'].str.count(\"F\")\n",
    "\n",
    "# use a count function to enumerate the # of G's (glycines) in each peptide\n",
    "peaks278_50['G'] = peaks278_50['Peptide'].str.count(\"G\")\n",
    "\n",
    "# use a count function to enumerate the # of H's (histidines) in each peptide\n",
    "peaks278_50['H'] = peaks278_50['Peptide'].str.count(\"H\")\n",
    "\n",
    "# use a count function to enumerate the # of I's (isoleucines) in each peptide\n",
    "# in peaks278_50 output, there will be no isoleucines (they're lumped in with leucines)\n",
    "peaks278_50['I'] = peaks278_50['Peptide'].str.count(\"I\")\n",
    "\n",
    "# use a count function to enumerate the # of K's (lysines) in each peptide\n",
    "peaks278_50['K'] = peaks278_50['Peptide'].str.count(\"K\")\n",
    "\n",
    "# use a count function to enumerate the # of L's (leucines) in each peptide\n",
    "# also these include the isoleucines\n",
    "peaks278_50['L'] = peaks278_50['Peptide'].str.count(\"L\")\n",
    "\n",
    "# use a count function to enumerate the # of M's (methionines) in each peptide\n",
    "peaks278_50['M'] = peaks278_50['Peptide'].str.count(\"M\")\n",
    "\n",
    "# use a count function to enumerate the # of N's (asparagines) in each peptide\n",
    "peaks278_50['N'] = peaks278_50['Peptide'].str.count(\"N\")\n",
    "\n",
    "# use a count function to enumerate the # of P's ([prolines]) in each peptide\n",
    "peaks278_50['P'] = peaks278_50['Peptide'].str.count(\"P\")\n",
    "\n",
    "# use a count function to enumerate the # of Q's (glutamines) in each peptide\n",
    "peaks278_50['Q'] = peaks278_50['Peptide'].str.count(\"Q\")\n",
    "\n",
    "# use a count function to enumerate the # of R's (arginines) in each peptide\n",
    "peaks278_50['R'] = peaks278_50['Peptide'].str.count(\"R\")\n",
    "\n",
    "# use a count function to enumerate the # of S's (serines) in each peptide\n",
    "peaks278_50['S'] = peaks278_50['Peptide'].str.count(\"S\")\n",
    "\n",
    "# use a count function to enumerate the # of T's (threonines) in each peptide\n",
    "peaks278_50['T'] = peaks278_50['Peptide'].str.count(\"T\")\n",
    "\n",
    "# use a count function to enumerate the # of V's (valines) in each peptide\n",
    "peaks278_50['V'] = peaks278_50['Peptide'].str.count(\"V\")\n",
    "\n",
    "# use a count function to enumerate the # of W's (tryptophans) in each peptide\n",
    "peaks278_50['W'] = peaks278_50['Peptide'].str.count(\"W\")\n",
    "\n",
    "# use a count function to enumerate the # of Y's (tyrosines) in each peptide\n",
    "peaks278_50['Y'] = peaks278_50['Peptide'].str.count(\"Y\")\n",
    "\n",
    "# use a count function to enumerate the # of carbamidomethylated C's in each peptide\n",
    "peaks278_50['c-carb'] = peaks278_50['Peptide'].str.count(\"57.02\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "peaks278_50['m-oxid'] = peaks278_50['Peptide'].apply(lambda x: x.count('M(+15.99)'))\n",
    "\n",
    "# use a lamba function to enumerate the # of deamidated N's in each peptide\n",
    "peaks278_50['n-deam'] = peaks278_50['Peptide'].apply(lambda x: x.count('N(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "peaks278_50['q-deam'] = peaks278_50['Peptide'].apply(lambda x: x.count('Q(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of hydroxylated K's in each peptide\n",
    "peaks278_50['k-hydr'] = peaks278_50['Peptide'].apply(lambda x: x.count('K(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of hydroxylated P's in each peptide\n",
    "# no more P hydroxylation in final searches\n",
    "# peaks278_50['p-hydr'] = peaks278_50['Peptide'].apply(lambda x: x.count('P(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "peaks278_50['r-meth'] = peaks278_50['Peptide'].apply(lambda x: x.count('R(+14.02)'))\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "peaks278_50['stripped peptide'] = peaks278_50['Peptide'].str.replace(r\"\\(.*\\)\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "peaks278_50['stripped length'] = peaks278_50['stripped peptide'].apply(len)\n",
    "\n",
    "# total the number of modifications in sequence\n",
    "peaks278_50['ptm-total'] = peaks278_50['c-carb'] + peaks278_50['m-oxid'] + peaks278_50['n-deam'] \\\n",
    "+ peaks278_50['q-deam'] + peaks278_50['k-hydr'] + peaks278_50['r-meth']\n",
    "\n",
    "# calculate NAAF numerator for each peptide k\n",
    "peaks278_50['NAAF num.'] = peaks278_50['Area'] / peaks278_50['stripped length']\n",
    "\n",
    "# write modified dataframe to new csv file\n",
    "peaks278_50.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/278/ETNP-SKQ17-278-265m-trap_PTMopt_15ppm_DN50.csv\")\n",
    "\n",
    "\n",
    "# check out the results\n",
    "peaks278_50.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>K</th>\n",
       "      <th>L</th>\n",
       "      <th>...</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>Total area</th>\n",
       "      <th>Total length</th>\n",
       "      <th>% C w/ carb</th>\n",
       "      <th>% M w/ oxid</th>\n",
       "      <th>% N w/ deam</th>\n",
       "      <th>% Q w/ deam</th>\n",
       "      <th>% K w/ hydr</th>\n",
       "      <th>% R w/ meth</th>\n",
       "      <th>NAAF denom.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>5564</td>\n",
       "      <td>1919</td>\n",
       "      <td>5229</td>\n",
       "      <td>6154</td>\n",
       "      <td>2787</td>\n",
       "      <td>4374</td>\n",
       "      <td>2690</td>\n",
       "      <td>0</td>\n",
       "      <td>9598</td>\n",
       "      <td>9028</td>\n",
       "      <td>...</td>\n",
       "      <td>2975</td>\n",
       "      <td>1.785712e+10</td>\n",
       "      <td>80491</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.428571</td>\n",
       "      <td>0.236156</td>\n",
       "      <td>0.026813</td>\n",
       "      <td>0.209419</td>\n",
       "      <td>0.479607</td>\n",
       "      <td>221852.426966</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 A     C     D     E     F     G     H  I     K     L  ...  \\\n",
       "sample total  5564  1919  5229  6154  2787  4374  2690  0  9598  9028  ...   \n",
       "\n",
       "              r-meth    Total area  Total length  % C w/ carb  % M w/ oxid  \\\n",
       "sample total    2975  1.785712e+10         80491          1.0     0.428571   \n",
       "\n",
       "              % N w/ deam  % Q w/ deam  % K w/ hydr  % R w/ meth  \\\n",
       "sample total     0.236156     0.026813     0.209419     0.479607   \n",
       "\n",
       "                NAAF denom.  \n",
       "sample total  221852.426966  \n",
       "\n",
       "[1 rows x 35 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of certain columns in the modified\n",
    "# peptide dataframe above (for >50% ALC)\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'A': peaks278_50['A'].sum(),\n",
    "        'C': peaks278_50['C'].sum(),\n",
    "        'D': peaks278_50['D'].sum(),\n",
    "        'E': peaks278_50['E'].sum(),\n",
    "        'F': peaks278_50['F'].sum(),\n",
    "        'G': peaks278_50['G'].sum(),\n",
    "        'H': peaks278_50['H'].sum(),\n",
    "        'I': peaks278_50['I'].sum(),\n",
    "        'K': peaks278_50['K'].sum(),\n",
    "        'L': peaks278_50['L'].sum(),\n",
    "        'M': peaks278_50['M'].sum(),\n",
    "        'N': peaks278_50['N'].sum(),\n",
    "        'P': peaks278_50['P'].sum(),\n",
    "        'Q': peaks278_50['Q'].sum(),\n",
    "        'R': peaks278_50['R'].sum(),\n",
    "        'S': peaks278_50['S'].sum(),\n",
    "        'T': peaks278_50['T'].sum(),\n",
    "        'V': peaks278_50['V'].sum(),\n",
    "        'W': peaks278_50['W'].sum(),\n",
    "        'Y': peaks278_50['Y'].sum(),\n",
    "        'c-carb': peaks278_50['c-carb'].sum(),\n",
    "        'm-oxid': peaks278_50['m-oxid'].sum(),\n",
    "        'n-deam': peaks278_50['n-deam'].sum(),\n",
    "        'q-deam': peaks278_50['q-deam'].sum(),\n",
    "        'k-hydr': peaks278_50['k-hydr'].sum(),\n",
    "        'r-meth': peaks278_50['r-meth'].sum(),\n",
    "        'Total area': peaks278_50['Area'].sum(),\n",
    "        'Total length': peaks278_50['stripped length'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaks278_50 = pd.DataFrame(data, columns=['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L',\\\n",
    "                                               'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', \\\n",
    "                                               'm-oxid', 'n-deam', 'q-deam', 'k-hydr', 'r-meth', \\\n",
    "                                               'Total area', 'Total length'], index=index)\n",
    "\n",
    "# calculate percentage of C's with carb (should be 1.0)\n",
    "totalpeaks278_50['% C w/ carb'] = totalpeaks278_50['c-carb'] / totalpeaks278_50['C'] \n",
    "\n",
    "# calculate percentage of M's that are oxidized\n",
    "totalpeaks278_50['% M w/ oxid'] = totalpeaks278_50['m-oxid'] / totalpeaks278_50['M'] \n",
    "\n",
    "# calculate percentage of N's that are deamidated\n",
    "totalpeaks278_50['% N w/ deam'] = totalpeaks278_50['n-deam'] / totalpeaks278_50['N'] \n",
    "\n",
    "# calculate percentage of Q's that are deamidated\n",
    "totalpeaks278_50['% Q w/ deam'] = totalpeaks278_50['q-deam'] / totalpeaks278_50['Q'] \n",
    "\n",
    "# calculate percentage of K's that are hydroxylated\n",
    "totalpeaks278_50['% K w/ hydr'] = totalpeaks278_50['k-hydr'] / totalpeaks278_50['K'] \n",
    "\n",
    "# calculate percentage of P's that are hydroxylated\n",
    "#totalpeaks278_50['% P w/ hydr'] = totalpeaks278_50['p-hydr'] / totalpeaks278_50['K'] \n",
    "\n",
    "# calculate percentage of R's that are methylated\n",
    "totalpeaks278_50['% R w/ meth'] = totalpeaks278_50['r-meth'] / totalpeaks278_50['R'] \n",
    "\n",
    "# calculate NAAF denominator for all peptides in dataset i\n",
    "totalpeaks278_50['NAAF denom.'] = totalpeaks278_50['Total area'] / totalpeaks278_50['Total length']\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "totalpeaks278_50.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/278/ETNP-SKQ17-278-265m-trap_PTMopt_15ppm_DN50_totals.csv\")\n",
    "\n",
    "totalpeaks278_50.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>NAAF factor</th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>...</th>\n",
       "      <th>T-NAAF50</th>\n",
       "      <th>V-NAAF50</th>\n",
       "      <th>W-NAAF50</th>\n",
       "      <th>Y-NAAF50</th>\n",
       "      <th>ccarb-NAAF50</th>\n",
       "      <th>moxid-NAAF50</th>\n",
       "      <th>ndeam-NAAF50</th>\n",
       "      <th>qdeam-NAAF50</th>\n",
       "      <th>khydr-NAAF50</th>\n",
       "      <th>rmeth-NAAF50</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WLEEDK</td>\n",
       "      <td>0.691150</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.691150</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MTAEELKK</td>\n",
       "      <td>12.170252</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>12.170252</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>12.170252</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SCDWPK</td>\n",
       "      <td>3.170276</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.170276</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.170276</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>WNQQFLNAAK</td>\n",
       "      <td>18.300453</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>18.300453</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18.300453</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SEDVYAK</td>\n",
       "      <td>0.535105</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.535105</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.535105</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  stripped peptide  NAAF factor  A  C  D  E  F  G  H  I  ...   T-NAAF50  \\\n",
       "0           WLEEDK     0.691150  0  0  1  2  0  0  0  0  ...   0.000000   \n",
       "1         MTAEELKK    12.170252  1  0  0  2  0  0  0  0  ...  12.170252   \n",
       "2           SCDWPK     3.170276  0  1  1  0  0  0  0  0  ...   0.000000   \n",
       "3       WNQQFLNAAK    18.300453  2  0  0  0  1  0  0  0  ...   0.000000   \n",
       "4          SEDVYAK     0.535105  1  0  1  1  0  0  0  0  ...   0.000000   \n",
       "\n",
       "   V-NAAF50   W-NAAF50  Y-NAAF50  ccarb-NAAF50  moxid-NAAF50  ndeam-NAAF50  \\\n",
       "0  0.000000   0.691150  0.000000      0.000000      0.000000           0.0   \n",
       "1  0.000000   0.000000  0.000000      0.000000     12.170252           0.0   \n",
       "2  0.000000   3.170276  0.000000      3.170276      0.000000           0.0   \n",
       "3  0.000000  18.300453  0.000000      0.000000      0.000000           0.0   \n",
       "4  0.535105   0.000000  0.535105      0.000000      0.000000           0.0   \n",
       "\n",
       "   qdeam-NAAF50  khydr-NAAF50  rmeth-NAAF50  \n",
       "0      0.000000           0.0           0.0  \n",
       "1      0.000000           0.0           0.0  \n",
       "2      0.000000           0.0           0.0  \n",
       "3     18.300453           0.0           0.0  \n",
       "4      0.000000           0.0           0.0  \n",
       "\n",
       "[5 rows x 54 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use the calculated NAAF factor (in totalpeaks278 dataframe, above) to caluclate the NAAF \n",
    "# NAAF: normalized normalized area abundance factor\n",
    "\n",
    "# don't have to worry here about DECOY hits messing with Area totals\n",
    "# but we will with Comet results\n",
    "\n",
    "NAAF50 = 221852.426966\n",
    "\n",
    "# use NAAF >50% ALC to get NAAF\n",
    "peaks278_50['NAAF factor'] = (peaks278_50['NAAF num.'])/NAAF50\n",
    "\n",
    "# make a dataframe that contains only what we need: sequences, AAs, PTMs\n",
    "peaks278_AA50 = peaks278_50[['stripped peptide', 'NAAF factor', 'A', 'C', 'D', 'E', 'F', 'G', 'H', \\\n",
    "                             'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', \\\n",
    "                             'm-oxid', 'n-deam', 'q-deam', 'k-hydr', 'r-meth']].copy()\n",
    "\n",
    "# multiply the NAAF50 factor by the AAs to normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaks278_AA50['A-NAAF50'] = peaks278_AA50['A'] * peaks278_50['NAAF factor']\n",
    "peaks278_AA50['C-NAAF50'] = peaks278_AA50['C'] * peaks278_50['NAAF factor']\n",
    "peaks278_AA50['D-NAAF50'] = peaks278_AA50['D'] * peaks278_50['NAAF factor']\n",
    "peaks278_AA50['E-NAAF50'] = peaks278_AA50['E'] * peaks278_50['NAAF factor']\n",
    "peaks278_AA50['F-NAAF50'] = peaks278_AA50['F'] * peaks278_50['NAAF factor']\n",
    "peaks278_AA50['G-NAAF50'] = peaks278_AA50['G'] * peaks278_50['NAAF factor']\n",
    "peaks278_AA50['H-NAAF50'] = peaks278_AA50['H'] * peaks278_50['NAAF factor']\n",
    "peaks278_AA50['I-NAAF50'] = peaks278_AA50['I'] * peaks278_50['NAAF factor']\n",
    "peaks278_AA50['K-NAAF50'] = peaks278_AA50['K'] * peaks278_50['NAAF factor']\n",
    "peaks278_AA50['L-NAAF50'] = peaks278_AA50['L'] * peaks278_50['NAAF factor']\n",
    "peaks278_AA50['M-NAAF50'] = peaks278_AA50['M'] * peaks278_50['NAAF factor']\n",
    "peaks278_AA50['N-NAAF50'] = peaks278_AA50['N'] * peaks278_50['NAAF factor']\n",
    "peaks278_AA50['P-NAAF50'] = peaks278_AA50['P'] * peaks278_50['NAAF factor']\n",
    "peaks278_AA50['Q-NAAF50'] = peaks278_AA50['Q'] * peaks278_50['NAAF factor']\n",
    "peaks278_AA50['R-NAAF50'] = peaks278_AA50['R'] * peaks278_50['NAAF factor']\n",
    "peaks278_AA50['S-NAAF50'] = peaks278_AA50['S'] * peaks278_50['NAAF factor']\n",
    "peaks278_AA50['T-NAAF50'] = peaks278_AA50['T'] * peaks278_50['NAAF factor']\n",
    "peaks278_AA50['V-NAAF50'] = peaks278_AA50['V'] * peaks278_50['NAAF factor']\n",
    "peaks278_AA50['W-NAAF50'] = peaks278_AA50['W'] * peaks278_50['NAAF factor']\n",
    "peaks278_AA50['Y-NAAF50'] = peaks278_AA50['Y'] * peaks278_50['NAAF factor']\n",
    "\n",
    "# multiply the NAAF50 factor by the PTMs normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaks278_AA50['ccarb-NAAF50'] = peaks278_AA50['c-carb'] * peaks278_AA50['NAAF factor']\n",
    "peaks278_AA50['moxid-NAAF50'] = peaks278_AA50['m-oxid'] * peaks278_AA50['NAAF factor']\n",
    "peaks278_AA50['ndeam-NAAF50'] = peaks278_AA50['n-deam'] * peaks278_AA50['NAAF factor']\n",
    "peaks278_AA50['qdeam-NAAF50'] = peaks278_AA50['q-deam'] * peaks278_AA50['NAAF factor']\n",
    "peaks278_AA50['khydr-NAAF50'] = peaks278_AA50['k-hydr'] * peaks278_AA50['NAAF factor']\n",
    "peaks278_AA50['rmeth-NAAF50'] = peaks278_AA50['r-meth'] * peaks278_AA50['NAAF factor']\n",
    "\n",
    "# write the dataframe to a new csv\n",
    "peaks278_AA50.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/278/ETNP-SKQ17-278-265m-trap_PTMopt_15ppm_DN50_NAAF.csv\")\n",
    "\n",
    "peaks278_AA50.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NAAF</th>\n",
       "      <th>A-NAAF</th>\n",
       "      <th>C-NAAF</th>\n",
       "      <th>D-NAAF</th>\n",
       "      <th>E-NAAF</th>\n",
       "      <th>F-NAAF</th>\n",
       "      <th>G-NAAF</th>\n",
       "      <th>H-NAAF</th>\n",
       "      <th>I-NAAF</th>\n",
       "      <th>K-NAAF</th>\n",
       "      <th>...</th>\n",
       "      <th>Q-deam-NAAF</th>\n",
       "      <th>K-hydr-NAAF</th>\n",
       "      <th>R-meth-NAAF</th>\n",
       "      <th>% C w/ carb. NAAF</th>\n",
       "      <th>% M w/ oxid. NAAF</th>\n",
       "      <th>% N w/ deam. NAAF</th>\n",
       "      <th>% Q w/ deam. NAAF</th>\n",
       "      <th>% K w/ hydr. NAAF</th>\n",
       "      <th>% R w/ meth. NAAF</th>\n",
       "      <th>NAAF check</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>12698.861516</td>\n",
       "      <td>7206.423438</td>\n",
       "      <td>1977.929736</td>\n",
       "      <td>5780.432065</td>\n",
       "      <td>6519.167339</td>\n",
       "      <td>2847.316116</td>\n",
       "      <td>5307.700617</td>\n",
       "      <td>2910.267226</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10063.583942</td>\n",
       "      <td>...</td>\n",
       "      <td>130.427599</td>\n",
       "      <td>3681.705649</td>\n",
       "      <td>4244.306896</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.481779</td>\n",
       "      <td>0.312771</td>\n",
       "      <td>0.056863</td>\n",
       "      <td>0.365844</td>\n",
       "      <td>0.530163</td>\n",
       "      <td>0.05724</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      NAAF       A-NAAF       C-NAAF       D-NAAF  \\\n",
       "sample total  12698.861516  7206.423438  1977.929736  5780.432065   \n",
       "\n",
       "                   E-NAAF       F-NAAF       G-NAAF       H-NAAF  I-NAAF  \\\n",
       "sample total  6519.167339  2847.316116  5307.700617  2910.267226     0.0   \n",
       "\n",
       "                    K-NAAF  ...  Q-deam-NAAF  K-hydr-NAAF  R-meth-NAAF  \\\n",
       "sample total  10063.583942  ...   130.427599  3681.705649  4244.306896   \n",
       "\n",
       "              % C w/ carb. NAAF  % M w/ oxid. NAAF  % N w/ deam. NAAF  \\\n",
       "sample total                1.0           0.481779           0.312771   \n",
       "\n",
       "              % Q w/ deam. NAAF  % K w/ hydr. NAAF  % R w/ meth. NAAF  \\\n",
       "sample total           0.056863           0.365844           0.530163   \n",
       "\n",
       "              NAAF check  \n",
       "sample total     0.05724  \n",
       "\n",
       "[1 rows x 34 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of NAAF normalized AAs for comet278 PeaksDN results\n",
    "# also contains the sums of the NAAF-corrected PTMs occurances for each affected residue\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'NAAF': peaks278_AA50['NAAF factor'].sum(),\n",
    "        'A-NAAF': peaks278_AA50['A-NAAF50'].sum(),\n",
    "        'C-NAAF': peaks278_AA50['C-NAAF50'].sum(),\n",
    "        'D-NAAF': peaks278_AA50['D-NAAF50'].sum(),\n",
    "        'E-NAAF': peaks278_AA50['E-NAAF50'].sum(),\n",
    "        'F-NAAF': peaks278_AA50['F-NAAF50'].sum(),\n",
    "        'G-NAAF': peaks278_AA50['G-NAAF50'].sum(),\n",
    "        'H-NAAF': peaks278_AA50['H-NAAF50'].sum(),\n",
    "        'I-NAAF': peaks278_AA50['I-NAAF50'].sum(),\n",
    "        'K-NAAF': peaks278_AA50['K-NAAF50'].sum(),\n",
    "        'L-NAAF': peaks278_AA50['L-NAAF50'].sum(),\n",
    "        'M-NAAF': peaks278_AA50['M-NAAF50'].sum(),\n",
    "        'N-NAAF': peaks278_AA50['N-NAAF50'].sum(),\n",
    "        'P-NAAF': peaks278_AA50['P-NAAF50'].sum(),\n",
    "        'Q-NAAF': peaks278_AA50['Q-NAAF50'].sum(),\n",
    "        'R-NAAF': peaks278_AA50['R-NAAF50'].sum(),\n",
    "        'S-NAAF': peaks278_AA50['S-NAAF50'].sum(),\n",
    "        'T-NAAF': peaks278_AA50['T-NAAF50'].sum(),\n",
    "        'V-NAAF': peaks278_AA50['V-NAAF50'].sum(),\n",
    "        'W-NAAF': peaks278_AA50['W-NAAF50'].sum(),\n",
    "        'Y-NAAF': peaks278_AA50['Y-NAAF50'].sum(),\n",
    "        'C-carb-NAAF': peaks278_AA50['ccarb-NAAF50'].sum(),\n",
    "        'M-oxid-NAAF': peaks278_AA50['moxid-NAAF50'].sum(),\n",
    "        'N-deam-NAAF': peaks278_AA50['ndeam-NAAF50'].sum(),\n",
    "        'Q-deam-NAAF': peaks278_AA50['qdeam-NAAF50'].sum(),\n",
    "        'K-hydr-NAAF': peaks278_AA50['khydr-NAAF50'].sum(),\n",
    "        'R-meth-NAAF': peaks278_AA50['rmeth-NAAF50'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaks278_AA50 = pd.DataFrame(data, columns=['NAAF', 'A-NAAF', 'C-NAAF', 'D-NAAF', 'E-NAAF', 'F-NAAF', \\\n",
    "                                                   'G-NAAF', 'H-NAAF', 'I-NAAF', 'K-NAAF', 'L-NAAF', 'M-NAAF', \\\n",
    "                                                   'N-NAAF', 'P-NAAF', 'Q-NAAF', 'R-NAAF', 'S-NAAF', \\\n",
    "                                                   'T-NAAF', 'V-NAAF', 'W-NAAF', 'Y-NAAF', 'C-carb-NAAF', \\\n",
    "                                                   'M-oxid-NAAF', 'N-deam-NAAF', 'Q-deam-NAAF', 'K-hydr-NAAF',\\\n",
    "                                                   'R-meth-NAAF'], index=index)\n",
    "\n",
    "# calculate the NAAF-corrected % modified C, M, N, Q, K, P, and Rs\n",
    "\n",
    "\n",
    "totalpeaks278_AA50['% C w/ carb. NAAF'] = totalpeaks278_AA50['C-carb-NAAF'] / totalpeaks278_AA50['C-NAAF']\n",
    "totalpeaks278_AA50['% M w/ oxid. NAAF'] = totalpeaks278_AA50['M-oxid-NAAF'] / totalpeaks278_AA50['M-NAAF']\n",
    "totalpeaks278_AA50['% N w/ deam. NAAF'] = totalpeaks278_AA50['N-deam-NAAF'] / totalpeaks278_AA50['N-NAAF']\n",
    "totalpeaks278_AA50['% Q w/ deam. NAAF'] = totalpeaks278_AA50['Q-deam-NAAF'] / totalpeaks278_AA50['Q-NAAF']\n",
    "totalpeaks278_AA50['% K w/ hydr. NAAF'] = totalpeaks278_AA50['K-hydr-NAAF'] / totalpeaks278_AA50['K-NAAF']\n",
    "totalpeaks278_AA50['% R w/ meth. NAAF'] = totalpeaks278_AA50['R-meth-NAAF'] / totalpeaks278_AA50['R-NAAF']\n",
    "\n",
    "# calculate NAAF summed numerator over denominator (in above cell) for all peptides in dataset i: a check\n",
    "totalpeaks278_AA50['NAAF check'] = totalpeaks278_AA50['NAAF'] / 221852.426966\n",
    "\n",
    "# write modified dataframe to new txt file, same name + totals\n",
    "totalpeaks278_AA50.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/278/ETNP-SKQ17-278-265m-trap_PTMopt_DN50_NAAF_totals.csv\")\n",
    "\n",
    "totalpeaks278_AA50.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Same process but for de novo peptide >80 % ALC:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Fraction</th>\n",
       "      <th>Scan</th>\n",
       "      <th>Source File</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Tag Length</th>\n",
       "      <th>ALC (%)</th>\n",
       "      <th>length</th>\n",
       "      <th>m/z</th>\n",
       "      <th>z</th>\n",
       "      <th>RT</th>\n",
       "      <th>...</th>\n",
       "      <th>c-carb</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>k-hydr</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>stripped length</th>\n",
       "      <th>ptm-total</th>\n",
       "      <th>NAAF num.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5</td>\n",
       "      <td>6054</td>\n",
       "      <td>20181004_4-19_265m_topP_278_90min.raw</td>\n",
       "      <td>WLEEDK</td>\n",
       "      <td>6</td>\n",
       "      <td>98</td>\n",
       "      <td>6</td>\n",
       "      <td>410.1956</td>\n",
       "      <td>2</td>\n",
       "      <td>21.80</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>WLEEDK</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>1.533333e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5</td>\n",
       "      <td>14925</td>\n",
       "      <td>20181004_4-19_265m_topP_278_90min.raw</td>\n",
       "      <td>M(+15.99)TAEELKK</td>\n",
       "      <td>8</td>\n",
       "      <td>98</td>\n",
       "      <td>8</td>\n",
       "      <td>483.2513</td>\n",
       "      <td>2</td>\n",
       "      <td>41.26</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>MTAEELKK</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>2.700000e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>13192</td>\n",
       "      <td>20181004_4-19_265m_topP_278_90min.raw</td>\n",
       "      <td>SC(+57.02)DWPK</td>\n",
       "      <td>6</td>\n",
       "      <td>97</td>\n",
       "      <td>6</td>\n",
       "      <td>396.6714</td>\n",
       "      <td>2</td>\n",
       "      <td>37.44</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SCDWPK</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>7.033333e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>16407</td>\n",
       "      <td>20181004_4-19_265m_topP_278_90min.raw</td>\n",
       "      <td>WNQ(+.98)QFLNAAK</td>\n",
       "      <td>10</td>\n",
       "      <td>97</td>\n",
       "      <td>10</td>\n",
       "      <td>610.7994</td>\n",
       "      <td>2</td>\n",
       "      <td>44.52</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>WNQQFLNAAK</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>4.060000e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>6599</td>\n",
       "      <td>20181004_4-19_265m_topP_278_90min.raw</td>\n",
       "      <td>SEDVYAK</td>\n",
       "      <td>7</td>\n",
       "      <td>97</td>\n",
       "      <td>7</td>\n",
       "      <td>406.1959</td>\n",
       "      <td>2</td>\n",
       "      <td>22.94</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SEDVYAK</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1.187143e+05</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Fraction   Scan                            Source File           Peptide  \\\n",
       "0         5   6054  20181004_4-19_265m_topP_278_90min.raw            WLEEDK   \n",
       "1         5  14925  20181004_4-19_265m_topP_278_90min.raw  M(+15.99)TAEELKK   \n",
       "2         5  13192  20181004_4-19_265m_topP_278_90min.raw    SC(+57.02)DWPK   \n",
       "3         5  16407  20181004_4-19_265m_topP_278_90min.raw  WNQ(+.98)QFLNAAK   \n",
       "4         5   6599  20181004_4-19_265m_topP_278_90min.raw           SEDVYAK   \n",
       "\n",
       "   Tag Length  ALC (%)  length       m/z  z     RT  ...  c-carb  m-oxid  \\\n",
       "0           6       98       6  410.1956  2  21.80  ...       0       0   \n",
       "1           8       98       8  483.2513  2  41.26  ...       0       1   \n",
       "2           6       97       6  396.6714  2  37.44  ...       1       0   \n",
       "3          10       97      10  610.7994  2  44.52  ...       0       0   \n",
       "4           7       97       7  406.1959  2  22.94  ...       0       0   \n",
       "\n",
       "   n-deam q-deam k-hydr r-meth stripped peptide  stripped length  ptm-total  \\\n",
       "0       0      0      0      0           WLEEDK                6          0   \n",
       "1       0      0      0      0         MTAEELKK                8          1   \n",
       "2       0      0      0      0           SCDWPK                6          1   \n",
       "3       0      1      0      0       WNQQFLNAAK               10          1   \n",
       "4       0      0      0      0          SEDVYAK                7          0   \n",
       "\n",
       "      NAAF num.  \n",
       "0  1.533333e+05  \n",
       "1  2.700000e+06  \n",
       "2  7.033333e+05  \n",
       "3  4.060000e+06  \n",
       "4  1.187143e+05  \n",
       "\n",
       "[5 rows x 47 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use a count function to enumerate the # of A's (alanines) in each peptide\n",
    "peaks278_80['A'] = peaks278_80['Peptide'].str.count(\"A\")\n",
    "\n",
    "# use a count function to enumerate the # of C's (cysteines) in each peptide\n",
    "peaks278_80['C'] = peaks278_80['Peptide'].str.count(\"C\")\n",
    "\n",
    "# use a count function to enumerate the # of D's (aspartic acids) in each peptide\n",
    "peaks278_80['D'] = peaks278_80['Peptide'].str.count(\"D\")\n",
    "\n",
    "# use a count function to enumerate the # of E's (glutamic acids) in each peptide\n",
    "peaks278_80['E'] = peaks278_80['Peptide'].str.count(\"E\")\n",
    "\n",
    "# use a count function to enumerate the # of F's (phenylalanines) in each peptide\n",
    "peaks278_80['F'] = peaks278_80['Peptide'].str.count(\"F\")\n",
    "\n",
    "# use a count function to enumerate the # of G's (glycines) in each peptide\n",
    "peaks278_80['G'] = peaks278_80['Peptide'].str.count(\"G\")\n",
    "\n",
    "# use a count function to enumerate the # of H's (histidines) in each peptide\n",
    "peaks278_80['H'] = peaks278_80['Peptide'].str.count(\"H\")\n",
    "\n",
    "# use a count function to enumerate the # of I's (isoleucines) in each peptide\n",
    "# in peaks278_80 output, there will be no isoleucines (they're lumped in with leucines)\n",
    "peaks278_80['I'] = peaks278_80['Peptide'].str.count(\"I\")\n",
    "\n",
    "# use a count function to enumerate the # of K's (lysines) in each peptide\n",
    "peaks278_80['K'] = peaks278_80['Peptide'].str.count(\"K\")\n",
    "\n",
    "# use a count function to enumerate the # of L's (leucines) in each peptide\n",
    "# also these include the isoleucines\n",
    "peaks278_80['L'] = peaks278_80['Peptide'].str.count(\"L\")\n",
    "\n",
    "# use a count function to enumerate the # of M's (methionines) in each peptide\n",
    "peaks278_80['M'] = peaks278_80['Peptide'].str.count(\"M\")\n",
    "\n",
    "# use a count function to enumerate the # of N's (asparagines) in each peptide\n",
    "peaks278_80['N'] = peaks278_80['Peptide'].str.count(\"N\")\n",
    "\n",
    "# use a count function to enumerate the # of P's ([prolines]) in each peptide\n",
    "peaks278_80['P'] = peaks278_80['Peptide'].str.count(\"P\")\n",
    "\n",
    "# use a count function to enumerate the # of Q's (glutamines) in each peptide\n",
    "peaks278_80['Q'] = peaks278_80['Peptide'].str.count(\"Q\")\n",
    "\n",
    "# use a count function to enumerate the # of R's (arginines) in each peptide\n",
    "peaks278_80['R'] = peaks278_80['Peptide'].str.count(\"R\")\n",
    "\n",
    "# use a count function to enumerate the # of S's (serines) in each peptide\n",
    "peaks278_80['S'] = peaks278_80['Peptide'].str.count(\"S\")\n",
    "\n",
    "# use a count function to enumerate the # of T's (threonines) in each peptide\n",
    "peaks278_80['T'] = peaks278_80['Peptide'].str.count(\"T\")\n",
    "\n",
    "# use a count function to enumerate the # of V's (valines) in each peptide\n",
    "peaks278_80['V'] = peaks278_80['Peptide'].str.count(\"V\")\n",
    "\n",
    "# use a count function to enumerate the # of W's (tryptophans) in each peptide\n",
    "peaks278_80['W'] = peaks278_80['Peptide'].str.count(\"W\")\n",
    "\n",
    "# use a count function to enumerate the # of Y's (tyrosines) in each peptide\n",
    "peaks278_80['Y'] = peaks278_80['Peptide'].str.count(\"Y\")\n",
    "\n",
    "# use a count function to enumerate the # of carbamidomethylated C's in each peptide\n",
    "peaks278_80['c-carb'] = peaks278_80['Peptide'].str.count(\"57.02\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "peaks278_80['m-oxid'] = peaks278_80['Peptide'].apply(lambda x: x.count('M(+15.99)'))\n",
    "\n",
    "# use a lamba function to enumerate the # of deamidated N's in each peptide\n",
    "peaks278_80['n-deam'] = peaks278_80['Peptide'].apply(lambda x: x.count('N(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "peaks278_80['q-deam'] = peaks278_80['Peptide'].apply(lambda x: x.count('Q(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of hydroxylated K's in each peptide\n",
    "peaks278_80['k-hydr'] = peaks278_80['Peptide'].apply(lambda x: x.count('K(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of hydroxylated P's in each peptide\n",
    "#peaks278_80['p-hydr'] = peaks278_80['Peptide'].apply(lambda x: x.count('P(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "peaks278_80['r-meth'] = peaks278_80['Peptide'].apply(lambda x: x.count('R(+14.02)'))\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "peaks278_80['stripped peptide'] = peaks278_80['Peptide'].str.replace(r\"\\(.*\\)\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "peaks278_80['stripped length'] = peaks278_80['stripped peptide'].apply(len)\n",
    "\n",
    "# total the number of modifications in sequence\n",
    "peaks278_80['ptm-total'] = peaks278_80['c-carb'] + peaks278_80['m-oxid'] + peaks278_80['n-deam'] + \\\n",
    "peaks278_80['q-deam'] + peaks278_80['k-hydr'] + peaks278_80['r-meth']\n",
    "\n",
    "# calculate NAAF numerator for each peptide k\n",
    "peaks278_80['NAAF num.'] = peaks278_80['Area'] / peaks278_80['stripped length']\n",
    "\n",
    "# write modified dataframe to new csv file\n",
    "peaks278_80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/278/ETNP-SKQ17-278-265m-trap_PTMopt_15ppm_DN80.csv\")\n",
    "\n",
    "# check out the results\n",
    "peaks278_80.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>K</th>\n",
       "      <th>L</th>\n",
       "      <th>...</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>Total area</th>\n",
       "      <th>Total length</th>\n",
       "      <th>% C w/ carb</th>\n",
       "      <th>% M w/ oxid</th>\n",
       "      <th>% N w/ deam</th>\n",
       "      <th>% Q w/ deam</th>\n",
       "      <th>% K w/ hydr</th>\n",
       "      <th>% R w/ meth</th>\n",
       "      <th>NAAF denom.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>887</td>\n",
       "      <td>169</td>\n",
       "      <td>785</td>\n",
       "      <td>1286</td>\n",
       "      <td>385</td>\n",
       "      <td>557</td>\n",
       "      <td>167</td>\n",
       "      <td>0</td>\n",
       "      <td>1173</td>\n",
       "      <td>1654</td>\n",
       "      <td>...</td>\n",
       "      <td>255</td>\n",
       "      <td>5.529521e+09</td>\n",
       "      <td>11548</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.483696</td>\n",
       "      <td>0.331516</td>\n",
       "      <td>0.071053</td>\n",
       "      <td>0.094629</td>\n",
       "      <td>0.328608</td>\n",
       "      <td>478829.326697</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                A    C    D     E    F    G    H  I     K     L  ...  r-meth  \\\n",
       "sample total  887  169  785  1286  385  557  167  0  1173  1654  ...     255   \n",
       "\n",
       "                Total area  Total length  % C w/ carb  % M w/ oxid  \\\n",
       "sample total  5.529521e+09         11548          1.0     0.483696   \n",
       "\n",
       "              % N w/ deam  % Q w/ deam  % K w/ hydr  % R w/ meth  \\\n",
       "sample total     0.331516     0.071053     0.094629     0.328608   \n",
       "\n",
       "                NAAF denom.  \n",
       "sample total  478829.326697  \n",
       "\n",
       "[1 rows x 35 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of certain columns in modified \n",
    "#peptide dataframe above (for >80% ALC)\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'A': peaks278_80['A'].sum(),\n",
    "        'C': peaks278_80['C'].sum(),\n",
    "        'D': peaks278_80['D'].sum(),\n",
    "        'E': peaks278_80['E'].sum(),\n",
    "        'F': peaks278_80['F'].sum(),\n",
    "        'G': peaks278_80['G'].sum(),\n",
    "        'H': peaks278_80['H'].sum(),\n",
    "        'I': peaks278_80['I'].sum(),\n",
    "        'K': peaks278_80['K'].sum(),\n",
    "        'L': peaks278_80['L'].sum(),\n",
    "        'M': peaks278_80['M'].sum(),\n",
    "        'N': peaks278_80['N'].sum(),\n",
    "        'P': peaks278_80['P'].sum(),\n",
    "        'Q': peaks278_80['Q'].sum(),\n",
    "        'R': peaks278_80['R'].sum(),\n",
    "        'S': peaks278_80['S'].sum(),\n",
    "        'T': peaks278_80['T'].sum(),\n",
    "        'V': peaks278_80['V'].sum(),\n",
    "        'W': peaks278_80['W'].sum(),\n",
    "        'Y': peaks278_80['Y'].sum(),\n",
    "        'c-carb': peaks278_80['c-carb'].sum(),\n",
    "        'm-oxid': peaks278_80['m-oxid'].sum(),\n",
    "        'n-deam': peaks278_80['n-deam'].sum(),\n",
    "        'q-deam': peaks278_80['q-deam'].sum(),\n",
    "        'k-hydr': peaks278_80['k-hydr'].sum(),\n",
    "        'r-meth': peaks278_80['r-meth'].sum(),\n",
    "        'Total area': peaks278_80['Area'].sum(),\n",
    "        'Total length': peaks278_80['stripped length'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaks278_80 = pd.DataFrame(data, columns=['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M',\\\n",
    "                                               'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb',\\\n",
    "                                               'm-oxid', 'n-deam', 'q-deam', 'k-hydr', 'r-meth',\\\n",
    "                                               'Total area', 'Total length'], index=index)\n",
    "\n",
    "# calculate percentage of C's with carb (should be 1.0)\n",
    "totalpeaks278_80['% C w/ carb'] = totalpeaks278_80['c-carb'] / totalpeaks278_80['C'] \n",
    "\n",
    "# calculate percentage of M's that are oxidized\n",
    "totalpeaks278_80['% M w/ oxid'] = totalpeaks278_80['m-oxid'] / totalpeaks278_80['M'] \n",
    "\n",
    "# calculate percentage of N's that are deamidated\n",
    "totalpeaks278_80['% N w/ deam'] = totalpeaks278_80['n-deam'] / totalpeaks278_80['N'] \n",
    "\n",
    "# calculate percentage of Q's that are deamidated\n",
    "totalpeaks278_80['% Q w/ deam'] = totalpeaks278_80['q-deam'] / totalpeaks278_80['Q'] \n",
    "\n",
    "# calculate percentage of K's that are hydroxylated\n",
    "totalpeaks278_80['% K w/ hydr'] = totalpeaks278_80['k-hydr'] / totalpeaks278_80['K'] \n",
    "\n",
    "# calculate percentage of P's that are hydroxylated\n",
    "#totalpeaks278_80['% P w/ hydr'] = totalpeaks278_80['p-hydr'] / totalpeaks278_80['K'] \n",
    "\n",
    "# calculate percentage of R's that are methylated\n",
    "totalpeaks278_80['% R w/ meth'] = totalpeaks278_80['r-meth'] / totalpeaks278_80['R'] \n",
    "\n",
    "# calculate NAAF denominator for all peptides in dataset i\n",
    "totalpeaks278_80['NAAF denom.'] = totalpeaks278_80['Total area'] / totalpeaks278_80['Total length']\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "totalpeaks278_80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/278/ETNP-SKQ17-278-265m-trap_PTMopt_15ppm_DN80_totals.csv\")\n",
    "\n",
    "totalpeaks278_80.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>NAAF factor</th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>...</th>\n",
       "      <th>T-NAAF80</th>\n",
       "      <th>V-NAAF80</th>\n",
       "      <th>W-NAAF80</th>\n",
       "      <th>Y-NAAF80</th>\n",
       "      <th>ccarb-NAAF80</th>\n",
       "      <th>moxid-NAAF80</th>\n",
       "      <th>ndeam-NAAF80</th>\n",
       "      <th>qdeam-NAAF80</th>\n",
       "      <th>khydr-NAAF80</th>\n",
       "      <th>rmeth-NAAF80</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>WLEEDK</td>\n",
       "      <td>0.320225</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.320225</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MTAEELKK</td>\n",
       "      <td>5.638752</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>5.638752</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>5.638752</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SCDWPK</td>\n",
       "      <td>1.468860</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.468860</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.46886</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>WNQQFLNAAK</td>\n",
       "      <td>8.479013</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>8.479013</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.479013</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SEDVYAK</td>\n",
       "      <td>0.247926</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.247926</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.247926</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  stripped peptide  NAAF factor  A  C  D  E  F  G  H  I  ...  T-NAAF80  \\\n",
       "0           WLEEDK     0.320225  0  0  1  2  0  0  0  0  ...  0.000000   \n",
       "1         MTAEELKK     5.638752  1  0  0  2  0  0  0  0  ...  5.638752   \n",
       "2           SCDWPK     1.468860  0  1  1  0  0  0  0  0  ...  0.000000   \n",
       "3       WNQQFLNAAK     8.479013  2  0  0  0  1  0  0  0  ...  0.000000   \n",
       "4          SEDVYAK     0.247926  1  0  1  1  0  0  0  0  ...  0.000000   \n",
       "\n",
       "   V-NAAF80  W-NAAF80  Y-NAAF80  ccarb-NAAF80  moxid-NAAF80  ndeam-NAAF80  \\\n",
       "0  0.000000  0.320225  0.000000       0.00000      0.000000           0.0   \n",
       "1  0.000000  0.000000  0.000000       0.00000      5.638752           0.0   \n",
       "2  0.000000  1.468860  0.000000       1.46886      0.000000           0.0   \n",
       "3  0.000000  8.479013  0.000000       0.00000      0.000000           0.0   \n",
       "4  0.247926  0.000000  0.247926       0.00000      0.000000           0.0   \n",
       "\n",
       "   qdeam-NAAF80  khydr-NAAF80  rmeth-NAAF80  \n",
       "0      0.000000           0.0           0.0  \n",
       "1      0.000000           0.0           0.0  \n",
       "2      0.000000           0.0           0.0  \n",
       "3      8.479013           0.0           0.0  \n",
       "4      0.000000           0.0           0.0  \n",
       "\n",
       "[5 rows x 54 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use the calculated NAAF factor (in totalpeaks278 dataframe, above) to caluclate the NAAF \n",
    "# NAAF: normalized normalized area abundance factor\n",
    "\n",
    "# don't have to worry here about DECOY hits messing with Area totals\n",
    "# but we will with Comet results\n",
    "\n",
    "NAAF80 = 478829.326697\n",
    "\n",
    "# use NAAF >80% ALC to get NAAF\n",
    "peaks278_80['NAAF factor'] = (peaks278_80['NAAF num.'])/NAAF80\n",
    "\n",
    "# make a dataframe that contains only what we need: sequences, AAs, PTMs\n",
    "peaks278_AA80 = peaks278_80[['stripped peptide', 'NAAF factor', 'A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', \\\n",
    "                             'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', 'm-oxid', \\\n",
    "                             'n-deam', 'q-deam', 'k-hydr', 'r-meth']].copy()\n",
    "\n",
    "# multiply the NAAF80 factor by the AAs to normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaks278_AA80['A-NAAF80'] = peaks278_AA80['A'] * peaks278_80['NAAF factor']\n",
    "peaks278_AA80['C-NAAF80'] = peaks278_AA80['C'] * peaks278_80['NAAF factor']\n",
    "peaks278_AA80['D-NAAF80'] = peaks278_AA80['D'] * peaks278_80['NAAF factor']\n",
    "peaks278_AA80['E-NAAF80'] = peaks278_AA80['E'] * peaks278_80['NAAF factor']\n",
    "peaks278_AA80['F-NAAF80'] = peaks278_AA80['F'] * peaks278_80['NAAF factor']\n",
    "peaks278_AA80['G-NAAF80'] = peaks278_AA80['G'] * peaks278_80['NAAF factor']\n",
    "peaks278_AA80['H-NAAF80'] = peaks278_AA80['H'] * peaks278_80['NAAF factor']\n",
    "peaks278_AA80['I-NAAF80'] = peaks278_AA80['I'] * peaks278_80['NAAF factor']\n",
    "peaks278_AA80['K-NAAF80'] = peaks278_AA80['K'] * peaks278_80['NAAF factor']\n",
    "peaks278_AA80['L-NAAF80'] = peaks278_AA80['L'] * peaks278_80['NAAF factor']\n",
    "peaks278_AA80['M-NAAF80'] = peaks278_AA80['M'] * peaks278_80['NAAF factor']\n",
    "peaks278_AA80['N-NAAF80'] = peaks278_AA80['N'] * peaks278_80['NAAF factor']\n",
    "peaks278_AA80['P-NAAF80'] = peaks278_AA80['P'] * peaks278_80['NAAF factor']\n",
    "peaks278_AA80['Q-NAAF80'] = peaks278_AA80['Q'] * peaks278_80['NAAF factor']\n",
    "peaks278_AA80['R-NAAF80'] = peaks278_AA80['R'] * peaks278_80['NAAF factor']\n",
    "peaks278_AA80['S-NAAF80'] = peaks278_AA80['S'] * peaks278_80['NAAF factor']\n",
    "peaks278_AA80['T-NAAF80'] = peaks278_AA80['T'] * peaks278_80['NAAF factor']\n",
    "peaks278_AA80['V-NAAF80'] = peaks278_AA80['V'] * peaks278_80['NAAF factor']\n",
    "peaks278_AA80['W-NAAF80'] = peaks278_AA80['W'] * peaks278_80['NAAF factor']\n",
    "peaks278_AA80['Y-NAAF80'] = peaks278_AA80['Y'] * peaks278_80['NAAF factor']\n",
    "\n",
    "# multiply the NAAF80 factor by the PTMs normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaks278_AA80['ccarb-NAAF80'] = peaks278_AA80['c-carb'] * peaks278_AA80['NAAF factor']\n",
    "peaks278_AA80['moxid-NAAF80'] = peaks278_AA80['m-oxid'] * peaks278_AA80['NAAF factor']\n",
    "peaks278_AA80['ndeam-NAAF80'] = peaks278_AA80['n-deam'] * peaks278_AA80['NAAF factor']\n",
    "peaks278_AA80['qdeam-NAAF80'] = peaks278_AA80['q-deam'] * peaks278_AA80['NAAF factor']\n",
    "peaks278_AA80['khydr-NAAF80'] = peaks278_AA80['k-hydr'] * peaks278_AA80['NAAF factor']\n",
    "peaks278_AA80['rmeth-NAAF80'] = peaks278_AA80['r-meth'] * peaks278_AA80['NAAF factor']\n",
    "\n",
    "# write the dataframe to a new csv\n",
    "peaks278_AA80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/278/ETNP-SKQ17-278-265m-trap_PTMopt_15ppm_DN80_NAAF.csv\")\n",
    "\n",
    "peaks278_AA80.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NAAF</th>\n",
       "      <th>A-NAAF</th>\n",
       "      <th>C-NAAF</th>\n",
       "      <th>D-NAAF</th>\n",
       "      <th>E-NAAF</th>\n",
       "      <th>F-NAAF</th>\n",
       "      <th>G-NAAF</th>\n",
       "      <th>H-NAAF</th>\n",
       "      <th>I-NAAF</th>\n",
       "      <th>K-NAAF</th>\n",
       "      <th>...</th>\n",
       "      <th>Q-deam-NAAF</th>\n",
       "      <th>K-hydr-NAAF</th>\n",
       "      <th>R-meth-NAAF</th>\n",
       "      <th>% C w/ carb. NAAF</th>\n",
       "      <th>% M w/ oxid. NAAF</th>\n",
       "      <th>% N w/ deam. NAAF</th>\n",
       "      <th>% Q w/ deam. NAAF</th>\n",
       "      <th>% K w/ hydr. NAAF</th>\n",
       "      <th>% R w/ meth. NAAF</th>\n",
       "      <th>NAAF check</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>1529.318041</td>\n",
       "      <td>1082.98612</td>\n",
       "      <td>129.771677</td>\n",
       "      <td>625.052273</td>\n",
       "      <td>1109.793016</td>\n",
       "      <td>280.219158</td>\n",
       "      <td>592.252388</td>\n",
       "      <td>125.425359</td>\n",
       "      <td>0.0</td>\n",
       "      <td>957.8277</td>\n",
       "      <td>...</td>\n",
       "      <td>22.696391</td>\n",
       "      <td>178.812916</td>\n",
       "      <td>355.837166</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.668331</td>\n",
       "      <td>0.377249</td>\n",
       "      <td>0.088905</td>\n",
       "      <td>0.186686</td>\n",
       "      <td>0.374069</td>\n",
       "      <td>0.003194</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     NAAF      A-NAAF      C-NAAF      D-NAAF       E-NAAF  \\\n",
       "sample total  1529.318041  1082.98612  129.771677  625.052273  1109.793016   \n",
       "\n",
       "                  F-NAAF      G-NAAF      H-NAAF  I-NAAF    K-NAAF  ...  \\\n",
       "sample total  280.219158  592.252388  125.425359     0.0  957.8277  ...   \n",
       "\n",
       "              Q-deam-NAAF  K-hydr-NAAF  R-meth-NAAF  % C w/ carb. NAAF  \\\n",
       "sample total    22.696391   178.812916   355.837166                1.0   \n",
       "\n",
       "              % M w/ oxid. NAAF  % N w/ deam. NAAF  % Q w/ deam. NAAF  \\\n",
       "sample total           0.668331           0.377249           0.088905   \n",
       "\n",
       "              % K w/ hydr. NAAF  % R w/ meth. NAAF  NAAF check  \n",
       "sample total           0.186686           0.374069    0.003194  \n",
       "\n",
       "[1 rows x 34 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of NAAF normalized AAs for comet278 PeaksDN results\n",
    "# also contains the sums of the NAAF-corrected PTMs occurances for each affected residue\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'NAAF': peaks278_AA80['NAAF factor'].sum(),\n",
    "        'A-NAAF': peaks278_AA80['A-NAAF80'].sum(),\n",
    "        'C-NAAF': peaks278_AA80['C-NAAF80'].sum(),\n",
    "        'D-NAAF': peaks278_AA80['D-NAAF80'].sum(),\n",
    "        'E-NAAF': peaks278_AA80['E-NAAF80'].sum(),\n",
    "        'F-NAAF': peaks278_AA80['F-NAAF80'].sum(),\n",
    "        'G-NAAF': peaks278_AA80['G-NAAF80'].sum(),\n",
    "        'H-NAAF': peaks278_AA80['H-NAAF80'].sum(),\n",
    "        'I-NAAF': peaks278_AA80['I-NAAF80'].sum(),\n",
    "        'K-NAAF': peaks278_AA80['K-NAAF80'].sum(),\n",
    "        'L-NAAF': peaks278_AA80['L-NAAF80'].sum(),\n",
    "        'M-NAAF': peaks278_AA80['M-NAAF80'].sum(),\n",
    "        'N-NAAF': peaks278_AA80['N-NAAF80'].sum(),\n",
    "        'P-NAAF': peaks278_AA80['P-NAAF80'].sum(),\n",
    "        'Q-NAAF': peaks278_AA80['Q-NAAF80'].sum(),\n",
    "        'R-NAAF': peaks278_AA80['R-NAAF80'].sum(),\n",
    "        'S-NAAF': peaks278_AA80['S-NAAF80'].sum(),\n",
    "        'T-NAAF': peaks278_AA80['T-NAAF80'].sum(),\n",
    "        'V-NAAF': peaks278_AA80['V-NAAF80'].sum(),\n",
    "        'W-NAAF': peaks278_AA80['W-NAAF80'].sum(),\n",
    "        'Y-NAAF': peaks278_AA80['Y-NAAF80'].sum(),\n",
    "        'C-carb-NAAF': peaks278_AA80['ccarb-NAAF80'].sum(),\n",
    "        'M-oxid-NAAF': peaks278_AA80['moxid-NAAF80'].sum(),\n",
    "        'N-deam-NAAF': peaks278_AA80['ndeam-NAAF80'].sum(),\n",
    "        'Q-deam-NAAF': peaks278_AA80['qdeam-NAAF80'].sum(),\n",
    "        'K-hydr-NAAF': peaks278_AA80['khydr-NAAF80'].sum(),\n",
    "        'R-meth-NAAF': peaks278_AA80['rmeth-NAAF80'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaks278_AA80 = pd.DataFrame(data, columns=['NAAF', 'A-NAAF', 'C-NAAF', 'D-NAAF', 'E-NAAF', 'F-NAAF', \\\n",
    "                                                   'G-NAAF', 'H-NAAF', 'I-NAAF', 'K-NAAF', 'L-NAAF', 'M-NAAF', \\\n",
    "                                                   'N-NAAF', 'P-NAAF', 'Q-NAAF', 'R-NAAF', 'S-NAAF', \\\n",
    "                                                   'T-NAAF', 'V-NAAF', 'W-NAAF', 'Y-NAAF', 'C-carb-NAAF', \\\n",
    "                                                   'M-oxid-NAAF', 'N-deam-NAAF', 'Q-deam-NAAF', 'K-hydr-NAAF',\\\n",
    "                                                   'R-meth-NAAF'], index=index)\n",
    "\n",
    "# calculate the NAAF-corrected % modified C, M, N, Q, K, P, and Rs\n",
    "\n",
    "\n",
    "totalpeaks278_AA80['% C w/ carb. NAAF'] = totalpeaks278_AA80['C-carb-NAAF'] / totalpeaks278_AA80['C-NAAF']\n",
    "totalpeaks278_AA80['% M w/ oxid. NAAF'] = totalpeaks278_AA80['M-oxid-NAAF'] / totalpeaks278_AA80['M-NAAF']\n",
    "totalpeaks278_AA80['% N w/ deam. NAAF'] = totalpeaks278_AA80['N-deam-NAAF'] / totalpeaks278_AA80['N-NAAF']\n",
    "totalpeaks278_AA80['% Q w/ deam. NAAF'] = totalpeaks278_AA80['Q-deam-NAAF'] / totalpeaks278_AA80['Q-NAAF']\n",
    "totalpeaks278_AA80['% K w/ hydr. NAAF'] = totalpeaks278_AA80['K-hydr-NAAF'] / totalpeaks278_AA80['K-NAAF']\n",
    "totalpeaks278_AA80['% R w/ meth. NAAF'] = totalpeaks278_AA80['R-meth-NAAF'] / totalpeaks278_AA80['R-NAAF']\n",
    "\n",
    "# calculate NAAF summed numerator over denominator (in above cell) for all peptides in dataset i: a check\n",
    "totalpeaks278_AA80['NAAF check'] = totalpeaks278_AA80['NAAF'] / 478829.326697\n",
    "\n",
    "# write modified dataframe to new txt file, same name + totals\n",
    "totalpeaks278_AA80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/278/ETNP-SKQ17-278-265m-trap_PTMopt_15ppm_DN80_NAAF_totals.csv\")\n",
    "\n",
    "totalpeaks278_AA80.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Visualizing the results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ALC max:  98\n",
      "ALC min:  80\n"
     ]
    }
   ],
   "source": [
    "print(\"ALC max: \", peaks278_80['ALC (%)'].max())\n",
    "print(\"ALC min: \", peaks278_80['ALC (%)'].min())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exporting txt files of stripped peptides at confidence cutoffs:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total modified peptides in 278: 10488\n",
      "Deduplicated modified peptides in 278: 10424\n",
      "Total strippled peptides in 278: 10488\n",
      "Deduplicated stripped peptides in 278: 9735\n"
     ]
    }
   ],
   "source": [
    "# keep only peptide column >50% ALC\n",
    "pep278_50_moddup = peaks278_50[[\"Peptide\"]]\n",
    "\n",
    "# keep only the stripped peptide column for the >50% ALC\n",
    "# this is what we'll use for UniPept input, etc\n",
    "pep278_50_dup = peaks278_50[[\"stripped peptide\"]]\n",
    "\n",
    "# deduplicate both of these lists\n",
    "pep278_50_mod = pep278_50_moddup.drop_duplicates()\n",
    "pep278_50 = pep278_50_dup.drop_duplicates()\n",
    "\n",
    "# print out the #s of modified and stripped peptides, deduplicated and not\n",
    "\n",
    "print(\"Total modified peptides in 278:\", len(pep278_50_moddup))\n",
    "print(\"Deduplicated modified peptides in 278:\", len(pep278_50_mod))\n",
    "print(\"Total strippled peptides in 278:\", len(pep278_50_dup))\n",
    "print(\"Deduplicated stripped peptides in 278:\", len(pep278_50))\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "pep278_50.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/278/ETNP-SKQ17-278-265m-trap_PTMopt_15ppm_DN50_stripped_peptides.txt\", header=False, index=False)\n",
    "\n",
    "# made the text file into a FASTA \n",
    "\n",
    "!awk '{print \">\"NR\"\\n\"$0}' /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/278/ETNP-SKQ17-278-265m-trap_PTMopt_15ppm_DN50_stripped_peptides.txt > \\\n",
    "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/278/ETNP-SKQ17-278-265m-trap_PTMopt_15ppm_DN50_stripped_peptides.fas\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total modified peptides in 278: 1482\n",
      "Deduplicated modified peptides in 278: 1463\n",
      "Total strippled peptides in 278: 1482\n",
      "Deduplicated stripped peptides in 278: 1441\n"
     ]
    }
   ],
   "source": [
    "# keep only peptide column >80% ALC\n",
    "pep278_80_moddup = peaks278_80[[\"Peptide\"]]\n",
    "\n",
    "# keep only the stripped peptide column for the >80% ALC\n",
    "# this is what we'll use for UniPept input, etc\n",
    "pep278_80_dup = peaks278_80[[\"stripped peptide\"]]\n",
    "\n",
    "# deduplicate both of these lists\n",
    "pep278_80_mod = pep278_80_moddup.drop_duplicates()\n",
    "pep278_80 = pep278_80_dup.drop_duplicates()\n",
    "\n",
    "# print out the #s of modified and stripped peptides, deduplicated and not\n",
    "\n",
    "print(\"Total modified peptides in 278:\", len(pep278_80_moddup))\n",
    "print(\"Deduplicated modified peptides in 278:\", len(pep278_80_mod))\n",
    "print(\"Total strippled peptides in 278:\", len(pep278_80_dup))\n",
    "print(\"Deduplicated stripped peptides in 278:\", len(pep278_80))\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "pep278_80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/278/ETNP-SKQ17-278-265m-trap_PTMopt_15ppm_DN80_stripped_peptides.txt\", header=False, index=False)\n",
    "\n",
    "# made the text file into a FASTA \n",
    "\n",
    "!awk '{print \">\"NR\"\\n\"$0}' /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/278/ETNP-SKQ17-278-265m-trap_PTMopt_15ppm_DN80_stripped_peptides.txt > \\\n",
    "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/278/ETNP-SKQ17-278-265m-trap_PTMopt_15ppm_DN80_stripped_peptides.fas\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
