{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Manipulation of Peaks de novo results of ETNP 2017 P2 samples LC-MS/MS data using python.\n",
    "\n",
    "Starting with:\n",
    "\n",
    "    Peaks de novo results (.csv) of PTM-optimized database searches > ALC 50%\n",
    "\n",
    "Goal:\n",
    "\n",
    "    Files with stripped (no PTMs) peptide lists and\n",
    "    Columns with #'s of each modification in every sequence\n",
    "    Column with stripped peptide lengths (# amino acids)\n",
    "    \n",
    "### To use for a different file:\n",
    "\n",
    "#### 1. Change the input file name in *IN 4*\n",
    "#### 2. Use 'find + replace' (Esc + F) to replace the running # (e.g., 233) for another\n",
    "#### 3. Update the NAAF factor calculated in *IN 6* into *IN 7*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We don't have technical duplicates here, sadly, unlike the MED4 Pro samples. I exported PeaksDN search results CSVs into my ETNP 2017 git repo:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt\n"
     ]
    }
   ],
   "source": [
    "cd /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;34m231\u001b[0m/  \u001b[01;34m233\u001b[0m/  \u001b[01;34m243\u001b[0m/  \u001b[01;34m273\u001b[0m/  \u001b[01;34m278\u001b[0m/  \u001b[01;34m378\u001b[0m/\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ETNP_SKQ17_DENOVO_164_243-965m-0.3-JA14_DN50_15ppm.csv\r\n",
      "ETNP_SKQ17_DENOVO_164_243-965m-0.3-JA14_DN50_15ppm.xml\r\n"
     ]
    }
   ],
   "source": [
    "ls 243"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.0.5'"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# LIBRARIES\n",
    "#import pandas library for working with tabular data\n",
    "import os\n",
    "os.getcwd()\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "from scipy.stats import kde\n",
    "#import regular expresson (regex)\n",
    "import re\n",
    "#check pandas version\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 243: 965 m McLane pump filtered on 0.3 um GF-75"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# redundant Peaks peptides >50% ALC in dataframe 1694\n",
      "# redundant Peaks peptides >80% ALC in dataframe 262\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Fraction</th>\n",
       "      <th>Scan</th>\n",
       "      <th>Source File</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Tag Length</th>\n",
       "      <th>ALC (%)</th>\n",
       "      <th>length</th>\n",
       "      <th>m/z</th>\n",
       "      <th>z</th>\n",
       "      <th>RT</th>\n",
       "      <th>Area</th>\n",
       "      <th>Mass</th>\n",
       "      <th>ppm</th>\n",
       "      <th>PTM</th>\n",
       "      <th>local confidence (%)</th>\n",
       "      <th>tag (&gt;=0%)</th>\n",
       "      <th>mode</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>10698</td>\n",
       "      <td>20170531_ETNP_1000m-0.3um-JA14_01.raw</td>\n",
       "      <td>TDEFVR</td>\n",
       "      <td>6</td>\n",
       "      <td>98</td>\n",
       "      <td>6</td>\n",
       "      <td>383.6909</td>\n",
       "      <td>2</td>\n",
       "      <td>38.27</td>\n",
       "      <td>158000.0</td>\n",
       "      <td>765.3657</td>\n",
       "      <td>2.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>98 99 100 98 98 97</td>\n",
       "      <td>TDEFVR</td>\n",
       "      <td>CID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>11876</td>\n",
       "      <td>20170531_ETNP_1000m-0.3um-JA14_01.raw</td>\n",
       "      <td>KYVFVK</td>\n",
       "      <td>6</td>\n",
       "      <td>98</td>\n",
       "      <td>6</td>\n",
       "      <td>392.2419</td>\n",
       "      <td>2</td>\n",
       "      <td>42.26</td>\n",
       "      <td>225000.0</td>\n",
       "      <td>782.4691</td>\n",
       "      <td>0.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99 97 96 99 100 99</td>\n",
       "      <td>KYVFVK</td>\n",
       "      <td>CID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>10148</td>\n",
       "      <td>20170531_ETNP_1000m-0.3um-JA14_01.raw</td>\n",
       "      <td>TSGEVFR</td>\n",
       "      <td>7</td>\n",
       "      <td>97</td>\n",
       "      <td>7</td>\n",
       "      <td>398.2039</td>\n",
       "      <td>2</td>\n",
       "      <td>36.30</td>\n",
       "      <td>285000.0</td>\n",
       "      <td>794.3923</td>\n",
       "      <td>1.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99 100 95 99 96 97 99</td>\n",
       "      <td>TSGEVFR</td>\n",
       "      <td>CID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>10606</td>\n",
       "      <td>20170531_ETNP_1000m-0.3um-JA14_01.raw</td>\n",
       "      <td>LN(+.98)LAQNWR</td>\n",
       "      <td>8</td>\n",
       "      <td>97</td>\n",
       "      <td>8</td>\n",
       "      <td>508.2628</td>\n",
       "      <td>2</td>\n",
       "      <td>37.93</td>\n",
       "      <td>407000.0</td>\n",
       "      <td>1014.5247</td>\n",
       "      <td>-13.5</td>\n",
       "      <td>Deamidation (NQ)</td>\n",
       "      <td>97 98 100 100 96 96 99 97</td>\n",
       "      <td>LN(+.98)LAQNWR</td>\n",
       "      <td>CID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>8910</td>\n",
       "      <td>20170531_ETNP_1000m-0.3um-JA14_01.raw</td>\n",
       "      <td>M(+15.99)EAYC(+57.02)VK</td>\n",
       "      <td>7</td>\n",
       "      <td>97</td>\n",
       "      <td>7</td>\n",
       "      <td>458.6999</td>\n",
       "      <td>2</td>\n",
       "      <td>32.06</td>\n",
       "      <td>465000.0</td>\n",
       "      <td>915.3830</td>\n",
       "      <td>2.4</td>\n",
       "      <td>Oxidation (M); Carbamidomethylation</td>\n",
       "      <td>94 99 100 99 97 98 98</td>\n",
       "      <td>M(+15.99)EAYC(+57.02)VK</td>\n",
       "      <td>CID</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Fraction   Scan                            Source File  \\\n",
       "0         3  10698  20170531_ETNP_1000m-0.3um-JA14_01.raw   \n",
       "1         3  11876  20170531_ETNP_1000m-0.3um-JA14_01.raw   \n",
       "2         3  10148  20170531_ETNP_1000m-0.3um-JA14_01.raw   \n",
       "3         3  10606  20170531_ETNP_1000m-0.3um-JA14_01.raw   \n",
       "4         3   8910  20170531_ETNP_1000m-0.3um-JA14_01.raw   \n",
       "\n",
       "                   Peptide  Tag Length  ALC (%)  length       m/z  z     RT  \\\n",
       "0                   TDEFVR           6       98       6  383.6909  2  38.27   \n",
       "1                   KYVFVK           6       98       6  392.2419  2  42.26   \n",
       "2                  TSGEVFR           7       97       7  398.2039  2  36.30   \n",
       "3           LN(+.98)LAQNWR           8       97       8  508.2628  2  37.93   \n",
       "4  M(+15.99)EAYC(+57.02)VK           7       97       7  458.6999  2  32.06   \n",
       "\n",
       "       Area       Mass   ppm                                  PTM  \\\n",
       "0  158000.0   765.3657   2.2                                  NaN   \n",
       "1  225000.0   782.4691   0.4                                  NaN   \n",
       "2  285000.0   794.3923   1.3                                  NaN   \n",
       "3  407000.0  1014.5247 -13.5                     Deamidation (NQ)   \n",
       "4  465000.0   915.3830   2.4  Oxidation (M); Carbamidomethylation   \n",
       "\n",
       "        local confidence (%)               tag (>=0%) mode  \n",
       "0         98 99 100 98 98 97                   TDEFVR  CID  \n",
       "1         99 97 96 99 100 99                   KYVFVK  CID  \n",
       "2      99 100 95 99 96 97 99                  TSGEVFR  CID  \n",
       "3  97 98 100 100 96 96 99 97           LN(+.98)LAQNWR  CID  \n",
       "4      94 99 100 99 97 98 98  M(+15.99)EAYC(+57.02)VK  CID  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read the CSV into a dataframe using the pandas read_csv function\n",
    "peaks243_50 = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/243/ETNP_SKQ17_DENOVO_164_243-965m-0.3-JA14_DN50_15ppm.csv\")\n",
    "\n",
    "# also make a dataframe but keep only peptides >80% ALC\n",
    "peaks243_80 = peaks243_50.loc[peaks243_50['ALC (%)'] >= 80].copy()\n",
    "\n",
    "# how many de novo sequence candidates >50% ALC?\n",
    "print(\"# redundant Peaks peptides >50% ALC in dataframe\", len(peaks243_50))\n",
    "# how many de novo sequence candidates >80ALC?\n",
    "print(\"# redundant Peaks peptides >80% ALC in dataframe\", len(peaks243_80))\n",
    "\n",
    "#look at the dataframe\n",
    "peaks243_50.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Fraction</th>\n",
       "      <th>Scan</th>\n",
       "      <th>Source File</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Tag Length</th>\n",
       "      <th>ALC (%)</th>\n",
       "      <th>length</th>\n",
       "      <th>m/z</th>\n",
       "      <th>z</th>\n",
       "      <th>RT</th>\n",
       "      <th>...</th>\n",
       "      <th>c-carb</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>k-hydr</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>stripped length</th>\n",
       "      <th>ptm-total</th>\n",
       "      <th>NAAF num.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>10698</td>\n",
       "      <td>20170531_ETNP_1000m-0.3um-JA14_01.raw</td>\n",
       "      <td>TDEFVR</td>\n",
       "      <td>6</td>\n",
       "      <td>98</td>\n",
       "      <td>6</td>\n",
       "      <td>383.6909</td>\n",
       "      <td>2</td>\n",
       "      <td>38.27</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TDEFVR</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>26333.333333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>11876</td>\n",
       "      <td>20170531_ETNP_1000m-0.3um-JA14_01.raw</td>\n",
       "      <td>KYVFVK</td>\n",
       "      <td>6</td>\n",
       "      <td>98</td>\n",
       "      <td>6</td>\n",
       "      <td>392.2419</td>\n",
       "      <td>2</td>\n",
       "      <td>42.26</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>KYVFVK</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>37500.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>10148</td>\n",
       "      <td>20170531_ETNP_1000m-0.3um-JA14_01.raw</td>\n",
       "      <td>TSGEVFR</td>\n",
       "      <td>7</td>\n",
       "      <td>97</td>\n",
       "      <td>7</td>\n",
       "      <td>398.2039</td>\n",
       "      <td>2</td>\n",
       "      <td>36.30</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TSGEVFR</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>40714.285714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>10606</td>\n",
       "      <td>20170531_ETNP_1000m-0.3um-JA14_01.raw</td>\n",
       "      <td>LN(+.98)LAQNWR</td>\n",
       "      <td>8</td>\n",
       "      <td>97</td>\n",
       "      <td>8</td>\n",
       "      <td>508.2628</td>\n",
       "      <td>2</td>\n",
       "      <td>37.93</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>LNLAQNWR</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>50875.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>8910</td>\n",
       "      <td>20170531_ETNP_1000m-0.3um-JA14_01.raw</td>\n",
       "      <td>M(+15.99)EAYC(+57.02)VK</td>\n",
       "      <td>7</td>\n",
       "      <td>97</td>\n",
       "      <td>7</td>\n",
       "      <td>458.6999</td>\n",
       "      <td>2</td>\n",
       "      <td>32.06</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>MVK</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>155000.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Fraction   Scan                            Source File  \\\n",
       "0         3  10698  20170531_ETNP_1000m-0.3um-JA14_01.raw   \n",
       "1         3  11876  20170531_ETNP_1000m-0.3um-JA14_01.raw   \n",
       "2         3  10148  20170531_ETNP_1000m-0.3um-JA14_01.raw   \n",
       "3         3  10606  20170531_ETNP_1000m-0.3um-JA14_01.raw   \n",
       "4         3   8910  20170531_ETNP_1000m-0.3um-JA14_01.raw   \n",
       "\n",
       "                   Peptide  Tag Length  ALC (%)  length       m/z  z     RT  \\\n",
       "0                   TDEFVR           6       98       6  383.6909  2  38.27   \n",
       "1                   KYVFVK           6       98       6  392.2419  2  42.26   \n",
       "2                  TSGEVFR           7       97       7  398.2039  2  36.30   \n",
       "3           LN(+.98)LAQNWR           8       97       8  508.2628  2  37.93   \n",
       "4  M(+15.99)EAYC(+57.02)VK           7       97       7  458.6999  2  32.06   \n",
       "\n",
       "   ...  c-carb  m-oxid  n-deam q-deam k-hydr r-meth stripped peptide  \\\n",
       "0  ...       0       0       0      0      0      0           TDEFVR   \n",
       "1  ...       0       0       0      0      0      0           KYVFVK   \n",
       "2  ...       0       0       0      0      0      0          TSGEVFR   \n",
       "3  ...       0       0       1      0      0      0         LNLAQNWR   \n",
       "4  ...       1       1       0      0      0      0              MVK   \n",
       "\n",
       "   stripped length  ptm-total      NAAF num.  \n",
       "0                6          0   26333.333333  \n",
       "1                6          0   37500.000000  \n",
       "2                7          0   40714.285714  \n",
       "3                8          1   50875.000000  \n",
       "4                3          2  155000.000000  \n",
       "\n",
       "[5 rows x 47 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use a count function to enumerate the # of A's (alanines) in each peptide\n",
    "peaks243_50['A'] = peaks243_50['Peptide'].str.count(\"A\")\n",
    "\n",
    "# use a count function to enumerate the # of C's (cysteines) in each peptide\n",
    "peaks243_50['C'] = peaks243_50['Peptide'].str.count(\"C\")\n",
    "\n",
    "# use a count function to enumerate the # of D's (aspartic acids) in each peptide\n",
    "peaks243_50['D'] = peaks243_50['Peptide'].str.count(\"D\")\n",
    "\n",
    "# use a count function to enumerate the # of E's (glutamic acids) in each peptide\n",
    "peaks243_50['E'] = peaks243_50['Peptide'].str.count(\"E\")\n",
    "\n",
    "# use a count function to enumerate the # of F's (phenylalanines) in each peptide\n",
    "peaks243_50['F'] = peaks243_50['Peptide'].str.count(\"F\")\n",
    "\n",
    "# use a count function to enumerate the # of G's (glycines) in each peptide\n",
    "peaks243_50['G'] = peaks243_50['Peptide'].str.count(\"G\")\n",
    "\n",
    "# use a count function to enumerate the # of H's (histidines) in each peptide\n",
    "peaks243_50['H'] = peaks243_50['Peptide'].str.count(\"H\")\n",
    "\n",
    "# use a count function to enumerate the # of I's (isoleucines) in each peptide\n",
    "# in peaks243_50 output, there will be no isoleucines (they're lumped in with leucines)\n",
    "peaks243_50['I'] = peaks243_50['Peptide'].str.count(\"I\")\n",
    "\n",
    "# use a count function to enumerate the # of K's (lysines) in each peptide\n",
    "peaks243_50['K'] = peaks243_50['Peptide'].str.count(\"K\")\n",
    "\n",
    "# use a count function to enumerate the # of L's (leucines) in each peptide\n",
    "# also these include the isoleucines\n",
    "peaks243_50['L'] = peaks243_50['Peptide'].str.count(\"L\")\n",
    "\n",
    "# use a count function to enumerate the # of M's (methionines) in each peptide\n",
    "peaks243_50['M'] = peaks243_50['Peptide'].str.count(\"M\")\n",
    "\n",
    "# use a count function to enumerate the # of N's (asparagines) in each peptide\n",
    "peaks243_50['N'] = peaks243_50['Peptide'].str.count(\"N\")\n",
    "\n",
    "# use a count function to enumerate the # of P's ([prolines]) in each peptide\n",
    "peaks243_50['P'] = peaks243_50['Peptide'].str.count(\"P\")\n",
    "\n",
    "# use a count function to enumerate the # of Q's (glutamines) in each peptide\n",
    "peaks243_50['Q'] = peaks243_50['Peptide'].str.count(\"Q\")\n",
    "\n",
    "# use a count function to enumerate the # of R's (arginines) in each peptide\n",
    "peaks243_50['R'] = peaks243_50['Peptide'].str.count(\"R\")\n",
    "\n",
    "# use a count function to enumerate the # of S's (serines) in each peptide\n",
    "peaks243_50['S'] = peaks243_50['Peptide'].str.count(\"S\")\n",
    "\n",
    "# use a count function to enumerate the # of T's (threonines) in each peptide\n",
    "peaks243_50['T'] = peaks243_50['Peptide'].str.count(\"T\")\n",
    "\n",
    "# use a count function to enumerate the # of V's (valines) in each peptide\n",
    "peaks243_50['V'] = peaks243_50['Peptide'].str.count(\"V\")\n",
    "\n",
    "# use a count function to enumerate the # of W's (tryptophans) in each peptide\n",
    "peaks243_50['W'] = peaks243_50['Peptide'].str.count(\"W\")\n",
    "\n",
    "# use a count function to enumerate the # of Y's (tyrosines) in each peptide\n",
    "peaks243_50['Y'] = peaks243_50['Peptide'].str.count(\"Y\")\n",
    "\n",
    "# use a count function to enumerate the # of carbamidomethylated C's in each peptide\n",
    "peaks243_50['c-carb'] = peaks243_50['Peptide'].str.count(\"57.02\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "peaks243_50['m-oxid'] = peaks243_50['Peptide'].apply(lambda x: x.count('M(+15.99)'))\n",
    "\n",
    "# use a lamba function to enumerate the # of deamidated N's in each peptide\n",
    "peaks243_50['n-deam'] = peaks243_50['Peptide'].apply(lambda x: x.count('N(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "peaks243_50['q-deam'] = peaks243_50['Peptide'].apply(lambda x: x.count('Q(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of hydroxylated K's in each peptide\n",
    "peaks243_50['k-hydr'] = peaks243_50['Peptide'].apply(lambda x: x.count('K(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of hydroxylated P's in each peptide\n",
    "# no more P hydroxylation in final searches\n",
    "# peaks243_50['p-hydr'] = peaks243_50['Peptide'].apply(lambda x: x.count('P(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "peaks243_50['r-meth'] = peaks243_50['Peptide'].apply(lambda x: x.count('R(+14.02)'))\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "peaks243_50['stripped peptide'] = peaks243_50['Peptide'].str.replace(r\"\\(.*\\)\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "peaks243_50['stripped length'] = peaks243_50['stripped peptide'].apply(len)\n",
    "\n",
    "# total the number of modifications in sequence\n",
    "peaks243_50['ptm-total'] = peaks243_50['c-carb'] + peaks243_50['m-oxid'] + peaks243_50['n-deam'] \\\n",
    "+ peaks243_50['q-deam'] + peaks243_50['k-hydr'] + peaks243_50['r-meth']\n",
    "\n",
    "# calculate NAAF numerator for each peptide k\n",
    "peaks243_50['NAAF num.'] = peaks243_50['Area'] / peaks243_50['stripped length']\n",
    "\n",
    "# write modified dataframe to new csv file\n",
    "peaks243_50.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/243/ETNP-SKQ17-243-965m-0.3-JA14_PTMopt_15ppm_DN50.csv\")\n",
    "\n",
    "\n",
    "# check out the results\n",
    "peaks243_50.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>K</th>\n",
       "      <th>L</th>\n",
       "      <th>...</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>Total area</th>\n",
       "      <th>Total length</th>\n",
       "      <th>% C w/ carb</th>\n",
       "      <th>% M w/ oxid</th>\n",
       "      <th>% N w/ deam</th>\n",
       "      <th>% Q w/ deam</th>\n",
       "      <th>% K w/ hydr</th>\n",
       "      <th>% R w/ meth</th>\n",
       "      <th>NAAF denom.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>1007</td>\n",
       "      <td>235</td>\n",
       "      <td>539</td>\n",
       "      <td>698</td>\n",
       "      <td>474</td>\n",
       "      <td>539</td>\n",
       "      <td>438</td>\n",
       "      <td>0</td>\n",
       "      <td>1552</td>\n",
       "      <td>1481</td>\n",
       "      <td>...</td>\n",
       "      <td>563</td>\n",
       "      <td>4.006242e+09</td>\n",
       "      <td>12736</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.389805</td>\n",
       "      <td>0.186972</td>\n",
       "      <td>0.031447</td>\n",
       "      <td>0.239691</td>\n",
       "      <td>0.422039</td>\n",
       "      <td>314560.470236</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 A    C    D    E    F    G    H  I     K     L  ...  r-meth  \\\n",
       "sample total  1007  235  539  698  474  539  438  0  1552  1481  ...     563   \n",
       "\n",
       "                Total area  Total length  % C w/ carb  % M w/ oxid  \\\n",
       "sample total  4.006242e+09         12736          1.0     0.389805   \n",
       "\n",
       "              % N w/ deam  % Q w/ deam  % K w/ hydr  % R w/ meth  \\\n",
       "sample total     0.186972     0.031447     0.239691     0.422039   \n",
       "\n",
       "                NAAF denom.  \n",
       "sample total  314560.470236  \n",
       "\n",
       "[1 rows x 35 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of certain columns in the modified\n",
    "# peptide dataframe above (for >50% ALC)\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'A': peaks243_50['A'].sum(),\n",
    "        'C': peaks243_50['C'].sum(),\n",
    "        'D': peaks243_50['D'].sum(),\n",
    "        'E': peaks243_50['E'].sum(),\n",
    "        'F': peaks243_50['F'].sum(),\n",
    "        'G': peaks243_50['G'].sum(),\n",
    "        'H': peaks243_50['H'].sum(),\n",
    "        'I': peaks243_50['I'].sum(),\n",
    "        'K': peaks243_50['K'].sum(),\n",
    "        'L': peaks243_50['L'].sum(),\n",
    "        'M': peaks243_50['M'].sum(),\n",
    "        'N': peaks243_50['N'].sum(),\n",
    "        'P': peaks243_50['P'].sum(),\n",
    "        'Q': peaks243_50['Q'].sum(),\n",
    "        'R': peaks243_50['R'].sum(),\n",
    "        'S': peaks243_50['S'].sum(),\n",
    "        'T': peaks243_50['T'].sum(),\n",
    "        'V': peaks243_50['V'].sum(),\n",
    "        'W': peaks243_50['W'].sum(),\n",
    "        'Y': peaks243_50['Y'].sum(),\n",
    "        'c-carb': peaks243_50['c-carb'].sum(),\n",
    "        'm-oxid': peaks243_50['m-oxid'].sum(),\n",
    "        'n-deam': peaks243_50['n-deam'].sum(),\n",
    "        'q-deam': peaks243_50['q-deam'].sum(),\n",
    "        'k-hydr': peaks243_50['k-hydr'].sum(),\n",
    "        'r-meth': peaks243_50['r-meth'].sum(),\n",
    "        'Total area': peaks243_50['Area'].sum(),\n",
    "        'Total length': peaks243_50['stripped length'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaks243_50 = pd.DataFrame(data, columns=['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L',\\\n",
    "                                               'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', \\\n",
    "                                               'm-oxid', 'n-deam', 'q-deam', 'k-hydr', 'r-meth', \\\n",
    "                                               'Total area', 'Total length'], index=index)\n",
    "\n",
    "# calculate percentage of C's with carb (should be 1.0)\n",
    "totalpeaks243_50['% C w/ carb'] = totalpeaks243_50['c-carb'] / totalpeaks243_50['C'] \n",
    "\n",
    "# calculate percentage of M's that are oxidized\n",
    "totalpeaks243_50['% M w/ oxid'] = totalpeaks243_50['m-oxid'] / totalpeaks243_50['M'] \n",
    "\n",
    "# calculate percentage of N's that are deamidated\n",
    "totalpeaks243_50['% N w/ deam'] = totalpeaks243_50['n-deam'] / totalpeaks243_50['N'] \n",
    "\n",
    "# calculate percentage of Q's that are deamidated\n",
    "totalpeaks243_50['% Q w/ deam'] = totalpeaks243_50['q-deam'] / totalpeaks243_50['Q'] \n",
    "\n",
    "# calculate percentage of K's that are hydroxylated\n",
    "totalpeaks243_50['% K w/ hydr'] = totalpeaks243_50['k-hydr'] / totalpeaks243_50['K'] \n",
    "\n",
    "# calculate percentage of P's that are hydroxylated\n",
    "#totalpeaks243_50['% P w/ hydr'] = totalpeaks243_50['p-hydr'] / totalpeaks243_50['K'] \n",
    "\n",
    "# calculate percentage of R's that are methylated\n",
    "totalpeaks243_50['% R w/ meth'] = totalpeaks243_50['r-meth'] / totalpeaks243_50['R'] \n",
    "\n",
    "# calculate NAAF denominator for all peptides in dataset i\n",
    "totalpeaks243_50['NAAF denom.'] = totalpeaks243_50['Total area'] / totalpeaks243_50['Total length']\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "totalpeaks243_50.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/243/ETNP-SKQ17-243-965m-0.3-JA14_PTMopt_15ppm_DN50_totals.csv\")\n",
    "\n",
    "totalpeaks243_50.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>NAAF factor</th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>...</th>\n",
       "      <th>T-NAAF50</th>\n",
       "      <th>V-NAAF50</th>\n",
       "      <th>W-NAAF50</th>\n",
       "      <th>Y-NAAF50</th>\n",
       "      <th>ccarb-NAAF50</th>\n",
       "      <th>moxid-NAAF50</th>\n",
       "      <th>ndeam-NAAF50</th>\n",
       "      <th>qdeam-NAAF50</th>\n",
       "      <th>khydr-NAAF50</th>\n",
       "      <th>rmeth-NAAF50</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TDEFVR</td>\n",
       "      <td>0.083715</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.083715</td>\n",
       "      <td>0.083715</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KYVFVK</td>\n",
       "      <td>0.119214</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.238428</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.119214</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TSGEVFR</td>\n",
       "      <td>0.129432</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.129432</td>\n",
       "      <td>0.129432</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LNLAQNWR</td>\n",
       "      <td>0.161734</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.161734</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.161734</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MVK</td>\n",
       "      <td>0.492751</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.492751</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.492751</td>\n",
       "      <td>0.492751</td>\n",
       "      <td>0.492751</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  stripped peptide  NAAF factor  A  C  D  E  F  G  H  I  ...  T-NAAF50  \\\n",
       "0           TDEFVR     0.083715  0  0  1  1  1  0  0  0  ...  0.083715   \n",
       "1           KYVFVK     0.119214  0  0  0  0  1  0  0  0  ...  0.000000   \n",
       "2          TSGEVFR     0.129432  0  0  0  1  1  1  0  0  ...  0.129432   \n",
       "3         LNLAQNWR     0.161734  1  0  0  0  0  0  0  0  ...  0.000000   \n",
       "4              MVK     0.492751  1  1  0  1  0  0  0  0  ...  0.000000   \n",
       "\n",
       "   V-NAAF50  W-NAAF50  Y-NAAF50  ccarb-NAAF50  moxid-NAAF50  ndeam-NAAF50  \\\n",
       "0  0.083715  0.000000  0.000000      0.000000      0.000000      0.000000   \n",
       "1  0.238428  0.000000  0.119214      0.000000      0.000000      0.000000   \n",
       "2  0.129432  0.000000  0.000000      0.000000      0.000000      0.000000   \n",
       "3  0.000000  0.161734  0.000000      0.000000      0.000000      0.161734   \n",
       "4  0.492751  0.000000  0.492751      0.492751      0.492751      0.000000   \n",
       "\n",
       "   qdeam-NAAF50  khydr-NAAF50  rmeth-NAAF50  \n",
       "0           0.0           0.0           0.0  \n",
       "1           0.0           0.0           0.0  \n",
       "2           0.0           0.0           0.0  \n",
       "3           0.0           0.0           0.0  \n",
       "4           0.0           0.0           0.0  \n",
       "\n",
       "[5 rows x 54 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use the calculated NAAF factor (in totalpeaks243 dataframe, above) to caluclate the NAAF \n",
    "# NAAF: normalized normalized area abundance factor\n",
    "\n",
    "# don't have to worry here about DECOY hits messing with Area totals\n",
    "# but we will with Comet results\n",
    "\n",
    "NAAF50 = 314560.470236\n",
    "\n",
    "# use NAAF >50% ALC to get NAAF\n",
    "peaks243_50['NAAF factor'] = (peaks243_50['NAAF num.'])/NAAF50\n",
    "\n",
    "# make a dataframe that contains only what we need: sequences, AAs, PTMs\n",
    "peaks243_AA50 = peaks243_50[['stripped peptide', 'NAAF factor', 'A', 'C', 'D', 'E', 'F', 'G', 'H', \\\n",
    "                             'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', \\\n",
    "                             'm-oxid', 'n-deam', 'q-deam', 'k-hydr', 'r-meth']].copy()\n",
    "\n",
    "# multiply the NAAF50 factor by the AAs to normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaks243_AA50['A-NAAF50'] = peaks243_AA50['A'] * peaks243_50['NAAF factor']\n",
    "peaks243_AA50['C-NAAF50'] = peaks243_AA50['C'] * peaks243_50['NAAF factor']\n",
    "peaks243_AA50['D-NAAF50'] = peaks243_AA50['D'] * peaks243_50['NAAF factor']\n",
    "peaks243_AA50['E-NAAF50'] = peaks243_AA50['E'] * peaks243_50['NAAF factor']\n",
    "peaks243_AA50['F-NAAF50'] = peaks243_AA50['F'] * peaks243_50['NAAF factor']\n",
    "peaks243_AA50['G-NAAF50'] = peaks243_AA50['G'] * peaks243_50['NAAF factor']\n",
    "peaks243_AA50['H-NAAF50'] = peaks243_AA50['H'] * peaks243_50['NAAF factor']\n",
    "peaks243_AA50['I-NAAF50'] = peaks243_AA50['I'] * peaks243_50['NAAF factor']\n",
    "peaks243_AA50['K-NAAF50'] = peaks243_AA50['K'] * peaks243_50['NAAF factor']\n",
    "peaks243_AA50['L-NAAF50'] = peaks243_AA50['L'] * peaks243_50['NAAF factor']\n",
    "peaks243_AA50['M-NAAF50'] = peaks243_AA50['M'] * peaks243_50['NAAF factor']\n",
    "peaks243_AA50['N-NAAF50'] = peaks243_AA50['N'] * peaks243_50['NAAF factor']\n",
    "peaks243_AA50['P-NAAF50'] = peaks243_AA50['P'] * peaks243_50['NAAF factor']\n",
    "peaks243_AA50['Q-NAAF50'] = peaks243_AA50['Q'] * peaks243_50['NAAF factor']\n",
    "peaks243_AA50['R-NAAF50'] = peaks243_AA50['R'] * peaks243_50['NAAF factor']\n",
    "peaks243_AA50['S-NAAF50'] = peaks243_AA50['S'] * peaks243_50['NAAF factor']\n",
    "peaks243_AA50['T-NAAF50'] = peaks243_AA50['T'] * peaks243_50['NAAF factor']\n",
    "peaks243_AA50['V-NAAF50'] = peaks243_AA50['V'] * peaks243_50['NAAF factor']\n",
    "peaks243_AA50['W-NAAF50'] = peaks243_AA50['W'] * peaks243_50['NAAF factor']\n",
    "peaks243_AA50['Y-NAAF50'] = peaks243_AA50['Y'] * peaks243_50['NAAF factor']\n",
    "\n",
    "# multiply the NAAF50 factor by the PTMs normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaks243_AA50['ccarb-NAAF50'] = peaks243_AA50['c-carb'] * peaks243_AA50['NAAF factor']\n",
    "peaks243_AA50['moxid-NAAF50'] = peaks243_AA50['m-oxid'] * peaks243_AA50['NAAF factor']\n",
    "peaks243_AA50['ndeam-NAAF50'] = peaks243_AA50['n-deam'] * peaks243_AA50['NAAF factor']\n",
    "peaks243_AA50['qdeam-NAAF50'] = peaks243_AA50['q-deam'] * peaks243_AA50['NAAF factor']\n",
    "peaks243_AA50['khydr-NAAF50'] = peaks243_AA50['k-hydr'] * peaks243_AA50['NAAF factor']\n",
    "peaks243_AA50['rmeth-NAAF50'] = peaks243_AA50['r-meth'] * peaks243_AA50['NAAF factor']\n",
    "\n",
    "# write the dataframe to a new csv\n",
    "peaks243_AA50.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/243/ETNP-SKQ17-243-965m-0.3-JA14_PTMopt_15ppm_DN50_NAAF.csv\")\n",
    "\n",
    "peaks243_AA50.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NAAF</th>\n",
       "      <th>A-NAAF</th>\n",
       "      <th>C-NAAF</th>\n",
       "      <th>D-NAAF</th>\n",
       "      <th>E-NAAF</th>\n",
       "      <th>F-NAAF</th>\n",
       "      <th>G-NAAF</th>\n",
       "      <th>H-NAAF</th>\n",
       "      <th>I-NAAF</th>\n",
       "      <th>K-NAAF</th>\n",
       "      <th>...</th>\n",
       "      <th>Q-deam-NAAF</th>\n",
       "      <th>K-hydr-NAAF</th>\n",
       "      <th>R-meth-NAAF</th>\n",
       "      <th>% C w/ carb. NAAF</th>\n",
       "      <th>% M w/ oxid. NAAF</th>\n",
       "      <th>% N w/ deam. NAAF</th>\n",
       "      <th>% Q w/ deam. NAAF</th>\n",
       "      <th>% K w/ hydr. NAAF</th>\n",
       "      <th>% R w/ meth. NAAF</th>\n",
       "      <th>NAAF check</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>1631.723922</td>\n",
       "      <td>1533.63853</td>\n",
       "      <td>80.064869</td>\n",
       "      <td>144.939138</td>\n",
       "      <td>498.099862</td>\n",
       "      <td>83.979696</td>\n",
       "      <td>262.101298</td>\n",
       "      <td>46.826685</td>\n",
       "      <td>0.0</td>\n",
       "      <td>213.58155</td>\n",
       "      <td>...</td>\n",
       "      <td>34.57973</td>\n",
       "      <td>74.017647</td>\n",
       "      <td>1072.548538</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.671715</td>\n",
       "      <td>0.506963</td>\n",
       "      <td>0.157717</td>\n",
       "      <td>0.346554</td>\n",
       "      <td>0.56339</td>\n",
       "      <td>0.005187</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     NAAF      A-NAAF     C-NAAF      D-NAAF      E-NAAF  \\\n",
       "sample total  1631.723922  1533.63853  80.064869  144.939138  498.099862   \n",
       "\n",
       "                 F-NAAF      G-NAAF     H-NAAF  I-NAAF     K-NAAF  ...  \\\n",
       "sample total  83.979696  262.101298  46.826685     0.0  213.58155  ...   \n",
       "\n",
       "              Q-deam-NAAF  K-hydr-NAAF  R-meth-NAAF  % C w/ carb. NAAF  \\\n",
       "sample total     34.57973    74.017647  1072.548538                1.0   \n",
       "\n",
       "              % M w/ oxid. NAAF  % N w/ deam. NAAF  % Q w/ deam. NAAF  \\\n",
       "sample total           0.671715           0.506963           0.157717   \n",
       "\n",
       "              % K w/ hydr. NAAF  % R w/ meth. NAAF  NAAF check  \n",
       "sample total           0.346554            0.56339    0.005187  \n",
       "\n",
       "[1 rows x 34 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of NAAF normalized AAs for comet243 PeaksDN results\n",
    "# also contains the sums of the NAAF-corrected PTMs occurances for each affected residue\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'NAAF': peaks243_AA50['NAAF factor'].sum(),\n",
    "        'A-NAAF': peaks243_AA50['A-NAAF50'].sum(),\n",
    "        'C-NAAF': peaks243_AA50['C-NAAF50'].sum(),\n",
    "        'D-NAAF': peaks243_AA50['D-NAAF50'].sum(),\n",
    "        'E-NAAF': peaks243_AA50['E-NAAF50'].sum(),\n",
    "        'F-NAAF': peaks243_AA50['F-NAAF50'].sum(),\n",
    "        'G-NAAF': peaks243_AA50['G-NAAF50'].sum(),\n",
    "        'H-NAAF': peaks243_AA50['H-NAAF50'].sum(),\n",
    "        'I-NAAF': peaks243_AA50['I-NAAF50'].sum(),\n",
    "        'K-NAAF': peaks243_AA50['K-NAAF50'].sum(),\n",
    "        'L-NAAF': peaks243_AA50['L-NAAF50'].sum(),\n",
    "        'M-NAAF': peaks243_AA50['M-NAAF50'].sum(),\n",
    "        'N-NAAF': peaks243_AA50['N-NAAF50'].sum(),\n",
    "        'P-NAAF': peaks243_AA50['P-NAAF50'].sum(),\n",
    "        'Q-NAAF': peaks243_AA50['Q-NAAF50'].sum(),\n",
    "        'R-NAAF': peaks243_AA50['R-NAAF50'].sum(),\n",
    "        'S-NAAF': peaks243_AA50['S-NAAF50'].sum(),\n",
    "        'T-NAAF': peaks243_AA50['T-NAAF50'].sum(),\n",
    "        'V-NAAF': peaks243_AA50['V-NAAF50'].sum(),\n",
    "        'W-NAAF': peaks243_AA50['W-NAAF50'].sum(),\n",
    "        'Y-NAAF': peaks243_AA50['Y-NAAF50'].sum(),\n",
    "        'C-carb-NAAF': peaks243_AA50['ccarb-NAAF50'].sum(),\n",
    "        'M-oxid-NAAF': peaks243_AA50['moxid-NAAF50'].sum(),\n",
    "        'N-deam-NAAF': peaks243_AA50['ndeam-NAAF50'].sum(),\n",
    "        'Q-deam-NAAF': peaks243_AA50['qdeam-NAAF50'].sum(),\n",
    "        'K-hydr-NAAF': peaks243_AA50['khydr-NAAF50'].sum(),\n",
    "        'R-meth-NAAF': peaks243_AA50['rmeth-NAAF50'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaks243_AA50 = pd.DataFrame(data, columns=['NAAF', 'A-NAAF', 'C-NAAF', 'D-NAAF', 'E-NAAF', 'F-NAAF', \\\n",
    "                                                   'G-NAAF', 'H-NAAF', 'I-NAAF', 'K-NAAF', 'L-NAAF', 'M-NAAF', \\\n",
    "                                                   'N-NAAF', 'P-NAAF', 'Q-NAAF', 'R-NAAF', 'S-NAAF', \\\n",
    "                                                   'T-NAAF', 'V-NAAF', 'W-NAAF', 'Y-NAAF', 'C-carb-NAAF', \\\n",
    "                                                   'M-oxid-NAAF', 'N-deam-NAAF', 'Q-deam-NAAF', 'K-hydr-NAAF',\\\n",
    "                                                   'R-meth-NAAF'], index=index)\n",
    "\n",
    "# calculate the NAAF-corrected % modified C, M, N, Q, K, P, and Rs\n",
    "\n",
    "\n",
    "totalpeaks243_AA50['% C w/ carb. NAAF'] = totalpeaks243_AA50['C-carb-NAAF'] / totalpeaks243_AA50['C-NAAF']\n",
    "totalpeaks243_AA50['% M w/ oxid. NAAF'] = totalpeaks243_AA50['M-oxid-NAAF'] / totalpeaks243_AA50['M-NAAF']\n",
    "totalpeaks243_AA50['% N w/ deam. NAAF'] = totalpeaks243_AA50['N-deam-NAAF'] / totalpeaks243_AA50['N-NAAF']\n",
    "totalpeaks243_AA50['% Q w/ deam. NAAF'] = totalpeaks243_AA50['Q-deam-NAAF'] / totalpeaks243_AA50['Q-NAAF']\n",
    "totalpeaks243_AA50['% K w/ hydr. NAAF'] = totalpeaks243_AA50['K-hydr-NAAF'] / totalpeaks243_AA50['K-NAAF']\n",
    "totalpeaks243_AA50['% R w/ meth. NAAF'] = totalpeaks243_AA50['R-meth-NAAF'] / totalpeaks243_AA50['R-NAAF']\n",
    "\n",
    "# calculate NAAF summed numerator over denominator (in above cell) for all peptides in dataset i: a check\n",
    "totalpeaks243_AA50['NAAF check'] = totalpeaks243_AA50['NAAF'] / 314560.470236\n",
    "\n",
    "# write modified dataframe to new txt file, same name + totals\n",
    "totalpeaks243_AA50.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/243/ETNP-SKQ17-243-965m-0.3-JA14_PTMopt_DN50_NAAF_totals.csv\")\n",
    "\n",
    "totalpeaks243_AA50.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Same process but for de novo peptide >80 % ALC:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Fraction</th>\n",
       "      <th>Scan</th>\n",
       "      <th>Source File</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Tag Length</th>\n",
       "      <th>ALC (%)</th>\n",
       "      <th>length</th>\n",
       "      <th>m/z</th>\n",
       "      <th>z</th>\n",
       "      <th>RT</th>\n",
       "      <th>...</th>\n",
       "      <th>c-carb</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>k-hydr</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>stripped length</th>\n",
       "      <th>ptm-total</th>\n",
       "      <th>NAAF num.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>10698</td>\n",
       "      <td>20170531_ETNP_1000m-0.3um-JA14_01.raw</td>\n",
       "      <td>TDEFVR</td>\n",
       "      <td>6</td>\n",
       "      <td>98</td>\n",
       "      <td>6</td>\n",
       "      <td>383.6909</td>\n",
       "      <td>2</td>\n",
       "      <td>38.27</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TDEFVR</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>26333.333333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>11876</td>\n",
       "      <td>20170531_ETNP_1000m-0.3um-JA14_01.raw</td>\n",
       "      <td>KYVFVK</td>\n",
       "      <td>6</td>\n",
       "      <td>98</td>\n",
       "      <td>6</td>\n",
       "      <td>392.2419</td>\n",
       "      <td>2</td>\n",
       "      <td>42.26</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>KYVFVK</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>37500.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>10148</td>\n",
       "      <td>20170531_ETNP_1000m-0.3um-JA14_01.raw</td>\n",
       "      <td>TSGEVFR</td>\n",
       "      <td>7</td>\n",
       "      <td>97</td>\n",
       "      <td>7</td>\n",
       "      <td>398.2039</td>\n",
       "      <td>2</td>\n",
       "      <td>36.30</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TSGEVFR</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>40714.285714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>10606</td>\n",
       "      <td>20170531_ETNP_1000m-0.3um-JA14_01.raw</td>\n",
       "      <td>LN(+.98)LAQNWR</td>\n",
       "      <td>8</td>\n",
       "      <td>97</td>\n",
       "      <td>8</td>\n",
       "      <td>508.2628</td>\n",
       "      <td>2</td>\n",
       "      <td>37.93</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>LNLAQNWR</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>50875.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>8910</td>\n",
       "      <td>20170531_ETNP_1000m-0.3um-JA14_01.raw</td>\n",
       "      <td>M(+15.99)EAYC(+57.02)VK</td>\n",
       "      <td>7</td>\n",
       "      <td>97</td>\n",
       "      <td>7</td>\n",
       "      <td>458.6999</td>\n",
       "      <td>2</td>\n",
       "      <td>32.06</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>MVK</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>155000.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Fraction   Scan                            Source File  \\\n",
       "0         3  10698  20170531_ETNP_1000m-0.3um-JA14_01.raw   \n",
       "1         3  11876  20170531_ETNP_1000m-0.3um-JA14_01.raw   \n",
       "2         3  10148  20170531_ETNP_1000m-0.3um-JA14_01.raw   \n",
       "3         3  10606  20170531_ETNP_1000m-0.3um-JA14_01.raw   \n",
       "4         3   8910  20170531_ETNP_1000m-0.3um-JA14_01.raw   \n",
       "\n",
       "                   Peptide  Tag Length  ALC (%)  length       m/z  z     RT  \\\n",
       "0                   TDEFVR           6       98       6  383.6909  2  38.27   \n",
       "1                   KYVFVK           6       98       6  392.2419  2  42.26   \n",
       "2                  TSGEVFR           7       97       7  398.2039  2  36.30   \n",
       "3           LN(+.98)LAQNWR           8       97       8  508.2628  2  37.93   \n",
       "4  M(+15.99)EAYC(+57.02)VK           7       97       7  458.6999  2  32.06   \n",
       "\n",
       "   ...  c-carb  m-oxid  n-deam q-deam k-hydr r-meth stripped peptide  \\\n",
       "0  ...       0       0       0      0      0      0           TDEFVR   \n",
       "1  ...       0       0       0      0      0      0           KYVFVK   \n",
       "2  ...       0       0       0      0      0      0          TSGEVFR   \n",
       "3  ...       0       0       1      0      0      0         LNLAQNWR   \n",
       "4  ...       1       1       0      0      0      0              MVK   \n",
       "\n",
       "   stripped length  ptm-total      NAAF num.  \n",
       "0                6          0   26333.333333  \n",
       "1                6          0   37500.000000  \n",
       "2                7          0   40714.285714  \n",
       "3                8          1   50875.000000  \n",
       "4                3          2  155000.000000  \n",
       "\n",
       "[5 rows x 47 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use a count function to enumerate the # of A's (alanines) in each peptide\n",
    "peaks243_80['A'] = peaks243_80['Peptide'].str.count(\"A\")\n",
    "\n",
    "# use a count function to enumerate the # of C's (cysteines) in each peptide\n",
    "peaks243_80['C'] = peaks243_80['Peptide'].str.count(\"C\")\n",
    "\n",
    "# use a count function to enumerate the # of D's (aspartic acids) in each peptide\n",
    "peaks243_80['D'] = peaks243_80['Peptide'].str.count(\"D\")\n",
    "\n",
    "# use a count function to enumerate the # of E's (glutamic acids) in each peptide\n",
    "peaks243_80['E'] = peaks243_80['Peptide'].str.count(\"E\")\n",
    "\n",
    "# use a count function to enumerate the # of F's (phenylalanines) in each peptide\n",
    "peaks243_80['F'] = peaks243_80['Peptide'].str.count(\"F\")\n",
    "\n",
    "# use a count function to enumerate the # of G's (glycines) in each peptide\n",
    "peaks243_80['G'] = peaks243_80['Peptide'].str.count(\"G\")\n",
    "\n",
    "# use a count function to enumerate the # of H's (histidines) in each peptide\n",
    "peaks243_80['H'] = peaks243_80['Peptide'].str.count(\"H\")\n",
    "\n",
    "# use a count function to enumerate the # of I's (isoleucines) in each peptide\n",
    "# in peaks243_80 output, there will be no isoleucines (they're lumped in with leucines)\n",
    "peaks243_80['I'] = peaks243_80['Peptide'].str.count(\"I\")\n",
    "\n",
    "# use a count function to enumerate the # of K's (lysines) in each peptide\n",
    "peaks243_80['K'] = peaks243_80['Peptide'].str.count(\"K\")\n",
    "\n",
    "# use a count function to enumerate the # of L's (leucines) in each peptide\n",
    "# also these include the isoleucines\n",
    "peaks243_80['L'] = peaks243_80['Peptide'].str.count(\"L\")\n",
    "\n",
    "# use a count function to enumerate the # of M's (methionines) in each peptide\n",
    "peaks243_80['M'] = peaks243_80['Peptide'].str.count(\"M\")\n",
    "\n",
    "# use a count function to enumerate the # of N's (asparagines) in each peptide\n",
    "peaks243_80['N'] = peaks243_80['Peptide'].str.count(\"N\")\n",
    "\n",
    "# use a count function to enumerate the # of P's ([prolines]) in each peptide\n",
    "peaks243_80['P'] = peaks243_80['Peptide'].str.count(\"P\")\n",
    "\n",
    "# use a count function to enumerate the # of Q's (glutamines) in each peptide\n",
    "peaks243_80['Q'] = peaks243_80['Peptide'].str.count(\"Q\")\n",
    "\n",
    "# use a count function to enumerate the # of R's (arginines) in each peptide\n",
    "peaks243_80['R'] = peaks243_80['Peptide'].str.count(\"R\")\n",
    "\n",
    "# use a count function to enumerate the # of S's (serines) in each peptide\n",
    "peaks243_80['S'] = peaks243_80['Peptide'].str.count(\"S\")\n",
    "\n",
    "# use a count function to enumerate the # of T's (threonines) in each peptide\n",
    "peaks243_80['T'] = peaks243_80['Peptide'].str.count(\"T\")\n",
    "\n",
    "# use a count function to enumerate the # of V's (valines) in each peptide\n",
    "peaks243_80['V'] = peaks243_80['Peptide'].str.count(\"V\")\n",
    "\n",
    "# use a count function to enumerate the # of W's (tryptophans) in each peptide\n",
    "peaks243_80['W'] = peaks243_80['Peptide'].str.count(\"W\")\n",
    "\n",
    "# use a count function to enumerate the # of Y's (tyrosines) in each peptide\n",
    "peaks243_80['Y'] = peaks243_80['Peptide'].str.count(\"Y\")\n",
    "\n",
    "# use a count function to enumerate the # of carbamidomethylated C's in each peptide\n",
    "peaks243_80['c-carb'] = peaks243_80['Peptide'].str.count(\"57.02\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "peaks243_80['m-oxid'] = peaks243_80['Peptide'].apply(lambda x: x.count('M(+15.99)'))\n",
    "\n",
    "# use a lamba function to enumerate the # of deamidated N's in each peptide\n",
    "peaks243_80['n-deam'] = peaks243_80['Peptide'].apply(lambda x: x.count('N(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "peaks243_80['q-deam'] = peaks243_80['Peptide'].apply(lambda x: x.count('Q(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of hydroxylated K's in each peptide\n",
    "peaks243_80['k-hydr'] = peaks243_80['Peptide'].apply(lambda x: x.count('K(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of hydroxylated P's in each peptide\n",
    "#peaks243_80['p-hydr'] = peaks243_80['Peptide'].apply(lambda x: x.count('P(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "peaks243_80['r-meth'] = peaks243_80['Peptide'].apply(lambda x: x.count('R(+14.02)'))\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "peaks243_80['stripped peptide'] = peaks243_80['Peptide'].str.replace(r\"\\(.*\\)\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "peaks243_80['stripped length'] = peaks243_80['stripped peptide'].apply(len)\n",
    "\n",
    "# total the number of modifications in sequence\n",
    "peaks243_80['ptm-total'] = peaks243_80['c-carb'] + peaks243_80['m-oxid'] + peaks243_80['n-deam'] + \\\n",
    "peaks243_80['q-deam'] + peaks243_80['k-hydr'] + peaks243_80['r-meth']\n",
    "\n",
    "# calculate NAAF numerator for each peptide k\n",
    "peaks243_80['NAAF num.'] = peaks243_80['Area'] / peaks243_80['stripped length']\n",
    "\n",
    "# write modified dataframe to new csv file\n",
    "peaks243_80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/243/ETNP-SKQ17-243-965m-0.3-JA14_PTMopt_15ppm_DN80.csv\")\n",
    "\n",
    "# check out the results\n",
    "peaks243_80.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>K</th>\n",
       "      <th>L</th>\n",
       "      <th>...</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>Total area</th>\n",
       "      <th>Total length</th>\n",
       "      <th>% C w/ carb</th>\n",
       "      <th>% M w/ oxid</th>\n",
       "      <th>% N w/ deam</th>\n",
       "      <th>% Q w/ deam</th>\n",
       "      <th>% K w/ hydr</th>\n",
       "      <th>% R w/ meth</th>\n",
       "      <th>NAAF denom.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>197</td>\n",
       "      <td>20</td>\n",
       "      <td>89</td>\n",
       "      <td>137</td>\n",
       "      <td>70</td>\n",
       "      <td>95</td>\n",
       "      <td>21</td>\n",
       "      <td>0</td>\n",
       "      <td>151</td>\n",
       "      <td>322</td>\n",
       "      <td>...</td>\n",
       "      <td>51</td>\n",
       "      <td>2.933053e+09</td>\n",
       "      <td>2078</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.583333</td>\n",
       "      <td>0.330508</td>\n",
       "      <td>0.066667</td>\n",
       "      <td>0.15894</td>\n",
       "      <td>0.260204</td>\n",
       "      <td>1.411479e+06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                A   C   D    E   F   G   H  I    K    L  ...  r-meth  \\\n",
       "sample total  197  20  89  137  70  95  21  0  151  322  ...      51   \n",
       "\n",
       "                Total area  Total length  % C w/ carb  % M w/ oxid  \\\n",
       "sample total  2.933053e+09          2078          1.0     0.583333   \n",
       "\n",
       "              % N w/ deam  % Q w/ deam  % K w/ hydr  % R w/ meth   NAAF denom.  \n",
       "sample total     0.330508     0.066667      0.15894     0.260204  1.411479e+06  \n",
       "\n",
       "[1 rows x 35 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of certain columns in modified \n",
    "#peptide dataframe above (for >80% ALC)\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'A': peaks243_80['A'].sum(),\n",
    "        'C': peaks243_80['C'].sum(),\n",
    "        'D': peaks243_80['D'].sum(),\n",
    "        'E': peaks243_80['E'].sum(),\n",
    "        'F': peaks243_80['F'].sum(),\n",
    "        'G': peaks243_80['G'].sum(),\n",
    "        'H': peaks243_80['H'].sum(),\n",
    "        'I': peaks243_80['I'].sum(),\n",
    "        'K': peaks243_80['K'].sum(),\n",
    "        'L': peaks243_80['L'].sum(),\n",
    "        'M': peaks243_80['M'].sum(),\n",
    "        'N': peaks243_80['N'].sum(),\n",
    "        'P': peaks243_80['P'].sum(),\n",
    "        'Q': peaks243_80['Q'].sum(),\n",
    "        'R': peaks243_80['R'].sum(),\n",
    "        'S': peaks243_80['S'].sum(),\n",
    "        'T': peaks243_80['T'].sum(),\n",
    "        'V': peaks243_80['V'].sum(),\n",
    "        'W': peaks243_80['W'].sum(),\n",
    "        'Y': peaks243_80['Y'].sum(),\n",
    "        'c-carb': peaks243_80['c-carb'].sum(),\n",
    "        'm-oxid': peaks243_80['m-oxid'].sum(),\n",
    "        'n-deam': peaks243_80['n-deam'].sum(),\n",
    "        'q-deam': peaks243_80['q-deam'].sum(),\n",
    "        'k-hydr': peaks243_80['k-hydr'].sum(),\n",
    "        'r-meth': peaks243_80['r-meth'].sum(),\n",
    "        'Total area': peaks243_80['Area'].sum(),\n",
    "        'Total length': peaks243_80['stripped length'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaks243_80 = pd.DataFrame(data, columns=['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M',\\\n",
    "                                               'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb',\\\n",
    "                                               'm-oxid', 'n-deam', 'q-deam', 'k-hydr', 'r-meth',\\\n",
    "                                               'Total area', 'Total length'], index=index)\n",
    "\n",
    "# calculate percentage of C's with carb (should be 1.0)\n",
    "totalpeaks243_80['% C w/ carb'] = totalpeaks243_80['c-carb'] / totalpeaks243_80['C'] \n",
    "\n",
    "# calculate percentage of M's that are oxidized\n",
    "totalpeaks243_80['% M w/ oxid'] = totalpeaks243_80['m-oxid'] / totalpeaks243_80['M'] \n",
    "\n",
    "# calculate percentage of N's that are deamidated\n",
    "totalpeaks243_80['% N w/ deam'] = totalpeaks243_80['n-deam'] / totalpeaks243_80['N'] \n",
    "\n",
    "# calculate percentage of Q's that are deamidated\n",
    "totalpeaks243_80['% Q w/ deam'] = totalpeaks243_80['q-deam'] / totalpeaks243_80['Q'] \n",
    "\n",
    "# calculate percentage of K's that are hydroxylated\n",
    "totalpeaks243_80['% K w/ hydr'] = totalpeaks243_80['k-hydr'] / totalpeaks243_80['K'] \n",
    "\n",
    "# calculate percentage of P's that are hydroxylated\n",
    "#totalpeaks243_80['% P w/ hydr'] = totalpeaks243_80['p-hydr'] / totalpeaks243_80['K'] \n",
    "\n",
    "# calculate percentage of R's that are methylated\n",
    "totalpeaks243_80['% R w/ meth'] = totalpeaks243_80['r-meth'] / totalpeaks243_80['R'] \n",
    "\n",
    "# calculate NAAF denominator for all peptides in dataset i\n",
    "totalpeaks243_80['NAAF denom.'] = totalpeaks243_80['Total area'] / totalpeaks243_80['Total length']\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "totalpeaks243_80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/243/ETNP-SKQ17-243-965m-0.3-JA14_PTMopt_15ppm_DN80_totals.csv\")\n",
    "\n",
    "totalpeaks243_80.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>NAAF factor</th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>...</th>\n",
       "      <th>T-NAAF80</th>\n",
       "      <th>V-NAAF80</th>\n",
       "      <th>W-NAAF80</th>\n",
       "      <th>Y-NAAF80</th>\n",
       "      <th>ccarb-NAAF80</th>\n",
       "      <th>moxid-NAAF80</th>\n",
       "      <th>ndeam-NAAF80</th>\n",
       "      <th>qdeam-NAAF80</th>\n",
       "      <th>khydr-NAAF80</th>\n",
       "      <th>rmeth-NAAF80</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TDEFVR</td>\n",
       "      <td>0.018657</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.018657</td>\n",
       "      <td>0.018657</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KYVFVK</td>\n",
       "      <td>0.026568</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.053136</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.026568</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TSGEVFR</td>\n",
       "      <td>0.028845</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.028845</td>\n",
       "      <td>0.028845</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LNLAQNWR</td>\n",
       "      <td>0.036044</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.036044</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.036044</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MVK</td>\n",
       "      <td>0.109814</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.109814</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.109814</td>\n",
       "      <td>0.109814</td>\n",
       "      <td>0.109814</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  stripped peptide  NAAF factor  A  C  D  E  F  G  H  I  ...  T-NAAF80  \\\n",
       "0           TDEFVR     0.018657  0  0  1  1  1  0  0  0  ...  0.018657   \n",
       "1           KYVFVK     0.026568  0  0  0  0  1  0  0  0  ...  0.000000   \n",
       "2          TSGEVFR     0.028845  0  0  0  1  1  1  0  0  ...  0.028845   \n",
       "3         LNLAQNWR     0.036044  1  0  0  0  0  0  0  0  ...  0.000000   \n",
       "4              MVK     0.109814  1  1  0  1  0  0  0  0  ...  0.000000   \n",
       "\n",
       "   V-NAAF80  W-NAAF80  Y-NAAF80  ccarb-NAAF80  moxid-NAAF80  ndeam-NAAF80  \\\n",
       "0  0.018657  0.000000  0.000000      0.000000      0.000000      0.000000   \n",
       "1  0.053136  0.000000  0.026568      0.000000      0.000000      0.000000   \n",
       "2  0.028845  0.000000  0.000000      0.000000      0.000000      0.000000   \n",
       "3  0.000000  0.036044  0.000000      0.000000      0.000000      0.036044   \n",
       "4  0.109814  0.000000  0.109814      0.109814      0.109814      0.000000   \n",
       "\n",
       "   qdeam-NAAF80  khydr-NAAF80  rmeth-NAAF80  \n",
       "0           0.0           0.0           0.0  \n",
       "1           0.0           0.0           0.0  \n",
       "2           0.0           0.0           0.0  \n",
       "3           0.0           0.0           0.0  \n",
       "4           0.0           0.0           0.0  \n",
       "\n",
       "[5 rows x 54 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use the calculated NAAF factor (in totalpeaks243 dataframe, above) to caluclate the NAAF \n",
    "# NAAF: normalized normalized area abundance factor\n",
    "\n",
    "# don't have to worry here about DECOY hits messing with Area totals\n",
    "# but we will with Comet results\n",
    "\n",
    "NAAF80 = 1.411479e+06\n",
    "\n",
    "# use NAAF >80% ALC to get NAAF\n",
    "peaks243_80['NAAF factor'] = (peaks243_80['NAAF num.'])/NAAF80\n",
    "\n",
    "# make a dataframe that contains only what we need: sequences, AAs, PTMs\n",
    "peaks243_AA80 = peaks243_80[['stripped peptide', 'NAAF factor', 'A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', \\\n",
    "                             'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', 'm-oxid', \\\n",
    "                             'n-deam', 'q-deam', 'k-hydr', 'r-meth']].copy()\n",
    "\n",
    "# multiply the NAAF80 factor by the AAs to normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaks243_AA80['A-NAAF80'] = peaks243_AA80['A'] * peaks243_80['NAAF factor']\n",
    "peaks243_AA80['C-NAAF80'] = peaks243_AA80['C'] * peaks243_80['NAAF factor']\n",
    "peaks243_AA80['D-NAAF80'] = peaks243_AA80['D'] * peaks243_80['NAAF factor']\n",
    "peaks243_AA80['E-NAAF80'] = peaks243_AA80['E'] * peaks243_80['NAAF factor']\n",
    "peaks243_AA80['F-NAAF80'] = peaks243_AA80['F'] * peaks243_80['NAAF factor']\n",
    "peaks243_AA80['G-NAAF80'] = peaks243_AA80['G'] * peaks243_80['NAAF factor']\n",
    "peaks243_AA80['H-NAAF80'] = peaks243_AA80['H'] * peaks243_80['NAAF factor']\n",
    "peaks243_AA80['I-NAAF80'] = peaks243_AA80['I'] * peaks243_80['NAAF factor']\n",
    "peaks243_AA80['K-NAAF80'] = peaks243_AA80['K'] * peaks243_80['NAAF factor']\n",
    "peaks243_AA80['L-NAAF80'] = peaks243_AA80['L'] * peaks243_80['NAAF factor']\n",
    "peaks243_AA80['M-NAAF80'] = peaks243_AA80['M'] * peaks243_80['NAAF factor']\n",
    "peaks243_AA80['N-NAAF80'] = peaks243_AA80['N'] * peaks243_80['NAAF factor']\n",
    "peaks243_AA80['P-NAAF80'] = peaks243_AA80['P'] * peaks243_80['NAAF factor']\n",
    "peaks243_AA80['Q-NAAF80'] = peaks243_AA80['Q'] * peaks243_80['NAAF factor']\n",
    "peaks243_AA80['R-NAAF80'] = peaks243_AA80['R'] * peaks243_80['NAAF factor']\n",
    "peaks243_AA80['S-NAAF80'] = peaks243_AA80['S'] * peaks243_80['NAAF factor']\n",
    "peaks243_AA80['T-NAAF80'] = peaks243_AA80['T'] * peaks243_80['NAAF factor']\n",
    "peaks243_AA80['V-NAAF80'] = peaks243_AA80['V'] * peaks243_80['NAAF factor']\n",
    "peaks243_AA80['W-NAAF80'] = peaks243_AA80['W'] * peaks243_80['NAAF factor']\n",
    "peaks243_AA80['Y-NAAF80'] = peaks243_AA80['Y'] * peaks243_80['NAAF factor']\n",
    "\n",
    "# multiply the NAAF80 factor by the PTMs normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaks243_AA80['ccarb-NAAF80'] = peaks243_AA80['c-carb'] * peaks243_AA80['NAAF factor']\n",
    "peaks243_AA80['moxid-NAAF80'] = peaks243_AA80['m-oxid'] * peaks243_AA80['NAAF factor']\n",
    "peaks243_AA80['ndeam-NAAF80'] = peaks243_AA80['n-deam'] * peaks243_AA80['NAAF factor']\n",
    "peaks243_AA80['qdeam-NAAF80'] = peaks243_AA80['q-deam'] * peaks243_AA80['NAAF factor']\n",
    "peaks243_AA80['khydr-NAAF80'] = peaks243_AA80['k-hydr'] * peaks243_AA80['NAAF factor']\n",
    "peaks243_AA80['rmeth-NAAF80'] = peaks243_AA80['r-meth'] * peaks243_AA80['NAAF factor']\n",
    "\n",
    "# write the dataframe to a new csv\n",
    "peaks243_AA80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/243/ETNP-SKQ17-243-965m-0.3-JA14_PTMopt_15ppm_DN80_NAAF.csv\")\n",
    "\n",
    "peaks243_AA80.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NAAF</th>\n",
       "      <th>A-NAAF</th>\n",
       "      <th>C-NAAF</th>\n",
       "      <th>D-NAAF</th>\n",
       "      <th>E-NAAF</th>\n",
       "      <th>F-NAAF</th>\n",
       "      <th>G-NAAF</th>\n",
       "      <th>H-NAAF</th>\n",
       "      <th>I-NAAF</th>\n",
       "      <th>K-NAAF</th>\n",
       "      <th>...</th>\n",
       "      <th>Q-deam-NAAF</th>\n",
       "      <th>K-hydr-NAAF</th>\n",
       "      <th>R-meth-NAAF</th>\n",
       "      <th>% C w/ carb. NAAF</th>\n",
       "      <th>% M w/ oxid. NAAF</th>\n",
       "      <th>% N w/ deam. NAAF</th>\n",
       "      <th>% Q w/ deam. NAAF</th>\n",
       "      <th>% K w/ hydr. NAAF</th>\n",
       "      <th>% R w/ meth. NAAF</th>\n",
       "      <th>NAAF check</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>259.74375</td>\n",
       "      <td>290.615333</td>\n",
       "      <td>12.075153</td>\n",
       "      <td>15.923436</td>\n",
       "      <td>85.047125</td>\n",
       "      <td>8.721283</td>\n",
       "      <td>34.709539</td>\n",
       "      <td>0.326574</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.231374</td>\n",
       "      <td>...</td>\n",
       "      <td>7.567013</td>\n",
       "      <td>6.34669</td>\n",
       "      <td>151.632947</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.536576</td>\n",
       "      <td>0.651574</td>\n",
       "      <td>0.22409</td>\n",
       "      <td>0.416685</td>\n",
       "      <td>0.547185</td>\n",
       "      <td>0.000184</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   NAAF      A-NAAF     C-NAAF     D-NAAF     E-NAAF  \\\n",
       "sample total  259.74375  290.615333  12.075153  15.923436  85.047125   \n",
       "\n",
       "                F-NAAF     G-NAAF    H-NAAF  I-NAAF     K-NAAF  ...  \\\n",
       "sample total  8.721283  34.709539  0.326574     0.0  15.231374  ...   \n",
       "\n",
       "              Q-deam-NAAF  K-hydr-NAAF  R-meth-NAAF  % C w/ carb. NAAF  \\\n",
       "sample total     7.567013      6.34669   151.632947                1.0   \n",
       "\n",
       "              % M w/ oxid. NAAF  % N w/ deam. NAAF  % Q w/ deam. NAAF  \\\n",
       "sample total           0.536576           0.651574            0.22409   \n",
       "\n",
       "              % K w/ hydr. NAAF  % R w/ meth. NAAF  NAAF check  \n",
       "sample total           0.416685           0.547185    0.000184  \n",
       "\n",
       "[1 rows x 34 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of NAAF normalized AAs for comet243 PeaksDN results\n",
    "# also contains the sums of the NAAF-corrected PTMs occurances for each affected residue\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'NAAF': peaks243_AA80['NAAF factor'].sum(),\n",
    "        'A-NAAF': peaks243_AA80['A-NAAF80'].sum(),\n",
    "        'C-NAAF': peaks243_AA80['C-NAAF80'].sum(),\n",
    "        'D-NAAF': peaks243_AA80['D-NAAF80'].sum(),\n",
    "        'E-NAAF': peaks243_AA80['E-NAAF80'].sum(),\n",
    "        'F-NAAF': peaks243_AA80['F-NAAF80'].sum(),\n",
    "        'G-NAAF': peaks243_AA80['G-NAAF80'].sum(),\n",
    "        'H-NAAF': peaks243_AA80['H-NAAF80'].sum(),\n",
    "        'I-NAAF': peaks243_AA80['I-NAAF80'].sum(),\n",
    "        'K-NAAF': peaks243_AA80['K-NAAF80'].sum(),\n",
    "        'L-NAAF': peaks243_AA80['L-NAAF80'].sum(),\n",
    "        'M-NAAF': peaks243_AA80['M-NAAF80'].sum(),\n",
    "        'N-NAAF': peaks243_AA80['N-NAAF80'].sum(),\n",
    "        'P-NAAF': peaks243_AA80['P-NAAF80'].sum(),\n",
    "        'Q-NAAF': peaks243_AA80['Q-NAAF80'].sum(),\n",
    "        'R-NAAF': peaks243_AA80['R-NAAF80'].sum(),\n",
    "        'S-NAAF': peaks243_AA80['S-NAAF80'].sum(),\n",
    "        'T-NAAF': peaks243_AA80['T-NAAF80'].sum(),\n",
    "        'V-NAAF': peaks243_AA80['V-NAAF80'].sum(),\n",
    "        'W-NAAF': peaks243_AA80['W-NAAF80'].sum(),\n",
    "        'Y-NAAF': peaks243_AA80['Y-NAAF80'].sum(),\n",
    "        'C-carb-NAAF': peaks243_AA80['ccarb-NAAF80'].sum(),\n",
    "        'M-oxid-NAAF': peaks243_AA80['moxid-NAAF80'].sum(),\n",
    "        'N-deam-NAAF': peaks243_AA80['ndeam-NAAF80'].sum(),\n",
    "        'Q-deam-NAAF': peaks243_AA80['qdeam-NAAF80'].sum(),\n",
    "        'K-hydr-NAAF': peaks243_AA80['khydr-NAAF80'].sum(),\n",
    "        'R-meth-NAAF': peaks243_AA80['rmeth-NAAF80'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaks243_AA80 = pd.DataFrame(data, columns=['NAAF', 'A-NAAF', 'C-NAAF', 'D-NAAF', 'E-NAAF', 'F-NAAF', \\\n",
    "                                                   'G-NAAF', 'H-NAAF', 'I-NAAF', 'K-NAAF', 'L-NAAF', 'M-NAAF', \\\n",
    "                                                   'N-NAAF', 'P-NAAF', 'Q-NAAF', 'R-NAAF', 'S-NAAF', \\\n",
    "                                                   'T-NAAF', 'V-NAAF', 'W-NAAF', 'Y-NAAF', 'C-carb-NAAF', \\\n",
    "                                                   'M-oxid-NAAF', 'N-deam-NAAF', 'Q-deam-NAAF', 'K-hydr-NAAF',\\\n",
    "                                                   'R-meth-NAAF'], index=index)\n",
    "\n",
    "# calculate the NAAF-corrected % modified C, M, N, Q, K, P, and Rs\n",
    "\n",
    "\n",
    "totalpeaks243_AA80['% C w/ carb. NAAF'] = totalpeaks243_AA80['C-carb-NAAF'] / totalpeaks243_AA80['C-NAAF']\n",
    "totalpeaks243_AA80['% M w/ oxid. NAAF'] = totalpeaks243_AA80['M-oxid-NAAF'] / totalpeaks243_AA80['M-NAAF']\n",
    "totalpeaks243_AA80['% N w/ deam. NAAF'] = totalpeaks243_AA80['N-deam-NAAF'] / totalpeaks243_AA80['N-NAAF']\n",
    "totalpeaks243_AA80['% Q w/ deam. NAAF'] = totalpeaks243_AA80['Q-deam-NAAF'] / totalpeaks243_AA80['Q-NAAF']\n",
    "totalpeaks243_AA80['% K w/ hydr. NAAF'] = totalpeaks243_AA80['K-hydr-NAAF'] / totalpeaks243_AA80['K-NAAF']\n",
    "totalpeaks243_AA80['% R w/ meth. NAAF'] = totalpeaks243_AA80['R-meth-NAAF'] / totalpeaks243_AA80['R-NAAF']\n",
    "\n",
    "# calculate NAAF summed numerator over denominator (in above cell) for all peptides in dataset i: a check\n",
    "totalpeaks243_AA80['NAAF check'] = totalpeaks243_AA80['NAAF'] / 1.411479e+06\n",
    "\n",
    "# write modified dataframe to new txt file, same name + totals\n",
    "totalpeaks243_AA80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/243/ETNP-SKQ17-243-965m-0.3-JA14_PTMopt_15ppm_DN80_NAAF_totals.csv\")\n",
    "\n",
    "totalpeaks243_AA80.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Visualizing the results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ALC max:  98\n",
      "ALC min:  80\n"
     ]
    }
   ],
   "source": [
    "print(\"ALC max: \", peaks243_80['ALC (%)'].max())\n",
    "print(\"ALC min: \", peaks243_80['ALC (%)'].min())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exporting txt files of stripped peptides at confidence cutoffs:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total modified peptides in 243: 1694\n",
      "Deduplicated modified peptides in 243: 1670\n",
      "Total strippled peptides in 243: 1694\n",
      "Deduplicated stripped peptides in 243: 1615\n"
     ]
    }
   ],
   "source": [
    "# keep only peptide column >50% ALC\n",
    "pep243_50_moddup = peaks243_50[[\"Peptide\"]]\n",
    "\n",
    "# keep only the stripped peptide column for the >50% ALC\n",
    "# this is what we'll use for UniPept input, etc\n",
    "pep243_50_dup = peaks243_50[[\"stripped peptide\"]]\n",
    "\n",
    "# deduplicate both of these lists\n",
    "pep243_50_mod = pep243_50_moddup.drop_duplicates()\n",
    "pep243_50 = pep243_50_dup.drop_duplicates()\n",
    "\n",
    "# print out the #s of modified and stripped peptides, deduplicated and not\n",
    "\n",
    "print(\"Total modified peptides in 243:\", len(pep243_50_moddup))\n",
    "print(\"Deduplicated modified peptides in 243:\", len(pep243_50_mod))\n",
    "print(\"Total strippled peptides in 243:\", len(pep243_50_dup))\n",
    "print(\"Deduplicated stripped peptides in 243:\", len(pep243_50))\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "pep243_50.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/243/ETNP-SKQ17-243-965m-0.3-JA14_PTMopt_15ppm_DN50_stripped_peptides.txt\", header=False, index=False)\n",
    "\n",
    "# made the text file into a FASTA \n",
    "\n",
    "!awk '{print \">\"NR\"\\n\"$0}' /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/243/ETNP-SKQ17-243-965m-0.3-JA14_PTMopt_15ppm_DN50_stripped_peptides.txt > \\\n",
    "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/243/ETNP-SKQ17-243-965m-0.3-JA14_PTMopt_15ppm_DN50_stripped_peptides.fas\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total modified peptides in 243: 262\n",
      "Deduplicated modified peptides in 243: 255\n",
      "Total strippled peptides in 243: 262\n",
      "Deduplicated stripped peptides in 243: 254\n"
     ]
    }
   ],
   "source": [
    "# keep only peptide column >80% ALC\n",
    "pep243_80_moddup = peaks243_80[[\"Peptide\"]]\n",
    "\n",
    "# keep only the stripped peptide column for the >80% ALC\n",
    "# this is what we'll use for UniPept input, etc\n",
    "pep243_80_dup = peaks243_80[[\"stripped peptide\"]]\n",
    "\n",
    "# deduplicate both of these lists\n",
    "pep243_80_mod = pep243_80_moddup.drop_duplicates()\n",
    "pep243_80 = pep243_80_dup.drop_duplicates()\n",
    "\n",
    "# print out the #s of modified and stripped peptides, deduplicated and not\n",
    "\n",
    "print(\"Total modified peptides in 243:\", len(pep243_80_moddup))\n",
    "print(\"Deduplicated modified peptides in 243:\", len(pep243_80_mod))\n",
    "print(\"Total strippled peptides in 243:\", len(pep243_80_dup))\n",
    "print(\"Deduplicated stripped peptides in 243:\", len(pep243_80))\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "pep243_80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/243/ETNP-SKQ17-243-965m-0.3-JA14_PTMopt_15ppm_DN80_stripped_peptides.txt\", header=False, index=False)\n",
    "\n",
    "# made the text file into a FASTA \n",
    "\n",
    "!awk '{print \">\"NR\"\\n\"$0}' /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/243/ETNP-SKQ17-243-965m-0.3-JA14_PTMopt_15ppm_DN80_stripped_peptides.txt > \\\n",
    "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/243/ETNP-SKQ17-243-965m-0.3-JA14_PTMopt_15ppm_DN80_stripped_peptides.fas\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
