{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Manipulation of Peaks de novo results of ETNP 2017 P2 samples LC-MS/MS data using python.\n",
    "\n",
    "Starting with:\n",
    "\n",
    "    Peaks de novo results (.csv) of PTM-optimized database searches > ALC 50%\n",
    "\n",
    "Goal:\n",
    "\n",
    "    Files with stripped (no PTMs) peptide lists and\n",
    "    Columns with #'s of each modification in every sequence\n",
    "    Column with stripped peptide lengths (# amino acids)\n",
    "    \n",
    "### To use for a different file:\n",
    "\n",
    "#### 1. Change the input file name in *IN 4*\n",
    "#### 2. Use 'find + replace' (Esc + F) to replace the running # (e.g., 233) for another\n",
    "#### 3. Update the NAAF factor calculated in *IN 6* into *IN 7*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We don't have technical duplicates here, sadly, unlike the MED4 Pro samples. I exported PeaksDN search results CSVs into my ETNP 2017 git repo:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt\n"
     ]
    }
   ],
   "source": [
    "cd /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;34m231\u001b[0m/  \u001b[01;34m233\u001b[0m/  \u001b[01;34m243\u001b[0m/  \u001b[01;34m273\u001b[0m/  \u001b[01;34m278\u001b[0m/  \u001b[01;34m378\u001b[0m/\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ETNP_SKQ17_DENOVO_162_233-265m-0.3-JA4_DN50_15ppm.csv\r\n",
      "ETNP_SKQ17_DENOVO_162_233-265m-0.3-JA4_DN50_15ppm.xml\r\n"
     ]
    }
   ],
   "source": [
    "ls 233"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.0.5'"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# LIBRARIES\n",
    "#import pandas library for working with tabular data\n",
    "import os\n",
    "os.getcwd()\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "from scipy.stats import kde\n",
    "#import regular expresson (regex)\n",
    "import re\n",
    "#check pandas version\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 233: 265 m McLane pump filtered on 0.3 um GF-75"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# redundant Peaks peptides >50% ALC in dataframe 4235\n",
      "# redundant Peaks peptides >80% ALC in dataframe 730\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Fraction</th>\n",
       "      <th>Scan</th>\n",
       "      <th>Source File</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Tag Length</th>\n",
       "      <th>ALC (%)</th>\n",
       "      <th>length</th>\n",
       "      <th>m/z</th>\n",
       "      <th>z</th>\n",
       "      <th>RT</th>\n",
       "      <th>Area</th>\n",
       "      <th>Mass</th>\n",
       "      <th>ppm</th>\n",
       "      <th>PTM</th>\n",
       "      <th>local confidence (%)</th>\n",
       "      <th>tag (&gt;=0%)</th>\n",
       "      <th>mode</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>14553</td>\n",
       "      <td>20170410__ETNP-233-265m-0.3um-JA4_01.raw</td>\n",
       "      <td>EN(+.98)LAALEK</td>\n",
       "      <td>8</td>\n",
       "      <td>98</td>\n",
       "      <td>8</td>\n",
       "      <td>444.7376</td>\n",
       "      <td>2</td>\n",
       "      <td>49.92</td>\n",
       "      <td>8900000.0</td>\n",
       "      <td>887.4600</td>\n",
       "      <td>0.7</td>\n",
       "      <td>Deamidation (NQ)</td>\n",
       "      <td>97 99 99 99 99 99 100 97</td>\n",
       "      <td>EN(+.98)LAALEK</td>\n",
       "      <td>CID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>26516</td>\n",
       "      <td>20170410__ETNP-233-265m-0.3um-JA4_01.raw</td>\n",
       "      <td>VGC(+57.02)DEGLFEELPR</td>\n",
       "      <td>13</td>\n",
       "      <td>98</td>\n",
       "      <td>13</td>\n",
       "      <td>760.8570</td>\n",
       "      <td>2</td>\n",
       "      <td>81.30</td>\n",
       "      <td>9410000.0</td>\n",
       "      <td>1519.6978</td>\n",
       "      <td>1.1</td>\n",
       "      <td>Carbamidomethylation</td>\n",
       "      <td>92 95 100 100 100 98 99 99 99 100 100 99 99</td>\n",
       "      <td>VGC(+57.02)DEGLFEELPR</td>\n",
       "      <td>CID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>22453</td>\n",
       "      <td>20170410__ETNP-233-265m-0.3um-JA4_01.raw</td>\n",
       "      <td>WSVVFK</td>\n",
       "      <td>6</td>\n",
       "      <td>98</td>\n",
       "      <td>6</td>\n",
       "      <td>383.2187</td>\n",
       "      <td>2</td>\n",
       "      <td>72.23</td>\n",
       "      <td>322000.0</td>\n",
       "      <td>764.4221</td>\n",
       "      <td>1.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>96 98 98 99 99 99</td>\n",
       "      <td>WSVVFK</td>\n",
       "      <td>CID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>19513</td>\n",
       "      <td>20170410__ETNP-233-265m-0.3um-JA4_01.raw</td>\n",
       "      <td>FDLLVNK</td>\n",
       "      <td>7</td>\n",
       "      <td>97</td>\n",
       "      <td>7</td>\n",
       "      <td>424.7480</td>\n",
       "      <td>2</td>\n",
       "      <td>64.92</td>\n",
       "      <td>1520000.0</td>\n",
       "      <td>847.4803</td>\n",
       "      <td>1.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>94 97 100 99 98 97 98</td>\n",
       "      <td>FDLLVNK</td>\n",
       "      <td>CID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>13937</td>\n",
       "      <td>20170410__ETNP-233-265m-0.3um-JA4_01.raw</td>\n",
       "      <td>EPLGPVVR</td>\n",
       "      <td>8</td>\n",
       "      <td>97</td>\n",
       "      <td>8</td>\n",
       "      <td>433.7584</td>\n",
       "      <td>2</td>\n",
       "      <td>47.83</td>\n",
       "      <td>4020000.0</td>\n",
       "      <td>865.5021</td>\n",
       "      <td>0.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>97 98 98 96 99 97 99 95</td>\n",
       "      <td>EPLGPVVR</td>\n",
       "      <td>CID</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Fraction   Scan                               Source File  \\\n",
       "0         2  14553  20170410__ETNP-233-265m-0.3um-JA4_01.raw   \n",
       "1         2  26516  20170410__ETNP-233-265m-0.3um-JA4_01.raw   \n",
       "2         2  22453  20170410__ETNP-233-265m-0.3um-JA4_01.raw   \n",
       "3         2  19513  20170410__ETNP-233-265m-0.3um-JA4_01.raw   \n",
       "4         2  13937  20170410__ETNP-233-265m-0.3um-JA4_01.raw   \n",
       "\n",
       "                 Peptide  Tag Length  ALC (%)  length       m/z  z     RT  \\\n",
       "0         EN(+.98)LAALEK           8       98       8  444.7376  2  49.92   \n",
       "1  VGC(+57.02)DEGLFEELPR          13       98      13  760.8570  2  81.30   \n",
       "2                 WSVVFK           6       98       6  383.2187  2  72.23   \n",
       "3                FDLLVNK           7       97       7  424.7480  2  64.92   \n",
       "4               EPLGPVVR           8       97       8  433.7584  2  47.83   \n",
       "\n",
       "        Area       Mass  ppm                   PTM  \\\n",
       "0  8900000.0   887.4600  0.7      Deamidation (NQ)   \n",
       "1  9410000.0  1519.6978  1.1  Carbamidomethylation   \n",
       "2   322000.0   764.4221  1.1                   NaN   \n",
       "3  1520000.0   847.4803  1.3                   NaN   \n",
       "4  4020000.0   865.5021  0.2                   NaN   \n",
       "\n",
       "                          local confidence (%)             tag (>=0%) mode  \n",
       "0                     97 99 99 99 99 99 100 97         EN(+.98)LAALEK  CID  \n",
       "1  92 95 100 100 100 98 99 99 99 100 100 99 99  VGC(+57.02)DEGLFEELPR  CID  \n",
       "2                            96 98 98 99 99 99                 WSVVFK  CID  \n",
       "3                        94 97 100 99 98 97 98                FDLLVNK  CID  \n",
       "4                      97 98 98 96 99 97 99 95               EPLGPVVR  CID  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read the CSV into a dataframe using the pandas read_csv function\n",
    "peaks233_50 = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/233/ETNP_SKQ17_DENOVO_162_233-265m-0.3-JA4_DN50_15ppm.csv\")\n",
    "\n",
    "# also make a dataframe but keep only peptides >80% ALC\n",
    "peaks233_80 = peaks233_50.loc[peaks233_50['ALC (%)'] >= 80].copy()\n",
    "\n",
    "# how many de novo sequence candidates >50% ALC?\n",
    "print(\"# redundant Peaks peptides >50% ALC in dataframe\", len(peaks233_50))\n",
    "# how many de novo sequence candidates >80ALC?\n",
    "print(\"# redundant Peaks peptides >80% ALC in dataframe\", len(peaks233_80))\n",
    "\n",
    "#look at the dataframe\n",
    "peaks233_50.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Fraction</th>\n",
       "      <th>Scan</th>\n",
       "      <th>Source File</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Tag Length</th>\n",
       "      <th>ALC (%)</th>\n",
       "      <th>length</th>\n",
       "      <th>m/z</th>\n",
       "      <th>z</th>\n",
       "      <th>RT</th>\n",
       "      <th>...</th>\n",
       "      <th>c-carb</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>k-hydr</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>stripped length</th>\n",
       "      <th>ptm-total</th>\n",
       "      <th>NAAF num.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>14553</td>\n",
       "      <td>20170410__ETNP-233-265m-0.3um-JA4_01.raw</td>\n",
       "      <td>EN(+.98)LAALEK</td>\n",
       "      <td>8</td>\n",
       "      <td>98</td>\n",
       "      <td>8</td>\n",
       "      <td>444.7376</td>\n",
       "      <td>2</td>\n",
       "      <td>49.92</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ENLAALEK</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>1.112500e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>26516</td>\n",
       "      <td>20170410__ETNP-233-265m-0.3um-JA4_01.raw</td>\n",
       "      <td>VGC(+57.02)DEGLFEELPR</td>\n",
       "      <td>13</td>\n",
       "      <td>98</td>\n",
       "      <td>13</td>\n",
       "      <td>760.8570</td>\n",
       "      <td>2</td>\n",
       "      <td>81.30</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>VGCDEGLFEELPR</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>7.238462e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>22453</td>\n",
       "      <td>20170410__ETNP-233-265m-0.3um-JA4_01.raw</td>\n",
       "      <td>WSVVFK</td>\n",
       "      <td>6</td>\n",
       "      <td>98</td>\n",
       "      <td>6</td>\n",
       "      <td>383.2187</td>\n",
       "      <td>2</td>\n",
       "      <td>72.23</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>WSVVFK</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>5.366667e+04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>19513</td>\n",
       "      <td>20170410__ETNP-233-265m-0.3um-JA4_01.raw</td>\n",
       "      <td>FDLLVNK</td>\n",
       "      <td>7</td>\n",
       "      <td>97</td>\n",
       "      <td>7</td>\n",
       "      <td>424.7480</td>\n",
       "      <td>2</td>\n",
       "      <td>64.92</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>FDLLVNK</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>2.171429e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>13937</td>\n",
       "      <td>20170410__ETNP-233-265m-0.3um-JA4_01.raw</td>\n",
       "      <td>EPLGPVVR</td>\n",
       "      <td>8</td>\n",
       "      <td>97</td>\n",
       "      <td>8</td>\n",
       "      <td>433.7584</td>\n",
       "      <td>2</td>\n",
       "      <td>47.83</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>EPLGPVVR</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>5.025000e+05</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Fraction   Scan                               Source File  \\\n",
       "0         2  14553  20170410__ETNP-233-265m-0.3um-JA4_01.raw   \n",
       "1         2  26516  20170410__ETNP-233-265m-0.3um-JA4_01.raw   \n",
       "2         2  22453  20170410__ETNP-233-265m-0.3um-JA4_01.raw   \n",
       "3         2  19513  20170410__ETNP-233-265m-0.3um-JA4_01.raw   \n",
       "4         2  13937  20170410__ETNP-233-265m-0.3um-JA4_01.raw   \n",
       "\n",
       "                 Peptide  Tag Length  ALC (%)  length       m/z  z     RT  \\\n",
       "0         EN(+.98)LAALEK           8       98       8  444.7376  2  49.92   \n",
       "1  VGC(+57.02)DEGLFEELPR          13       98      13  760.8570  2  81.30   \n",
       "2                 WSVVFK           6       98       6  383.2187  2  72.23   \n",
       "3                FDLLVNK           7       97       7  424.7480  2  64.92   \n",
       "4               EPLGPVVR           8       97       8  433.7584  2  47.83   \n",
       "\n",
       "   ...  c-carb  m-oxid  n-deam q-deam k-hydr r-meth stripped peptide  \\\n",
       "0  ...       0       0       1      0      0      0         ENLAALEK   \n",
       "1  ...       1       0       0      0      0      0    VGCDEGLFEELPR   \n",
       "2  ...       0       0       0      0      0      0           WSVVFK   \n",
       "3  ...       0       0       0      0      0      0          FDLLVNK   \n",
       "4  ...       0       0       0      0      0      0         EPLGPVVR   \n",
       "\n",
       "   stripped length  ptm-total     NAAF num.  \n",
       "0                8          1  1.112500e+06  \n",
       "1               13          1  7.238462e+05  \n",
       "2                6          0  5.366667e+04  \n",
       "3                7          0  2.171429e+05  \n",
       "4                8          0  5.025000e+05  \n",
       "\n",
       "[5 rows x 47 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use a count function to enumerate the # of A's (alanines) in each peptide\n",
    "peaks233_50['A'] = peaks233_50['Peptide'].str.count(\"A\")\n",
    "\n",
    "# use a count function to enumerate the # of C's (cysteines) in each peptide\n",
    "peaks233_50['C'] = peaks233_50['Peptide'].str.count(\"C\")\n",
    "\n",
    "# use a count function to enumerate the # of D's (aspartic acids) in each peptide\n",
    "peaks233_50['D'] = peaks233_50['Peptide'].str.count(\"D\")\n",
    "\n",
    "# use a count function to enumerate the # of E's (glutamic acids) in each peptide\n",
    "peaks233_50['E'] = peaks233_50['Peptide'].str.count(\"E\")\n",
    "\n",
    "# use a count function to enumerate the # of F's (phenylalanines) in each peptide\n",
    "peaks233_50['F'] = peaks233_50['Peptide'].str.count(\"F\")\n",
    "\n",
    "# use a count function to enumerate the # of G's (glycines) in each peptide\n",
    "peaks233_50['G'] = peaks233_50['Peptide'].str.count(\"G\")\n",
    "\n",
    "# use a count function to enumerate the # of H's (histidines) in each peptide\n",
    "peaks233_50['H'] = peaks233_50['Peptide'].str.count(\"H\")\n",
    "\n",
    "# use a count function to enumerate the # of I's (isoleucines) in each peptide\n",
    "# in peaks233_50 output, there will be no isoleucines (they're lumped in with leucines)\n",
    "peaks233_50['I'] = peaks233_50['Peptide'].str.count(\"I\")\n",
    "\n",
    "# use a count function to enumerate the # of K's (lysines) in each peptide\n",
    "peaks233_50['K'] = peaks233_50['Peptide'].str.count(\"K\")\n",
    "\n",
    "# use a count function to enumerate the # of L's (leucines) in each peptide\n",
    "# also these include the isoleucines\n",
    "peaks233_50['L'] = peaks233_50['Peptide'].str.count(\"L\")\n",
    "\n",
    "# use a count function to enumerate the # of M's (methionines) in each peptide\n",
    "peaks233_50['M'] = peaks233_50['Peptide'].str.count(\"M\")\n",
    "\n",
    "# use a count function to enumerate the # of N's (asparagines) in each peptide\n",
    "peaks233_50['N'] = peaks233_50['Peptide'].str.count(\"N\")\n",
    "\n",
    "# use a count function to enumerate the # of P's ([prolines]) in each peptide\n",
    "peaks233_50['P'] = peaks233_50['Peptide'].str.count(\"P\")\n",
    "\n",
    "# use a count function to enumerate the # of Q's (glutamines) in each peptide\n",
    "peaks233_50['Q'] = peaks233_50['Peptide'].str.count(\"Q\")\n",
    "\n",
    "# use a count function to enumerate the # of R's (arginines) in each peptide\n",
    "peaks233_50['R'] = peaks233_50['Peptide'].str.count(\"R\")\n",
    "\n",
    "# use a count function to enumerate the # of S's (serines) in each peptide\n",
    "peaks233_50['S'] = peaks233_50['Peptide'].str.count(\"S\")\n",
    "\n",
    "# use a count function to enumerate the # of T's (threonines) in each peptide\n",
    "peaks233_50['T'] = peaks233_50['Peptide'].str.count(\"T\")\n",
    "\n",
    "# use a count function to enumerate the # of V's (valines) in each peptide\n",
    "peaks233_50['V'] = peaks233_50['Peptide'].str.count(\"V\")\n",
    "\n",
    "# use a count function to enumerate the # of W's (tryptophans) in each peptide\n",
    "peaks233_50['W'] = peaks233_50['Peptide'].str.count(\"W\")\n",
    "\n",
    "# use a count function to enumerate the # of Y's (tyrosines) in each peptide\n",
    "peaks233_50['Y'] = peaks233_50['Peptide'].str.count(\"Y\")\n",
    "\n",
    "# use a count function to enumerate the # of carbamidomethylated C's in each peptide\n",
    "peaks233_50['c-carb'] = peaks233_50['Peptide'].str.count(\"57.02\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "peaks233_50['m-oxid'] = peaks233_50['Peptide'].apply(lambda x: x.count('M(+15.99)'))\n",
    "\n",
    "# use a lamba function to enumerate the # of deamidated N's in each peptide\n",
    "peaks233_50['n-deam'] = peaks233_50['Peptide'].apply(lambda x: x.count('N(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "peaks233_50['q-deam'] = peaks233_50['Peptide'].apply(lambda x: x.count('Q(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of hydroxylated K's in each peptide\n",
    "peaks233_50['k-hydr'] = peaks233_50['Peptide'].apply(lambda x: x.count('K(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of hydroxylated P's in each peptide\n",
    "# no more P hydroxylation in final searches\n",
    "# peaks233_50['p-hydr'] = peaks233_50['Peptide'].apply(lambda x: x.count('P(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "peaks233_50['r-meth'] = peaks233_50['Peptide'].apply(lambda x: x.count('R(+14.02)'))\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "peaks233_50['stripped peptide'] = peaks233_50['Peptide'].str.replace(r\"\\(.*\\)\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "peaks233_50['stripped length'] = peaks233_50['stripped peptide'].apply(len)\n",
    "\n",
    "# total the number of modifications in sequence\n",
    "peaks233_50['ptm-total'] = peaks233_50['c-carb'] + peaks233_50['m-oxid'] + peaks233_50['n-deam'] \\\n",
    "+ peaks233_50['q-deam'] + peaks233_50['k-hydr'] + peaks233_50['r-meth']\n",
    "\n",
    "# calculate NAAF numerator for each peptide k\n",
    "peaks233_50['NAAF num.'] = peaks233_50['Area'] / peaks233_50['stripped length']\n",
    "\n",
    "# write modified dataframe to new csv file\n",
    "peaks233_50.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/233/ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_15ppm_DN50.csv\")\n",
    "\n",
    "\n",
    "# check out the results\n",
    "peaks233_50.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>K</th>\n",
       "      <th>L</th>\n",
       "      <th>...</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>Total area</th>\n",
       "      <th>Total length</th>\n",
       "      <th>% C w/ carb</th>\n",
       "      <th>% M w/ oxid</th>\n",
       "      <th>% N w/ deam</th>\n",
       "      <th>% Q w/ deam</th>\n",
       "      <th>% K w/ hydr</th>\n",
       "      <th>% R w/ meth</th>\n",
       "      <th>NAAF denom.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>3021</td>\n",
       "      <td>591</td>\n",
       "      <td>1546</td>\n",
       "      <td>2121</td>\n",
       "      <td>1356</td>\n",
       "      <td>1648</td>\n",
       "      <td>1019</td>\n",
       "      <td>0</td>\n",
       "      <td>4038</td>\n",
       "      <td>4376</td>\n",
       "      <td>...</td>\n",
       "      <td>1085</td>\n",
       "      <td>9.568026e+09</td>\n",
       "      <td>35570</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.402795</td>\n",
       "      <td>0.196872</td>\n",
       "      <td>0.020505</td>\n",
       "      <td>0.165676</td>\n",
       "      <td>0.341517</td>\n",
       "      <td>268991.466947</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 A    C     D     E     F     G     H  I     K     L  ...  \\\n",
       "sample total  3021  591  1546  2121  1356  1648  1019  0  4038  4376  ...   \n",
       "\n",
       "              r-meth    Total area  Total length  % C w/ carb  % M w/ oxid  \\\n",
       "sample total    1085  9.568026e+09         35570          1.0     0.402795   \n",
       "\n",
       "              % N w/ deam  % Q w/ deam  % K w/ hydr  % R w/ meth  \\\n",
       "sample total     0.196872     0.020505     0.165676     0.341517   \n",
       "\n",
       "                NAAF denom.  \n",
       "sample total  268991.466947  \n",
       "\n",
       "[1 rows x 35 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of certain columns in the modified\n",
    "# peptide dataframe above (for >50% ALC)\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'A': peaks233_50['A'].sum(),\n",
    "        'C': peaks233_50['C'].sum(),\n",
    "        'D': peaks233_50['D'].sum(),\n",
    "        'E': peaks233_50['E'].sum(),\n",
    "        'F': peaks233_50['F'].sum(),\n",
    "        'G': peaks233_50['G'].sum(),\n",
    "        'H': peaks233_50['H'].sum(),\n",
    "        'I': peaks233_50['I'].sum(),\n",
    "        'K': peaks233_50['K'].sum(),\n",
    "        'L': peaks233_50['L'].sum(),\n",
    "        'M': peaks233_50['M'].sum(),\n",
    "        'N': peaks233_50['N'].sum(),\n",
    "        'P': peaks233_50['P'].sum(),\n",
    "        'Q': peaks233_50['Q'].sum(),\n",
    "        'R': peaks233_50['R'].sum(),\n",
    "        'S': peaks233_50['S'].sum(),\n",
    "        'T': peaks233_50['T'].sum(),\n",
    "        'V': peaks233_50['V'].sum(),\n",
    "        'W': peaks233_50['W'].sum(),\n",
    "        'Y': peaks233_50['Y'].sum(),\n",
    "        'c-carb': peaks233_50['c-carb'].sum(),\n",
    "        'm-oxid': peaks233_50['m-oxid'].sum(),\n",
    "        'n-deam': peaks233_50['n-deam'].sum(),\n",
    "        'q-deam': peaks233_50['q-deam'].sum(),\n",
    "        'k-hydr': peaks233_50['k-hydr'].sum(),\n",
    "        'r-meth': peaks233_50['r-meth'].sum(),\n",
    "        'Total area': peaks233_50['Area'].sum(),\n",
    "        'Total length': peaks233_50['stripped length'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaks233_50 = pd.DataFrame(data, columns=['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L',\\\n",
    "                                               'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', \\\n",
    "                                               'm-oxid', 'n-deam', 'q-deam', 'k-hydr', 'r-meth', \\\n",
    "                                               'Total area', 'Total length'], index=index)\n",
    "\n",
    "# calculate percentage of C's with carb (should be 1.0)\n",
    "totalpeaks233_50['% C w/ carb'] = totalpeaks233_50['c-carb'] / totalpeaks233_50['C'] \n",
    "\n",
    "# calculate percentage of M's that are oxidized\n",
    "totalpeaks233_50['% M w/ oxid'] = totalpeaks233_50['m-oxid'] / totalpeaks233_50['M'] \n",
    "\n",
    "# calculate percentage of N's that are deamidated\n",
    "totalpeaks233_50['% N w/ deam'] = totalpeaks233_50['n-deam'] / totalpeaks233_50['N'] \n",
    "\n",
    "# calculate percentage of Q's that are deamidated\n",
    "totalpeaks233_50['% Q w/ deam'] = totalpeaks233_50['q-deam'] / totalpeaks233_50['Q'] \n",
    "\n",
    "# calculate percentage of K's that are hydroxylated\n",
    "totalpeaks233_50['% K w/ hydr'] = totalpeaks233_50['k-hydr'] / totalpeaks233_50['K'] \n",
    "\n",
    "# calculate percentage of P's that are hydroxylated\n",
    "#totalpeaks233_50['% P w/ hydr'] = totalpeaks233_50['p-hydr'] / totalpeaks233_50['K'] \n",
    "\n",
    "# calculate percentage of R's that are methylated\n",
    "totalpeaks233_50['% R w/ meth'] = totalpeaks233_50['r-meth'] / totalpeaks233_50['R'] \n",
    "\n",
    "# calculate NAAF denominator for all peptides in dataset i\n",
    "totalpeaks233_50['NAAF denom.'] = totalpeaks233_50['Total area'] / totalpeaks233_50['Total length']\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "totalpeaks233_50.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/233/ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_15ppm_DN50_totals.csv\")\n",
    "\n",
    "totalpeaks233_50.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>NAAF factor</th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>...</th>\n",
       "      <th>T-NAAF50</th>\n",
       "      <th>V-NAAF50</th>\n",
       "      <th>W-NAAF50</th>\n",
       "      <th>Y-NAAF50</th>\n",
       "      <th>ccarb-NAAF50</th>\n",
       "      <th>moxid-NAAF50</th>\n",
       "      <th>ndeam-NAAF50</th>\n",
       "      <th>qdeam-NAAF50</th>\n",
       "      <th>khydr-NAAF50</th>\n",
       "      <th>rmeth-NAAF50</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENLAALEK</td>\n",
       "      <td>4.135819</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.135819</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>VGCDEGLFEELPR</td>\n",
       "      <td>2.690963</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.690963</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.690963</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>WSVVFK</td>\n",
       "      <td>0.199511</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.399021</td>\n",
       "      <td>0.199511</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FDLLVNK</td>\n",
       "      <td>0.807248</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.807248</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EPLGPVVR</td>\n",
       "      <td>1.868089</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.736178</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  stripped peptide  NAAF factor  A  C  D  E  F  G  H  I  ...  T-NAAF50  \\\n",
       "0         ENLAALEK     4.135819  2  0  0  2  0  0  0  0  ...       0.0   \n",
       "1    VGCDEGLFEELPR     2.690963  0  1  1  3  1  2  0  0  ...       0.0   \n",
       "2           WSVVFK     0.199511  0  0  0  0  1  0  0  0  ...       0.0   \n",
       "3          FDLLVNK     0.807248  0  0  1  0  1  0  0  0  ...       0.0   \n",
       "4         EPLGPVVR     1.868089  0  0  0  1  0  1  0  0  ...       0.0   \n",
       "\n",
       "   V-NAAF50  W-NAAF50  Y-NAAF50  ccarb-NAAF50  moxid-NAAF50  ndeam-NAAF50  \\\n",
       "0  0.000000  0.000000       0.0      0.000000           0.0      4.135819   \n",
       "1  2.690963  0.000000       0.0      2.690963           0.0      0.000000   \n",
       "2  0.399021  0.199511       0.0      0.000000           0.0      0.000000   \n",
       "3  0.807248  0.000000       0.0      0.000000           0.0      0.000000   \n",
       "4  3.736178  0.000000       0.0      0.000000           0.0      0.000000   \n",
       "\n",
       "   qdeam-NAAF50  khydr-NAAF50  rmeth-NAAF50  \n",
       "0           0.0           0.0           0.0  \n",
       "1           0.0           0.0           0.0  \n",
       "2           0.0           0.0           0.0  \n",
       "3           0.0           0.0           0.0  \n",
       "4           0.0           0.0           0.0  \n",
       "\n",
       "[5 rows x 54 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use the calculated NAAF factor (in totalpeaks233 dataframe, above) to caluclate the NAAF \n",
    "# NAAF: normalized normalized area abundance factor\n",
    "\n",
    "# don't have to worry here about DECOY hits messing with Area totals\n",
    "# but we will with Comet results\n",
    "\n",
    "NAAF50 = 268991.466947\n",
    "\n",
    "# use NAAF >50% ALC to get NAAF\n",
    "peaks233_50['NAAF factor'] = (peaks233_50['NAAF num.'])/NAAF50\n",
    "\n",
    "# make a dataframe that contains only what we need: sequences, AAs, PTMs\n",
    "peaks233_AA50 = peaks233_50[['stripped peptide', 'NAAF factor', 'A', 'C', 'D', 'E', 'F', 'G', 'H', \\\n",
    "                             'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', \\\n",
    "                             'm-oxid', 'n-deam', 'q-deam', 'k-hydr', 'r-meth']].copy()\n",
    "\n",
    "# multiply the NAAF50 factor by the AAs to normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaks233_AA50['A-NAAF50'] = peaks233_AA50['A'] * peaks233_50['NAAF factor']\n",
    "peaks233_AA50['C-NAAF50'] = peaks233_AA50['C'] * peaks233_50['NAAF factor']\n",
    "peaks233_AA50['D-NAAF50'] = peaks233_AA50['D'] * peaks233_50['NAAF factor']\n",
    "peaks233_AA50['E-NAAF50'] = peaks233_AA50['E'] * peaks233_50['NAAF factor']\n",
    "peaks233_AA50['F-NAAF50'] = peaks233_AA50['F'] * peaks233_50['NAAF factor']\n",
    "peaks233_AA50['G-NAAF50'] = peaks233_AA50['G'] * peaks233_50['NAAF factor']\n",
    "peaks233_AA50['H-NAAF50'] = peaks233_AA50['H'] * peaks233_50['NAAF factor']\n",
    "peaks233_AA50['I-NAAF50'] = peaks233_AA50['I'] * peaks233_50['NAAF factor']\n",
    "peaks233_AA50['K-NAAF50'] = peaks233_AA50['K'] * peaks233_50['NAAF factor']\n",
    "peaks233_AA50['L-NAAF50'] = peaks233_AA50['L'] * peaks233_50['NAAF factor']\n",
    "peaks233_AA50['M-NAAF50'] = peaks233_AA50['M'] * peaks233_50['NAAF factor']\n",
    "peaks233_AA50['N-NAAF50'] = peaks233_AA50['N'] * peaks233_50['NAAF factor']\n",
    "peaks233_AA50['P-NAAF50'] = peaks233_AA50['P'] * peaks233_50['NAAF factor']\n",
    "peaks233_AA50['Q-NAAF50'] = peaks233_AA50['Q'] * peaks233_50['NAAF factor']\n",
    "peaks233_AA50['R-NAAF50'] = peaks233_AA50['R'] * peaks233_50['NAAF factor']\n",
    "peaks233_AA50['S-NAAF50'] = peaks233_AA50['S'] * peaks233_50['NAAF factor']\n",
    "peaks233_AA50['T-NAAF50'] = peaks233_AA50['T'] * peaks233_50['NAAF factor']\n",
    "peaks233_AA50['V-NAAF50'] = peaks233_AA50['V'] * peaks233_50['NAAF factor']\n",
    "peaks233_AA50['W-NAAF50'] = peaks233_AA50['W'] * peaks233_50['NAAF factor']\n",
    "peaks233_AA50['Y-NAAF50'] = peaks233_AA50['Y'] * peaks233_50['NAAF factor']\n",
    "\n",
    "# multiply the NAAF50 factor by the PTMs normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaks233_AA50['ccarb-NAAF50'] = peaks233_AA50['c-carb'] * peaks233_AA50['NAAF factor']\n",
    "peaks233_AA50['moxid-NAAF50'] = peaks233_AA50['m-oxid'] * peaks233_AA50['NAAF factor']\n",
    "peaks233_AA50['ndeam-NAAF50'] = peaks233_AA50['n-deam'] * peaks233_AA50['NAAF factor']\n",
    "peaks233_AA50['qdeam-NAAF50'] = peaks233_AA50['q-deam'] * peaks233_AA50['NAAF factor']\n",
    "peaks233_AA50['khydr-NAAF50'] = peaks233_AA50['k-hydr'] * peaks233_AA50['NAAF factor']\n",
    "peaks233_AA50['rmeth-NAAF50'] = peaks233_AA50['r-meth'] * peaks233_AA50['NAAF factor']\n",
    "\n",
    "# write the dataframe to a new csv\n",
    "peaks233_AA50.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/233/ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_15ppm_DN50_NAAF.csv\")\n",
    "\n",
    "peaks233_AA50.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NAAF</th>\n",
       "      <th>A-NAAF</th>\n",
       "      <th>C-NAAF</th>\n",
       "      <th>D-NAAF</th>\n",
       "      <th>E-NAAF</th>\n",
       "      <th>F-NAAF</th>\n",
       "      <th>G-NAAF</th>\n",
       "      <th>H-NAAF</th>\n",
       "      <th>I-NAAF</th>\n",
       "      <th>K-NAAF</th>\n",
       "      <th>...</th>\n",
       "      <th>Q-deam-NAAF</th>\n",
       "      <th>K-hydr-NAAF</th>\n",
       "      <th>R-meth-NAAF</th>\n",
       "      <th>% C w/ carb. NAAF</th>\n",
       "      <th>% M w/ oxid. NAAF</th>\n",
       "      <th>% N w/ deam. NAAF</th>\n",
       "      <th>% Q w/ deam. NAAF</th>\n",
       "      <th>% K w/ hydr. NAAF</th>\n",
       "      <th>% R w/ meth. NAAF</th>\n",
       "      <th>NAAF check</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>4469.010808</td>\n",
       "      <td>3847.40697</td>\n",
       "      <td>672.870558</td>\n",
       "      <td>656.326796</td>\n",
       "      <td>1651.830591</td>\n",
       "      <td>532.272591</td>\n",
       "      <td>1281.508363</td>\n",
       "      <td>421.829531</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1708.434088</td>\n",
       "      <td>...</td>\n",
       "      <td>112.51498</td>\n",
       "      <td>484.359101</td>\n",
       "      <td>1111.94236</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.295306</td>\n",
       "      <td>0.463614</td>\n",
       "      <td>0.143355</td>\n",
       "      <td>0.283511</td>\n",
       "      <td>0.249789</td>\n",
       "      <td>0.016614</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     NAAF      A-NAAF      C-NAAF      D-NAAF       E-NAAF  \\\n",
       "sample total  4469.010808  3847.40697  672.870558  656.326796  1651.830591   \n",
       "\n",
       "                  F-NAAF       G-NAAF      H-NAAF  I-NAAF       K-NAAF  ...  \\\n",
       "sample total  532.272591  1281.508363  421.829531     0.0  1708.434088  ...   \n",
       "\n",
       "              Q-deam-NAAF  K-hydr-NAAF  R-meth-NAAF  % C w/ carb. NAAF  \\\n",
       "sample total    112.51498   484.359101   1111.94236                1.0   \n",
       "\n",
       "              % M w/ oxid. NAAF  % N w/ deam. NAAF  % Q w/ deam. NAAF  \\\n",
       "sample total           0.295306           0.463614           0.143355   \n",
       "\n",
       "              % K w/ hydr. NAAF  % R w/ meth. NAAF  NAAF check  \n",
       "sample total           0.283511           0.249789    0.016614  \n",
       "\n",
       "[1 rows x 34 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of NAAF normalized AAs for comet233 PeaksDN results\n",
    "# also contains the sums of the NAAF-corrected PTMs occurances for each affected residue\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'NAAF': peaks233_AA50['NAAF factor'].sum(),\n",
    "        'A-NAAF': peaks233_AA50['A-NAAF50'].sum(),\n",
    "        'C-NAAF': peaks233_AA50['C-NAAF50'].sum(),\n",
    "        'D-NAAF': peaks233_AA50['D-NAAF50'].sum(),\n",
    "        'E-NAAF': peaks233_AA50['E-NAAF50'].sum(),\n",
    "        'F-NAAF': peaks233_AA50['F-NAAF50'].sum(),\n",
    "        'G-NAAF': peaks233_AA50['G-NAAF50'].sum(),\n",
    "        'H-NAAF': peaks233_AA50['H-NAAF50'].sum(),\n",
    "        'I-NAAF': peaks233_AA50['I-NAAF50'].sum(),\n",
    "        'K-NAAF': peaks233_AA50['K-NAAF50'].sum(),\n",
    "        'L-NAAF': peaks233_AA50['L-NAAF50'].sum(),\n",
    "        'M-NAAF': peaks233_AA50['M-NAAF50'].sum(),\n",
    "        'N-NAAF': peaks233_AA50['N-NAAF50'].sum(),\n",
    "        'P-NAAF': peaks233_AA50['P-NAAF50'].sum(),\n",
    "        'Q-NAAF': peaks233_AA50['Q-NAAF50'].sum(),\n",
    "        'R-NAAF': peaks233_AA50['R-NAAF50'].sum(),\n",
    "        'S-NAAF': peaks233_AA50['S-NAAF50'].sum(),\n",
    "        'T-NAAF': peaks233_AA50['T-NAAF50'].sum(),\n",
    "        'V-NAAF': peaks233_AA50['V-NAAF50'].sum(),\n",
    "        'W-NAAF': peaks233_AA50['W-NAAF50'].sum(),\n",
    "        'Y-NAAF': peaks233_AA50['Y-NAAF50'].sum(),\n",
    "        'C-carb-NAAF': peaks233_AA50['ccarb-NAAF50'].sum(),\n",
    "        'M-oxid-NAAF': peaks233_AA50['moxid-NAAF50'].sum(),\n",
    "        'N-deam-NAAF': peaks233_AA50['ndeam-NAAF50'].sum(),\n",
    "        'Q-deam-NAAF': peaks233_AA50['qdeam-NAAF50'].sum(),\n",
    "        'K-hydr-NAAF': peaks233_AA50['khydr-NAAF50'].sum(),\n",
    "        'R-meth-NAAF': peaks233_AA50['rmeth-NAAF50'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaks233_AA50 = pd.DataFrame(data, columns=['NAAF', 'A-NAAF', 'C-NAAF', 'D-NAAF', 'E-NAAF', 'F-NAAF', \\\n",
    "                                                   'G-NAAF', 'H-NAAF', 'I-NAAF', 'K-NAAF', 'L-NAAF', 'M-NAAF', \\\n",
    "                                                   'N-NAAF', 'P-NAAF', 'Q-NAAF', 'R-NAAF', 'S-NAAF', \\\n",
    "                                                   'T-NAAF', 'V-NAAF', 'W-NAAF', 'Y-NAAF', 'C-carb-NAAF', \\\n",
    "                                                   'M-oxid-NAAF', 'N-deam-NAAF', 'Q-deam-NAAF', 'K-hydr-NAAF',\\\n",
    "                                                   'R-meth-NAAF'], index=index)\n",
    "\n",
    "# calculate the NAAF-corrected % modified C, M, N, Q, K, P, and Rs\n",
    "\n",
    "\n",
    "totalpeaks233_AA50['% C w/ carb. NAAF'] = totalpeaks233_AA50['C-carb-NAAF'] / totalpeaks233_AA50['C-NAAF']\n",
    "totalpeaks233_AA50['% M w/ oxid. NAAF'] = totalpeaks233_AA50['M-oxid-NAAF'] / totalpeaks233_AA50['M-NAAF']\n",
    "totalpeaks233_AA50['% N w/ deam. NAAF'] = totalpeaks233_AA50['N-deam-NAAF'] / totalpeaks233_AA50['N-NAAF']\n",
    "totalpeaks233_AA50['% Q w/ deam. NAAF'] = totalpeaks233_AA50['Q-deam-NAAF'] / totalpeaks233_AA50['Q-NAAF']\n",
    "totalpeaks233_AA50['% K w/ hydr. NAAF'] = totalpeaks233_AA50['K-hydr-NAAF'] / totalpeaks233_AA50['K-NAAF']\n",
    "totalpeaks233_AA50['% R w/ meth. NAAF'] = totalpeaks233_AA50['R-meth-NAAF'] / totalpeaks233_AA50['R-NAAF']\n",
    "\n",
    "# calculate NAAF summed numerator over denominator (in above cell) for all peptides in dataset i: a check\n",
    "totalpeaks233_AA50['NAAF check'] = totalpeaks233_AA50['NAAF'] / 268991.466947\n",
    "\n",
    "# write modified dataframe to new txt file, same name + totals\n",
    "totalpeaks233_AA50.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/233/ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_DN50_NAAF_totals.csv\")\n",
    "\n",
    "totalpeaks233_AA50.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Same process but for de novo peptide >80 % ALC:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Fraction</th>\n",
       "      <th>Scan</th>\n",
       "      <th>Source File</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Tag Length</th>\n",
       "      <th>ALC (%)</th>\n",
       "      <th>length</th>\n",
       "      <th>m/z</th>\n",
       "      <th>z</th>\n",
       "      <th>RT</th>\n",
       "      <th>...</th>\n",
       "      <th>c-carb</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>k-hydr</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>stripped length</th>\n",
       "      <th>ptm-total</th>\n",
       "      <th>NAAF num.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>14553</td>\n",
       "      <td>20170410__ETNP-233-265m-0.3um-JA4_01.raw</td>\n",
       "      <td>EN(+.98)LAALEK</td>\n",
       "      <td>8</td>\n",
       "      <td>98</td>\n",
       "      <td>8</td>\n",
       "      <td>444.7376</td>\n",
       "      <td>2</td>\n",
       "      <td>49.92</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ENLAALEK</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>1.112500e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>26516</td>\n",
       "      <td>20170410__ETNP-233-265m-0.3um-JA4_01.raw</td>\n",
       "      <td>VGC(+57.02)DEGLFEELPR</td>\n",
       "      <td>13</td>\n",
       "      <td>98</td>\n",
       "      <td>13</td>\n",
       "      <td>760.8570</td>\n",
       "      <td>2</td>\n",
       "      <td>81.30</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>VGCDEGLFEELPR</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>7.238462e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>22453</td>\n",
       "      <td>20170410__ETNP-233-265m-0.3um-JA4_01.raw</td>\n",
       "      <td>WSVVFK</td>\n",
       "      <td>6</td>\n",
       "      <td>98</td>\n",
       "      <td>6</td>\n",
       "      <td>383.2187</td>\n",
       "      <td>2</td>\n",
       "      <td>72.23</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>WSVVFK</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>5.366667e+04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>19513</td>\n",
       "      <td>20170410__ETNP-233-265m-0.3um-JA4_01.raw</td>\n",
       "      <td>FDLLVNK</td>\n",
       "      <td>7</td>\n",
       "      <td>97</td>\n",
       "      <td>7</td>\n",
       "      <td>424.7480</td>\n",
       "      <td>2</td>\n",
       "      <td>64.92</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>FDLLVNK</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>2.171429e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>13937</td>\n",
       "      <td>20170410__ETNP-233-265m-0.3um-JA4_01.raw</td>\n",
       "      <td>EPLGPVVR</td>\n",
       "      <td>8</td>\n",
       "      <td>97</td>\n",
       "      <td>8</td>\n",
       "      <td>433.7584</td>\n",
       "      <td>2</td>\n",
       "      <td>47.83</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>EPLGPVVR</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>5.025000e+05</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Fraction   Scan                               Source File  \\\n",
       "0         2  14553  20170410__ETNP-233-265m-0.3um-JA4_01.raw   \n",
       "1         2  26516  20170410__ETNP-233-265m-0.3um-JA4_01.raw   \n",
       "2         2  22453  20170410__ETNP-233-265m-0.3um-JA4_01.raw   \n",
       "3         2  19513  20170410__ETNP-233-265m-0.3um-JA4_01.raw   \n",
       "4         2  13937  20170410__ETNP-233-265m-0.3um-JA4_01.raw   \n",
       "\n",
       "                 Peptide  Tag Length  ALC (%)  length       m/z  z     RT  \\\n",
       "0         EN(+.98)LAALEK           8       98       8  444.7376  2  49.92   \n",
       "1  VGC(+57.02)DEGLFEELPR          13       98      13  760.8570  2  81.30   \n",
       "2                 WSVVFK           6       98       6  383.2187  2  72.23   \n",
       "3                FDLLVNK           7       97       7  424.7480  2  64.92   \n",
       "4               EPLGPVVR           8       97       8  433.7584  2  47.83   \n",
       "\n",
       "   ...  c-carb  m-oxid  n-deam q-deam k-hydr r-meth stripped peptide  \\\n",
       "0  ...       0       0       1      0      0      0         ENLAALEK   \n",
       "1  ...       1       0       0      0      0      0    VGCDEGLFEELPR   \n",
       "2  ...       0       0       0      0      0      0           WSVVFK   \n",
       "3  ...       0       0       0      0      0      0          FDLLVNK   \n",
       "4  ...       0       0       0      0      0      0         EPLGPVVR   \n",
       "\n",
       "   stripped length  ptm-total     NAAF num.  \n",
       "0                8          1  1.112500e+06  \n",
       "1               13          1  7.238462e+05  \n",
       "2                6          0  5.366667e+04  \n",
       "3                7          0  2.171429e+05  \n",
       "4                8          0  5.025000e+05  \n",
       "\n",
       "[5 rows x 47 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use a count function to enumerate the # of A's (alanines) in each peptide\n",
    "peaks233_80['A'] = peaks233_80['Peptide'].str.count(\"A\")\n",
    "\n",
    "# use a count function to enumerate the # of C's (cysteines) in each peptide\n",
    "peaks233_80['C'] = peaks233_80['Peptide'].str.count(\"C\")\n",
    "\n",
    "# use a count function to enumerate the # of D's (aspartic acids) in each peptide\n",
    "peaks233_80['D'] = peaks233_80['Peptide'].str.count(\"D\")\n",
    "\n",
    "# use a count function to enumerate the # of E's (glutamic acids) in each peptide\n",
    "peaks233_80['E'] = peaks233_80['Peptide'].str.count(\"E\")\n",
    "\n",
    "# use a count function to enumerate the # of F's (phenylalanines) in each peptide\n",
    "peaks233_80['F'] = peaks233_80['Peptide'].str.count(\"F\")\n",
    "\n",
    "# use a count function to enumerate the # of G's (glycines) in each peptide\n",
    "peaks233_80['G'] = peaks233_80['Peptide'].str.count(\"G\")\n",
    "\n",
    "# use a count function to enumerate the # of H's (histidines) in each peptide\n",
    "peaks233_80['H'] = peaks233_80['Peptide'].str.count(\"H\")\n",
    "\n",
    "# use a count function to enumerate the # of I's (isoleucines) in each peptide\n",
    "# in peaks233_80 output, there will be no isoleucines (they're lumped in with leucines)\n",
    "peaks233_80['I'] = peaks233_80['Peptide'].str.count(\"I\")\n",
    "\n",
    "# use a count function to enumerate the # of K's (lysines) in each peptide\n",
    "peaks233_80['K'] = peaks233_80['Peptide'].str.count(\"K\")\n",
    "\n",
    "# use a count function to enumerate the # of L's (leucines) in each peptide\n",
    "# also these include the isoleucines\n",
    "peaks233_80['L'] = peaks233_80['Peptide'].str.count(\"L\")\n",
    "\n",
    "# use a count function to enumerate the # of M's (methionines) in each peptide\n",
    "peaks233_80['M'] = peaks233_80['Peptide'].str.count(\"M\")\n",
    "\n",
    "# use a count function to enumerate the # of N's (asparagines) in each peptide\n",
    "peaks233_80['N'] = peaks233_80['Peptide'].str.count(\"N\")\n",
    "\n",
    "# use a count function to enumerate the # of P's ([prolines]) in each peptide\n",
    "peaks233_80['P'] = peaks233_80['Peptide'].str.count(\"P\")\n",
    "\n",
    "# use a count function to enumerate the # of Q's (glutamines) in each peptide\n",
    "peaks233_80['Q'] = peaks233_80['Peptide'].str.count(\"Q\")\n",
    "\n",
    "# use a count function to enumerate the # of R's (arginines) in each peptide\n",
    "peaks233_80['R'] = peaks233_80['Peptide'].str.count(\"R\")\n",
    "\n",
    "# use a count function to enumerate the # of S's (serines) in each peptide\n",
    "peaks233_80['S'] = peaks233_80['Peptide'].str.count(\"S\")\n",
    "\n",
    "# use a count function to enumerate the # of T's (threonines) in each peptide\n",
    "peaks233_80['T'] = peaks233_80['Peptide'].str.count(\"T\")\n",
    "\n",
    "# use a count function to enumerate the # of V's (valines) in each peptide\n",
    "peaks233_80['V'] = peaks233_80['Peptide'].str.count(\"V\")\n",
    "\n",
    "# use a count function to enumerate the # of W's (tryptophans) in each peptide\n",
    "peaks233_80['W'] = peaks233_80['Peptide'].str.count(\"W\")\n",
    "\n",
    "# use a count function to enumerate the # of Y's (tyrosines) in each peptide\n",
    "peaks233_80['Y'] = peaks233_80['Peptide'].str.count(\"Y\")\n",
    "\n",
    "# use a count function to enumerate the # of carbamidomethylated C's in each peptide\n",
    "peaks233_80['c-carb'] = peaks233_80['Peptide'].str.count(\"57.02\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "peaks233_80['m-oxid'] = peaks233_80['Peptide'].apply(lambda x: x.count('M(+15.99)'))\n",
    "\n",
    "# use a lamba function to enumerate the # of deamidated N's in each peptide\n",
    "peaks233_80['n-deam'] = peaks233_80['Peptide'].apply(lambda x: x.count('N(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "peaks233_80['q-deam'] = peaks233_80['Peptide'].apply(lambda x: x.count('Q(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of hydroxylated K's in each peptide\n",
    "peaks233_80['k-hydr'] = peaks233_80['Peptide'].apply(lambda x: x.count('K(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of hydroxylated P's in each peptide\n",
    "#peaks233_80['p-hydr'] = peaks233_80['Peptide'].apply(lambda x: x.count('P(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "peaks233_80['r-meth'] = peaks233_80['Peptide'].apply(lambda x: x.count('R(+14.02)'))\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "peaks233_80['stripped peptide'] = peaks233_80['Peptide'].str.replace(r\"\\(.*\\)\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "peaks233_80['stripped length'] = peaks233_80['stripped peptide'].apply(len)\n",
    "\n",
    "# total the number of modifications in sequence\n",
    "peaks233_80['ptm-total'] = peaks233_80['c-carb'] + peaks233_80['m-oxid'] + peaks233_80['n-deam'] + \\\n",
    "peaks233_80['q-deam'] + peaks233_80['k-hydr'] + peaks233_80['r-meth']\n",
    "\n",
    "# calculate NAAF numerator for each peptide k\n",
    "peaks233_80['NAAF num.'] = peaks233_80['Area'] / peaks233_80['stripped length']\n",
    "\n",
    "# write modified dataframe to new csv file\n",
    "peaks233_80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/233/ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_15ppm_DN80.csv\")\n",
    "\n",
    "# check out the results\n",
    "peaks233_80.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>K</th>\n",
       "      <th>L</th>\n",
       "      <th>...</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>Total area</th>\n",
       "      <th>Total length</th>\n",
       "      <th>% C w/ carb</th>\n",
       "      <th>% M w/ oxid</th>\n",
       "      <th>% N w/ deam</th>\n",
       "      <th>% Q w/ deam</th>\n",
       "      <th>% K w/ hydr</th>\n",
       "      <th>% R w/ meth</th>\n",
       "      <th>NAAF denom.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>637</td>\n",
       "      <td>57</td>\n",
       "      <td>255</td>\n",
       "      <td>448</td>\n",
       "      <td>258</td>\n",
       "      <td>280</td>\n",
       "      <td>67</td>\n",
       "      <td>0</td>\n",
       "      <td>535</td>\n",
       "      <td>940</td>\n",
       "      <td>...</td>\n",
       "      <td>93</td>\n",
       "      <td>7.082521e+09</td>\n",
       "      <td>6274</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.508929</td>\n",
       "      <td>0.241117</td>\n",
       "      <td>0.030303</td>\n",
       "      <td>0.076636</td>\n",
       "      <td>0.207127</td>\n",
       "      <td>1.128868e+06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                A   C    D    E    F    G   H  I    K    L  ...  r-meth  \\\n",
       "sample total  637  57  255  448  258  280  67  0  535  940  ...      93   \n",
       "\n",
       "                Total area  Total length  % C w/ carb  % M w/ oxid  \\\n",
       "sample total  7.082521e+09          6274          1.0     0.508929   \n",
       "\n",
       "              % N w/ deam  % Q w/ deam  % K w/ hydr  % R w/ meth   NAAF denom.  \n",
       "sample total     0.241117     0.030303     0.076636     0.207127  1.128868e+06  \n",
       "\n",
       "[1 rows x 35 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of certain columns in modified \n",
    "#peptide dataframe above (for >80% ALC)\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'A': peaks233_80['A'].sum(),\n",
    "        'C': peaks233_80['C'].sum(),\n",
    "        'D': peaks233_80['D'].sum(),\n",
    "        'E': peaks233_80['E'].sum(),\n",
    "        'F': peaks233_80['F'].sum(),\n",
    "        'G': peaks233_80['G'].sum(),\n",
    "        'H': peaks233_80['H'].sum(),\n",
    "        'I': peaks233_80['I'].sum(),\n",
    "        'K': peaks233_80['K'].sum(),\n",
    "        'L': peaks233_80['L'].sum(),\n",
    "        'M': peaks233_80['M'].sum(),\n",
    "        'N': peaks233_80['N'].sum(),\n",
    "        'P': peaks233_80['P'].sum(),\n",
    "        'Q': peaks233_80['Q'].sum(),\n",
    "        'R': peaks233_80['R'].sum(),\n",
    "        'S': peaks233_80['S'].sum(),\n",
    "        'T': peaks233_80['T'].sum(),\n",
    "        'V': peaks233_80['V'].sum(),\n",
    "        'W': peaks233_80['W'].sum(),\n",
    "        'Y': peaks233_80['Y'].sum(),\n",
    "        'c-carb': peaks233_80['c-carb'].sum(),\n",
    "        'm-oxid': peaks233_80['m-oxid'].sum(),\n",
    "        'n-deam': peaks233_80['n-deam'].sum(),\n",
    "        'q-deam': peaks233_80['q-deam'].sum(),\n",
    "        'k-hydr': peaks233_80['k-hydr'].sum(),\n",
    "        'r-meth': peaks233_80['r-meth'].sum(),\n",
    "        'Total area': peaks233_80['Area'].sum(),\n",
    "        'Total length': peaks233_80['stripped length'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaks233_80 = pd.DataFrame(data, columns=['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M',\\\n",
    "                                               'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb',\\\n",
    "                                               'm-oxid', 'n-deam', 'q-deam', 'k-hydr', 'r-meth',\\\n",
    "                                               'Total area', 'Total length'], index=index)\n",
    "\n",
    "# calculate percentage of C's with carb (should be 1.0)\n",
    "totalpeaks233_80['% C w/ carb'] = totalpeaks233_80['c-carb'] / totalpeaks233_80['C'] \n",
    "\n",
    "# calculate percentage of M's that are oxidized\n",
    "totalpeaks233_80['% M w/ oxid'] = totalpeaks233_80['m-oxid'] / totalpeaks233_80['M'] \n",
    "\n",
    "# calculate percentage of N's that are deamidated\n",
    "totalpeaks233_80['% N w/ deam'] = totalpeaks233_80['n-deam'] / totalpeaks233_80['N'] \n",
    "\n",
    "# calculate percentage of Q's that are deamidated\n",
    "totalpeaks233_80['% Q w/ deam'] = totalpeaks233_80['q-deam'] / totalpeaks233_80['Q'] \n",
    "\n",
    "# calculate percentage of K's that are hydroxylated\n",
    "totalpeaks233_80['% K w/ hydr'] = totalpeaks233_80['k-hydr'] / totalpeaks233_80['K'] \n",
    "\n",
    "# calculate percentage of P's that are hydroxylated\n",
    "#totalpeaks233_80['% P w/ hydr'] = totalpeaks233_80['p-hydr'] / totalpeaks233_80['K'] \n",
    "\n",
    "# calculate percentage of R's that are methylated\n",
    "totalpeaks233_80['% R w/ meth'] = totalpeaks233_80['r-meth'] / totalpeaks233_80['R'] \n",
    "\n",
    "# calculate NAAF denominator for all peptides in dataset i\n",
    "totalpeaks233_80['NAAF denom.'] = totalpeaks233_80['Total area'] / totalpeaks233_80['Total length']\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "totalpeaks233_80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/233/ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_15ppm_DN80_totals.csv\")\n",
    "\n",
    "totalpeaks233_80.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>NAAF factor</th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>...</th>\n",
       "      <th>T-NAAF80</th>\n",
       "      <th>V-NAAF80</th>\n",
       "      <th>W-NAAF80</th>\n",
       "      <th>Y-NAAF80</th>\n",
       "      <th>ccarb-NAAF80</th>\n",
       "      <th>moxid-NAAF80</th>\n",
       "      <th>ndeam-NAAF80</th>\n",
       "      <th>qdeam-NAAF80</th>\n",
       "      <th>khydr-NAAF80</th>\n",
       "      <th>rmeth-NAAF80</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENLAALEK</td>\n",
       "      <td>0.985501</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.985501</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>VGCDEGLFEELPR</td>\n",
       "      <td>0.641214</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.641214</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.641214</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>WSVVFK</td>\n",
       "      <td>0.047540</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.095080</td>\n",
       "      <td>0.04754</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FDLLVNK</td>\n",
       "      <td>0.192355</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.192355</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EPLGPVVR</td>\n",
       "      <td>0.445136</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.890272</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  stripped peptide  NAAF factor  A  C  D  E  F  G  H  I  ...  T-NAAF80  \\\n",
       "0         ENLAALEK     0.985501  2  0  0  2  0  0  0  0  ...       0.0   \n",
       "1    VGCDEGLFEELPR     0.641214  0  1  1  3  1  2  0  0  ...       0.0   \n",
       "2           WSVVFK     0.047540  0  0  0  0  1  0  0  0  ...       0.0   \n",
       "3          FDLLVNK     0.192355  0  0  1  0  1  0  0  0  ...       0.0   \n",
       "4         EPLGPVVR     0.445136  0  0  0  1  0  1  0  0  ...       0.0   \n",
       "\n",
       "   V-NAAF80  W-NAAF80  Y-NAAF80  ccarb-NAAF80  moxid-NAAF80  ndeam-NAAF80  \\\n",
       "0  0.000000   0.00000       0.0      0.000000           0.0      0.985501   \n",
       "1  0.641214   0.00000       0.0      0.641214           0.0      0.000000   \n",
       "2  0.095080   0.04754       0.0      0.000000           0.0      0.000000   \n",
       "3  0.192355   0.00000       0.0      0.000000           0.0      0.000000   \n",
       "4  0.890272   0.00000       0.0      0.000000           0.0      0.000000   \n",
       "\n",
       "   qdeam-NAAF80  khydr-NAAF80  rmeth-NAAF80  \n",
       "0           0.0           0.0           0.0  \n",
       "1           0.0           0.0           0.0  \n",
       "2           0.0           0.0           0.0  \n",
       "3           0.0           0.0           0.0  \n",
       "4           0.0           0.0           0.0  \n",
       "\n",
       "[5 rows x 54 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use the calculated NAAF factor (in totalpeaks233 dataframe, above) to caluclate the NAAF \n",
    "# NAAF: normalized normalized area abundance factor\n",
    "\n",
    "# don't have to worry here about DECOY hits messing with Area totals\n",
    "# but we will with Comet results\n",
    "\n",
    "NAAF80 = 1.128868e+06\n",
    "\n",
    "# use NAAF >80% ALC to get NAAF\n",
    "peaks233_80['NAAF factor'] = (peaks233_80['NAAF num.'])/NAAF80\n",
    "\n",
    "# make a dataframe that contains only what we need: sequences, AAs, PTMs\n",
    "peaks233_AA80 = peaks233_80[['stripped peptide', 'NAAF factor', 'A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', \\\n",
    "                             'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', 'm-oxid', \\\n",
    "                             'n-deam', 'q-deam', 'k-hydr', 'r-meth']].copy()\n",
    "\n",
    "# multiply the NAAF80 factor by the AAs to normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaks233_AA80['A-NAAF80'] = peaks233_AA80['A'] * peaks233_80['NAAF factor']\n",
    "peaks233_AA80['C-NAAF80'] = peaks233_AA80['C'] * peaks233_80['NAAF factor']\n",
    "peaks233_AA80['D-NAAF80'] = peaks233_AA80['D'] * peaks233_80['NAAF factor']\n",
    "peaks233_AA80['E-NAAF80'] = peaks233_AA80['E'] * peaks233_80['NAAF factor']\n",
    "peaks233_AA80['F-NAAF80'] = peaks233_AA80['F'] * peaks233_80['NAAF factor']\n",
    "peaks233_AA80['G-NAAF80'] = peaks233_AA80['G'] * peaks233_80['NAAF factor']\n",
    "peaks233_AA80['H-NAAF80'] = peaks233_AA80['H'] * peaks233_80['NAAF factor']\n",
    "peaks233_AA80['I-NAAF80'] = peaks233_AA80['I'] * peaks233_80['NAAF factor']\n",
    "peaks233_AA80['K-NAAF80'] = peaks233_AA80['K'] * peaks233_80['NAAF factor']\n",
    "peaks233_AA80['L-NAAF80'] = peaks233_AA80['L'] * peaks233_80['NAAF factor']\n",
    "peaks233_AA80['M-NAAF80'] = peaks233_AA80['M'] * peaks233_80['NAAF factor']\n",
    "peaks233_AA80['N-NAAF80'] = peaks233_AA80['N'] * peaks233_80['NAAF factor']\n",
    "peaks233_AA80['P-NAAF80'] = peaks233_AA80['P'] * peaks233_80['NAAF factor']\n",
    "peaks233_AA80['Q-NAAF80'] = peaks233_AA80['Q'] * peaks233_80['NAAF factor']\n",
    "peaks233_AA80['R-NAAF80'] = peaks233_AA80['R'] * peaks233_80['NAAF factor']\n",
    "peaks233_AA80['S-NAAF80'] = peaks233_AA80['S'] * peaks233_80['NAAF factor']\n",
    "peaks233_AA80['T-NAAF80'] = peaks233_AA80['T'] * peaks233_80['NAAF factor']\n",
    "peaks233_AA80['V-NAAF80'] = peaks233_AA80['V'] * peaks233_80['NAAF factor']\n",
    "peaks233_AA80['W-NAAF80'] = peaks233_AA80['W'] * peaks233_80['NAAF factor']\n",
    "peaks233_AA80['Y-NAAF80'] = peaks233_AA80['Y'] * peaks233_80['NAAF factor']\n",
    "\n",
    "# multiply the NAAF80 factor by the PTMs normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaks233_AA80['ccarb-NAAF80'] = peaks233_AA80['c-carb'] * peaks233_AA80['NAAF factor']\n",
    "peaks233_AA80['moxid-NAAF80'] = peaks233_AA80['m-oxid'] * peaks233_AA80['NAAF factor']\n",
    "peaks233_AA80['ndeam-NAAF80'] = peaks233_AA80['n-deam'] * peaks233_AA80['NAAF factor']\n",
    "peaks233_AA80['qdeam-NAAF80'] = peaks233_AA80['q-deam'] * peaks233_AA80['NAAF factor']\n",
    "peaks233_AA80['khydr-NAAF80'] = peaks233_AA80['k-hydr'] * peaks233_AA80['NAAF factor']\n",
    "peaks233_AA80['rmeth-NAAF80'] = peaks233_AA80['r-meth'] * peaks233_AA80['NAAF factor']\n",
    "\n",
    "# write the dataframe to a new csv\n",
    "peaks233_AA80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/233/ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_15ppm_DN80_NAAF.csv\")\n",
    "\n",
    "peaks233_AA80.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NAAF</th>\n",
       "      <th>A-NAAF</th>\n",
       "      <th>C-NAAF</th>\n",
       "      <th>D-NAAF</th>\n",
       "      <th>E-NAAF</th>\n",
       "      <th>F-NAAF</th>\n",
       "      <th>G-NAAF</th>\n",
       "      <th>H-NAAF</th>\n",
       "      <th>I-NAAF</th>\n",
       "      <th>K-NAAF</th>\n",
       "      <th>...</th>\n",
       "      <th>Q-deam-NAAF</th>\n",
       "      <th>K-hydr-NAAF</th>\n",
       "      <th>R-meth-NAAF</th>\n",
       "      <th>% C w/ carb. NAAF</th>\n",
       "      <th>% M w/ oxid. NAAF</th>\n",
       "      <th>% N w/ deam. NAAF</th>\n",
       "      <th>% Q w/ deam. NAAF</th>\n",
       "      <th>% K w/ hydr. NAAF</th>\n",
       "      <th>% R w/ meth. NAAF</th>\n",
       "      <th>NAAF check</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>729.042031</td>\n",
       "      <td>659.799741</td>\n",
       "      <td>64.905681</td>\n",
       "      <td>65.848432</td>\n",
       "      <td>256.832922</td>\n",
       "      <td>63.603617</td>\n",
       "      <td>148.393768</td>\n",
       "      <td>8.960728</td>\n",
       "      <td>0.0</td>\n",
       "      <td>169.382693</td>\n",
       "      <td>...</td>\n",
       "      <td>25.855878</td>\n",
       "      <td>35.800928</td>\n",
       "      <td>70.427309</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.585299</td>\n",
       "      <td>0.224428</td>\n",
       "      <td>0.211361</td>\n",
       "      <td>0.10607</td>\n",
       "      <td>0.000646</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    NAAF      A-NAAF     C-NAAF     D-NAAF      E-NAAF  \\\n",
       "sample total  729.042031  659.799741  64.905681  65.848432  256.832922   \n",
       "\n",
       "                 F-NAAF      G-NAAF    H-NAAF  I-NAAF      K-NAAF  ...  \\\n",
       "sample total  63.603617  148.393768  8.960728     0.0  169.382693  ...   \n",
       "\n",
       "              Q-deam-NAAF  K-hydr-NAAF  R-meth-NAAF  % C w/ carb. NAAF  \\\n",
       "sample total    25.855878    35.800928    70.427309                1.0   \n",
       "\n",
       "              % M w/ oxid. NAAF  % N w/ deam. NAAF  % Q w/ deam. NAAF  \\\n",
       "sample total                0.1           0.585299           0.224428   \n",
       "\n",
       "              % K w/ hydr. NAAF  % R w/ meth. NAAF  NAAF check  \n",
       "sample total           0.211361            0.10607    0.000646  \n",
       "\n",
       "[1 rows x 34 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of NAAF normalized AAs for comet233 PeaksDN results\n",
    "# also contains the sums of the NAAF-corrected PTMs occurances for each affected residue\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'NAAF': peaks233_AA80['NAAF factor'].sum(),\n",
    "        'A-NAAF': peaks233_AA80['A-NAAF80'].sum(),\n",
    "        'C-NAAF': peaks233_AA80['C-NAAF80'].sum(),\n",
    "        'D-NAAF': peaks233_AA80['D-NAAF80'].sum(),\n",
    "        'E-NAAF': peaks233_AA80['E-NAAF80'].sum(),\n",
    "        'F-NAAF': peaks233_AA80['F-NAAF80'].sum(),\n",
    "        'G-NAAF': peaks233_AA80['G-NAAF80'].sum(),\n",
    "        'H-NAAF': peaks233_AA80['H-NAAF80'].sum(),\n",
    "        'I-NAAF': peaks233_AA80['I-NAAF80'].sum(),\n",
    "        'K-NAAF': peaks233_AA80['K-NAAF80'].sum(),\n",
    "        'L-NAAF': peaks233_AA80['L-NAAF80'].sum(),\n",
    "        'M-NAAF': peaks233_AA80['M-NAAF80'].sum(),\n",
    "        'N-NAAF': peaks233_AA80['N-NAAF80'].sum(),\n",
    "        'P-NAAF': peaks233_AA80['P-NAAF80'].sum(),\n",
    "        'Q-NAAF': peaks233_AA80['Q-NAAF80'].sum(),\n",
    "        'R-NAAF': peaks233_AA80['R-NAAF80'].sum(),\n",
    "        'S-NAAF': peaks233_AA80['S-NAAF80'].sum(),\n",
    "        'T-NAAF': peaks233_AA80['T-NAAF80'].sum(),\n",
    "        'V-NAAF': peaks233_AA80['V-NAAF80'].sum(),\n",
    "        'W-NAAF': peaks233_AA80['W-NAAF80'].sum(),\n",
    "        'Y-NAAF': peaks233_AA80['Y-NAAF80'].sum(),\n",
    "        'C-carb-NAAF': peaks233_AA80['ccarb-NAAF80'].sum(),\n",
    "        'M-oxid-NAAF': peaks233_AA80['moxid-NAAF80'].sum(),\n",
    "        'N-deam-NAAF': peaks233_AA80['ndeam-NAAF80'].sum(),\n",
    "        'Q-deam-NAAF': peaks233_AA80['qdeam-NAAF80'].sum(),\n",
    "        'K-hydr-NAAF': peaks233_AA80['khydr-NAAF80'].sum(),\n",
    "        'R-meth-NAAF': peaks233_AA80['rmeth-NAAF80'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaks233_AA80 = pd.DataFrame(data, columns=['NAAF', 'A-NAAF', 'C-NAAF', 'D-NAAF', 'E-NAAF', 'F-NAAF', \\\n",
    "                                                   'G-NAAF', 'H-NAAF', 'I-NAAF', 'K-NAAF', 'L-NAAF', 'M-NAAF', \\\n",
    "                                                   'N-NAAF', 'P-NAAF', 'Q-NAAF', 'R-NAAF', 'S-NAAF', \\\n",
    "                                                   'T-NAAF', 'V-NAAF', 'W-NAAF', 'Y-NAAF', 'C-carb-NAAF', \\\n",
    "                                                   'M-oxid-NAAF', 'N-deam-NAAF', 'Q-deam-NAAF', 'K-hydr-NAAF',\\\n",
    "                                                   'R-meth-NAAF'], index=index)\n",
    "\n",
    "# calculate the NAAF-corrected % modified C, M, N, Q, K, P, and Rs\n",
    "\n",
    "\n",
    "totalpeaks233_AA80['% C w/ carb. NAAF'] = totalpeaks233_AA80['C-carb-NAAF'] / totalpeaks233_AA80['C-NAAF']\n",
    "totalpeaks233_AA80['% M w/ oxid. NAAF'] = totalpeaks233_AA80['M-oxid-NAAF'] / totalpeaks233_AA80['M-NAAF']\n",
    "totalpeaks233_AA80['% N w/ deam. NAAF'] = totalpeaks233_AA80['N-deam-NAAF'] / totalpeaks233_AA80['N-NAAF']\n",
    "totalpeaks233_AA80['% Q w/ deam. NAAF'] = totalpeaks233_AA80['Q-deam-NAAF'] / totalpeaks233_AA80['Q-NAAF']\n",
    "totalpeaks233_AA80['% K w/ hydr. NAAF'] = totalpeaks233_AA80['K-hydr-NAAF'] / totalpeaks233_AA80['K-NAAF']\n",
    "totalpeaks233_AA80['% R w/ meth. NAAF'] = totalpeaks233_AA80['R-meth-NAAF'] / totalpeaks233_AA80['R-NAAF']\n",
    "\n",
    "# calculate NAAF summed numerator over denominator (in above cell) for all peptides in dataset i: a check\n",
    "totalpeaks233_AA80['NAAF check'] = totalpeaks233_AA80['NAAF'] / 1.128868e+06\n",
    "\n",
    "# write modified dataframe to new txt file, same name + totals\n",
    "totalpeaks233_AA80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/233/ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_15ppm_DN80_NAAF_totals.csv\")\n",
    "\n",
    "totalpeaks233_AA80.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Visualizing the results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ALC max:  98\n",
      "ALC min:  80\n"
     ]
    }
   ],
   "source": [
    "print(\"ALC max: \", peaks233_80['ALC (%)'].max())\n",
    "print(\"ALC min: \", peaks233_80['ALC (%)'].min())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exporting txt files of stripped peptides at confidence cutoffs:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total modified peptides in 233: 4235\n",
      "Deduplicated modified peptides in 233: 4144\n",
      "Total strippled peptides in 233: 4235\n",
      "Deduplicated stripped peptides in 233: 4048\n"
     ]
    }
   ],
   "source": [
    "# keep only peptide column >50% ALC\n",
    "pep233_50_moddup = peaks233_50[[\"Peptide\"]]\n",
    "\n",
    "# keep only the stripped peptide column for the >50% ALC\n",
    "# this is what we'll use for UniPept input, etc\n",
    "pep233_50_dup = peaks233_50[[\"stripped peptide\"]]\n",
    "\n",
    "# deduplicate both of these lists\n",
    "pep233_50_mod = pep233_50_moddup.drop_duplicates()\n",
    "pep233_50 = pep233_50_dup.drop_duplicates()\n",
    "\n",
    "# print out the #s of modified and stripped peptides, deduplicated and not\n",
    "\n",
    "print(\"Total modified peptides in 233:\", len(pep233_50_moddup))\n",
    "print(\"Deduplicated modified peptides in 233:\", len(pep233_50_mod))\n",
    "print(\"Total strippled peptides in 233:\", len(pep233_50_dup))\n",
    "print(\"Deduplicated stripped peptides in 233:\", len(pep233_50))\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "pep233_50.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/233/ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_15ppm_DN50_stripped_peptides.txt\", header=False, index=False)\n",
    "\n",
    "# made the text file into a FASTA \n",
    "\n",
    "!awk '{print \">\"NR\"\\n\"$0}' /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/233/ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_15ppm_DN50_stripped_peptides.txt > \\\n",
    "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/233/ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_15ppm_DN50_stripped_peptides.fas\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total modified peptides in 233: 730\n",
      "Deduplicated modified peptides in 233: 686\n",
      "Total strippled peptides in 233: 730\n",
      "Deduplicated stripped peptides in 233: 684\n"
     ]
    }
   ],
   "source": [
    "# keep only peptide column >80% ALC\n",
    "pep233_80_moddup = peaks233_80[[\"Peptide\"]]\n",
    "\n",
    "# keep only the stripped peptide column for the >80% ALC\n",
    "# this is what we'll use for UniPept input, etc\n",
    "pep233_80_dup = peaks233_80[[\"stripped peptide\"]]\n",
    "\n",
    "# deduplicate both of these lists\n",
    "pep233_80_mod = pep233_80_moddup.drop_duplicates()\n",
    "pep233_80 = pep233_80_dup.drop_duplicates()\n",
    "\n",
    "# print out the #s of modified and stripped peptides, deduplicated and not\n",
    "\n",
    "print(\"Total modified peptides in 233:\", len(pep233_80_moddup))\n",
    "print(\"Deduplicated modified peptides in 233:\", len(pep233_80_mod))\n",
    "print(\"Total strippled peptides in 233:\", len(pep233_80_dup))\n",
    "print(\"Deduplicated stripped peptides in 233:\", len(pep233_80))\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "pep233_80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/233/ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_15ppm_DN80_stripped_peptides.txt\", header=False, index=False)\n",
    "\n",
    "# made the text file into a FASTA \n",
    "\n",
    "!awk '{print \">\"NR\"\\n\"$0}' /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/233/ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_15ppm_DN80_stripped_peptides.txt > \\\n",
    "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDN-PTMopt/233/ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_15ppm_DN80_stripped_peptides.fas\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
