{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sequence to protein mapping with PepExplorer\n",
    "\n",
    "### PepExplorer (Leprevost et al., 2014) is a sequence similarity-driven tool that takes the output of our de novo algorithm (PEAK), which contain candidate sequences with PTMs and confidence scores, and maps them to a user-defined target-decoy sequence database.\n",
    "\n",
    "### PepExplorer is part of the Pattern Lab for Proteomics suite of tools available for free download [here](http://proteomics.fiocruz.br/software/pepexplorer/).\n",
    "\n",
    "### I used PepExplorer 0.1.0.78 on my PC (x64 Windows 10). The parameters for searching de novo results (>50% ALC) were:\n",
    "\n",
    "    - Min AAs in peptide: 5\n",
    "    - Decoy method: reverse\n",
    "    - PEAKS 8.0 parameters\n",
    "    - MinIdent: 80%\n",
    "    \n",
    "### I exported the results of the alignments of each MED4 replicate vs. the MED4 sequence database (proteome from GenBank) and moved the .txt results to my local machine in this git directory. Using LibreOffice Calc I also took out only the protein IDs/# of alignments/spec counts/description and saved as a .csv:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.0.5'"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# LIBRARIES\n",
    "#import pandas library for working with tabular data\n",
    "import os\n",
    "os.getcwd()\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "from scipy.stats import kde\n",
    "#import regular expresson (regex)\n",
    "import re\n",
    "#check pandas version\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/millieginty/Documents/git-repos/2017-etnp/analyses/pronovo-2020/pepexplorer/med4-PTMopt\n"
     ]
    }
   ],
   "source": [
    "cd /home/millieginty/Documents/git-repos/2017-etnp/analyses/pronovo-2020/pepexplorer/med4-PTMopt/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['ProteinID', 'Alignments', 'SpecCounts', 'Unique', 'Coverage',\n",
      "       'Protein description'],\n",
      "      dtype='object')\n",
      "Index(['ProteinID', 'Alignments', 'SpecCounts', 'Unique', 'Coverage',\n",
      "       'Protein description'],\n",
      "      dtype='object')\n",
      "# redundant PeaksDN80 proteins =  1167\n",
      "# nonredundant PeaksDN80 proteins =  1167\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0    PMM0001\n",
       "1    PMM0002\n",
       "2    PMM0003\n",
       "3    PMM0004\n",
       "4    PMM0005\n",
       "Name: ProteinID, dtype: object"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# combining and exporting results for PepExplorer results from de novo peptide > 80% ALC\n",
    "# read the protein files into dataframes and combined the replicates\n",
    "\n",
    "pe1 = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/analyses/pronovo-2020/pepexplorer/med4-PTMopt/RAL95_MED2_trypsin_1_PTMopt_DN80_PepExplorer-vs-MED4Graa_15ppm.csv\")\n",
    "pe2 = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/analyses/pronovo-2020/pepexplorer/med4-PTMopt/RAL95_MED2_trypsin_2_PTMopt_DN80_PepExplorer-vs-MED4Graa_15ppm.csv\")\n",
    "\n",
    "frames = [pe1, pe2]\n",
    "\n",
    "print(pe1.columns)\n",
    "print(pe2.columns)\n",
    "\n",
    "pe80prot = pe1.merge(pe2,  on='ProteinID', how= 'outer')\n",
    "\n",
    "# how many redundant proteins?\n",
    "print(\"# redundant PeaksDN80 proteins = \", len(pe80prot))\n",
    "\n",
    "# export as a csv\n",
    "pe80prot.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/analyses/pronovo-2020/pepexplorer/med4-PTMopt/RAL4_MED2_combine_DN80PepEx.csv\")\n",
    "\n",
    "# get just protein list\n",
    "pe80dup = pe80prot['ProteinID']\n",
    "\n",
    "# deduplicate protein list\n",
    "pe80 = pe80dup.drop_duplicates()\n",
    "\n",
    "# how many nonredundant proteins?\n",
    "print(\"# nonredundant PeaksDN80 proteins = \", len(pe80))\n",
    "\n",
    "# export as a .txt file without headers\n",
    "pe80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/analyses/pronovo-2020/pepexplorer/med4-PTMopt/RAL4_MED2_combine_DN80_15ppm_PepEx_proteins.txt\", header=False, index=False)\n",
    "\n",
    "# take a look\n",
    "pe80.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Let's compare to our Comet results!\n",
    "\n",
    "We've already exported a .txt file that's a list of all the accession #'s found in the MED4 Comet nb.\n",
    "We need to export a full list of proteins and identifying peptides in ProteinProphet."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP\n"
     ]
    }
   ],
   "source": [
    "cd /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PSMs in Comet results < 1% FDR: 19809\n",
      "peptide in full Comet results < 1% FDR: 7190\n",
      "proteins in full Comet results < 1% FDR: 1217\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>spectrum</th>\n",
       "      <th>xcorr</th>\n",
       "      <th>deltacn</th>\n",
       "      <th>expect</th>\n",
       "      <th>peptide</th>\n",
       "      <th>ions</th>\n",
       "      <th>calc_neutral_pep_mass</th>\n",
       "      <th>precursor_intensity</th>\n",
       "      <th>protein</th>\n",
       "      <th>protein_descr</th>\n",
       "      <th>...</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>k-iron</th>\n",
       "      <th>k-meth</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>stripped length</th>\n",
       "      <th>ptm-total</th>\n",
       "      <th>NAAF num.</th>\n",
       "      <th>binned</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.37926.37926.2</td>\n",
       "      <td>6.713</td>\n",
       "      <td>0.789</td>\n",
       "      <td>3.740000e-16</td>\n",
       "      <td>K.SKPFNDGFNAATGDYVDMSAAGIVDPAK.V</td>\n",
       "      <td>41/54</td>\n",
       "      <td>2857.3229</td>\n",
       "      <td>5578550.0</td>\n",
       "      <td>PMM1436</td>\n",
       "      <td>| groEL | chaperonin GroEL</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SKPFNDGFNAATGDYVDMSAAGIVDPAK</td>\n",
       "      <td>28</td>\n",
       "      <td>0</td>\n",
       "      <td>199233.928571</td>\n",
       "      <td>6.5-7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.31081.31081.2</td>\n",
       "      <td>6.630</td>\n",
       "      <td>0.708</td>\n",
       "      <td>1.080000e-05</td>\n",
       "      <td>K.IDIEDGGIVTIASHDGAAAEEAQK.L</td>\n",
       "      <td>36/46</td>\n",
       "      <td>2409.1660</td>\n",
       "      <td>1448720.0</td>\n",
       "      <td>PMM1191</td>\n",
       "      <td>| pnp | polynucleotide phosphorylase\\polyadeny...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>IDIEDGGIVTIASHDGAAAEEAQK</td>\n",
       "      <td>24</td>\n",
       "      <td>0</td>\n",
       "      <td>60363.333333</td>\n",
       "      <td>6.5-7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_1.33045.33045.2</td>\n",
       "      <td>6.521</td>\n",
       "      <td>0.768</td>\n",
       "      <td>2.670000e-13</td>\n",
       "      <td>R.AKNTGVEYAQ[129.04]GLVALGGDDEELAK.E</td>\n",
       "      <td>37/46</td>\n",
       "      <td>2448.2020</td>\n",
       "      <td>3484900.0</td>\n",
       "      <td>PMM0228</td>\n",
       "      <td>| psbO | photosystem II manganese-stabilizing ...</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>AKNTGVEYAQGLVALGGDDEELAK</td>\n",
       "      <td>24</td>\n",
       "      <td>1</td>\n",
       "      <td>145204.166667</td>\n",
       "      <td>6.5-7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_2.32518.32518.4</td>\n",
       "      <td>6.318</td>\n",
       "      <td>1.000</td>\n",
       "      <td>1.270000e-07</td>\n",
       "      <td>K.LAIDDSSIN[115.03]LDQVDYINAHGTSTTANDKNETSAIK.S</td>\n",
       "      <td>49/204</td>\n",
       "      <td>3734.7759</td>\n",
       "      <td>5246400.0</td>\n",
       "      <td>PMM1609</td>\n",
       "      <td>| fabF | 3-oxoacyl-[acyl-carrier-protein] synt...</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>LAIDDSSINLDQVDYINAHGTSTTANDKNETSAIK</td>\n",
       "      <td>35</td>\n",
       "      <td>1</td>\n",
       "      <td>149897.142857</td>\n",
       "      <td>6-6.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>022016_RAL4_95_MED2_trypsin_2.44677.44677.3</td>\n",
       "      <td>6.048</td>\n",
       "      <td>1.000</td>\n",
       "      <td>4.980000e-11</td>\n",
       "      <td>R.SNLNTVIIDKNPSVGALAITHQIAN[115.03]YPGVPVDISGEK.L</td>\n",
       "      <td>48/144</td>\n",
       "      <td>3845.0211</td>\n",
       "      <td>1562260.0</td>\n",
       "      <td>PMM0211</td>\n",
       "      <td>| trxB | FAD-dependent pyridine nucleotide-dis...</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>SNLNTVIIDKNPSVGALAITHQIANYPGVPVDISGEK</td>\n",
       "      <td>37</td>\n",
       "      <td>1</td>\n",
       "      <td>42223.243243</td>\n",
       "      <td>6-6.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 44 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       spectrum  xcorr  deltacn        expect  \\\n",
       "0   022016_RAL4_95_MED2_trypsin_1.37926.37926.2  6.713    0.789  3.740000e-16   \n",
       "1   022016_RAL4_95_MED2_trypsin_1.31081.31081.2  6.630    0.708  1.080000e-05   \n",
       "3   022016_RAL4_95_MED2_trypsin_1.33045.33045.2  6.521    0.768  2.670000e-13   \n",
       "5   022016_RAL4_95_MED2_trypsin_2.32518.32518.4  6.318    1.000  1.270000e-07   \n",
       "12  022016_RAL4_95_MED2_trypsin_2.44677.44677.3  6.048    1.000  4.980000e-11   \n",
       "\n",
       "                                              peptide    ions  \\\n",
       "0                    K.SKPFNDGFNAATGDYVDMSAAGIVDPAK.V   41/54   \n",
       "1                        K.IDIEDGGIVTIASHDGAAAEEAQK.L   36/46   \n",
       "3                R.AKNTGVEYAQ[129.04]GLVALGGDDEELAK.E   37/46   \n",
       "5     K.LAIDDSSIN[115.03]LDQVDYINAHGTSTTANDKNETSAIK.S  49/204   \n",
       "12  R.SNLNTVIIDKNPSVGALAITHQIAN[115.03]YPGVPVDISGEK.L  48/144   \n",
       "\n",
       "    calc_neutral_pep_mass  precursor_intensity  protein  \\\n",
       "0               2857.3229            5578550.0  PMM1436   \n",
       "1               2409.1660            1448720.0  PMM1191   \n",
       "3               2448.2020            3484900.0  PMM0228   \n",
       "5               3734.7759            5246400.0  PMM1609   \n",
       "12              3845.0211            1562260.0  PMM0211   \n",
       "\n",
       "                                        protein_descr  ... n-deam q-deam  \\\n",
       "0                          | groEL | chaperonin GroEL  ...      0      0   \n",
       "1   | pnp | polynucleotide phosphorylase\\polyadeny...  ...      0      0   \n",
       "3   | psbO | photosystem II manganese-stabilizing ...  ...      0      1   \n",
       "5   | fabF | 3-oxoacyl-[acyl-carrier-protein] synt...  ...      1      0   \n",
       "12  | trxB | FAD-dependent pyridine nucleotide-dis...  ...      1      0   \n",
       "\n",
       "    k-iron  k-meth  r-meth                       stripped peptide  \\\n",
       "0        0       0       0           SKPFNDGFNAATGDYVDMSAAGIVDPAK   \n",
       "1        0       0       0               IDIEDGGIVTIASHDGAAAEEAQK   \n",
       "3        0       0       0               AKNTGVEYAQGLVALGGDDEELAK   \n",
       "5        0       0       0    LAIDDSSINLDQVDYINAHGTSTTANDKNETSAIK   \n",
       "12       0       0       0  SNLNTVIIDKNPSVGALAITHQIANYPGVPVDISGEK   \n",
       "\n",
       "    stripped length  ptm-total      NAAF num.  binned  \n",
       "0                28          0  199233.928571   6.5-7  \n",
       "1                24          0   60363.333333   6.5-7  \n",
       "3                24          1  145204.166667   6.5-7  \n",
       "5                35          1  149897.142857   6-6.5  \n",
       "12               37          1   42223.243243   6-6.5  \n",
       "\n",
       "[5 rows x 44 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read in Comet <1% FDR peptides created in MED4 Comet notebook\n",
    "# @15 ppm tolerance against GenBank MED4 db with PTM optimzations, its XCorr > 2.17\n",
    "# read in the version without DECOY proteins (there were just a couple, but still)\n",
    "\n",
    "cometfull = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_combine_PTMopt_Comet15_2.17XCorr_noDECOY.csv\")\n",
    "\n",
    "print(\"PSMs in Comet results < 1% FDR:\", len(cometfull))\n",
    "\n",
    "# deduplicate based on modified peptides (so, get all peptides)\n",
    "cometpep = cometfull.drop_duplicates(subset='peptide')\n",
    "\n",
    "# deduplicate based on protein name (PMM)\n",
    "cometpro = cometfull.drop_duplicates(subset='protein')\n",
    "\n",
    "\n",
    "print(\"peptide in full Comet results < 1% FDR:\", len(cometpep))\n",
    "print(\"proteins in full Comet results < 1% FDR:\", len(cometpro))\n",
    "\n",
    "# write to a csv\n",
    "\n",
    "cometpro.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/TPP/RAL4_MED2_trypsin_combine_PTMopt_Comet15_2.17XCorr_noDECOY_proteins.csv\")\n",
    "\n",
    "cometpro.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Compare to PeaksDB proteins:\n",
    "\n",
    "In the PeaksDB output there's a protein csv. \n",
    "We'll read that in as a pandas dataframe and see what it looks like."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/PeaksDB/MED_1_MED4-RAL_QE_PEAKS_163\n"
     ]
    }
   ],
   "source": [
    "cd /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/PeaksDB/MED_1_MED4-RAL_QE_PEAKS_163/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "'MED_1_MED4-RAL_QE_PEAKS_163_DB search psm.csv'\r\n",
      "'MED_1_MED4-RAL_QE_PEAKS_163_de novo only peptides.csv'\r\n",
      " MED_1_MED4-RAL_QE_PEAKS_163_peptide.csv\r\n",
      " MED_1_MED4-RAL_QE_PEAKS_163_peptides.pep.xml\r\n",
      " MED_1_MED4-RAL_QE_PEAKS_163_protein-peptides.csv\r\n",
      " MED_1_MED4-RAL_QE_PEAKS_163_proteins.csv\r\n",
      " MED_1_MED4-RAL_QE_PEAKS_163_proteins.fasta\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Protein Group', 'Protein ID', 'Accession', '-10lgP', 'Coverage (%)',\n",
      "       'Coverage (%) RAL95_trypsin_MED_1', 'Area RAL95_trypsin_MED_1',\n",
      "       '#Peptides', '#Unique', '#Spec RAL95_trypsin_MED_1', 'PTM', 'Avg. Mass',\n",
      "       'Description'],\n",
      "      dtype='object')\n",
      "Index(['Protein Group', 'Protein ID', 'Accession', '-10lgP', 'Coverage (%)',\n",
      "       'Coverage (%) RAL95_trypsin_MED_2', 'Area RAL95_trypsin_MED_2',\n",
      "       '#Peptides', '#Unique', '#Spec RAL95_trypsin_MED_2', 'PTM', 'Avg. Mass',\n",
      "       'Description'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Protein Group</th>\n",
       "      <th>Protein ID</th>\n",
       "      <th>Accession</th>\n",
       "      <th>-10lgP</th>\n",
       "      <th>Coverage (%)</th>\n",
       "      <th>Coverage (%) RAL95_trypsin_MED_1</th>\n",
       "      <th>Area RAL95_trypsin_MED_1</th>\n",
       "      <th>#Peptides</th>\n",
       "      <th>#Unique</th>\n",
       "      <th>#Spec RAL95_trypsin_MED_1</th>\n",
       "      <th>PTM</th>\n",
       "      <th>Avg. Mass</th>\n",
       "      <th>Description</th>\n",
       "      <th>Coverage (%) RAL95_trypsin_MED_2</th>\n",
       "      <th>Area RAL95_trypsin_MED_2</th>\n",
       "      <th>#Spec RAL95_trypsin_MED_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>9880</td>\n",
       "      <td>PMM1436</td>\n",
       "      <td>409.57</td>\n",
       "      <td>94</td>\n",
       "      <td>94.0</td>\n",
       "      <td>5.030000e+10</td>\n",
       "      <td>182</td>\n",
       "      <td>181</td>\n",
       "      <td>1295.0</td>\n",
       "      <td>Carbamidomethylation; Oxidation (M); Deamidati...</td>\n",
       "      <td>57441</td>\n",
       "      <td>| groEL | chaperonin GroEL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>10366</td>\n",
       "      <td>PMM0452</td>\n",
       "      <td>339.33</td>\n",
       "      <td>68</td>\n",
       "      <td>68.0</td>\n",
       "      <td>7.650000e+09</td>\n",
       "      <td>66</td>\n",
       "      <td>65</td>\n",
       "      <td>259.0</td>\n",
       "      <td>Carbamidomethylation; Oxidation (M); Deamidati...</td>\n",
       "      <td>61441</td>\n",
       "      <td>| groEL | chaperonin GroEL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>5</td>\n",
       "      <td>12773</td>\n",
       "      <td>PMM0228</td>\n",
       "      <td>322.95</td>\n",
       "      <td>84</td>\n",
       "      <td>84.0</td>\n",
       "      <td>4.360000e+09</td>\n",
       "      <td>51</td>\n",
       "      <td>51</td>\n",
       "      <td>164.0</td>\n",
       "      <td>Carbamidomethylation; Oxidation (M); Deamidati...</td>\n",
       "      <td>28114</td>\n",
       "      <td>| psbO | photosystem II manganese-stabilizing ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14</td>\n",
       "      <td>9877</td>\n",
       "      <td>PMM1485</td>\n",
       "      <td>316.97</td>\n",
       "      <td>57</td>\n",
       "      <td>57.0</td>\n",
       "      <td>1.120000e+09</td>\n",
       "      <td>55</td>\n",
       "      <td>54</td>\n",
       "      <td>105.0</td>\n",
       "      <td>Carbamidomethylation; Oxidation (M); Deamidati...</td>\n",
       "      <td>122908</td>\n",
       "      <td>| rpoB | DNA-directed RNA polymerase subunit beta</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>9777</td>\n",
       "      <td>PMM1704</td>\n",
       "      <td>314.23</td>\n",
       "      <td>63</td>\n",
       "      <td>63.0</td>\n",
       "      <td>3.730000e+09</td>\n",
       "      <td>46</td>\n",
       "      <td>46</td>\n",
       "      <td>176.0</td>\n",
       "      <td>Carbamidomethylation; Oxidation (M); Deamidati...</td>\n",
       "      <td>68244</td>\n",
       "      <td>| dnaK | molecular chaperone DnaK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Protein Group  Protein ID Accession  -10lgP  Coverage (%)  \\\n",
       "0              1        9880   PMM1436  409.57            94   \n",
       "1              3       10366   PMM0452  339.33            68   \n",
       "2              5       12773   PMM0228  322.95            84   \n",
       "3             14        9877   PMM1485  316.97            57   \n",
       "4              4        9777   PMM1704  314.23            63   \n",
       "\n",
       "   Coverage (%) RAL95_trypsin_MED_1  Area RAL95_trypsin_MED_1  #Peptides  \\\n",
       "0                              94.0              5.030000e+10        182   \n",
       "1                              68.0              7.650000e+09         66   \n",
       "2                              84.0              4.360000e+09         51   \n",
       "3                              57.0              1.120000e+09         55   \n",
       "4                              63.0              3.730000e+09         46   \n",
       "\n",
       "   #Unique  #Spec RAL95_trypsin_MED_1  \\\n",
       "0      181                     1295.0   \n",
       "1       65                      259.0   \n",
       "2       51                      164.0   \n",
       "3       54                      105.0   \n",
       "4       46                      176.0   \n",
       "\n",
       "                                                 PTM  Avg. Mass  \\\n",
       "0  Carbamidomethylation; Oxidation (M); Deamidati...      57441   \n",
       "1  Carbamidomethylation; Oxidation (M); Deamidati...      61441   \n",
       "2  Carbamidomethylation; Oxidation (M); Deamidati...      28114   \n",
       "3  Carbamidomethylation; Oxidation (M); Deamidati...     122908   \n",
       "4  Carbamidomethylation; Oxidation (M); Deamidati...      68244   \n",
       "\n",
       "                                         Description  \\\n",
       "0                         | groEL | chaperonin GroEL   \n",
       "1                         | groEL | chaperonin GroEL   \n",
       "2  | psbO | photosystem II manganese-stabilizing ...   \n",
       "3  | rpoB | DNA-directed RNA polymerase subunit beta   \n",
       "4                  | dnaK | molecular chaperone DnaK   \n",
       "\n",
       "   Coverage (%) RAL95_trypsin_MED_2  Area RAL95_trypsin_MED_2  \\\n",
       "0                               NaN                       NaN   \n",
       "1                               NaN                       NaN   \n",
       "2                               NaN                       NaN   \n",
       "3                               NaN                       NaN   \n",
       "4                               NaN                       NaN   \n",
       "\n",
       "   #Spec RAL95_trypsin_MED_2  \n",
       "0                        NaN  \n",
       "1                        NaN  \n",
       "2                        NaN  \n",
       "3                        NaN  \n",
       "4                        NaN  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read the protein files into dataframes and combined the replicates\n",
    "\n",
    "pdb1 = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/PeaksDB/MED_1_MED4-RAL_QE_PEAKS_163/MED_1_MED4-RAL_QE_PEAKS_163_proteins.csv\")\n",
    "pdb2 = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/PeaksDB/MED_2_MED4-RAL_QE_PEAKS_165/MED_2_MED4-RAL_QE_PEAKS_165_proteins.csv\")\n",
    "\n",
    "frames = [pdb1, pdb2]\n",
    "\n",
    "print(pdb1.columns)\n",
    "print(pdb2.columns)\n",
    "\n",
    "# concatenate dataframes\n",
    "pdb = pd.concat(frames, sort=False, ignore_index=True)\n",
    "\n",
    "pdb.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/RAL4_95_MED2_trypsin/PeaksDB/RAL4_MED2_trypsin_combine_PTMopt_PeaksDB20_15ppm_proteins.csv\")\n",
    "\n",
    "pdb.head()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
