{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Manipulation of Peaks de novo results of ETNP 2017 P2 samples LC-MS/MS data using python.\n",
    "\n",
    "Starting with:\n",
    "\n",
    "    Peaks de novo results (.csv) of PTM-optimized database searches\n",
    "\n",
    "Goal:\n",
    "\n",
    "    Files with stripped (no PTMs) peptide lists and\n",
    "    Columns with #'s of each modification in every sequence\n",
    "    Column with stripped peptide lengths (# amino acids)\n",
    "    \n",
    "### To use:\n",
    "\n",
    "#### 1. Change the input file name in *IN 4*\n",
    "#### 2. Change output file name in *IN 6*, *IN 7*, *IN 8*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We don't have technical duplicates here, sadly, unlike the MED4 Pro samples. I exported PeaksDN search results CSVs into my ETNP 2017 git repo:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt\n"
     ]
    }
   ],
   "source": [
    "cd /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ETNP-SKQ17-231-100m-0.3-JA2_DN50.csv\r\n",
      "ETNP-SKQ17-231-100m-0.3-JA2_DN50_ptm.csv\r\n",
      "ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped.csv\r\n",
      "ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas\r\n",
      "ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.txt\r\n",
      "ETNP-SKQ17-231-100m-0.3-JA2_DN50_totals.csv\r\n",
      "ETNP-SKQ17-231-100m-0.3-JA2_DN80_stripped_peptides.fas\r\n",
      "ETNP-SKQ17-231-100m-0.3-JA2_DN80_stripped_peptides.txt\r\n",
      "ETNP-SKQ17-231-100m-0.3-JA2_PTMopt_DN50_AA_NAAF.csv\r\n",
      "ETNP-SKQ17-231-100m-0.3-JA2_PTMopt_DN50.csv\r\n",
      "ETNP-SKQ17-231-100m-0.3-JA2__PTMopt_DN50_totals.csv\r\n",
      "ETNP-SKQ17-231-100m-0.3-JA2_PTMopt_DN80_AA_NAAF.csv\r\n",
      "ETNP-SKQ17-231-100m-0.3-JA2_PTMopt_DN80.csv\r\n",
      "ETNP-SKQ17-231-100m-0.3-JA2__PTMopt_DN80_totals.csv\r\n",
      "ETNP-SKQ17-233-265m-0.3-JA2_DN50_stripped.csv\r\n",
      "ETNP-SKQ17-233-265m-0.3-JA4_DN50.csv\r\n",
      "ETNP-SKQ17-233-265m-0.3-JA4_DN50_ptm.csv\r\n",
      "ETNP-SKQ17-233-265m-0.3-JA4_DN50_stripped.csv\r\n",
      "ETNP-SKQ17-233-265m-0.3-JA4_DN50_stripped_peptides.txt\r\n",
      "ETNP-SKQ17-233-265m-0.3-JA4_DN50_totals.csv\r\n",
      "ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_DN50_AA_NAAF.csv\r\n",
      "ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_DN50.csv\r\n",
      "ETNP-SKQ17-233-265m-0.3-JA4__PTMopt_DN50_totals.csv\r\n",
      "ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_DN80_AA_NAAF.csv\r\n",
      "ETNP-SKQ17-233-265m-0.3-JA4_PTMopt_DN80.csv\r\n",
      "ETNP-SKQ17-233-265m-0.3-JA4__PTMopt_DN80_totals.csv\r\n",
      "ETNP-SKQ17-243-965m-0.3-JA14_DN50.csv\r\n",
      "ETNP-SKQ17-243-965m-0.3-JA14_DN50_ptm.csv\r\n",
      "ETNP-SKQ17-243-965m-0.3-JA14_DN50_stripped.csv\r\n",
      "ETNP-SKQ17-243-965m-0.3-JA14_DN50_stripped_peptides.txt\r\n",
      "ETNP-SKQ17-243-965m-0.3-JA14_DN50_totals.csv\r\n",
      "ETNP-SKQ17-243-965m-0.3-JA14_DN80_stripped_peptides.txt\r\n",
      "ETNP-SKQ17-243-965m-0.3-JA14_PTMopt_DN50_AA_NAAF.csv\r\n",
      "ETNP-SKQ17-243-965m-0.3-JA14_PTMopt_DN50.csv\r\n",
      "ETNP-SKQ17-243-965m-0.3-JA14__PTMopt_DN50_totals.csv\r\n",
      "ETNP-SKQ17-243-965m-0.3-JA14_PTMopt_DN80_AA_NAAF.csv\r\n",
      "ETNP-SKQ17-243-965m-0.3-JA14_PTMopt_DN80.csv\r\n",
      "ETNP-SKQ17-243-965m-0.3-JA14__PTMopt_DN80_totals.csv\r\n",
      "ETNP-SKQ17-273-965m-trap_DN50.csv\r\n",
      "ETNP-SKQ17-273-965m-trap_DN50_ptm.csv\r\n",
      "ETNP-SKQ17-273-965m-trap_DN50_stripped.csv\r\n",
      "ETNP-SKQ17-273-965m-trap_DN50_stripped_peptides.txt\r\n",
      "ETNP-SKQ17-273-965m-trap_DN50_totals.csv\r\n",
      "ETNP-SKQ17-273-965m-trap_DN80_stripped_peptides.txt\r\n",
      "ETNP-SKQ17-278-265m-trap_DN50.csv\r\n",
      "ETNP-SKQ17-278-265m-trap_DN50_ptm.csv\r\n",
      "ETNP-SKQ17-278-265m-trap_DN50_stripped.csv\r\n",
      "ETNP-SKQ17-278-265m-trap_DN50_stripped_peptides.txt\r\n",
      "ETNP-SKQ17-278-265m-trap_DN50_totals.csv\r\n",
      "ETNP-SKQ17-278-265m-trap_DN80_stripped_peptides.txt\r\n",
      "ETNP-SKQ17-378-100m-trap_DN50.csv\r\n",
      "ETNP-SKQ17-378-100m-trap_DN50_ptm.csv\r\n",
      "ETNP-SKQ17-378-100m-trap_DN50_stripped.csv\r\n",
      "ETNP-SKQ17-378-100m-trap_DN50_stripped_peptides.txt\r\n",
      "ETNP-SKQ17-378-100m-trap_DN50_totals.csv\r\n",
      "ETNP-SKQ17-378-100m-trap_DN80_stripped_peptides.txt\r\n",
      "ETNP-SKQ17-combine_totals.csv\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.0.5'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# LIBRARIES\n",
    "#import pandas library for working with tabular data\n",
    "import os\n",
    "os.getcwd()\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "from scipy.stats import kde\n",
    "#import regular expresson (regex)\n",
    "import re\n",
    "#check pandas version\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 231: 100 m McLane pump filtered on 0.3 um GF-75"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# redundant Peaks peptides >50% ALC in dataframe 5312\n",
      "# redundant Peaks peptides >80% ALC in dataframe 1140\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Fraction</th>\n",
       "      <th>Scan</th>\n",
       "      <th>Source File</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Tag Length</th>\n",
       "      <th>ALC (%)</th>\n",
       "      <th>length</th>\n",
       "      <th>m/z</th>\n",
       "      <th>z</th>\n",
       "      <th>RT</th>\n",
       "      <th>Area</th>\n",
       "      <th>Mass</th>\n",
       "      <th>ppm</th>\n",
       "      <th>PTM</th>\n",
       "      <th>local confidence (%)</th>\n",
       "      <th>tag (&gt;=0%)</th>\n",
       "      <th>mode</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>14924</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>EN(+.98)LAALEK</td>\n",
       "      <td>8</td>\n",
       "      <td>98</td>\n",
       "      <td>8</td>\n",
       "      <td>444.7382</td>\n",
       "      <td>2</td>\n",
       "      <td>50.59</td>\n",
       "      <td>38500000.0</td>\n",
       "      <td>887.4600</td>\n",
       "      <td>2.0</td>\n",
       "      <td>Deamidation (NQ)</td>\n",
       "      <td>98 99 100 99 98 99 100 96</td>\n",
       "      <td>EN(+.98)LAALEK</td>\n",
       "      <td>CID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>11873</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>EN(+.98)N(+.98)LLAK</td>\n",
       "      <td>7</td>\n",
       "      <td>98</td>\n",
       "      <td>7</td>\n",
       "      <td>402.2114</td>\n",
       "      <td>2</td>\n",
       "      <td>41.36</td>\n",
       "      <td>7910000.0</td>\n",
       "      <td>802.4072</td>\n",
       "      <td>1.2</td>\n",
       "      <td>Deamidation (NQ)</td>\n",
       "      <td>97 98 97 98 99 99 99</td>\n",
       "      <td>EN(+.98)N(+.98)LLAK</td>\n",
       "      <td>CID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>16508</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>TGN(+.98)FLDPK</td>\n",
       "      <td>8</td>\n",
       "      <td>98</td>\n",
       "      <td>8</td>\n",
       "      <td>446.7253</td>\n",
       "      <td>2</td>\n",
       "      <td>55.16</td>\n",
       "      <td>5860000.0</td>\n",
       "      <td>891.4338</td>\n",
       "      <td>2.6</td>\n",
       "      <td>Deamidation (NQ)</td>\n",
       "      <td>94 97 99 99 99 99 99 98</td>\n",
       "      <td>TGN(+.98)FLDPK</td>\n",
       "      <td>CID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>27852</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>WLVNHPR</td>\n",
       "      <td>7</td>\n",
       "      <td>97</td>\n",
       "      <td>7</td>\n",
       "      <td>461.2513</td>\n",
       "      <td>2</td>\n",
       "      <td>81.49</td>\n",
       "      <td>1550000.0</td>\n",
       "      <td>920.4980</td>\n",
       "      <td>-10.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99 100 100 96 97 97 98</td>\n",
       "      <td>WLVNHPR</td>\n",
       "      <td>CID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>15671</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>TDENLPLGPK</td>\n",
       "      <td>10</td>\n",
       "      <td>97</td>\n",
       "      <td>10</td>\n",
       "      <td>542.2884</td>\n",
       "      <td>2</td>\n",
       "      <td>52.76</td>\n",
       "      <td>10900000.0</td>\n",
       "      <td>1082.5608</td>\n",
       "      <td>1.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>98 99 100 99 100 97 98 95 98 95</td>\n",
       "      <td>TDENLPLGPK</td>\n",
       "      <td>CID</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Fraction   Scan                              Source File  \\\n",
       "0         1  14924  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "1         1  11873  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "2         1  16508  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "3         1  27852  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "4         1  15671  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "\n",
       "               Peptide  Tag Length  ALC (%)  length       m/z  z     RT  \\\n",
       "0       EN(+.98)LAALEK           8       98       8  444.7382  2  50.59   \n",
       "1  EN(+.98)N(+.98)LLAK           7       98       7  402.2114  2  41.36   \n",
       "2       TGN(+.98)FLDPK           8       98       8  446.7253  2  55.16   \n",
       "3              WLVNHPR           7       97       7  461.2513  2  81.49   \n",
       "4           TDENLPLGPK          10       97      10  542.2884  2  52.76   \n",
       "\n",
       "         Area       Mass   ppm               PTM  \\\n",
       "0  38500000.0   887.4600   2.0  Deamidation (NQ)   \n",
       "1   7910000.0   802.4072   1.2  Deamidation (NQ)   \n",
       "2   5860000.0   891.4338   2.6  Deamidation (NQ)   \n",
       "3   1550000.0   920.4980 -10.8               NaN   \n",
       "4  10900000.0  1082.5608   1.3               NaN   \n",
       "\n",
       "              local confidence (%)           tag (>=0%) mode  \n",
       "0        98 99 100 99 98 99 100 96       EN(+.98)LAALEK  CID  \n",
       "1             97 98 97 98 99 99 99  EN(+.98)N(+.98)LLAK  CID  \n",
       "2          94 97 99 99 99 99 99 98       TGN(+.98)FLDPK  CID  \n",
       "3           99 100 100 96 97 97 98              WLVNHPR  CID  \n",
       "4  98 99 100 99 100 97 98 95 98 95           TDENLPLGPK  CID  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#read the CSV into a datadrame we name 'comet' using the pandas read_csv function\n",
    "peaks231_50 = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50.csv\")\n",
    "\n",
    "# also make a dataframe but keep only peptides  >80% ALC\n",
    "peaks231_80 = peaks231_50.loc[peaks231_50['ALC (%)'] >= 80].copy()\n",
    "\n",
    "# how many de novo sequence candidates >50% ALC?\n",
    "print(\"# redundant Peaks peptides >50% ALC in dataframe\", len(peaks231_50))\n",
    "# how many de novo sequence candidates >80ALC?\n",
    "print(\"# redundant Peaks peptides >80% ALC in dataframe\", len(peaks231_80))\n",
    "\n",
    "#look at the dataframe\n",
    "peaks231_50.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Fraction</th>\n",
       "      <th>Scan</th>\n",
       "      <th>Source File</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Tag Length</th>\n",
       "      <th>ALC (%)</th>\n",
       "      <th>length</th>\n",
       "      <th>m/z</th>\n",
       "      <th>z</th>\n",
       "      <th>RT</th>\n",
       "      <th>...</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>k-iron</th>\n",
       "      <th>k-meth</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>stripped length</th>\n",
       "      <th>ptm-total</th>\n",
       "      <th>NAAF num.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>14924</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>EN(+.98)LAALEK</td>\n",
       "      <td>8</td>\n",
       "      <td>98</td>\n",
       "      <td>8</td>\n",
       "      <td>444.7382</td>\n",
       "      <td>2</td>\n",
       "      <td>50.59</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ENLAALEK</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>4.812500e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>11873</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>EN(+.98)N(+.98)LLAK</td>\n",
       "      <td>7</td>\n",
       "      <td>98</td>\n",
       "      <td>7</td>\n",
       "      <td>402.2114</td>\n",
       "      <td>2</td>\n",
       "      <td>41.36</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ENLLAK</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>1.318333e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>16508</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>TGN(+.98)FLDPK</td>\n",
       "      <td>8</td>\n",
       "      <td>98</td>\n",
       "      <td>8</td>\n",
       "      <td>446.7253</td>\n",
       "      <td>2</td>\n",
       "      <td>55.16</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TGNFLDPK</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>7.325000e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>27852</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>WLVNHPR</td>\n",
       "      <td>7</td>\n",
       "      <td>97</td>\n",
       "      <td>7</td>\n",
       "      <td>461.2513</td>\n",
       "      <td>2</td>\n",
       "      <td>81.49</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>WLVNHPR</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>2.214286e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>15671</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>TDENLPLGPK</td>\n",
       "      <td>10</td>\n",
       "      <td>97</td>\n",
       "      <td>10</td>\n",
       "      <td>542.2884</td>\n",
       "      <td>2</td>\n",
       "      <td>52.76</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TDENLPLGPK</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>1.090000e+06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 48 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Fraction   Scan                              Source File  \\\n",
       "0         1  14924  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "1         1  11873  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "2         1  16508  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "3         1  27852  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "4         1  15671  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "\n",
       "               Peptide  Tag Length  ALC (%)  length       m/z  z     RT  ...  \\\n",
       "0       EN(+.98)LAALEK           8       98       8  444.7382  2  50.59  ...   \n",
       "1  EN(+.98)N(+.98)LLAK           7       98       7  402.2114  2  41.36  ...   \n",
       "2       TGN(+.98)FLDPK           8       98       8  446.7253  2  55.16  ...   \n",
       "3              WLVNHPR           7       97       7  461.2513  2  81.49  ...   \n",
       "4           TDENLPLGPK          10       97      10  542.2884  2  52.76  ...   \n",
       "\n",
       "   m-oxid  n-deam  q-deam k-iron k-meth r-meth stripped peptide  \\\n",
       "0       0       1       0      0      0      0         ENLAALEK   \n",
       "1       0       2       0      0      0      0           ENLLAK   \n",
       "2       0       1       0      0      0      0         TGNFLDPK   \n",
       "3       0       0       0      0      0      0          WLVNHPR   \n",
       "4       0       0       0      0      0      0       TDENLPLGPK   \n",
       "\n",
       "   stripped length  ptm-total     NAAF num.  \n",
       "0                8          1  4.812500e+06  \n",
       "1                6          2  1.318333e+06  \n",
       "2                8          1  7.325000e+05  \n",
       "3                7          0  2.214286e+05  \n",
       "4               10          0  1.090000e+06  \n",
       "\n",
       "[5 rows x 48 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use a count function to enumerate the # of A's (alanines) in each peptide\n",
    "peaks231_50['A'] = peaks231_50['Peptide'].str.count(\"A\")\n",
    "\n",
    "# use a count function to enumerate the # of C's (cysteines) in each peptide\n",
    "peaks231_50['C'] = peaks231_50['Peptide'].str.count(\"C\")\n",
    "\n",
    "# use a count function to enumerate the # of D's (aspartic acids) in each peptide\n",
    "peaks231_50['D'] = peaks231_50['Peptide'].str.count(\"D\")\n",
    "\n",
    "# use a count function to enumerate the # of E's (glutamic acids) in each peptide\n",
    "peaks231_50['E'] = peaks231_50['Peptide'].str.count(\"E\")\n",
    "\n",
    "# use a count function to enumerate the # of F's (phenylalanines) in each peptide\n",
    "peaks231_50['F'] = peaks231_50['Peptide'].str.count(\"F\")\n",
    "\n",
    "# use a count function to enumerate the # of G's (glycines) in each peptide\n",
    "peaks231_50['G'] = peaks231_50['Peptide'].str.count(\"G\")\n",
    "\n",
    "# use a count function to enumerate the # of H's (histidines) in each peptide\n",
    "peaks231_50['H'] = peaks231_50['Peptide'].str.count(\"H\")\n",
    "\n",
    "# use a count function to enumerate the # of I's (isoleucines) in each peptide\n",
    "# in peaks231_50 output, there will be no isoleucines (they're lumped in with leucines)\n",
    "peaks231_50['I'] = peaks231_50['Peptide'].str.count(\"I\")\n",
    "\n",
    "# use a count function to enumerate the # of K's (lysines) in each peptide\n",
    "peaks231_50['K'] = peaks231_50['Peptide'].str.count(\"K\")\n",
    "\n",
    "# use a count function to enumerate the # of L's (leucines) in each peptide\n",
    "# also these include the isoleucines\n",
    "peaks231_50['I/L'] = peaks231_50['Peptide'].str.count(\"L\")\n",
    "\n",
    "# use a count function to enumerate the # of M's (methionines) in each peptide\n",
    "peaks231_50['M'] = peaks231_50['Peptide'].str.count(\"M\")\n",
    "\n",
    "# use a count function to enumerate the # of N's (asparagines) in each peptide\n",
    "peaks231_50['N'] = peaks231_50['Peptide'].str.count(\"N\")\n",
    "\n",
    "# use a count function to enumerate the # of P's ([prolines]) in each peptide\n",
    "peaks231_50['P'] = peaks231_50['Peptide'].str.count(\"P\")\n",
    "\n",
    "# use a count function to enumerate the # of Q's (glutamines) in each peptide\n",
    "peaks231_50['Q'] = peaks231_50['Peptide'].str.count(\"Q\")\n",
    "\n",
    "# use a count function to enumerate the # of R's (arginines) in each peptide\n",
    "peaks231_50['R'] = peaks231_50['Peptide'].str.count(\"R\")\n",
    "\n",
    "# use a count function to enumerate the # of S's (serines) in each peptide\n",
    "peaks231_50['S'] = peaks231_50['Peptide'].str.count(\"S\")\n",
    "\n",
    "# use a count function to enumerate the # of T's (threonines) in each peptide\n",
    "peaks231_50['T'] = peaks231_50['Peptide'].str.count(\"T\")\n",
    "\n",
    "# use a count function to enumerate the # of V's (valines) in each peptide\n",
    "peaks231_50['V'] = peaks231_50['Peptide'].str.count(\"V\")\n",
    "\n",
    "# use a count function to enumerate the # of W's (tryptophans) in each peptide\n",
    "peaks231_50['W'] = peaks231_50['Peptide'].str.count(\"W\")\n",
    "\n",
    "# use a count function to enumerate the # of Y's (tyrosines) in each peptide\n",
    "peaks231_50['Y'] = peaks231_50['Peptide'].str.count(\"Y\")\n",
    "\n",
    "# use a count function to enumerate the # of carbamidomethylated C's in each peptide\n",
    "peaks231_50['c-carb'] = peaks231_50['Peptide'].str.count(\"57.02\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "peaks231_50['m-oxid'] = peaks231_50['Peptide'].apply(lambda x: x.count('M(+15.99)'))\n",
    "\n",
    "# use a lamba function to enumerate the # of deamidated N's in each peptide\n",
    "# used to use peaks231_50['n-deam'] = peaks231_50['Peptide'].str.count(\"N\\(+.98\") but that didn't work with the 'N'\n",
    "\n",
    "peaks231_50['n-deam'] = peaks231_50['Peptide'].apply(lambda x: x.count('N(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "peaks231_50['q-deam'] = peaks231_50['Peptide'].apply(lambda x: x.count('Q(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of iron adducted K's in each peptide\n",
    "peaks231_50['k-iron'] = peaks231_50['Peptide'].str.count(\"53.92\")\n",
    "\n",
    "# use a count function to enumerate the # of methylated K's in each peptide\n",
    "peaks231_50['k-meth'] = peaks231_50['Peptide'].apply(lambda x: x.count('K(+14.02)'))\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "peaks231_50['r-meth'] = peaks231_50['Peptide'].apply(lambda x: x.count('R(+14.02)'))\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "peaks231_50['stripped peptide'] = peaks231_50['Peptide'].str.replace(r\"\\(.*\\)\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "peaks231_50['stripped length'] = peaks231_50['stripped peptide'].apply(len)\n",
    "\n",
    "# total the number of modifications in sequence\n",
    "peaks231_50['ptm-total'] = peaks231_50['c-carb'] + peaks231_50['m-oxid'] + peaks231_50['n-deam'] + peaks231_50['q-deam'] + peaks231_50['k-iron'] + peaks231_50['k-meth'] + peaks231_50['r-meth']\n",
    "\n",
    "# calculate NAAF numerator for each peptide k\n",
    "peaks231_50['NAAF num.'] = peaks231_50['Area'] / peaks231_50['stripped length']\n",
    "\n",
    "# write modified dataframe to new txt file, same name + 'stripped'\n",
    "peaks231_50.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_PTMopt_DN50.csv\")\n",
    "\n",
    "\n",
    "# check out the results\n",
    "peaks231_50.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>K</th>\n",
       "      <th>I/L</th>\n",
       "      <th>...</th>\n",
       "      <th>Total area</th>\n",
       "      <th>Total length</th>\n",
       "      <th>% C w/ carb</th>\n",
       "      <th>% M w/ oxid</th>\n",
       "      <th>% N w/ deam</th>\n",
       "      <th>% Q w/ deam</th>\n",
       "      <th>% K w/ iron</th>\n",
       "      <th>% K w/ meth</th>\n",
       "      <th>% R w/ meth</th>\n",
       "      <th>NAAF denom.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>3927</td>\n",
       "      <td>626</td>\n",
       "      <td>2010</td>\n",
       "      <td>2762</td>\n",
       "      <td>1551</td>\n",
       "      <td>2101</td>\n",
       "      <td>1048</td>\n",
       "      <td>0</td>\n",
       "      <td>5403</td>\n",
       "      <td>4784</td>\n",
       "      <td>...</td>\n",
       "      <td>1.264679e+10</td>\n",
       "      <td>41900</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.43424</td>\n",
       "      <td>0.202742</td>\n",
       "      <td>0.029278</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.147326</td>\n",
       "      <td>0.287637</td>\n",
       "      <td>301832.734206</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 A    C     D     E     F     G     H  I     K   I/L  ...  \\\n",
       "sample total  3927  626  2010  2762  1551  2101  1048  0  5403  4784  ...   \n",
       "\n",
       "                Total area  Total length  % C w/ carb  % M w/ oxid  \\\n",
       "sample total  1.264679e+10         41900          1.0      0.43424   \n",
       "\n",
       "              % N w/ deam  % Q w/ deam  % K w/ iron  % K w/ meth  % R w/ meth  \\\n",
       "sample total     0.202742     0.029278          0.0     0.147326     0.287637   \n",
       "\n",
       "                NAAF denom.  \n",
       "sample total  301832.734206  \n",
       "\n",
       "[1 rows x 37 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of certain columns in the stripped peptide dataframe above (for >50% ALC)\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'A': peaks231_50['A'].sum(),\n",
    "        'C': peaks231_50['C'].sum(),\n",
    "        'D': peaks231_50['D'].sum(),\n",
    "        'E': peaks231_50['E'].sum(),\n",
    "        'F': peaks231_50['F'].sum(),\n",
    "        'G': peaks231_50['G'].sum(),\n",
    "        'H': peaks231_50['H'].sum(),\n",
    "        'I': peaks231_50['I'].sum(),\n",
    "        'K': peaks231_50['K'].sum(),\n",
    "        'I/L': peaks231_50['I/L'].sum(),\n",
    "        'M': peaks231_50['M'].sum(),\n",
    "        'N': peaks231_50['N'].sum(),\n",
    "        'P': peaks231_50['P'].sum(),\n",
    "        'Q': peaks231_50['Q'].sum(),\n",
    "        'R': peaks231_50['R'].sum(),\n",
    "        'S': peaks231_50['S'].sum(),\n",
    "        'T': peaks231_50['T'].sum(),\n",
    "        'V': peaks231_50['V'].sum(),\n",
    "        'W': peaks231_50['W'].sum(),\n",
    "        'Y': peaks231_50['Y'].sum(),\n",
    "        'c-carb': peaks231_50['c-carb'].sum(),\n",
    "        'm-oxid': peaks231_50['m-oxid'].sum(),\n",
    "        'n-deam': peaks231_50['n-deam'].sum(),\n",
    "        'q-deam': peaks231_50['q-deam'].sum(),\n",
    "        'k-iron': peaks231_50['k-iron'].sum(),\n",
    "        'k-meth': peaks231_50['k-meth'].sum(),\n",
    "        'r-meth': peaks231_50['r-meth'].sum(),\n",
    "        'Total area': peaks231_50['Area'].sum(),\n",
    "        'Total length': peaks231_50['stripped length'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaks231_50 = pd.DataFrame(data, columns=['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'I/L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', 'm-oxid', 'n-deam', 'q-deam', 'k-iron', 'k-meth', 'r-meth', 'Total area', 'Total length'], index=index)\n",
    "\n",
    "# calculate percentage of C's with carb (should be 1.0)\n",
    "totalpeaks231_50['% C w/ carb'] = totalpeaks231_50['c-carb'] / totalpeaks231_50['C'] \n",
    "\n",
    "# calculate percentage of M's that are oxidized\n",
    "totalpeaks231_50['% M w/ oxid'] = totalpeaks231_50['m-oxid'] / totalpeaks231_50['M'] \n",
    "\n",
    "# calculate percentage of N's that are deamidated\n",
    "totalpeaks231_50['% N w/ deam'] = totalpeaks231_50['n-deam'] / totalpeaks231_50['N'] \n",
    "\n",
    "# calculate percentage of Q's that are deamidated\n",
    "totalpeaks231_50['% Q w/ deam'] = totalpeaks231_50['q-deam'] / totalpeaks231_50['Q'] \n",
    "\n",
    "# calculate percentage of K's that are hydroxylated\n",
    "totalpeaks231_50['% K w/ iron'] = totalpeaks231_50['k-iron'] / totalpeaks231_50['K'] \n",
    "\n",
    "# calculate percentage of K's that are methylated\n",
    "totalpeaks231_50['% K w/ meth'] = totalpeaks231_50['k-meth'] / totalpeaks231_50['K'] \n",
    "\n",
    "# calculate percentage of R's that are methylated\n",
    "totalpeaks231_50['% R w/ meth'] = totalpeaks231_50['r-meth'] / totalpeaks231_50['R'] \n",
    "\n",
    "# calculate NAAF denominator for all peptides in dataset i\n",
    "totalpeaks231_50['NAAF denom.'] = totalpeaks231_50['Total area'] / totalpeaks231_50['Total length']\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "totalpeaks231_50.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2__PTMopt_DN50_totals.csv\")\n",
    "\n",
    "totalpeaks231_50.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>NAAF factor</th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>K</th>\n",
       "      <th>...</th>\n",
       "      <th>M-NAAF50</th>\n",
       "      <th>N-NAAF50</th>\n",
       "      <th>P-NAAF50</th>\n",
       "      <th>Q-NAAF50</th>\n",
       "      <th>R-NAAF50</th>\n",
       "      <th>S-NAAF50</th>\n",
       "      <th>T-NAAF50</th>\n",
       "      <th>V-NAAF50</th>\n",
       "      <th>W-NAAF50</th>\n",
       "      <th>Y-NAAF50</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENLAALEK</td>\n",
       "      <td>15.944261</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.944261</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENLLAK</td>\n",
       "      <td>4.367761</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.735523</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TGNFLDPK</td>\n",
       "      <td>2.426841</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.426841</td>\n",
       "      <td>2.426841</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.426841</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>WLVNHPR</td>\n",
       "      <td>0.733614</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.733614</td>\n",
       "      <td>0.733614</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.733614</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.733614</td>\n",
       "      <td>0.733614</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TDENLPLGPK</td>\n",
       "      <td>3.611272</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.611272</td>\n",
       "      <td>7.222543</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.611272</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  stripped peptide  NAAF factor  A  C  D  E  F  G  H  K  ...  M-NAAF50  \\\n",
       "0         ENLAALEK    15.944261  2  0  0  2  0  0  0  1  ...       0.0   \n",
       "1           ENLLAK     4.367761  1  0  0  1  0  0  0  1  ...       0.0   \n",
       "2         TGNFLDPK     2.426841  0  0  1  0  1  1  0  1  ...       0.0   \n",
       "3          WLVNHPR     0.733614  0  0  0  0  0  0  1  0  ...       0.0   \n",
       "4       TDENLPLGPK     3.611272  0  0  1  1  0  1  0  1  ...       0.0   \n",
       "\n",
       "    N-NAAF50  P-NAAF50  Q-NAAF50  R-NAAF50  S-NAAF50  T-NAAF50  V-NAAF50  \\\n",
       "0  15.944261  0.000000       0.0  0.000000       0.0  0.000000  0.000000   \n",
       "1   8.735523  0.000000       0.0  0.000000       0.0  0.000000  0.000000   \n",
       "2   2.426841  2.426841       0.0  0.000000       0.0  2.426841  0.000000   \n",
       "3   0.733614  0.733614       0.0  0.733614       0.0  0.000000  0.733614   \n",
       "4   3.611272  7.222543       0.0  0.000000       0.0  3.611272  0.000000   \n",
       "\n",
       "   W-NAAF50  Y-NAAF50  \n",
       "0  0.000000       0.0  \n",
       "1  0.000000       0.0  \n",
       "2  0.000000       0.0  \n",
       "3  0.733614       0.0  \n",
       "4  0.000000       0.0  \n",
       "\n",
       "[5 rows x 40 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use the calculated NAAF factor (in totalpeaks231_ dataframe, above) to caluclate the NAAF \n",
    "# NAAF: normalized normalized area abundance factor\n",
    "\n",
    "# don't have to worry here about DECOY hits messing with Area totals\n",
    "# but we will with Comet results\n",
    "\n",
    "NAAF50 = 301832.734206\n",
    "\n",
    "# use NAAF >50% ALC to get NAAF\n",
    "peaks231_50['NAAF factor'] = (peaks231_50['NAAF num.'])/NAAF50\n",
    "\n",
    "# separate out the dataframe into AAs \n",
    "peaks231_AA50 = peaks231_50[['stripped peptide', 'NAAF factor', 'A', 'C', 'D', 'E', 'F', 'G', 'H', 'K', 'I/L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y']].copy()\n",
    "\n",
    "# multiply the NAAF50 factor by the AA total to normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaks231_AA50['A-NAAF50'] = peaks231_AA50['A'] * peaks231_50['NAAF factor']\n",
    "peaks231_AA50['C-NAAF50'] = peaks231_AA50['C'] * peaks231_50['NAAF factor']\n",
    "peaks231_AA50['D-NAAF50'] = peaks231_AA50['D'] * peaks231_50['NAAF factor']\n",
    "peaks231_AA50['E-NAAF50'] = peaks231_AA50['E'] * peaks231_50['NAAF factor']\n",
    "peaks231_AA50['F-NAAF50'] = peaks231_AA50['F'] * peaks231_50['NAAF factor']\n",
    "peaks231_AA50['G-NAAF50'] = peaks231_AA50['G'] * peaks231_50['NAAF factor']\n",
    "peaks231_AA50['H-NAAF50'] = peaks231_AA50['H'] * peaks231_50['NAAF factor']\n",
    "peaks231_AA50['K-NAAF50'] = peaks231_AA50['K'] * peaks231_50['NAAF factor']\n",
    "peaks231_AA50['I/L-NAAF50'] = peaks231_AA50['I/L'] * peaks231_50['NAAF factor']\n",
    "peaks231_AA50['M-NAAF50'] = peaks231_AA50['M'] * peaks231_50['NAAF factor']\n",
    "peaks231_AA50['N-NAAF50'] = peaks231_AA50['N'] * peaks231_50['NAAF factor']\n",
    "peaks231_AA50['P-NAAF50'] = peaks231_AA50['P'] * peaks231_50['NAAF factor']\n",
    "peaks231_AA50['Q-NAAF50'] = peaks231_AA50['Q'] * peaks231_50['NAAF factor']\n",
    "peaks231_AA50['R-NAAF50'] = peaks231_AA50['R'] * peaks231_50['NAAF factor']\n",
    "peaks231_AA50['S-NAAF50'] = peaks231_AA50['S'] * peaks231_50['NAAF factor']\n",
    "peaks231_AA50['T-NAAF50'] = peaks231_AA50['T'] * peaks231_50['NAAF factor']\n",
    "peaks231_AA50['V-NAAF50'] = peaks231_AA50['V'] * peaks231_50['NAAF factor']\n",
    "peaks231_AA50['W-NAAF50'] = peaks231_AA50['W'] * peaks231_50['NAAF factor']\n",
    "peaks231_AA50['Y-NAAF50'] = peaks231_AA50['Y'] * peaks231_50['NAAF factor']\n",
    "\n",
    "# write the dataframe to a new csv\n",
    "peaks231_AA50.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_PTMopt_DN50_AA_NAAF.csv\")\n",
    "\n",
    "peaks231_AA50.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Same process but for de novo peptide >80 % ALC:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Fraction</th>\n",
       "      <th>Scan</th>\n",
       "      <th>Source File</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Tag Length</th>\n",
       "      <th>ALC (%)</th>\n",
       "      <th>length</th>\n",
       "      <th>m/z</th>\n",
       "      <th>z</th>\n",
       "      <th>RT</th>\n",
       "      <th>...</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>k-iron</th>\n",
       "      <th>k-meth</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>stripped length</th>\n",
       "      <th>ptm-total</th>\n",
       "      <th>NAAF num.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>14924</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>EN(+.98)LAALEK</td>\n",
       "      <td>8</td>\n",
       "      <td>98</td>\n",
       "      <td>8</td>\n",
       "      <td>444.7382</td>\n",
       "      <td>2</td>\n",
       "      <td>50.59</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ENLAALEK</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>4.812500e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>11873</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>EN(+.98)N(+.98)LLAK</td>\n",
       "      <td>7</td>\n",
       "      <td>98</td>\n",
       "      <td>7</td>\n",
       "      <td>402.2114</td>\n",
       "      <td>2</td>\n",
       "      <td>41.36</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ENLLAK</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>1.318333e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>16508</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>TGN(+.98)FLDPK</td>\n",
       "      <td>8</td>\n",
       "      <td>98</td>\n",
       "      <td>8</td>\n",
       "      <td>446.7253</td>\n",
       "      <td>2</td>\n",
       "      <td>55.16</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TGNFLDPK</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>7.325000e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>27852</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>WLVNHPR</td>\n",
       "      <td>7</td>\n",
       "      <td>97</td>\n",
       "      <td>7</td>\n",
       "      <td>461.2513</td>\n",
       "      <td>2</td>\n",
       "      <td>81.49</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>WLVNHPR</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>2.214286e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>15671</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>TDENLPLGPK</td>\n",
       "      <td>10</td>\n",
       "      <td>97</td>\n",
       "      <td>10</td>\n",
       "      <td>542.2884</td>\n",
       "      <td>2</td>\n",
       "      <td>52.76</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TDENLPLGPK</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>1.090000e+06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 48 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Fraction   Scan                              Source File  \\\n",
       "0         1  14924  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "1         1  11873  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "2         1  16508  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "3         1  27852  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "4         1  15671  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "\n",
       "               Peptide  Tag Length  ALC (%)  length       m/z  z     RT  ...  \\\n",
       "0       EN(+.98)LAALEK           8       98       8  444.7382  2  50.59  ...   \n",
       "1  EN(+.98)N(+.98)LLAK           7       98       7  402.2114  2  41.36  ...   \n",
       "2       TGN(+.98)FLDPK           8       98       8  446.7253  2  55.16  ...   \n",
       "3              WLVNHPR           7       97       7  461.2513  2  81.49  ...   \n",
       "4           TDENLPLGPK          10       97      10  542.2884  2  52.76  ...   \n",
       "\n",
       "   m-oxid  n-deam  q-deam k-iron k-meth r-meth stripped peptide  \\\n",
       "0       0       1       0      0      0      0         ENLAALEK   \n",
       "1       0       2       0      0      0      0           ENLLAK   \n",
       "2       0       1       0      0      0      0         TGNFLDPK   \n",
       "3       0       0       0      0      0      0          WLVNHPR   \n",
       "4       0       0       0      0      0      0       TDENLPLGPK   \n",
       "\n",
       "   stripped length  ptm-total     NAAF num.  \n",
       "0                8          1  4.812500e+06  \n",
       "1                6          2  1.318333e+06  \n",
       "2                8          1  7.325000e+05  \n",
       "3                7          0  2.214286e+05  \n",
       "4               10          0  1.090000e+06  \n",
       "\n",
       "[5 rows x 48 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use a count function to enumerate the # of A's (alanines) in each peptide\n",
    "peaks231_80['A'] = peaks231_80['Peptide'].str.count(\"A\")\n",
    "\n",
    "# use a count function to enumerate the # of C's (cysteines) in each peptide\n",
    "peaks231_80['C'] = peaks231_80['Peptide'].str.count(\"C\")\n",
    "\n",
    "# use a count function to enumerate the # of D's (aspartic acids) in each peptide\n",
    "peaks231_80['D'] = peaks231_80['Peptide'].str.count(\"D\")\n",
    "\n",
    "# use a count function to enumerate the # of E's (glutamic acids) in each peptide\n",
    "peaks231_80['E'] = peaks231_80['Peptide'].str.count(\"E\")\n",
    "\n",
    "# use a count function to enumerate the # of F's (phenylalanines) in each peptide\n",
    "peaks231_80['F'] = peaks231_80['Peptide'].str.count(\"F\")\n",
    "\n",
    "# use a count function to enumerate the # of G's (glycines) in each peptide\n",
    "peaks231_80['G'] = peaks231_80['Peptide'].str.count(\"G\")\n",
    "\n",
    "# use a count function to enumerate the # of H's (histidines) in each peptide\n",
    "peaks231_80['H'] = peaks231_80['Peptide'].str.count(\"H\")\n",
    "\n",
    "# use a count function to enumerate the # of I's (isoleucines) in each peptide\n",
    "# in peaks231_80 output, there will be no isoleucines (they're lumped in with leucines)\n",
    "peaks231_80['I'] = peaks231_80['Peptide'].str.count(\"I\")\n",
    "\n",
    "# use a count function to enumerate the # of K's (lysines) in each peptide\n",
    "peaks231_80['K'] = peaks231_80['Peptide'].str.count(\"K\")\n",
    "\n",
    "# use a count function to enumerate the # of L's (leucines) in each peptide\n",
    "# also these include the isoleucines\n",
    "peaks231_80['I/L'] = peaks231_80['Peptide'].str.count(\"L\")\n",
    "\n",
    "# use a count function to enumerate the # of M's (methionines) in each peptide\n",
    "peaks231_80['M'] = peaks231_80['Peptide'].str.count(\"M\")\n",
    "\n",
    "# use a count function to enumerate the # of N's (asparagines) in each peptide\n",
    "peaks231_80['N'] = peaks231_80['Peptide'].str.count(\"N\")\n",
    "\n",
    "# use a count function to enumerate the # of P's ([prolines]) in each peptide\n",
    "peaks231_80['P'] = peaks231_80['Peptide'].str.count(\"P\")\n",
    "\n",
    "# use a count function to enumerate the # of Q's (glutamines) in each peptide\n",
    "peaks231_80['Q'] = peaks231_80['Peptide'].str.count(\"Q\")\n",
    "\n",
    "# use a count function to enumerate the # of R's (arginines) in each peptide\n",
    "peaks231_80['R'] = peaks231_80['Peptide'].str.count(\"R\")\n",
    "\n",
    "# use a count function to enumerate the # of S's (serines) in each peptide\n",
    "peaks231_80['S'] = peaks231_80['Peptide'].str.count(\"S\")\n",
    "\n",
    "# use a count function to enumerate the # of T's (threonines) in each peptide\n",
    "peaks231_80['T'] = peaks231_80['Peptide'].str.count(\"T\")\n",
    "\n",
    "# use a count function to enumerate the # of V's (valines) in each peptide\n",
    "peaks231_80['V'] = peaks231_80['Peptide'].str.count(\"V\")\n",
    "\n",
    "# use a count function to enumerate the # of W's (tryptophans) in each peptide\n",
    "peaks231_80['W'] = peaks231_80['Peptide'].str.count(\"W\")\n",
    "\n",
    "# use a count function to enumerate the # of Y's (tyrosines) in each peptide\n",
    "peaks231_80['Y'] = peaks231_80['Peptide'].str.count(\"Y\")\n",
    "\n",
    "# use a count function to enumerate the # of carbamidomethylated C's in each peptide\n",
    "peaks231_80['c-carb'] = peaks231_80['Peptide'].str.count(\"57.02\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "peaks231_80['m-oxid'] = peaks231_80['Peptide'].apply(lambda x: x.count('M(+15.99)'))\n",
    "\n",
    "# use a lamba function to enumerate the # of deamidated N's in each peptide\n",
    "# used to use peaks231_80['n-deam'] = peaks231_80['Peptide'].str.count(\"N\\(+.98\") but that didn't work with the 'N'\n",
    "\n",
    "peaks231_80['n-deam'] = peaks231_80['Peptide'].apply(lambda x: x.count('N(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "peaks231_80['q-deam'] = peaks231_80['Peptide'].apply(lambda x: x.count('Q(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of iron adducted K's in each peptide\n",
    "peaks231_80['k-iron'] = peaks231_80['Peptide'].str.count(\"53.92\")\n",
    "\n",
    "# use a count function to enumerate the # of methylated K's in each peptide\n",
    "peaks231_80['k-meth'] = peaks231_80['Peptide'].apply(lambda x: x.count('K(+14.02)'))\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "peaks231_80['r-meth'] = peaks231_80['Peptide'].apply(lambda x: x.count('R(+14.02)'))\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "peaks231_80['stripped peptide'] = peaks231_80['Peptide'].str.replace(r\"\\(.*\\)\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "peaks231_80['stripped length'] = peaks231_80['stripped peptide'].apply(len)\n",
    "\n",
    "# total the number of modifications in sequence\n",
    "peaks231_80['ptm-total'] = peaks231_80['c-carb'] + peaks231_80['m-oxid'] + peaks231_80['n-deam'] + peaks231_80['q-deam'] + peaks231_80['k-iron'] + peaks231_80['k-meth'] + peaks231_80['r-meth']\n",
    "\n",
    "# calculate NAAF numerator for each peptide k\n",
    "peaks231_80['NAAF num.'] = peaks231_80['Area'] / peaks231_80['stripped length']\n",
    "\n",
    "# write modified dataframe to new txt file, same name + 'stripped'\n",
    "peaks231_80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_PTMopt_DN80.csv\")\n",
    "\n",
    "\n",
    "# check out the results\n",
    "peaks231_80.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Fraction</th>\n",
       "      <th>Scan</th>\n",
       "      <th>Source File</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Tag Length</th>\n",
       "      <th>ALC (%)</th>\n",
       "      <th>length</th>\n",
       "      <th>m/z</th>\n",
       "      <th>z</th>\n",
       "      <th>RT</th>\n",
       "      <th>...</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>k-iron</th>\n",
       "      <th>k-meth</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>stripped length</th>\n",
       "      <th>ptm-total</th>\n",
       "      <th>NAAF num.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>14924</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>EN(+.98)LAALEK</td>\n",
       "      <td>8</td>\n",
       "      <td>98</td>\n",
       "      <td>8</td>\n",
       "      <td>444.7382</td>\n",
       "      <td>2</td>\n",
       "      <td>50.59</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ENLAALEK</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>4.812500e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>11873</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>EN(+.98)N(+.98)LLAK</td>\n",
       "      <td>7</td>\n",
       "      <td>98</td>\n",
       "      <td>7</td>\n",
       "      <td>402.2114</td>\n",
       "      <td>2</td>\n",
       "      <td>41.36</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>ENLLAK</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>1.318333e+06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>16508</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>TGN(+.98)FLDPK</td>\n",
       "      <td>8</td>\n",
       "      <td>98</td>\n",
       "      <td>8</td>\n",
       "      <td>446.7253</td>\n",
       "      <td>2</td>\n",
       "      <td>55.16</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TGNFLDPK</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>7.325000e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>27852</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>WLVNHPR</td>\n",
       "      <td>7</td>\n",
       "      <td>97</td>\n",
       "      <td>7</td>\n",
       "      <td>461.2513</td>\n",
       "      <td>2</td>\n",
       "      <td>81.49</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>WLVNHPR</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>2.214286e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>15671</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>TDENLPLGPK</td>\n",
       "      <td>10</td>\n",
       "      <td>97</td>\n",
       "      <td>10</td>\n",
       "      <td>542.2884</td>\n",
       "      <td>2</td>\n",
       "      <td>52.76</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>TDENLPLGPK</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>1.090000e+06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 48 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Fraction   Scan                              Source File  \\\n",
       "0         1  14924  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "1         1  11873  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "2         1  16508  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "3         1  27852  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "4         1  15671  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "\n",
       "               Peptide  Tag Length  ALC (%)  length       m/z  z     RT  ...  \\\n",
       "0       EN(+.98)LAALEK           8       98       8  444.7382  2  50.59  ...   \n",
       "1  EN(+.98)N(+.98)LLAK           7       98       7  402.2114  2  41.36  ...   \n",
       "2       TGN(+.98)FLDPK           8       98       8  446.7253  2  55.16  ...   \n",
       "3              WLVNHPR           7       97       7  461.2513  2  81.49  ...   \n",
       "4           TDENLPLGPK          10       97      10  542.2884  2  52.76  ...   \n",
       "\n",
       "   m-oxid  n-deam  q-deam k-iron k-meth r-meth stripped peptide  \\\n",
       "0       0       1       0      0      0      0         ENLAALEK   \n",
       "1       0       2       0      0      0      0           ENLLAK   \n",
       "2       0       1       0      0      0      0         TGNFLDPK   \n",
       "3       0       0       0      0      0      0          WLVNHPR   \n",
       "4       0       0       0      0      0      0       TDENLPLGPK   \n",
       "\n",
       "   stripped length  ptm-total     NAAF num.  \n",
       "0                8          1  4.812500e+06  \n",
       "1                6          2  1.318333e+06  \n",
       "2                8          1  7.325000e+05  \n",
       "3                7          0  2.214286e+05  \n",
       "4               10          0  1.090000e+06  \n",
       "\n",
       "[5 rows x 48 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use a count function to enumerate the # of A's (alanines) in each peptide\n",
    "peaks231_80['A'] = peaks231_80['Peptide'].str.count(\"A\")\n",
    "\n",
    "# use a count function to enumerate the # of C's (cysteines) in each peptide\n",
    "peaks231_80['C'] = peaks231_80['Peptide'].str.count(\"C\")\n",
    "\n",
    "# use a count function to enumerate the # of D's (aspartic acids) in each peptide\n",
    "peaks231_80['D'] = peaks231_80['Peptide'].str.count(\"D\")\n",
    "\n",
    "# use a count function to enumerate the # of E's (glutamic acids) in each peptide\n",
    "peaks231_80['E'] = peaks231_80['Peptide'].str.count(\"E\")\n",
    "\n",
    "# use a count function to enumerate the # of F's (phenylalanines) in each peptide\n",
    "peaks231_80['F'] = peaks231_80['Peptide'].str.count(\"F\")\n",
    "\n",
    "# use a count function to enumerate the # of G's (glycines) in each peptide\n",
    "peaks231_80['G'] = peaks231_80['Peptide'].str.count(\"G\")\n",
    "\n",
    "# use a count function to enumerate the # of H's (histidines) in each peptide\n",
    "peaks231_80['H'] = peaks231_80['Peptide'].str.count(\"H\")\n",
    "\n",
    "# use a count function to enumerate the # of I's (isoleucines) in each peptide\n",
    "# in peaks231_80 output, there will be no isoleucines (they're lumped in with leucines)\n",
    "peaks231_80['I'] = peaks231_80['Peptide'].str.count(\"I\")\n",
    "\n",
    "# use a count function to enumerate the # of K's (lysines) in each peptide\n",
    "peaks231_80['K'] = peaks231_80['Peptide'].str.count(\"K\")\n",
    "\n",
    "# use a count function to enumerate the # of L's (leucines) in each peptide\n",
    "# also these include the isoleucines\n",
    "peaks231_80['I/L'] = peaks231_80['Peptide'].str.count(\"L\")\n",
    "\n",
    "# use a count function to enumerate the # of M's (methionines) in each peptide\n",
    "peaks231_80['M'] = peaks231_80['Peptide'].str.count(\"M\")\n",
    "\n",
    "# use a count function to enumerate the # of N's (asparagines) in each peptide\n",
    "peaks231_80['N'] = peaks231_80['Peptide'].str.count(\"N\")\n",
    "\n",
    "# use a count function to enumerate the # of P's ([prolines]) in each peptide\n",
    "peaks231_80['P'] = peaks231_80['Peptide'].str.count(\"P\")\n",
    "\n",
    "# use a count function to enumerate the # of Q's (glutamines) in each peptide\n",
    "peaks231_80['Q'] = peaks231_80['Peptide'].str.count(\"Q\")\n",
    "\n",
    "# use a count function to enumerate the # of R's (arginines) in each peptide\n",
    "peaks231_80['R'] = peaks231_80['Peptide'].str.count(\"R\")\n",
    "\n",
    "# use a count function to enumerate the # of S's (serines) in each peptide\n",
    "peaks231_80['S'] = peaks231_80['Peptide'].str.count(\"S\")\n",
    "\n",
    "# use a count function to enumerate the # of T's (threonines) in each peptide\n",
    "peaks231_80['T'] = peaks231_80['Peptide'].str.count(\"T\")\n",
    "\n",
    "# use a count function to enumerate the # of V's (valines) in each peptide\n",
    "peaks231_80['V'] = peaks231_80['Peptide'].str.count(\"V\")\n",
    "\n",
    "# use a count function to enumerate the # of W's (tryptophans) in each peptide\n",
    "peaks231_80['W'] = peaks231_80['Peptide'].str.count(\"W\")\n",
    "\n",
    "# use a count function to enumerate the # of Y's (tyrosines) in each peptide\n",
    "peaks231_80['Y'] = peaks231_80['Peptide'].str.count(\"Y\")\n",
    "\n",
    "# use a count function to enumerate the # of carbamidomethylated C's in each peptide\n",
    "peaks231_80['c-carb'] = peaks231_80['Peptide'].str.count(\"57.02\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "peaks231_80['m-oxid'] = peaks231_80['Peptide'].apply(lambda x: x.count('M(+15.99)'))\n",
    "\n",
    "# use a lamba function to enumerate the # of deamidated N's in each peptide\n",
    "# used to use peaks231_80['n-deam'] = peaks231_80['Peptide'].str.count(\"N\\(+.98\") but that didn't work with the 'N'\n",
    "\n",
    "peaks231_80['n-deam'] = peaks231_80['Peptide'].apply(lambda x: x.count('N(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "peaks231_80['q-deam'] = peaks231_80['Peptide'].apply(lambda x: x.count('Q(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of iron adducted K's in each peptide\n",
    "peaks231_80['k-iron'] = peaks231_80['Peptide'].str.count(\"53.92\")\n",
    "\n",
    "# use a count function to enumerate the # of methylated K's in each peptide\n",
    "peaks231_80['k-meth'] = peaks231_80['Peptide'].apply(lambda x: x.count('K(+14.02)'))\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "peaks231_80['r-meth'] = peaks231_80['Peptide'].apply(lambda x: x.count('R(+14.02)'))\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "peaks231_80['stripped peptide'] = peaks231_80['Peptide'].str.replace(r\"\\(.*\\)\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "peaks231_80['stripped length'] = peaks231_80['stripped peptide'].apply(len)\n",
    "\n",
    "# total the number of modifications in sequence\n",
    "peaks231_80['ptm-total'] = peaks231_80['c-carb'] + peaks231_80['m-oxid'] + peaks231_80['n-deam'] + peaks231_80['q-deam'] + peaks231_80['k-iron'] + peaks231_80['k-meth'] + peaks231_80['r-meth']\n",
    "\n",
    "# calculate NAAF numerator for each peptide k\n",
    "peaks231_80['NAAF num.'] = peaks231_80['Area'] / peaks231_80['stripped length']\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "peaks231_80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_PTMopt_DN80.csv\")\n",
    "\n",
    "\n",
    "# check out the results\n",
    "peaks231_80.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>K</th>\n",
       "      <th>I/L</th>\n",
       "      <th>...</th>\n",
       "      <th>Total area</th>\n",
       "      <th>Total length</th>\n",
       "      <th>% C w/ carb</th>\n",
       "      <th>% M w/ oxid</th>\n",
       "      <th>% N w/ deam</th>\n",
       "      <th>% Q w/ deam</th>\n",
       "      <th>% K w/ iron</th>\n",
       "      <th>% K w/ meth</th>\n",
       "      <th>% R w/ meth</th>\n",
       "      <th>NAAF denom.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>941</td>\n",
       "      <td>42</td>\n",
       "      <td>454</td>\n",
       "      <td>746</td>\n",
       "      <td>350</td>\n",
       "      <td>486</td>\n",
       "      <td>81</td>\n",
       "      <td>0</td>\n",
       "      <td>836</td>\n",
       "      <td>1415</td>\n",
       "      <td>...</td>\n",
       "      <td>8.121100e+09</td>\n",
       "      <td>9476</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.60241</td>\n",
       "      <td>0.265203</td>\n",
       "      <td>0.091837</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.084928</td>\n",
       "      <td>0.157969</td>\n",
       "      <td>857017.763508</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                A   C    D    E    F    G   H  I    K   I/L  ...  \\\n",
       "sample total  941  42  454  746  350  486  81  0  836  1415  ...   \n",
       "\n",
       "                Total area  Total length  % C w/ carb  % M w/ oxid  \\\n",
       "sample total  8.121100e+09          9476          1.0      0.60241   \n",
       "\n",
       "              % N w/ deam  % Q w/ deam  % K w/ iron  % K w/ meth  % R w/ meth  \\\n",
       "sample total     0.265203     0.091837          0.0     0.084928     0.157969   \n",
       "\n",
       "                NAAF denom.  \n",
       "sample total  857017.763508  \n",
       "\n",
       "[1 rows x 37 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of certain columns in the stripped peptide dataframe above (for >50% ALC)\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'A': peaks231_80['A'].sum(),\n",
    "        'C': peaks231_80['C'].sum(),\n",
    "        'D': peaks231_80['D'].sum(),\n",
    "        'E': peaks231_80['E'].sum(),\n",
    "        'F': peaks231_80['F'].sum(),\n",
    "        'G': peaks231_80['G'].sum(),\n",
    "        'H': peaks231_80['H'].sum(),\n",
    "        'I': peaks231_80['I'].sum(),\n",
    "        'K': peaks231_80['K'].sum(),\n",
    "        'I/L': peaks231_80['I/L'].sum(),\n",
    "        'M': peaks231_80['M'].sum(),\n",
    "        'N': peaks231_80['N'].sum(),\n",
    "        'P': peaks231_80['P'].sum(),\n",
    "        'Q': peaks231_80['Q'].sum(),\n",
    "        'R': peaks231_80['R'].sum(),\n",
    "        'S': peaks231_80['S'].sum(),\n",
    "        'T': peaks231_80['T'].sum(),\n",
    "        'V': peaks231_80['V'].sum(),\n",
    "        'W': peaks231_80['W'].sum(),\n",
    "        'Y': peaks231_80['Y'].sum(),\n",
    "        'c-carb': peaks231_80['c-carb'].sum(),\n",
    "        'm-oxid': peaks231_80['m-oxid'].sum(),\n",
    "        'n-deam': peaks231_80['n-deam'].sum(),\n",
    "        'q-deam': peaks231_80['q-deam'].sum(),\n",
    "        'k-iron': peaks231_80['k-iron'].sum(),\n",
    "        'k-meth': peaks231_80['k-meth'].sum(),\n",
    "        'r-meth': peaks231_80['r-meth'].sum(),\n",
    "        'Total area': peaks231_80['Area'].sum(),\n",
    "        'Total length': peaks231_80['stripped length'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaks231_80 = pd.DataFrame(data, columns=['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'I/L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', 'm-oxid', 'n-deam', 'q-deam', 'k-iron', 'k-meth', 'r-meth', 'Total area', 'Total length'], index=index)\n",
    "\n",
    "# calculate percentage of C's with carb (should be 1.0)\n",
    "totalpeaks231_80['% C w/ carb'] = totalpeaks231_80['c-carb'] / totalpeaks231_80['C'] \n",
    "\n",
    "# calculate percentage of M's that are oxidized\n",
    "totalpeaks231_80['% M w/ oxid'] = totalpeaks231_80['m-oxid'] / totalpeaks231_80['M'] \n",
    "\n",
    "# calculate percentage of N's that are deamidated\n",
    "totalpeaks231_80['% N w/ deam'] = totalpeaks231_80['n-deam'] / totalpeaks231_80['N'] \n",
    "\n",
    "# calculate percentage of Q's that are deamidated\n",
    "totalpeaks231_80['% Q w/ deam'] = totalpeaks231_80['q-deam'] / totalpeaks231_80['Q'] \n",
    "\n",
    "# calculate percentage of K's that are hydroxylated\n",
    "totalpeaks231_80['% K w/ iron'] = totalpeaks231_80['k-iron'] / totalpeaks231_80['K'] \n",
    "\n",
    "# calculate percentage of K's that are methylated\n",
    "totalpeaks231_80['% K w/ meth'] = totalpeaks231_80['k-meth'] / totalpeaks231_80['K'] \n",
    "\n",
    "# calculate percentage of R's that are methylated\n",
    "totalpeaks231_80['% R w/ meth'] = totalpeaks231_80['r-meth'] / totalpeaks231_80['R'] \n",
    "\n",
    "# calculate NAAF denominator for all peptides in dataset i\n",
    "totalpeaks231_80['NAAF denom.'] = totalpeaks231_80['Total area'] / totalpeaks231_80['Total length']\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "totalpeaks231_80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2__PTMopt_DN80_totals.csv\")\n",
    "\n",
    "totalpeaks231_80.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>NAAF factor</th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>K</th>\n",
       "      <th>...</th>\n",
       "      <th>M-NAAF80</th>\n",
       "      <th>N-NAAF80</th>\n",
       "      <th>P-NAAF80</th>\n",
       "      <th>Q-NAAF80</th>\n",
       "      <th>R-NAAF80</th>\n",
       "      <th>S-NAAF80</th>\n",
       "      <th>T-NAAF80</th>\n",
       "      <th>V-NAAF80</th>\n",
       "      <th>W-NAAF80</th>\n",
       "      <th>Y-NAAF80</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENLAALEK</td>\n",
       "      <td>5.615403</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.615403</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENLLAK</td>\n",
       "      <td>1.538280</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.076560</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TGNFLDPK</td>\n",
       "      <td>0.854708</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.854708</td>\n",
       "      <td>0.854708</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.854708</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>WLVNHPR</td>\n",
       "      <td>0.258371</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.258371</td>\n",
       "      <td>0.258371</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.258371</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.258371</td>\n",
       "      <td>0.258371</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TDENLPLGPK</td>\n",
       "      <td>1.271852</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.271852</td>\n",
       "      <td>2.543705</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.271852</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  stripped peptide  NAAF factor  A  C  D  E  F  G  H  K  ...  M-NAAF80  \\\n",
       "0         ENLAALEK     5.615403  2  0  0  2  0  0  0  1  ...       0.0   \n",
       "1           ENLLAK     1.538280  1  0  0  1  0  0  0  1  ...       0.0   \n",
       "2         TGNFLDPK     0.854708  0  0  1  0  1  1  0  1  ...       0.0   \n",
       "3          WLVNHPR     0.258371  0  0  0  0  0  0  1  0  ...       0.0   \n",
       "4       TDENLPLGPK     1.271852  0  0  1  1  0  1  0  1  ...       0.0   \n",
       "\n",
       "   N-NAAF80  P-NAAF80  Q-NAAF80  R-NAAF80  S-NAAF80  T-NAAF80  V-NAAF80  \\\n",
       "0  5.615403  0.000000       0.0  0.000000       0.0  0.000000  0.000000   \n",
       "1  3.076560  0.000000       0.0  0.000000       0.0  0.000000  0.000000   \n",
       "2  0.854708  0.854708       0.0  0.000000       0.0  0.854708  0.000000   \n",
       "3  0.258371  0.258371       0.0  0.258371       0.0  0.000000  0.258371   \n",
       "4  1.271852  2.543705       0.0  0.000000       0.0  1.271852  0.000000   \n",
       "\n",
       "   W-NAAF80  Y-NAAF80  \n",
       "0  0.000000       0.0  \n",
       "1  0.000000       0.0  \n",
       "2  0.000000       0.0  \n",
       "3  0.258371       0.0  \n",
       "4  0.000000       0.0  \n",
       "\n",
       "[5 rows x 40 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use the calculated NAAF factor (in totalpeaks231_ dataframe, above) to caluclate the NAAF \n",
    "# NAAF: normalized normalized area abundance factor\n",
    "\n",
    "# don't have to worry here about DECOY hits messing with Area totals\n",
    "# but we will with Comet results\n",
    "\n",
    "NAAF80 = 857017.763508\n",
    "\n",
    "# use NAAF >80% ALC to get NAAF\n",
    "peaks231_80['NAAF factor'] = (peaks231_80['NAAF num.'])/NAAF80\n",
    "\n",
    "# separate out the dataframe into AAs \n",
    "peaks231_AA80 = peaks231_80[['stripped peptide', 'NAAF factor', 'A', 'C', 'D', 'E', 'F', 'G', 'H', 'K', 'I/L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y']].copy()\n",
    "\n",
    "# multiply the NAAF80 factor by the AA total to normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaks231_AA80['A-NAAF80'] = peaks231_AA80['A'] * peaks231_80['NAAF factor']\n",
    "peaks231_AA80['C-NAAF80'] = peaks231_AA80['C'] * peaks231_80['NAAF factor']\n",
    "peaks231_AA80['D-NAAF80'] = peaks231_AA80['D'] * peaks231_80['NAAF factor']\n",
    "peaks231_AA80['E-NAAF80'] = peaks231_AA80['E'] * peaks231_80['NAAF factor']\n",
    "peaks231_AA80['F-NAAF80'] = peaks231_AA80['F'] * peaks231_80['NAAF factor']\n",
    "peaks231_AA80['G-NAAF80'] = peaks231_AA80['G'] * peaks231_80['NAAF factor']\n",
    "peaks231_AA80['H-NAAF80'] = peaks231_AA80['H'] * peaks231_80['NAAF factor']\n",
    "peaks231_AA80['K-NAAF80'] = peaks231_AA80['K'] * peaks231_80['NAAF factor']\n",
    "peaks231_AA80['I/L-NAAF80'] = peaks231_AA80['I/L'] * peaks231_80['NAAF factor']\n",
    "peaks231_AA80['M-NAAF80'] = peaks231_AA80['M'] * peaks231_80['NAAF factor']\n",
    "peaks231_AA80['N-NAAF80'] = peaks231_AA80['N'] * peaks231_80['NAAF factor']\n",
    "peaks231_AA80['P-NAAF80'] = peaks231_AA80['P'] * peaks231_80['NAAF factor']\n",
    "peaks231_AA80['Q-NAAF80'] = peaks231_AA80['Q'] * peaks231_80['NAAF factor']\n",
    "peaks231_AA80['R-NAAF80'] = peaks231_AA80['R'] * peaks231_80['NAAF factor']\n",
    "peaks231_AA80['S-NAAF80'] = peaks231_AA80['S'] * peaks231_80['NAAF factor']\n",
    "peaks231_AA80['T-NAAF80'] = peaks231_AA80['T'] * peaks231_80['NAAF factor']\n",
    "peaks231_AA80['V-NAAF80'] = peaks231_AA80['V'] * peaks231_80['NAAF factor']\n",
    "peaks231_AA80['W-NAAF80'] = peaks231_AA80['W'] * peaks231_80['NAAF factor']\n",
    "peaks231_AA80['Y-NAAF80'] = peaks231_AA80['Y'] * peaks231_80['NAAF factor']\n",
    "\n",
    "# write the dataframe to a new csv\n",
    "peaks231_AA80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_PTMopt_DN80_AA_NAAF.csv\")\n",
    "\n",
    "peaks231_AA80.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Visualizing the results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(\"ALC max: \", peaks['ALC (%)'].max())\n",
    "print(\"ALC min: \", peaks['ALC (%)'].min())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# take only AA totals and transpose for easier bar plotting in matplotlib\n",
    "\n",
    "peaksaatot = totalpeaks[['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y']].copy().T\n",
    "\n",
    "# take only AA %s and transpose for easier bar plotting in matplotlib\n",
    "\n",
    "peaksreltot = totalpeaks[['% C w/ carb.', '% M w/ oxid', '% N w/ deam', '% Q w/ deam', '% K w/ hydr', '% P w/ hydr', '% K w/ meth', '% R w/ meth']].copy().T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# bar plot of residue totals\n",
    "# there is no isoleucine (I) in Peaks data, which is why L is really big and I is 0\n",
    "\n",
    "\n",
    "x_labels = ['sample total']\n",
    "\n",
    "ax = totalpeaks.plot(y=['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y'], kind=\"bar\", title = '100 m suspended')\n",
    "plt.xticks(rotation=0)\n",
    "ax.get_legend().remove()\n",
    "ax.set_xticklabels(x_labels)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# bar plot of residue totals\n",
    "# there is no isoleucine (I) in Peaks data, which is why L is really big and I is 0\n",
    "\n",
    "my_colors = [(x/10.0, x/20.0, 0.75) for x in range(len(peaksaatot))] # <-- Quick gradient example along the Red/Green dimensions.\n",
    "\n",
    "ax = peaksaatot.plot(y=['sample total'], kind=\"bar\", color = 'green', title = '100 m suspended')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# bar plot of relative modifications\n",
    "\n",
    "ax = totalpeaks.plot(y=['% C w/ carb.', '% M w/ oxid', '% N w/ deam', '% Q w/ deam', '% K w/ hydr', '% P w/ hydr', '% K w/ meth', '% R w/ meth'], kind=\"bar\", title = '100 m suspended')\n",
    "ax.set_xticklabels([])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# bar plot of relative mods\n",
    "\n",
    "\n",
    "ax = peaksreltot.plot(y=['sample total'], kind=\"bar\", title = '100 m suspended')\n",
    "plt.xticks(rotation=45)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# making evenly spaced bins for the ALC data based on the min and max, called above\n",
    "bins = [50, 55, 60, 65, 70, 75, 80, 85, 90, 95, 100]\n",
    "labels = ['50-55', '55-60', '60-65', '65-70', '70-75', '75-80', '80-85', '85-90', '90-95', '95-100']\n",
    "\n",
    "# use pandas cut function to do the binning itself\n",
    "peaks['binned'] = pd.cut(peaks['ALC (%)'], bins=bins, labels=labels)\n",
    "\n",
    "# bar plots of binned PTM data\n",
    "\n",
    "index = ['50-55', '55-60', '60-65', '65-70', '70-75', '75-80', '80-85', '85-90', '90-95', '95-100']\n",
    "data = {'Total PTMs': [peaks.groupby('binned')['ptm-total'].sum()['50-55'], peaks.groupby('binned')['ptm-total'].sum()['55-60'], peaks.groupby('binned')['ptm-total'].sum()['60-65'], peaks.groupby('binned')['ptm-total'].sum()['65-70'], peaks.groupby('binned')['ptm-total'].sum()['70-75'], peaks.groupby('binned')['ptm-total'].sum()['75-80'], peaks.groupby('binned')['ptm-total'].sum()['80-85'], peaks.groupby('binned')['ptm-total'].sum()['85-90'], peaks.groupby('binned')['ptm-total'].sum()['90-95'], peaks.groupby('binned')['ptm-total'].sum()['95-100']],\n",
    "        'Cys carb.': [peaks.groupby('binned')['c-carb'].sum()['50-55'], peaks.groupby('binned')['c-carb'].sum()['55-60'], peaks.groupby('binned')['c-carb'].sum()['60-65'], peaks.groupby('binned')['c-carb'].sum()['65-70'], peaks.groupby('binned')['c-carb'].sum()['70-75'], peaks.groupby('binned')['c-carb'].sum()['75-80'], peaks.groupby('binned')['c-carb'].sum()['80-85'], peaks.groupby('binned')['c-carb'].sum()['85-90'], peaks.groupby('binned')['c-carb'].sum()['90-95'], peaks.groupby('binned')['c-carb'].sum()['95-100']],\n",
    "        'Met oxi.': [peaks.groupby('binned')['m-oxid'].sum()['50-55'], peaks.groupby('binned')['m-oxid'].sum()['55-60'], peaks.groupby('binned')['m-oxid'].sum()['60-65'], peaks.groupby('binned')['m-oxid'].sum()['65-70'], peaks.groupby('binned')['m-oxid'].sum()['70-75'], peaks.groupby('binned')['m-oxid'].sum()['75-80'], peaks.groupby('binned')['m-oxid'].sum()['80-85'], peaks.groupby('binned')['m-oxid'].sum()['85-90'], peaks.groupby('binned')['m-oxid'].sum()['90-95'], peaks.groupby('binned')['m-oxid'].sum()['95-100']],\n",
    "        'Asp deam.': [peaks.groupby('binned')['n-deam'].sum()['50-55'], peaks.groupby('binned')['n-deam'].sum()['55-60'], peaks.groupby('binned')['n-deam'].sum()['60-65'], peaks.groupby('binned')['n-deam'].sum()['65-70'], peaks.groupby('binned')['n-deam'].sum()['70-75'], peaks.groupby('binned')['n-deam'].sum()['75-80'], peaks.groupby('binned')['n-deam'].sum()['80-85'], peaks.groupby('binned')['n-deam'].sum()['85-90'], peaks.groupby('binned')['n-deam'].sum()['90-95'], peaks.groupby('binned')['n-deam'].sum()['95-100']],\n",
    "        'Glut deam.': [peaks.groupby('binned')['q-deam'].sum()['50-55'], peaks.groupby('binned')['q-deam'].sum()['55-60'], peaks.groupby('binned')['q-deam'].sum()['60-65'], peaks.groupby('binned')['q-deam'].sum()['65-70'], peaks.groupby('binned')['q-deam'].sum()['70-75'], peaks.groupby('binned')['q-deam'].sum()['75-80'], peaks.groupby('binned')['q-deam'].sum()['80-85'], peaks.groupby('binned')['q-deam'].sum()['85-90'], peaks.groupby('binned')['q-deam'].sum()['90-95'], peaks.groupby('binned')['q-deam'].sum()['95-100']],\n",
    "        'Lys hydr': [peaks.groupby('binned')['k-hydr'].sum()['50-55'], peaks.groupby('binned')['k-hydr'].sum()['55-60'], peaks.groupby('binned')['k-hydr'].sum()['60-65'], peaks.groupby('binned')['k-hydr'].sum()['65-70'], peaks.groupby('binned')['k-hydr'].sum()['70-75'], peaks.groupby('binned')['k-hydr'].sum()['75-80'], peaks.groupby('binned')['k-hydr'].sum()['80-85'], peaks.groupby('binned')['k-hydr'].sum()['85-90'], peaks.groupby('binned')['k-hydr'].sum()['90-95'], peaks.groupby('binned')['k-hydr'].sum()['95-100']],\n",
    "        'Pro hydr': [peaks.groupby('binned')['p-hydr'].sum()['50-55'], peaks.groupby('binned')['p-hydr'].sum()['55-60'], peaks.groupby('binned')['p-hydr'].sum()['60-65'], peaks.groupby('binned')['p-hydr'].sum()['65-70'], peaks.groupby('binned')['p-hydr'].sum()['70-75'], peaks.groupby('binned')['p-hydr'].sum()['75-80'], peaks.groupby('binned')['p-hydr'].sum()['80-85'], peaks.groupby('binned')['p-hydr'].sum()['85-90'], peaks.groupby('binned')['p-hydr'].sum()['90-95'], peaks.groupby('binned')['p-hydr'].sum()['95-100']],\n",
    "        'Lys meth.': [peaks.groupby('binned')['k-meth'].sum()['50-55'], peaks.groupby('binned')['k-meth'].sum()['55-60'], peaks.groupby('binned')['k-meth'].sum()['60-65'], peaks.groupby('binned')['k-meth'].sum()['65-70'], peaks.groupby('binned')['k-meth'].sum()['70-75'], peaks.groupby('binned')['k-meth'].sum()['75-80'], peaks.groupby('binned')['k-meth'].sum()['80-85'], peaks.groupby('binned')['k-meth'].sum()['85-90'], peaks.groupby('binned')['k-meth'].sum()['90-95'], peaks.groupby('binned')['k-meth'].sum()['95-100']],\n",
    "        'Arg meth.': [peaks.groupby('binned')['r-meth'].sum()['50-55'], peaks.groupby('binned')['r-meth'].sum()['55-60'], peaks.groupby('binned')['r-meth'].sum()['60-65'], peaks.groupby('binned')['r-meth'].sum()['65-70'], peaks.groupby('binned')['r-meth'].sum()['70-75'], peaks.groupby('binned')['r-meth'].sum()['75-80'], peaks.groupby('binned')['r-meth'].sum()['80-85'], peaks.groupby('binned')['r-meth'].sum()['85-90'], peaks.groupby('binned')['r-meth'].sum()['90-95'], peaks.groupby('binned')['r-meth'].sum()['95-100']]\n",
    "        }\n",
    "\n",
    "peaksbin = pd.DataFrame(data, columns=['Total PTMs','Cys carb.','Met oxi.','Asp deam.', 'Glut deam.', 'Lys hydr', 'Pro hydr', 'Lys meth.', 'Arg meth.'], index=index)\n",
    "\n",
    "# write the peaks bin ptm dataframe to a csv:\n",
    "peaksbin.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_ptm.csv\")\n",
    "\n",
    "ax1 = peaksbin.plot.bar(y='Total PTMs', rot=45)\n",
    "ax1.set_title('Total PTMs')\n",
    "\n",
    "ax2 = peaksbin.plot.bar(y='Cys carb.', rot=45)\n",
    "ax2.set_title('Cysteine carbamidomethylation')\n",
    "\n",
    "ax3 = peaksbin.plot.bar(y='Met oxi.', rot=45)\n",
    "ax3.set_title('Methionine oxidation')\n",
    "\n",
    "ax4 = peaksbin.plot.bar(y='Asp deam.', rot=45)\n",
    "ax4.set_title('Asparagine deamidation')\n",
    "\n",
    "ax5 = peaksbin.plot.bar(y='Glut deam.', rot=45)\n",
    "ax5.set_title('Glutamine deamidation')\n",
    "\n",
    "ax6 = peaksbin.plot.bar(y='Lys hydr', rot=45)\n",
    "ax6.set_title('Lysine hydroxylation')\n",
    "\n",
    "ax7 = peaksbin.plot.bar(y='Pro hydr', rot=45)\n",
    "ax7.set_title('Proline hydroxylation')\n",
    "\n",
    "ax8 = peaksbin.plot.bar(y='Lys meth.', rot=45)\n",
    "ax8.set_title('Lysine methylation')\n",
    "\n",
    "ax9 = peaksbin.plot.bar(y='Arg meth.', rot=45)\n",
    "ax9.set_title('Arginine methylation')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exporting txt files of stripped peptides at confidence cutoffs:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# keep only peptide column >50% ALC\n",
    "pep = peaks[[\"stripped peptide\"]]\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "pep.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.txt\", header=False, index=False)\n",
    "\n",
    "# made the text file into a FASTA \n",
    "\n",
    "!awk '{print \">\"NR\"\\n\"$0}' /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.txt > /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas\n",
    "\n",
    "# look\n",
    "\n",
    "print(\"# of DN peptide >50% ALC\", len(pep))\n",
    "pep.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# keep only peptides  >80% ALC\n",
    "peaks80 = peaks.loc[peaks['ALC (%)'] >= 80]\n",
    "\n",
    "# see how many rows and double check\n",
    "# peaks80.head(-10)\n",
    "\n",
    "# keep only peptide column \n",
    "pep80 = peaks80[[\"stripped peptide\"]]\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "pep80.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN80_stripped_peptides.txt\", header=False, index=False)\n",
    "\n",
    "# made the text file into a FASTA \n",
    "\n",
    "!awk '{print \">\"NR\"\\n\"$0}' /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN80_stripped_peptides.txt > /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN80_stripped_peptides.fas\n",
    "\n",
    "print(\"# of DN peptide >80% ALC\", len(pep80))\n",
    "pep80.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Using BioPython to query peptide sequences\n",
    "\n",
    "I installed the BioPython package using `pip install biopython`. All instructions and information [here](https://www.tutorialspoint.com/biopython/index.htm). \n",
    "\n",
    "GitHub project: https://github.com/biopython/biopython\n",
    "\n",
    "I'm relying on the ProtParam module to parse sequences for relative AA composition, instability, secondary structure, instability, and hydrophobicity. You can read more about that module and the studies the indecies are derived from here:\n",
    "\n",
    "https://biopython.org/wiki/ProtParam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# Bio.SeqIO is the standard Sequence Input/Output interface for BioPython 1.43 and later\n",
    "# Bio.SeqIO provides a simple uniform interface to input and output assorted sequence file formats.\n",
    "# (including multiple sequence alignments), but will only deal with sequences as SeqRecord objects\n",
    "\n",
    "# for accepted file formats see https://biopython.org/wiki/SeqIO\n",
    "\n",
    "from Bio import SeqIO\n",
    "#for seq_record in SeqIO.parse(\"/home/millieginty/Documents/git-repos/2017-etnp/data/MED4/MED2_tryp_1raw_db_peptides_nmod.fasta\", \"fasta\"):\n",
    "    #print(seq_record.id)\n",
    "    #print(repr(seq_record.seq))\n",
    "    #print(len(seq_record))\n",
    "    \n",
    "# I commented the print functions out so the output doesn't take up too much space. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# seeing what the ProtParam module can do with a single protein sequence:\n",
    "\n",
    "from Bio.SeqUtils.ProtParam import ProteinAnalysis\n",
    "\n",
    "test_seq = \"MAEGEITTFTALTEKFNLPPGNYKKPKLLYCSNGGHFLRILPDGTVDGTRDRSDQHIQLQLSAESVGEVYIKSTETGQYLAMDTSGLLYGSQTPSEECLFLERLEENHYNTYTSKKHAEKNWFVGLKKNGSCKRGPRTHYGQKAILFLPLPV\"\n",
    "\n",
    "analysed_seq = ProteinAnalysis(test_seq)\n",
    "print(\"molecular weight of seq =\", analysed_seq.molecular_weight())\n",
    "\n",
    "# calculates the aromaticity value of a protein according to Lobry & Gautier (1994, Nucleic Acids Res., 22, 3174-3180). \n",
    "# it's simply the relative frequency of Phe+Trp+Tyr.\n",
    "\n",
    "analysed_seq.aromaticity()\n",
    "print(\"aromaticity of seq =\", analysed_seq.aromaticity())\n",
    "\n",
    "# secondary_structure_fraction:\n",
    "# this methods returns a list of the fraction of amino acids which tend to be in helix, turn or sheet. \n",
    "# AAs in helix: V, I, Y, F, W, L\n",
    "# AAs in turn: N, P, G, S\n",
    "# AAs in sheet: E, M, A, L\n",
    "# the returned list contains 3 values: [Helix, Turn, Sheet]\n",
    "\n",
    "analysed_seq.secondary_structure_fraction()\n",
    "print(\"frac in H T S =\", analysed_seq.secondary_structure_fraction())\n",
    "\n",
    "# the instability index, an implementation of the method of Guruprasad et al. (1990, Protein Engineering, 4, 155-161).\n",
    "# this method tests a protein for stability. \n",
    "# any value above 40 means the protein is unstable (=has a short half life)\n",
    "# NOT SURE WHAT THIS MEANS FOR PEPTIDES, BUT WE COULD DO THIS FOR PROTEINS\n",
    "\n",
    "analysed_seq.instability_index()\n",
    "print(\"instability =\", analysed_seq.instability_index())\n",
    "\n",
    "# count_amino_acids will do just that, and get_amino_acids_percent will return %'s for each AA across the sequence. \n",
    "analysed_seq.get_amino_acids_percent()\n",
    "\n",
    "# taking the returned dictionary and converting to a dataframe\n",
    "\n",
    "aadict = analysed_seq.get_amino_acids_percent()\n",
    "aadf = pd.DataFrame(list(aadict.items()),columns = ['residue','% occurance']) \n",
    "\n",
    "aadf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# use SeqIO and a loop to apply count_amino_acids to each sequence in the file\n",
    "# aatot will give us the total number of each residue in the entire sample output\n",
    "\n",
    "import collections\n",
    "from Bio import SeqIO\n",
    "from Bio.SeqUtils.ProtParam import ProteinAnalysis\n",
    "\n",
    "all_aas = collections.defaultdict(int)\n",
    "for record in SeqIO.parse(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas\", \"fasta\"):\n",
    "    x = ProteinAnalysis(str(record.seq))\n",
    "    #print(record.id, x.count_amino_acids())\n",
    "    for aa, count in x.count_amino_acids().items():\n",
    "        all_aas[aa] += count        \n",
    "        \n",
    "# made a dataframe for amino acid total counts        \n",
    "data = (all_aas)\n",
    "aatot = pd.DataFrame(data, index = ['sample sequence total'])\n",
    "aatot.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pandas import Series, DataFrame\n",
    "\n",
    "with open('/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas') as fasta_file:  # Will close handle cleanly\n",
    "    identifiers = []\n",
    "    lengths = []\n",
    "    for seq_record in SeqIO.parse(fasta_file, 'fasta'):  # (generator)\n",
    "        identifiers.append(seq_record.id)\n",
    "        lengths.append(len(seq_record.seq))\n",
    "        \n",
    "        \n",
    "#converting lists to pandas Series    \n",
    "s1 = Series(identifiers, name='ID')\n",
    "s2 = Series(lengths, name='length')\n",
    "\n",
    "#Gathering Series into a pandas DataFrame and rename index as ID column\n",
    "idseq = DataFrame(dict(ID=s1, length=s2)).set_index(['ID'])\n",
    "\n",
    "idseq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pandas import Series, DataFrame\n",
    "\n",
    "\n",
    "\n",
    "with open('/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas) as fasta_file:  # Will close handle cleanly\n",
    "    identifiers = []\n",
    "    lengths = []\n",
    "    aa = []\n",
    "    for seq_record in SeqIO.parse(fasta_file, 'fasta'):  # (generator)\n",
    "        identifiers.append(seq_record.id)\n",
    "        lengths.append(len(seq_record.seq))\n",
    "        aa.count_amino_acids(seq_record.seq)\n",
    "        \n",
    "        \n",
    "#converting lists to pandas Series    \n",
    "s1 = Series(identifiers, name='ID')\n",
    "s2 = Series(lengths, name='length')\n",
    "s3 = Series(aa, name='AAs')\n",
    "\n",
    "#Gathering Series into a pandas DataFrame and rename index as ID column\n",
    "idseq = DataFrame(dict(ID=s1, length=s2, AAs=s3)).set_index(['ID'])\n",
    "\n",
    "idseq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio.SeqUtils import ProtParam\n",
    "\n",
    "handle = open(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas\") \n",
    "for record in SeqIO.parse(handle, \"fasta\"): \n",
    "    seq = str(record.seq)\n",
    "    X = ProtParam.ProteinAnalysis(seq)\n",
    "    print(X.count_amino_acids()) \n",
    "    #print X.get_amino_acids_percent() \n",
    "    #print X.molecular_weight() \n",
    "    #print X.aromaticity() \n",
    "    #print X.instability_index() \n",
    "    #print X.flexibility() \n",
    "    #print X.isoelectric_point() \n",
    "    #print X.secondary_structure_fraction()\n",
    "    \n",
    "# made a data series from the count_amino_acids function\n",
    "# aacount = {X.count_amino_acids()}\n",
    "\n",
    "# made a pandas dataframe from the series generated above\n",
    "# aacount = pd.DataFrame(list(data.items()),columns = ['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y']) \n",
    "\n",
    "aacount = pd.DataFrame(X.count_amino_acids(), index=[0])\n",
    "\n",
    "# look at new dataframe\n",
    "\n",
    "# aacount.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
