{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Tue Jun 26 15:37:26 PDT 2018\r\n"
     ]
    }
   ],
   "source": [
    "!date"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Tuesday, June 26, 2018 ###\n",
    "\n",
    "This notebook is to document my use of the MetaGOmics web application (http://www.yeastrc.org/metagomics/home.do) to perform a peptide-based functional and taxonomic analysis of my metaproteomic data, as described in [Riffle et al., 2018](https://www.ncbi.nlm.nih.gov/pubmed/29280960). The MetaGOmics algorithim is pretty close to Unipept in its approach to a peptide-based least common ancestor evaluation (LCA) that is tied to the UniProtKB database, but it also does a similar evaluation of protein functionality using GO terms from the protein annotations, and it also uses spetral counts to approach some quanitiation of these analyses.\n",
    "\n",
    "The input needed for MetaGOmics:\n",
    "\n",
    "- the queried sequence database in FASTA format. It's noted in the Riffle paper that one only needs a FASTA file containing the proteins matched by any of the uploaded peptides - hence there's some trimming that will make the analysis much quicker. To achieve this I'm uploading the exported FASTA from PEAKS 8.5 Spider searches. This was simple for PEAKS output, but apparently there's some tips for this trimming in the MetaGOmics's GitHub repo\n",
    "\n",
    "- a text file containing the identified peptides and their corresponding spectral counts. Though, it would work just as well to use spectral areas over spectral counts, it's a relative comparison (and would compare a peptide's spectral area to the total, just as the number of spectral counts attributed to a peptide would be compared to the total number of spectral counts for all peptides).\n",
    "\n",
    "I went to MetaGOmics and uploaded ja2-proteins.fasta in the web application and named it _2017 P2 100 m suspended (JA2)_ I chose to use the SwissProt (Unisprot) database for the GO annotations database and I selected a for the BLAST filters: 1E-10, using the top hit only. I recieved an email several seconds later with a link to a page where I could access the database I'd set up and load peptide lists. It looks like this: ![metagomics database page screengrab]( https://raw.githubusercontent.com/MeganEDuffy/2017-etnp/master/images/06.26.18-metgomics-ss.png)\n",
    "\n",
    "Then I need to actually upload the peptide sequences and their corresponding spectral counts (or areas, I chose counts to start). I took the Spider output xlxs file I'd made from the PEAKS exported .csv, and I just made a new Excel file containing the sequences and the # spectra columns, with no headers. It looked like this:\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IVVGGPYSSVSDAASVLDGSQK\t7\r",
      "\r\n",
      "IVVGGPYSSVSDAASSLDSSQK\t6\r",
      "\r\n",
      "AIQQQIENPLAQQILSGELVPGK\t6\r",
      "\r\n",
      "LPQVEGTGGDVQPSQDLVR\t13\r",
      "\r\n",
      "QAVSADSSGSFIGGAELASLK\t7\r",
      "\r\n",
      "LGEHNIDVLEGNEQFINAAK\t11\r",
      "\r\n",
      "YLGSTGGLLNSAETEEK\t6\r",
      "\r\n",
      "YIGSTGGLLNSAETEEK\t6\r",
      "\r\n",
      "AISADSSGGFIGGAELSQLK\t3\r",
      "\r\n",
      "AGAGDDEVNAGSGDDIVR\t3\r",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!head /Users/meganduffy/Documents/git-repos/2017-etnp/data/ja1-ja6/MED_ETNP_JA2_uwpr201704_SPIDER_19/ja2-metagomics-input.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note that I needed to have it saved as tab delimted (.txt). When I tried saving it as a .csv, I didn't get any meaninggul output (no error messages, just no matched GO terms for any peptides). Then I got an email saying I could view my data. You can see that the two (one failed .csv, one suscessful .txt uploads look like in the screengrab above). When I click on the GO annotation download button for the sucessful one (100 m suspended), I get this: ![metagomics results download window](https://raw.githubusercontent.com/MeganEDuffy/2017-etnp/master/images/06.27.18-metagomics-results-ss.png)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I don't understand why I don't have GO images. Downloading the GO report, I get this below. Plenty of content! I made a new `metagomics` directory in `analyses` and renamed the report with a more descriptive title. See here:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# MetaGOmics GO report\r\n",
      "# MetaGOmics version: 0.1.1\r\n",
      "# Run date: Tue Jun 26 21:21:21 PDT 2018\r\n",
      "GO acc\tGO aspect\tGO name\tcount\ttotal count\tratio\r\n",
      "unknownprc\tbiological_process\tunknown biological process\t1960\t2961\t0.6619385342789598\r\n",
      "GO:0038061\tbiological_process\tNIK/NF-kappaB signaling\t2\t2961\t6.754474839581223E-4\r\n",
      "unknownfun\tmolecular_function\tunknown molecular function\t2225\t2961\t0.751435325903411\r\n",
      "GO:0009733\tbiological_process\tresponse to auxin\t4\t2961\t0.0013508949679162446\r\n",
      "GO:0032392\tbiological_process\tDNA geometric change\t2\t2961\t6.754474839581223E-4\r\n",
      "unknowncmp\tcellular_component\tunknown cellular component\t1982\t2961\t0.6693684566024991\r\n",
      "GO:0009735\tbiological_process\tresponse to cytokinin\t102\t2961\t0.034447821681864235\r\n",
      "GO:0009737\tbiological_process\tresponse to abscisic acid\t6\t2961\t0.002026342451874367\r\n",
      "GO:0032403\tmolecular_function\tprotein complex binding\t16\t2961\t0.005403579871664978\r\n",
      "GO:0009741\tbiological_process\tresponse to brassinosteroid\t10\t2961\t0.003377237419790611\r\n",
      "GO:0051591\tbiological_process\tresponse to cAMP\t4\t2961\t0.0013508949679162446\r\n",
      "GO:0051592\tbiological_process\tresponse to calcium ion\t4\t2961\t0.0013508949679162446\r\n",
      "GO:0098771\tbiological_process\tinorganic ion homeostasis\t2\t2961\t6.754474839581223E-4\r\n",
      "GO:0098772\tmolecular_function\tmolecular function regulator\t12\t2961\t0.004052684903748734\r\n",
      "GO:0009751\tbiological_process\tresponse to salicylic acid\t2\t2961\t6.754474839581223E-4\r\n",
      "GO:0098776\tbiological_process\tprotein transport across the cell outer membrane\t2\t2961\t6.754474839581223E-4\r\n",
      "GO:0051602\tbiological_process\tresponse to electrical stimulus\t4\t2961\t0.0013508949679162446\r\n",
      "GO:0051604\tbiological_process\tprotein maturation\t100\t2961\t0.033772374197906116\r\n",
      "GO:0051606\tbiological_process\tdetection of stimulus\t27\t2961\t0.00911854103343465\r\n",
      "GO:0000001\tbiological_process\tmitochondrion inheritance\t4\t2961\t0.0013508949679162446\r\n",
      "GO:0000002\tbiological_process\tmitochondrial genome maintenance\t97\t2961\t0.03275920297196893\r\n",
      "GO:0000003\tbiological_process\treproduction\t10\t2961\t0.003377237419790611\r\n",
      "GO:0038093\tbiological_process\tFc receptor signaling pathway\t6\t2961\t0.002026342451874367\r\n",
      "GO:0038094\tbiological_process\tFc-gamma receptor signaling pathway\t4\t2961\t0.0013508949679162446\r\n",
      "GO:0038095\tbiological_process\tFc-epsilon receptor signaling pathway\t2\t2961\t6.754474839581223E-4\r\n",
      "GO:0042254\tbiological_process\tribosome biogenesis\t5\t2961\t0.0016886187098953055\r\n",
      "GO:0038096\tbiological_process\tFc-gamma receptor signaling pathway involved in phagocytosis\t4\t2961\t0.0013508949679162446\r\n",
      "GO:0044183\tmolecular_function\tprotein binding involved in protein folding\t150\t2961\t0.05065856129685917\r\n",
      "GO:0051082\tmolecular_function\tunfolded protein binding\t159\t2961\t0.05369807497467072\r\n",
      "GO:0032432\tcellular_component\tactin filament bundle\t4\t2961\t0.0013508949679162446\r\n",
      "GO:0009772\tbiological_process\tphotosynthetic electron transport in photosystem II\t20\t2961\t0.006754474839581222\r\n",
      "GO:0098796\tcellular_component\tmembrane protein complex\t458\t2961\t0.15467747382641\r\n",
      "GO:0032434\tbiological_process\tregulation of proteasomal ubiquitin-dependent protein catabolic process\t15\t2961\t0.005065856129685917\r\n",
      "GO:0009773\tbiological_process\tphotosynthetic electron transport in photosystem I\t7\t2961\t0.002364066193853428\r\n",
      "GO:0098797\tcellular_component\tplasma membrane protein complex\t41\t2961\t0.013846673421141506\r\n",
      "GO:0032435\tbiological_process\tnegative regulation of proteasomal ubiquitin-dependent protein catabolic process\t15\t2961\t0.005065856129685917\r\n",
      "GO:0015036\tmolecular_function\tdisulfide oxidoreductase activity\t5\t2961\t0.0016886187098953055\r\n",
      "GO:0000011\tbiological_process\tvacuole inheritance\t4\t2961\t0.0013508949679162446\r\n",
      "GO:0098805\tcellular_component\twhole membrane\t17\t2961\t0.005741303613644039\r\n",
      "GO:0038111\tbiological_process\tinterleukin-7-mediated signaling pathway\t6\t2961\t0.002026342451874367\r\n",
      "GO:0098807\tcellular_component\tchloroplast thylakoid membrane protein complex\t7\t2961\t0.002364066193853428\r\n",
      "GO:0046128\tbiological_process\tpurine ribonucleoside metabolic process\t27\t2961\t0.00911854103343465\r\n",
      "GO:0032446\tbiological_process\tprotein modification by small protein conjugation\t2\t2961\t6.754474839581223E-4\r\n",
      "GO:0046129\tbiological_process\tpurine ribonucleoside biosynthetic process\t18\t2961\t0.0060790273556231\r\n",
      "GO:0009792\tbiological_process\tembryo development ending in birth or egg hatching\t19\t2961\t0.006416751097602162\r\n",
      "GO:0098813\tbiological_process\tnuclear chromosome segregation\t10\t2961\t0.003377237419790611\r\n"
     ]
    }
   ],
   "source": [
    "!head -50 /Users/meganduffy/Documents/git-repos/2017-etnp/analyses/metagomics/susvsink/ja2/go_report_ja2_06.26.18.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The taxonomy report looks like this:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# MetaGOmics taxonomy report\r\n",
      "# MetaGOmics version: 0.1.1\r\n",
      "# Run date: Tue Jun 26 21:24:25 PDT 2018\r\n",
      "GO acc\tGO aspect\tGO name\ttaxon name\tNCBI taxonomy id\ttaxonomy rank\tPSM count\tratio of GO\tratio of run\r\n",
      "unknownprc\tbiological_process\tunknown biological process\troot\t1\tno rank\t51\t0.026020408163265306\t0.017223910840932118\r\n",
      "unknownprc\tbiological_process\tunknown biological process\tBacteria\t2\tsuperkingdom\t32\t0.0163265306122449\t0.010807159743329957\r\n",
      "unknownprc\tbiological_process\tunknown biological process\tHominidae\t9604\tfamily\t2\t0.0010204081632653062\t6.754474839581223E-4\r\n",
      "unknownprc\tbiological_process\tunknown biological process\tCyanothecaceae\t1892249\tfamily\t13\t0.0066326530612244895\t0.004390408645727794\r\n",
      "unknownprc\tbiological_process\tunknown biological process\tRodentia\t9989\torder\t2\t0.0010204081632653062\t6.754474839581223E-4\r\n",
      "unknownprc\tbiological_process\tunknown biological process\tHomo\t9605\tgenus\t2\t0.0010204081632653062\t6.754474839581223E-4\r\n"
     ]
    }
   ],
   "source": [
    "!head /Users/meganduffy/Documents/git-repos/2017-etnp/analyses/metagomics/susvsink/ja2/taxonomy_report_ja2_06.26.18.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There were some modifications in parentheses in my peptide input, I need to fix that step of this process to continue. And I need to figure out why I'm not getting any images from the MetaGOmics app."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IVVGGPYSSVSDAASVLDGSQK\t7\r",
      "\r\n",
      "IVVGGPYSSVSDAASSLDSSQK\t6\r",
      "\r\n",
      "AIQQQIENPLAQQILSGELVPGK\t6\r",
      "\r\n",
      "LPQVEGTGGDVQPSQDLVR\t13\r",
      "\r\n",
      "QAVSADSSGSFIGGAELASLK\t7\r",
      "\r\n",
      "LGEHNIDVLEGNEQFINAAK\t11\r",
      "\r\n",
      "YLGSTGGLLNSAETEEK\t6\r",
      "\r\n",
      "YIGSTGGLLNSAETEEK\t6\r",
      "\r\n",
      "AISADSSGGFIGGAELSQLK\t3\r",
      "\r\n",
      "AGAGDDEVNAGSGDDIVR\t3\r",
      "\r\n",
      "IVVGGPYSSVSDASSGLDGSQK\t3\r",
      "\r\n",
      "YKDNFTVTAPEIALNEGFDR\t3\r",
      "\r\n",
      "SC(+57.02)AAAGTEC(+57.02)LISGWGNTK(+28.03)\t10\r",
      "\r\n",
      "TNYFGIQGTDNGNLTNSFAESELER\t3\r",
      "\r\n",
      "TNYFGLQGTDNGNLTNSFAESELER\t3\r",
      "\r\n",
      "AIQQQIE(+21.98)NPLAQQILSGELVPGK\t1\r",
      "\r\n",
      "LGSDSGM(+15.99)LAFEPSNIK\t2\r",
      "\r\n",
      "LPQVEGTGGDVQPSQ(+.98)DLVR\t1\r",
      "\r\n",
      "VIGQNEAVDAVSNAIR\t19\r",
      "\r\n",
      "ALTTGVDYAQGLVALGGDDK\t3\r",
      "\r\n",
      "ILSTINDADLNASAAEIK\t2\r",
      "\r\n",
      "GGQPLFFGEGTYANLSQTAR\t3\r",
      "\r\n",
      "TNYFGIQGTDNGNLTNSFAESELERA\t6\r",
      "\r\n",
      "TNYFGLQGTDNGNLTNSFAESELERA\t6\r",
      "\r\n",
      "TGEIGDGKIFISPVDSVVR\t2\r",
      "\r\n",
      "DNFTVTAPEIALNEGFDR\t2\r",
      "\r\n",
      "TNQNVGLDPETLALATPAR\t3\r",
      "\r\n",
      "YNSGEGGC(+57.02)FYSVDTIEAPWNSGR\t2\r",
      "\r\n",
      "SGGGGGGGLGSGGSIR\t1\r",
      "\r\n",
      "AGAGDDEVNAGS(-15.99)GDDIVR\t1\r",
      "\r\n",
      "LPQVEGTGGD(+21.98)VQPSQDLVR\t2\r",
      "\r\n",
      "AISSGAGNSDFVVGPER\t3\r",
      "\r\n",
      "LGSDAGM(+15.99)LAFEPSNIK\t3\r",
      "\r\n",
      "L(+28.03)GEHNIDVLEGNEQFINAAK\t1\r",
      "\r\n",
      "GGSGNDTINGGGGDDLIR\t2\r",
      "\r\n",
      "IVVGGPYSSVSDAASSLDSSQKSVFDK\t2\r",
      "\r\n",
      "LVPGFEAPVNLVYSQGNR\t4\r",
      "\r\n",
      "LVPGFEAPVNLVYSK\t3\r",
      "\r\n",
      "AYGGETVTAGSILIR\t2\r",
      "\r\n",
      "TGGDPIELFETAVK\t2\r",
      "\r\n",
      "TGGDPLELFETAVK\t2\r",
      "\r\n",
      "IAENAGANGAVVAENVK\t2\r",
      "\r\n",
      "VAQEAVDAVYAVHPNGAVGR\t2\r",
      "\r\n",
      "IVVGGPYSSVSDAASVLD(+14.02)GSQK\t1\r",
      "\r\n",
      "M(+15.99)QTYGNPNPTYGWWAGNAGTVNR(+14.02)\t2\r",
      "\r\n",
      "ALTTGVEYAQGLVALGGDDK\t3\r",
      "\r\n",
      "TQFYNDEPEAIEYGENFIVHR\t1\r",
      "\r\n",
      "LGSDEYVLLSEK\t2\r",
      "\r\n",
      "DNFTVTASEIALNEGFDR\t3\r",
      "\r\n",
      "LGEAGEKIDLNNSSVR\t6\r",
      "\r\n"
     ]
    }
   ],
   "source": [
    "#What sequence modifications are there?\n",
    "\n",
    "!head -50 /Users/meganduffy/Documents/git-repos/2017-etnp/data/ja1-ja6/MED_ETNP_JA2_uwpr201704_SPIDER_19/ja2-metagomics-input.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# remove '(+15.99)' and '(+57.02)', (+.98) from sequences\n",
    "!sed 's/(+15.99)//g' /Users/meganduffy/Documents/git-repos/2017-etnp/data/ja1-ja6/MED_ETNP_JA2_uwpr201704_SPIDER_19/ja2-metagomics-input.txt \\\n",
    "| sed 's/(+57.02)//g'\\\n",
    "| sed 's/(+.98)//g'\\\n",
    "> /Users/meganduffy/Documents/git-repos/2017-etnp/data/ja1-ja6/MED_ETNP_JA2_uwpr201704_SPIDER_19/ja2-metagomics--nomod-input.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IVVGGPYSSVSDAASVLDGSQK\t7\r",
      "\r\n",
      "IVVGGPYSSVSDAASSLDSSQK\t6\r",
      "\r\n",
      "AIQQQIENPLAQQILSGELVPGK\t6\r",
      "\r\n",
      "LPQVEGTGGDVQPSQDLVR\t13\r",
      "\r\n",
      "QAVSADSSGSFIGGAELASLK\t7\r",
      "\r\n",
      "LGEHNIDVLEGNEQFINAAK\t11\r",
      "\r\n",
      "YLGSTGGLLNSAETEEK\t6\r",
      "\r\n",
      "YIGSTGGLLNSAETEEK\t6\r",
      "\r\n",
      "AISADSSGGFIGGAELSQLK\t3\r",
      "\r\n",
      "AGAGDDEVNAGSGDDIVR\t3\r",
      "\r\n",
      "IVVGGPYSSVSDASSGLDGSQK\t3\r",
      "\r\n",
      "YKDNFTVTAPEIALNEGFDR\t3\r",
      "\r\n",
      "SCAAAGTECLISGWGNTK(+28.03)\t10\r",
      "\r\n",
      "TNYFGIQGTDNGNLTNSFAESELER\t3\r",
      "\r\n",
      "TNYFGLQGTDNGNLTNSFAESELER\t3\r",
      "\r\n",
      "AIQQQIE(+21.98)NPLAQQILSGELVPGK\t1\r",
      "\r\n",
      "LGSDSGMLAFEPSNIK\t2\r",
      "\r\n",
      "LPQVEGTGGDVQPSQDLVR\t1\r",
      "\r\n",
      "VIGQNEAVDAVSNAIR\t19\r",
      "\r\n",
      "ALTTGVDYAQGLVALGGDDK\t3\r",
      "\r\n",
      "ILSTINDADLNASAAEIK\t2\r",
      "\r\n",
      "GGQPLFFGEGTYANLSQTAR\t3\r",
      "\r\n",
      "TNYFGIQGTDNGNLTNSFAESELERA\t6\r",
      "\r\n",
      "TNYFGLQGTDNGNLTNSFAESELERA\t6\r",
      "\r\n",
      "TGEIGDGKIFISPVDSVVR\t2\r",
      "\r\n",
      "DNFTVTAPEIALNEGFDR\t2\r",
      "\r\n",
      "TNQNVGLDPETLALATPAR\t3\r",
      "\r\n",
      "YNSGEGGCFYSVDTIEAPWNSGR\t2\r",
      "\r\n",
      "SGGGGGGGLGSGGSIR\t1\r",
      "\r\n",
      "AGAGDDEVNAGS(-15.99)GDDIVR\t1\r",
      "\r\n",
      "LPQVEGTGGD(+21.98)VQPSQDLVR\t2\r",
      "\r\n",
      "AISSGAGNSDFVVGPER\t3\r",
      "\r\n",
      "LGSDAGMLAFEPSNIK\t3\r",
      "\r\n",
      "L(+28.03)GEHNIDVLEGNEQFINAAK\t1\r",
      "\r\n",
      "GGSGNDTINGGGGDDLIR\t2\r",
      "\r\n",
      "IVVGGPYSSVSDAASSLDSSQKSVFDK\t2\r",
      "\r\n",
      "LVPGFEAPVNLVYSQGNR\t4\r",
      "\r\n",
      "LVPGFEAPVNLVYSK\t3\r",
      "\r\n",
      "AYGGETVTAGSILIR\t2\r",
      "\r\n",
      "TGGDPIELFETAVK\t2\r",
      "\r\n",
      "TGGDPLELFETAVK\t2\r",
      "\r\n",
      "IAENAGANGAVVAENVK\t2\r",
      "\r\n",
      "VAQEAVDAVYAVHPNGAVGR\t2\r",
      "\r\n",
      "IVVGGPYSSVSDAASVLD(+14.02)GSQK\t1\r",
      "\r\n",
      "MQTYGNPNPTYGWWAGNAGTVNR(+14.02)\t2\r",
      "\r\n",
      "ALTTGVEYAQGLVALGGDDK\t3\r",
      "\r\n",
      "TQFYNDEPEAIEYGENFIVHR\t1\r",
      "\r\n",
      "LGSDEYVLLSEK\t2\r",
      "\r\n",
      "DNFTVTASEIALNEGFDR\t3\r",
      "\r\n",
      "LGEAGEKIDLNNSSVR\t6\r",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!head -50 /Users/meganduffy/Documents/git-repos/2017-etnp/data/ja1-ja6/MED_ETNP_JA2_uwpr201704_SPIDER_19/ja2-metagomics--nomod-input.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#I have more to get. Looking at some old notebooks where I did this. Maybe I'll try printing out all the mods\n",
    "!awk -F \"[()]\" '{ for (i=2; i<NF; i+=2) print $i }' /Users/meganduffy/Documents/git-repos/2017-etnp/data/ja1-ja6/MED_ETNP_JA2_uwpr201704_SPIDER_19/ja2-metagomics-input.txt > /Users/meganduffy/Documents/git-repos/2017-etnp/data/ja1-ja6/MED_ETNP_JA2_uwpr201704_SPIDER_19/ja2-propeptides-mods.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+57.02\r\n",
      "+57.02\r\n",
      "+28.03\r\n",
      "+21.98\r\n",
      "+15.99\r\n",
      "+.98\r\n",
      "+57.02\r\n",
      "-15.99\r\n",
      "+21.98\r\n",
      "+15.99\r\n",
      "+28.03\r\n",
      "+14.02\r\n",
      "+15.99\r\n",
      "+14.02\r\n",
      "+14.02\r\n",
      "+57.02\r\n",
      "+57.02\r\n",
      "+15.99\r\n",
      "+57.02\r\n",
      "+21.98\r\n",
      "+57.02\r\n",
      "+57.02\r\n",
      "+57.02\r\n",
      "-17.03\r\n",
      "+21.98\r\n",
      "+15.99\r\n",
      "+15.99\r\n",
      "+15.99\r\n",
      "+21.98\r\n",
      "+15.99\r\n",
      "+15.99\r\n",
      "+37.95\r\n",
      "-15.99\r\n",
      "+21.98\r\n",
      "-17.03\r\n",
      "-17.03\r\n",
      "+57.02\r\n",
      "+57.02\r\n",
      "+57.02\r\n",
      "+57.02\r\n",
      "+15.99\r\n",
      "+37.95\r\n",
      "+57.02\r\n",
      "+57.02\r\n",
      "+.98\r\n",
      "+28.03\r\n",
      "+15.99\r\n",
      "+57.02\r\n",
      "+21.98\r\n",
      "+15.99\r\n"
     ]
    }
   ],
   "source": [
    "!head -50 /Users/meganduffy/Documents/git-repos/2017-etnp/data/ja1-ja6/MED_ETNP_JA2_uwpr201704_SPIDER_19/ja2-propeptides-mods.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# remove '(+15.99)' and '(+57.02)', (+.98), all other mods from sequences\n",
    "# I went through and manually found all the mods to remove them, checking that none were left\n",
    "!sed 's/(+15.99)//g' /Users/meganduffy/Documents/git-repos/2017-etnp/data/ja1-ja6/MED_ETNP_JA2_uwpr201704_SPIDER_19/ja2-metagomics-input.txt \\\n",
    "| sed 's/(+57.02)//g'\\\n",
    "| sed 's/(+.98)//g'\\\n",
    "| sed 's/(+21.98)//g'\\\n",
    "| sed 's/(+28.03)//g'\\\n",
    "| sed 's/(+37.95)//g'\\\n",
    "| sed 's/(-17.03)//g'\\\n",
    "| sed 's/(+14.02)//g'\\\n",
    "| sed 's/(-15.99)//g'\\\n",
    "| sed 's/(+42.01)//g'\\\n",
    "| sed 's/(sub Q)//g'\\\n",
    "| sed 's/(+71.04)//g'\\\n",
    "| sed 's/(+55.92)//g'\\\n",
    "| sed 's/(sub A)//g'\\\n",
    "| sed 's/(+30.01)//g'\\\n",
    "| sed 's/(+3.99)//g'\\\n",
    "| sed 's/(+58.01)//g'\\\n",
    "| sed 's/(+43.01)//g'\\\n",
    "| sed 's/(+72.02)//g'\\\n",
    "| sed 's/(-18.01)//g'\\\n",
    "| sed 's/(sub D)//g'\\\n",
    "| sed 's/(sub G)//g'\\\n",
    "| sed 's/(+27.99)//g'\\\n",
    "| sed 's/(sub K)//g'\\\n",
    "| sed 's/(+37.96)//g'\\\n",
    "| sed 's/(+43.99)//g'\\\n",
    "| sed 's/(+21.97)//g'\\\n",
    "| sed 's/(+31.99)//g'\\\n",
    "| sed 's/(sub I)//g'\\\n",
    "| sed 's/(sub S)//g'\\\n",
    "| sed 's/(sub N)//g'\\\n",
    "| sed 's/(+15.01)//g'\\\n",
    "| sed 's/(-.98)//g'\\\n",
    "| sed 's/(+114.04)//g'\\\n",
    "| sed 's/(sub Y)//g'\\\n",
    "| sed 's/(sub T)//g'\\\n",
    "| sed 's/(del Q)//g'\\\n",
    "| sed 's/(-48.00)//g'\\\n",
    "| sed 's/(+100.02)//g'\\\n",
    "| sed 's/(+17.03)//g'\\\n",
    "| sed 's/(+79.97)//g'\\\n",
    "| sed 's/(+17.03)//g'\\\n",
    "| sed 's/(sub L)//g'\\\n",
    "| sed 's/(+53.92)//g'\\\n",
    "| sed 's/(sub V)//g'\\\n",
    "| sed 's/(sub M)//g'\\\n",
    "| sed 's/(+15.00)//g'\\\n",
    "| sed 's/(+150.04)//g'\\\n",
    "| sed 's/(+60.00)//g'\\\n",
    "| sed 's/(+119.04)//g'\\\n",
    "| sed 's/(+41.03)//g'\\\n",
    "| sed 's/(+29.02)//g'\\\n",
    "| sed 's/(+79.96)//g'\\\n",
    "| sed 's/(+282.05)//g'\\\n",
    "| sed 's/(del N)//g'\\\n",
    "| sed 's/(+228.11)//g'\\\n",
    "| sed 's/(sub E)//g'\\\n",
    "| sed 's/(+27.05)//g'\\\n",
    "| sed 's/(+87.03)//g'\\\n",
    "| sed 's/(-15.01)//g'\\\n",
    "> /Users/meganduffy/Documents/git-repos/2017-etnp/data/ja1-ja6/MED_ETNP_JA2_uwpr201704_SPIDER_19/ja2-metagomics-nomod-input.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "IVVGGPYSSVSDAASVLDGSQK\t7\r",
      "\r\n",
      "IVVGGPYSSVSDAASSLDSSQK\t6\r",
      "\r\n",
      "AIQQQIENPLAQQILSGELVPGK\t6\r",
      "\r\n",
      "LPQVEGTGGDVQPSQDLVR\t13\r",
      "\r\n",
      "QAVSADSSGSFIGGAELASLK\t7\r",
      "\r\n",
      "LGEHNIDVLEGNEQFINAAK\t11\r",
      "\r\n",
      "YLGSTGGLLNSAETEEK\t6\r",
      "\r\n",
      "YIGSTGGLLNSAETEEK\t6\r",
      "\r\n",
      "AISADSSGGFIGGAELSQLK\t3\r",
      "\r\n",
      "AGAGDDEVNAGSGDDIVR\t3\r",
      "\r\n",
      "IVVGGPYSSVSDASSGLDGSQK\t3\r",
      "\r\n",
      "YKDNFTVTAPEIALNEGFDR\t3\r",
      "\r\n",
      "SCAAAGTECLISGWGNTK\t10\r",
      "\r\n",
      "TNYFGIQGTDNGNLTNSFAESELER\t3\r",
      "\r\n",
      "TNYFGLQGTDNGNLTNSFAESELER\t3\r",
      "\r\n",
      "AIQQQIENPLAQQILSGELVPGK\t1\r",
      "\r\n",
      "LGSDSGMLAFEPSNIK\t2\r",
      "\r\n",
      "LPQVEGTGGDVQPSQDLVR\t1\r",
      "\r\n",
      "VIGQNEAVDAVSNAIR\t19\r",
      "\r\n",
      "ALTTGVDYAQGLVALGGDDK\t3\r",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!head -20 /Users/meganduffy/Documents/git-repos/2017-etnp/data/ja1-ja6/MED_ETNP_JA2_uwpr201704_SPIDER_19/ja2-metagomics-nomod-input.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now I'm using the input file without modifications to search in MetaGOmics. It completed in a matter of minutes. Again, no downloadable images - so I guess this means there were no significant terms of that go aspect? I saved the files with today's date, 06.28.18. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "go_report_ja2_06.26.18.txt             taxonomy_report_ja2_06.26.18.txt\r\n",
      "go_report_ja2nomods_06.28.18.txt       taxonomy_report_ja2nomods_06.28.18.txt\r\n"
     ]
    }
   ],
   "source": [
    "!ls /Users/meganduffy/Documents/git-repos/2017-etnp/analyses/metagomics/susvsink/ja2/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# MetaGOmics GO report\r\n",
      "# MetaGOmics version: 0.1.1\r\n",
      "# Run date: Thu Jun 28 14:58:29 PDT 2018\r\n",
      "GO acc\tGO aspect\tGO name\tcount\ttotal count\tratio\r\n",
      "unknownprc\tbiological_process\tunknown biological process\t1518\t2509\t0.6050219210840972\r\n",
      "GO:0038061\tbiological_process\tNIK/NF-kappaB signaling\t2\t2509\t7.971303308090873E-4\r\n",
      "unknownfun\tmolecular_function\tunknown molecular function\t1752\t2509\t0.6982861697887605\r\n",
      "GO:0009733\tbiological_process\tresponse to auxin\t4\t2509\t0.0015942606616181746\r\n",
      "GO:0032392\tbiological_process\tDNA geometric change\t2\t2509\t7.971303308090873E-4\r\n",
      "unknowncmp\tcellular_component\tunknown cellular component\t1545\t2509\t0.6157831805500199\r\n"
     ]
    }
   ],
   "source": [
    "!head /Users/meganduffy/Documents/git-repos/2017-etnp/analyses/metagomics/susvsink/ja2/go_report_ja2nomods_06.28.18.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# MetaGOmics taxonomy report\r\n",
      "# MetaGOmics version: 0.1.1\r\n",
      "# Run date: Thu Jun 28 15:01:39 PDT 2018\r\n",
      "GO acc\tGO aspect\tGO name\ttaxon name\tNCBI taxonomy id\ttaxonomy rank\tPSM count\tratio of GO\tratio of run\r\n",
      "unknownprc\tbiological_process\tunknown biological process\troot\t1\tno rank\t68\t0.04479578392621871\t0.027102431247508968\r\n",
      "unknownprc\tbiological_process\tunknown biological process\tBacteria\t2\tsuperkingdom\t49\t0.032279314888010543\t0.019529693104822637\r\n",
      "unknownprc\tbiological_process\tunknown biological process\tHominidae\t9604\tfamily\t2\t0.0013175230566534915\t7.971303308090873E-4\r\n",
      "unknownprc\tbiological_process\tunknown biological process\tCyanothecaceae\t1892249\tfamily\t16\t0.010540184453227932\t0.006377042646472699\r\n",
      "unknownprc\tbiological_process\tunknown biological process\tRodentia\t9989\torder\t2\t0.0013175230566534915\t7.971303308090873E-4\r\n",
      "unknownprc\tbiological_process\tunknown biological process\tHomo\t9605\tgenus\t2\t0.0013175230566534915\t7.971303308090873E-4\r\n"
     ]
    }
   ],
   "source": [
    "!head /Users/meganduffy/Documents/git-repos/2017-etnp/analyses/metagomics/susvsink/ja2/taxonomy_report_ja2nomods_06.28.18.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now uploading the ja14 protein fasta (ja14-proteins-spider72.fasta) and doing the same process for that sample (1000 m suspended). I opened the csv file from the Spider search (72) and extracted just the peptide sequence column and the #spectra column, deleted the headers, and saved the file as a tab-delimited text file all in Excel. It now looks like this:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SC(+57.02)AAAGTEC(+57.02)LISGWGNTK(+28.03)\t4\r",
      "\r\n",
      "AIQQQIENPLAQQILSGELVPGK\t3\r",
      "\r\n",
      "LPQVEGTGGDVQPSQDLVR\t6\r",
      "\r\n",
      "AIQQQIE(+21.98)NPLAQQILSGELVPGK\t1\r",
      "\r\n",
      "VATVSLPRSC(+57.02)AAAGTEC(+57.02)LISGWGNTK(+28.03)\t1\r",
      "\r\n",
      "S(+28.03)SGSSYPSLLQC(+57.02)LK\t2\r",
      "\r\n",
      "LPQVEGTGGD(+37.96)VQPSQDLVR\t1\r",
      "\r\n",
      "LPQVEGTGGDVQ(+.98)PSQDLVR\t2\r",
      "\r\n",
      "SSGSSYPSLLQC(+57.02)LK(+28.03)APVLSDSSC(+57.02)K\t1\r",
      "\r\n",
      "LGEHNIDVLEGNEQFINAAK\t3\r",
      "\r\n"
     ]
    }
   ],
   "source": [
    "!head /Users/meganduffy/Documents/git-repos/2017-etnp/data/ja9-ja16/MED_ETNP_JA14_11.07.17_SPIDER_72/ja14-metagomics-input.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Now using the code from the previous file to remove modifications. \n",
    "# There will probably be some here I need to weed out as they didn't occur in JA2.\n",
    "# Update: there were only 3! Many fewer peptides, period, though.\n",
    "!sed 's/(+15.99)//g' /Users/meganduffy/Documents/git-repos/2017-etnp/data/ja9-ja16/MED_ETNP_JA14_11.07.17_SPIDER_72/ja14-metagomics-input.txt \\\n",
    "| sed 's/(+57.02)//g'\\\n",
    "| sed 's/(+.98)//g'\\\n",
    "| sed 's/(+21.98)//g'\\\n",
    "| sed 's/(+28.03)//g'\\\n",
    "| sed 's/(+37.95)//g'\\\n",
    "| sed 's/(-17.03)//g'\\\n",
    "| sed 's/(+14.02)//g'\\\n",
    "| sed 's/(-15.99)//g'\\\n",
    "| sed 's/(+42.01)//g'\\\n",
    "| sed 's/(sub Q)//g'\\\n",
    "| sed 's/(+71.04)//g'\\\n",
    "| sed 's/(+55.92)//g'\\\n",
    "| sed 's/(sub A)//g'\\\n",
    "| sed 's/(+30.01)//g'\\\n",
    "| sed 's/(+3.99)//g'\\\n",
    "| sed 's/(+58.01)//g'\\\n",
    "| sed 's/(+43.01)//g'\\\n",
    "| sed 's/(+72.02)//g'\\\n",
    "| sed 's/(-18.01)//g'\\\n",
    "| sed 's/(sub D)//g'\\\n",
    "| sed 's/(sub G)//g'\\\n",
    "| sed 's/(+27.99)//g'\\\n",
    "| sed 's/(sub K)//g'\\\n",
    "| sed 's/(+37.96)//g'\\\n",
    "| sed 's/(+43.99)//g'\\\n",
    "| sed 's/(+21.97)//g'\\\n",
    "| sed 's/(+31.99)//g'\\\n",
    "| sed 's/(sub I)//g'\\\n",
    "| sed 's/(sub S)//g'\\\n",
    "| sed 's/(sub N)//g'\\\n",
    "| sed 's/(+15.01)//g'\\\n",
    "| sed 's/(-.98)//g'\\\n",
    "| sed 's/(+114.04)//g'\\\n",
    "| sed 's/(sub Y)//g'\\\n",
    "| sed 's/(sub T)//g'\\\n",
    "| sed 's/(del Q)//g'\\\n",
    "| sed 's/(-48.00)//g'\\\n",
    "| sed 's/(+100.02)//g'\\\n",
    "| sed 's/(+17.03)//g'\\\n",
    "| sed 's/(+79.97)//g'\\\n",
    "| sed 's/(+17.03)//g'\\\n",
    "| sed 's/(sub L)//g'\\\n",
    "| sed 's/(+53.92)//g'\\\n",
    "| sed 's/(sub V)//g'\\\n",
    "| sed 's/(sub M)//g'\\\n",
    "| sed 's/(+15.00)//g'\\\n",
    "| sed 's/(+150.04)//g'\\\n",
    "| sed 's/(+60.00)//g'\\\n",
    "| sed 's/(+119.04)//g'\\\n",
    "| sed 's/(+41.03)//g'\\\n",
    "| sed 's/(+29.02)//g'\\\n",
    "| sed 's/(+79.96)//g'\\\n",
    "| sed 's/(+282.05)//g'\\\n",
    "| sed 's/(del N)//g'\\\n",
    "| sed 's/(+228.11)//g'\\\n",
    "| sed 's/(sub E)//g'\\\n",
    "| sed 's/(+27.05)//g'\\\n",
    "| sed 's/(+87.03)//g'\\\n",
    "| sed 's/(-15.01)//g'\\\n",
    "| sed 's/(-1.03)//g'\\\n",
    "| sed 's/(sub F)//g'\\\n",
    "| sed 's/(del A)//g'\\\n",
    "> /Users/meganduffy/Documents/git-repos/2017-etnp/data/ja9-ja16/MED_ETNP_JA14_11.07.17_SPIDER_72/ja14-metagomics-nomod-input.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SCAAAGTECLISGWGNTK\t4\r",
      "\r\n",
      "AIQQQIENPLAQQILSGELVPGK\t3\r",
      "\r\n",
      "LPQVEGTGGDVQPSQDLVR\t6\r",
      "\r\n",
      "AIQQQIENPLAQQILSGELVPGK\t1\r",
      "\r\n",
      "VATVSLPRSCAAAGTECLISGWGNTK\t1\r",
      "\r\n",
      "SSGSSYPSLLQCLK\t2\r",
      "\r\n",
      "LPQVEGTGGDVQPSQDLVR\t1\r",
      "\r\n",
      "LPQVEGTGGDVQPSQDLVR\t2\r",
      "\r\n",
      "SSGSSYPSLLQCLKAPVLSDSSCK\t1\r",
      "\r\n",
      "LGEHNIDVLEGNEQFINAAK\t3\r",
      "\r\n",
      "VIGQNEAVDAVSNAIR\t4\r",
      "\r\n",
      "LPQVEGTGGDVQPSQDLVR\t1\r",
      "\r\n",
      "LGEHNIDVLEGNEQFINAAK\t1\r",
      "\r\n",
      "LPQVEGTGGDVQPSQDLVR\t1\r",
      "\r\n",
      "IITHPNFNGNTLDNDIMLIK\t1\r",
      "\r\n",
      "VIGQNEAVDAVSNAIR\t1\r",
      "\r\n",
      "VIGQNEAVDAVSNAIR\t2\r",
      "\r\n",
      "VATVSLPR\t1\r",
      "\r\n",
      "SSGSSYPSLLQCLK\t2\r",
      "\r\n",
      "GQNEAVDAVSNAIR\t1\r",
      "\r\n",
      "SCAAAGTECLISGWGNTK\t3\r",
      "\r\n",
      "SCAAAGTECLISGWGNTK\t1\r",
      "\r\n",
      "VIGQNEAVDAVSNAIR\t1\r",
      "\r\n",
      "VIGQNEAVDAVSNAIR\t1\r",
      "\r\n",
      "SSGSSYPSLLQCLK\t1\r",
      "\r\n",
      "AIDLIDEAASSIR\t7\r",
      "\r\n",
      "AIDLIDEAASSLR\t7\r",
      "\r\n",
      "VIGQNEAVDAVSNAIR\t1\r",
      "\r\n",
      "VTDAEIAEVLAR\t6\r",
      "\r\n",
      "SCAAAGTECLISGWGNTK\t1\r",
      "\r\n",
      "EEEVGLDIAQNGER\t1\r",
      "\r\n",
      "SCAAAGTECLISGWGNTK\t2\r",
      "\r\n",
      "LGEHNIDVLEGNEQFINAAK\t1\r",
      "\r\n",
      "DLYGNQADASWNPR\t1\r",
      "\r\n",
      "ADIKAVKDK\t1\r",
      "\r\n",
      "LSSPATLNSR\t2\r",
      "\r\n",
      "TQFYNDEPEAIEYGENFIVHR\t1\r",
      "\r\n",
      "VIGQNEAVDAVSNAIR\t1\r",
      "\r\n",
      "SCAAAGTECLISGWGNTK\t1\r",
      "\r\n",
      "VTVEEPFYVRPEEHPGAI\t1\r",
      "\r\n",
      "VIGQNEPVDAVSNAIR\t1\r",
      "\r\n",
      "PGGYQANALAQR\t1\r",
      "\r\n",
      "LPQVEGTGGDVQPSQDLVR\t2\r",
      "\r\n",
      "ADEVVAAYDSGR\t1\r",
      "\r\n",
      "NNPVLIGEPGVGK\t4\r",
      "\r\n",
      "SSGSSYPSLLQCLK\t1\r",
      "\r\n",
      "VIGQNEAVDAVSNAIR\t1\r",
      "\r\n",
      "DITFVSWGGAYTASQQK\t1\r",
      "\r\n",
      "VIGQNEVDAVSNAIR\t1\r",
      "\r\n",
      "AVAAAVLGDANK\t1\r",
      "\r\n",
      "TTTWTLLR\t1\r",
      "\r\n",
      "PETISKEPLGPVTR\t1\r",
      "\r\n",
      "NGQKAVTQWNTLHY\t3\r",
      "\r\n",
      "TVEEAQVR\t1\r",
      "\r\n",
      "LPQVEGTGGDVQ\t1\r",
      "\r\n",
      "AIDLIDEAASSIR\t1\r",
      "\r\n",
      "DALKAVKDAK\t1\r",
      "\r\n",
      "VTDAEIAEVLAR\t1\r",
      "\r\n",
      "EDLAALEK\t1\r",
      "\r\n",
      "GDSIGDISGEVR\t1\r",
      "\r\n",
      "GDSLGDISGEVR\t1\r",
      "\r\n",
      "PEQLISGKEDAANNFAR\t1\r",
      "\r\n",
      "VATVSLPR\t4\r",
      "\r\n",
      "AVTQWNTLHY\t1\r",
      "\r\n",
      "DVLVTGGYSNVDQSVMGAR\t2\r",
      "\r\n",
      "EEEIGLDLAQNGER\t1\r",
      "\r\n",
      "AIDLIDEAASSIR\t1\r",
      "\r\n",
      "DVLVTGGYSNVDQSVMGPR\t2\r",
      "\r\n",
      "YVGEGMEEGEFSEAR\t1\r",
      "\r\n",
      "YGENFIVHR\t1\r",
      "\r\n",
      "DVNASVATIKTKR\t1\r",
      "\r\n",
      "IGAAILSEGK\t1\r",
      "\r\n",
      "VGNPLDTYPDRR\t1\r",
      "\r\n",
      "LPQVEGTGGDVQPSQDLVR\t1\r",
      "\r\n",
      "ADQIYLAGLVYR\t1\r",
      "\r\n",
      "VTDAEIAEVLAR\t1\r",
      "\r\n",
      "PKEEEVGLDIAQNGER\t1\r",
      "\r\n",
      "VIGQNEAVDAVSNAIR\t1\r",
      "\r\n",
      "NNPVLIGEPGVGK\t1\r",
      "\r\n",
      "GMEEGEFSEAR\t1\r",
      "\r\n",
      "QVGNYGESYKR\t1\r",
      "\r\n",
      "LSSPATLNSRVATVSLPR\t1\r",
      "\r\n",
      "VFAEYAADGER\t1\r",
      "\r\n",
      "AQWINGGILYANPTR\t1\r",
      "\r\n",
      "SGGGGGGGLGSGGSIR\t1\r",
      "\r\n",
      "WEEVYRDQYK\t1\r",
      "\r\n",
      "GGLLYVPPAR\t3\r",
      "\r\n",
      "GGILYVPPAR\t3\r",
      "\r\n",
      "VATVSLPR\t3\r",
      "\r\n",
      "IFLIPGPSK\t1\r",
      "\r\n",
      "EFQTNLVPYPR\t1\r",
      "\r\n",
      "LGSLLLER\t1\r",
      "\r\n",
      "LGSLLIER\t1\r",
      "\r\n",
      "IGSILLER\t1\r",
      "\r\n",
      "IGSLLIER\t1\r",
      "\r\n",
      "AIQQQIENPLAQQILSGELVPGK\t1\r",
      "\r\n",
      "VATVSLPRSCAAAGTECLISGWGNTK\t1\r",
      "\r\n",
      "ANNLDIEPVTFEDKNVR\t1\r",
      "\r\n",
      "PEVYNWQLGR\t1\r",
      "\r\n",
      "LPQVEGTGGDVRPSQDL\t1\r",
      "\r\n",
      "SCAAAGTECLISGWGNTK\t1\r",
      "\r\n",
      "VEEPFYVRPEEHPGAI\t1\r",
      "\r\n",
      "QITDQAER\t1\r",
      "\r\n",
      "QLTDQAER\t1\r",
      "\r\n",
      "DASFYGALKPR\t1\r",
      "\r\n",
      "PFYVRPEEHPGAI\t1\r",
      "\r\n",
      "EAIEYGENFIVHR\t1\r",
      "\r\n",
      "TAIVEGLAQR\t1\r",
      "\r\n",
      "IGQNELVDAVSNAIR\t1\r",
      "\r\n",
      "VGNYGESYKR\t1\r",
      "\r\n",
      "PPTVVPGGDLAK\t1\r",
      "\r\n",
      "AEIATKIAR\t1\r",
      "\r\n",
      "GEGMEEGEFSEAR\t1\r",
      "\r\n",
      "LSVDYGK\t1\r",
      "\r\n",
      "ANNLDIEPVTFEDK\t1\r",
      "\r\n",
      "PDNFVFGQTGAGNNWAK\t1\r",
      "\r\n",
      "GPNAQWINGGILYANPTR\t1\r",
      "\r\n",
      "SSAESLALQAK\t1\r",
      "\r\n",
      "ELNLADYSR\t1\r",
      "\r\n",
      "NGESDVLSR\t1\r",
      "\r\n",
      "TALADLIAR\t1\r",
      "\r\n",
      "DLIDEAASSIR\t1\r",
      "\r\n",
      "PTVIDEVR\t1\r",
      "\r\n",
      "NGQKAVTQWNTLHY\t1\r",
      "\r\n",
      "RGDSIGDISGEVR\t1\r",
      "\r\n",
      "RGDSLGDISGEVR\t1\r",
      "\r\n",
      "VTDAEIAEVLAR\t1\r",
      "\r\n",
      "GDSDLQLER\t1\r",
      "\r\n",
      "AKTVEEAQVR\t1\r",
      "\r\n",
      "CAAAGTECLISGWGNTK\t1\r",
      "\r\n",
      "TLAPGGYQANALAQR\t1\r",
      "\r\n",
      "GGVPRPHGR\t1\r",
      "\r\n",
      "ISFFVPSR\t1\r",
      "\r\n",
      "RTVENYNGGLGEIK\t1\r",
      "\r\n",
      "NFATADSGGKGR\t1\r",
      "\r\n",
      "PDAAIAEAER\t1\r",
      "\r\n",
      "ANLGNDIIIR\t1\r",
      "\r\n",
      "EPLGPVVR\t1\r",
      "\r\n",
      "GTDEFVR\t1\r",
      "\r\n",
      "FYVRPEEHPGAI\t1\r",
      "\r\n",
      "AQIDLWKN\t1\r",
      "\r\n",
      "TQVQIGSFDR\t1\r",
      "\r\n",
      "ALQPLGDR\t1\r",
      "\r\n",
      "ETISKEPLGPVVR\t1\r",
      "\r\n",
      "ALLGLAAK\t1\r",
      "\r\n",
      "TSGEAFRTQ\t1\r",
      "\r\n",
      "TVLAVEIAR\t1\r",
      "\r\n",
      "TPKEEEVGLDIAQNGER\t1\r",
      "\r\n",
      "HILISSKR\t1\r",
      "\r\n",
      "DSIGDISGEVR\t1\r",
      "\r\n",
      "DSLGDISGEVR\t1\r",
      "\r\n",
      "DAADITEVLPK\t1\r",
      "\r\n",
      "AFLYGLGAR\t1\r",
      "\r\n",
      "EVLEEDAAK\t1\r",
      "\r\n",
      "TGTAQLTVK\t1\r",
      "\r\n",
      "GTLEDVQAR\t1\r",
      "\r\n",
      "IDLIDEAASSIR\t1\r",
      "\r\n",
      "IDLIDEAASSLR\t1\r",
      "\r\n",
      "AGPLAPHR\t1\r",
      "\r\n",
      "WEEVYR\t1\r\n"
     ]
    }
   ],
   "source": [
    "!head -2000 /Users/meganduffy/Documents/git-repos/2017-etnp/data/ja9-ja16/MED_ETNP_JA14_11.07.17_SPIDER_72/ja14-metagomics-nomod-input.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Did the MetaGOmics search with the same (default) parameters are before and here are the results below. Again, no images available to download. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# MetaGOmics GO report\r\n",
      "# MetaGOmics version: 0.1.1\r\n",
      "# Run date: Thu Jun 28 16:13:24 PDT 2018\r\n",
      "GO acc\tGO aspect\tGO name\tcount\ttotal count\tratio\r\n",
      "GO:0016817\tmolecular_function\thydrolase activity, acting on acid anhydrides\t17\t138\t0.12318840579710146\r\n",
      "GO:0031328\tbiological_process\tpositive regulation of cellular biosynthetic process\t15\t138\t0.10869565217391304\r\n",
      "unknownprc\tbiological_process\tunknown biological process\t103\t138\t0.7463768115942029\r\n",
      "GO:0016818\tmolecular_function\thydrolase activity, acting on acid anhydrides, in phosphorus-containing anhydrides\t17\t138\t0.12318840579710146\r\n",
      "unknownfun\tmolecular_function\tunknown molecular function\t114\t138\t0.8260869565217391\r\n",
      "GO:0031329\tbiological_process\tregulation of cellular catabolic process\t2\t138\t0.014492753623188406\r\n"
     ]
    }
   ],
   "source": [
    "!head /Users/meganduffy/Documents/git-repos/2017-etnp/analyses/metagomics/susvsink/ja14/go_report_ja14_06.28.18.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# MetaGOmics taxonomy report\r\n",
      "# MetaGOmics version: 0.1.1\r\n",
      "# Run date: Thu Jun 28 16:13:31 PDT 2018\r\n",
      "GO acc\tGO aspect\tGO name\ttaxon name\tNCBI taxonomy id\ttaxonomy rank\tPSM count\tratio of GO\tratio of run\r\n",
      "GO:0016817\tmolecular_function\thydrolase activity, acting on acid anhydrides\troot\t1\tno rank\t17\t1.0\t0.12318840579710146\r\n",
      "GO:0016817\tmolecular_function\thydrolase activity, acting on acid anhydrides\tEukaryota\t2759\tsuperkingdom\t2\t0.11764705882352941\t0.014492753623188406\r\n",
      "GO:0031328\tbiological_process\tpositive regulation of cellular biosynthetic process\troot\t1\tno rank\t15\t1.0\t0.10869565217391304\r\n",
      "GO:0016818\tmolecular_function\thydrolase activity, acting on acid anhydrides, in phosphorus-containing anhydrides\troot\t1\tno rank\t17\t1.0\t0.12318840579710146\r\n",
      "GO:0016818\tmolecular_function\thydrolase activity, acting on acid anhydrides, in phosphorus-containing anhydrides\tEukaryota\t2759\tsuperkingdom\t2\t0.11764705882352941\t0.014492753623188406\r\n",
      "unknownfun\tmolecular_function\tunknown molecular function\tRhizobium leguminosarum\t384\tspecies\t11\t0.09649122807017543\t0.07971014492753623\r\n"
     ]
    }
   ],
   "source": [
    "!head /Users/meganduffy/Documents/git-repos/2017-etnp/analyses/metagomics/susvsink/ja14/taxonomy_report_ja14_06.28.18.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
