{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Manipulation of PeaskDB *de novo*-assisted database search results of ETNP 2017 P2 samples LC-MS/MS data using python.\n",
    "\n",
    "Starting with:\n",
    "\n",
    "    PeaksDB search results (.csv) of PTM-optimized database searches\n",
    "    These were all searched with 15 ppm precursor tolerance and 0.5 ppm fragement ion tolerance\n",
    "    Search database included marine fungi and labrinthulomyces discovered using _de novo_ peptide sequencing\n",
    "    XInteract file includes precursor intensities and protein descriptions\n",
    "\n",
    "Goal:\n",
    "\n",
    "    Files with stripped (no PTMs) peptide lists and\n",
    "    Columns with #'s of each modification in every sequence\n",
    "    Column with stripped peptide lengths (# amino acids)\n",
    "    \n",
    "### To use for a different file:\n",
    "\n",
    "#### 1. Change the input file name in *IN 4*\n",
    "#### 2. Use 'find + replace' (Esc + F) to replace the running # (e.g., 233) for another\n",
    "#### 3. Update the NAAF factor calculated in *IN 6* into *IN 7*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We don't have technical duplicates here, sadly, unlike the MED4 Pro samples. I exported PeaksDB search results as `csv` files into my ETNP 2017 git repo:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDB-PTMopt\n"
     ]
    }
   ],
   "source": [
    "cd /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDB-PTMopt/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.0.5'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# LIBRARIES\n",
    "#import pandas library for working with tabular data\n",
    "import os\n",
    "os.getcwd()\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "from scipy.stats import kde\n",
    "#import regular expresson (regex)\n",
    "import re\n",
    "#check pandas version\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "'ETNP_SKQ17_PEAKSDB_167_378-100m-trap_15ppm_DB search psm.csv'\r\n",
      "'ETNP_SKQ17_PEAKSDB_167_378-100m-trap_15ppm_de novo only peptides.csv'\r\n",
      " ETNP_SKQ17_PEAKSDB_167_378-100m-trap_15ppm_peptide.csv\r\n",
      " ETNP_SKQ17_PEAKSDB_167_378-100m-trap_15ppm_peptides.pep.xml\r\n",
      " ETNP_SKQ17_PEAKSDB_167_378-100m-trap_15ppm_protein-peptides.csv\r\n",
      " ETNP_SKQ17_PEAKSDB_167_378-100m-trap_15ppm_proteins.csv\r\n",
      " ETNP_SKQ17_PEAKSDB_167_378-100m-trap_15ppm_proteins.fasta\r\n",
      " ETNP_SKQ17_PEAKSDB20_378-100m-trap_15ppm.csv\r\n",
      " ETNP_SKQ17_PEAKSDB20_378-100m-trap_15ppm_NAAF.csv\r\n",
      " ETNP_SKQ17_PEAKSDB20_378-100m-trap_15ppm_NAAF_totals\r\n",
      " ETNP_SKQ17_PEAKSDB20_378-100m-trap_15ppm_stripped_peptides.fas\r\n",
      " ETNP_SKQ17_PEAKSDB20_378-100m-trap_15ppm_stripped_peptides.txt\r\n",
      " ETNP_SKQ17_PEAKSDB20_378-100m-trap_15ppm_totals.csv\r\n"
     ]
    }
   ],
   "source": [
    "ls 378/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 378: 100 m trap filtered on 0.3 um GF-75"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Peptide', '-10lgP', 'Mass', 'Length', 'ppm', 'm/z', 'RT',\n",
      "       'Area 378_100m_trap_SKQ17_P2', 'Fraction', 'Scan', 'Source File',\n",
      "       '#Spec', '#Spec 378_100m_trap_SKQ17_P2', 'Accession', 'PTM', 'AScore'],\n",
      "      dtype='object')\n",
      "# redundant peaksdb peptides in combined dataframe 120\n",
      "# nonredundant peaksdb peptides in combined dataframe 120\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide</th>\n",
       "      <th>-10lgP</th>\n",
       "      <th>Mass</th>\n",
       "      <th>Length</th>\n",
       "      <th>ppm</th>\n",
       "      <th>m/z</th>\n",
       "      <th>RT</th>\n",
       "      <th>Area</th>\n",
       "      <th>Fraction</th>\n",
       "      <th>Scan</th>\n",
       "      <th>Source File</th>\n",
       "      <th>#Spec</th>\n",
       "      <th>#Spec</th>\n",
       "      <th>Accession</th>\n",
       "      <th>PTM</th>\n",
       "      <th>AScore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LPQVEGTGGDVQPSQDLVR</td>\n",
       "      <td>86.46</td>\n",
       "      <td>1994.0068</td>\n",
       "      <td>19</td>\n",
       "      <td>0.2</td>\n",
       "      <td>998.0109</td>\n",
       "      <td>54.03</td>\n",
       "      <td>63800000.0</td>\n",
       "      <td>4</td>\n",
       "      <td>15243</td>\n",
       "      <td>20181214_378_etnp2017_100m_trap.raw</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>gi|54036848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>VIGQNEAVDAVSNAIR</td>\n",
       "      <td>67.23</td>\n",
       "      <td>1654.8638</td>\n",
       "      <td>16</td>\n",
       "      <td>0.7</td>\n",
       "      <td>828.4398</td>\n",
       "      <td>64.54</td>\n",
       "      <td>41700000.0</td>\n",
       "      <td>4</td>\n",
       "      <td>18159</td>\n",
       "      <td>20181214_378_etnp2017_100m_trap.raw</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>gi|54036848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AIQQQIENPLAQQILSGELVPGK</td>\n",
       "      <td>64.26</td>\n",
       "      <td>2473.3540</td>\n",
       "      <td>23</td>\n",
       "      <td>0.2</td>\n",
       "      <td>825.4587</td>\n",
       "      <td>86.70</td>\n",
       "      <td>774000.0</td>\n",
       "      <td>4</td>\n",
       "      <td>24111</td>\n",
       "      <td>20181214_378_etnp2017_100m_trap.raw</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>gi|54036848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AIDLIDEAASSIR</td>\n",
       "      <td>45.64</td>\n",
       "      <td>1372.7197</td>\n",
       "      <td>13</td>\n",
       "      <td>0.6</td>\n",
       "      <td>458.5808</td>\n",
       "      <td>76.55</td>\n",
       "      <td>26000000.0</td>\n",
       "      <td>4</td>\n",
       "      <td>21495</td>\n",
       "      <td>20181214_378_etnp2017_100m_trap.raw</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>gi|54036848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AIDLIDEAASSLR</td>\n",
       "      <td>45.64</td>\n",
       "      <td>1372.7197</td>\n",
       "      <td>13</td>\n",
       "      <td>0.6</td>\n",
       "      <td>458.5808</td>\n",
       "      <td>76.55</td>\n",
       "      <td>26000000.0</td>\n",
       "      <td>4</td>\n",
       "      <td>21495</td>\n",
       "      <td>20181214_378_etnp2017_100m_trap.raw</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Peptide  -10lgP       Mass  Length  ppm       m/z     RT  \\\n",
       "0      LPQVEGTGGDVQPSQDLVR   86.46  1994.0068      19  0.2  998.0109  54.03   \n",
       "1         VIGQNEAVDAVSNAIR   67.23  1654.8638      16  0.7  828.4398  64.54   \n",
       "2  AIQQQIENPLAQQILSGELVPGK   64.26  2473.3540      23  0.2  825.4587  86.70   \n",
       "3            AIDLIDEAASSIR   45.64  1372.7197      13  0.6  458.5808  76.55   \n",
       "4            AIDLIDEAASSLR   45.64  1372.7197      13  0.6  458.5808  76.55   \n",
       "\n",
       "         Area  Fraction   Scan                          Source File  #Spec  \\\n",
       "0  63800000.0         4  15243  20181214_378_etnp2017_100m_trap.raw      4   \n",
       "1  41700000.0         4  18159  20181214_378_etnp2017_100m_trap.raw      5   \n",
       "2    774000.0         4  24111  20181214_378_etnp2017_100m_trap.raw      2   \n",
       "3  26000000.0         4  21495  20181214_378_etnp2017_100m_trap.raw      6   \n",
       "4  26000000.0         4  21495  20181214_378_etnp2017_100m_trap.raw      6   \n",
       "\n",
       "   #Spec    Accession  PTM AScore  \n",
       "0      4  gi|54036848  NaN    NaN  \n",
       "1      5  gi|54036848  NaN    NaN  \n",
       "2      2  gi|54036848  NaN    NaN  \n",
       "3      6  gi|54036848  NaN    NaN  \n",
       "4      6          NaN  NaN    NaN  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read the CSV into a dataframe using the pandas read_csv function\n",
    "peaksdbdup378 = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDB-PTMopt/378/ETNP_SKQ17_PEAKSDB_167_378-100m-trap_15ppm_peptide.csv\")\n",
    "\n",
    "# remove redundant rows\n",
    "peaksdb378 = pd.DataFrame.drop_duplicates(peaksdbdup378)\n",
    "\n",
    "print(peaksdb378.columns)\n",
    "\n",
    "columns = ['Peptide', '-10lgP', 'Mass', 'Length', 'ppm', 'm/z', 'RT',\n",
    "       'Area', 'Fraction', 'Scan', 'Source File',\n",
    "       '#Spec', '#Spec', 'Accession', 'PTM',\n",
    "       'AScore']\n",
    "\n",
    "peaksdb378.columns = columns\n",
    "\n",
    "print(\"# redundant peaksdb peptides in combined dataframe\", len(peaksdbdup378))\n",
    "print(\"# nonredundant peaksdb peptides in combined dataframe\", len(peaksdb378))\n",
    "\n",
    "#look at the dataframe\n",
    "peaksdb378.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide</th>\n",
       "      <th>-10lgP</th>\n",
       "      <th>Mass</th>\n",
       "      <th>Length</th>\n",
       "      <th>ppm</th>\n",
       "      <th>m/z</th>\n",
       "      <th>RT</th>\n",
       "      <th>Area</th>\n",
       "      <th>Fraction</th>\n",
       "      <th>Scan</th>\n",
       "      <th>...</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>k-hydr</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>stripped length</th>\n",
       "      <th>NAAF num.</th>\n",
       "      <th>ptm-total</th>\n",
       "      <th>stripped I-L</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LPQVEGTGGDVQPSQDLVR</td>\n",
       "      <td>86.46</td>\n",
       "      <td>1994.0068</td>\n",
       "      <td>19</td>\n",
       "      <td>0.2</td>\n",
       "      <td>998.0109</td>\n",
       "      <td>54.03</td>\n",
       "      <td>63800000.0</td>\n",
       "      <td>4</td>\n",
       "      <td>15243</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>LPQVEGTGGDVQPSQDLVR</td>\n",
       "      <td>19</td>\n",
       "      <td>3.357895e+06</td>\n",
       "      <td>0</td>\n",
       "      <td>LPQVEGTGGDVQPSQDLVR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>VIGQNEAVDAVSNAIR</td>\n",
       "      <td>67.23</td>\n",
       "      <td>1654.8638</td>\n",
       "      <td>16</td>\n",
       "      <td>0.7</td>\n",
       "      <td>828.4398</td>\n",
       "      <td>64.54</td>\n",
       "      <td>41700000.0</td>\n",
       "      <td>4</td>\n",
       "      <td>18159</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>VIGQNEAVDAVSNAIR</td>\n",
       "      <td>16</td>\n",
       "      <td>2.606250e+06</td>\n",
       "      <td>0</td>\n",
       "      <td>VLGQNEAVDAVSNALR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AIQQQIENPLAQQILSGELVPGK</td>\n",
       "      <td>64.26</td>\n",
       "      <td>2473.3540</td>\n",
       "      <td>23</td>\n",
       "      <td>0.2</td>\n",
       "      <td>825.4587</td>\n",
       "      <td>86.70</td>\n",
       "      <td>774000.0</td>\n",
       "      <td>4</td>\n",
       "      <td>24111</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>AIQQQIENPLAQQILSGELVPGK</td>\n",
       "      <td>23</td>\n",
       "      <td>3.365217e+04</td>\n",
       "      <td>0</td>\n",
       "      <td>ALQQQLENPLAQQLLSGELVPGK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AIDLIDEAASSIR</td>\n",
       "      <td>45.64</td>\n",
       "      <td>1372.7197</td>\n",
       "      <td>13</td>\n",
       "      <td>0.6</td>\n",
       "      <td>458.5808</td>\n",
       "      <td>76.55</td>\n",
       "      <td>26000000.0</td>\n",
       "      <td>4</td>\n",
       "      <td>21495</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>AIDLIDEAASSIR</td>\n",
       "      <td>13</td>\n",
       "      <td>2.000000e+06</td>\n",
       "      <td>0</td>\n",
       "      <td>ALDLLDEAASSLR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AIDLIDEAASSLR</td>\n",
       "      <td>45.64</td>\n",
       "      <td>1372.7197</td>\n",
       "      <td>13</td>\n",
       "      <td>0.6</td>\n",
       "      <td>458.5808</td>\n",
       "      <td>76.55</td>\n",
       "      <td>26000000.0</td>\n",
       "      <td>4</td>\n",
       "      <td>21495</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>AIDLIDEAASSLR</td>\n",
       "      <td>13</td>\n",
       "      <td>2.000000e+06</td>\n",
       "      <td>0</td>\n",
       "      <td>ALDLLDEAASSLR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Peptide  -10lgP       Mass  Length  ppm       m/z     RT  \\\n",
       "0      LPQVEGTGGDVQPSQDLVR   86.46  1994.0068      19  0.2  998.0109  54.03   \n",
       "1         VIGQNEAVDAVSNAIR   67.23  1654.8638      16  0.7  828.4398  64.54   \n",
       "2  AIQQQIENPLAQQILSGELVPGK   64.26  2473.3540      23  0.2  825.4587  86.70   \n",
       "3            AIDLIDEAASSIR   45.64  1372.7197      13  0.6  458.5808  76.55   \n",
       "4            AIDLIDEAASSLR   45.64  1372.7197      13  0.6  458.5808  76.55   \n",
       "\n",
       "         Area  Fraction   Scan  ... m-oxid  n-deam  q-deam k-hydr r-meth  \\\n",
       "0  63800000.0         4  15243  ...      0       0       0      0      0   \n",
       "1  41700000.0         4  18159  ...      0       0       0      0      0   \n",
       "2    774000.0         4  24111  ...      0       0       0      0      0   \n",
       "3  26000000.0         4  21495  ...      0       0       0      0      0   \n",
       "4  26000000.0         4  21495  ...      0       0       0      0      0   \n",
       "\n",
       "          stripped peptide  stripped length     NAAF num.  ptm-total  \\\n",
       "0      LPQVEGTGGDVQPSQDLVR               19  3.357895e+06          0   \n",
       "1         VIGQNEAVDAVSNAIR               16  2.606250e+06          0   \n",
       "2  AIQQQIENPLAQQILSGELVPGK               23  3.365217e+04          0   \n",
       "3            AIDLIDEAASSIR               13  2.000000e+06          0   \n",
       "4            AIDLIDEAASSLR               13  2.000000e+06          0   \n",
       "\n",
       "              stripped I-L  \n",
       "0      LPQVEGTGGDVQPSQDLVR  \n",
       "1         VLGQNEAVDAVSNALR  \n",
       "2  ALQQQLENPLAQQLLSGELVPGK  \n",
       "3            ALDLLDEAASSLR  \n",
       "4            ALDLLDEAASSLR  \n",
       "\n",
       "[5 rows x 47 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use a count function to enumerate the # of A's (alanines) in each peptide\n",
    "peaksdb378['A'] = peaksdb378['Peptide'].str.count(\"A\")\n",
    "\n",
    "# use a count function to enumerate the # of C's (cysteines) in each peptide\n",
    "peaksdb378['C'] = peaksdb378['Peptide'].str.count(\"C\")\n",
    "\n",
    "# use a count function to enumerate the # of D's (aspartic acids) in each peptide\n",
    "peaksdb378['D'] = peaksdb378['Peptide'].str.count(\"D\")\n",
    "\n",
    "# use a count function to enumerate the # of E's (glutamic acids) in each peptide\n",
    "peaksdb378['E'] = peaksdb378['Peptide'].str.count(\"E\")\n",
    "\n",
    "# use a count function to enumerate the # of F's (phenylalanines) in each peptide\n",
    "peaksdb378['F'] = peaksdb378['Peptide'].str.count(\"F\")\n",
    "\n",
    "# use a count function to enumerate the # of G's (glycines) in each peptide\n",
    "peaksdb378['G'] = peaksdb378['Peptide'].str.count(\"G\")\n",
    "\n",
    "# use a count function to enumerate the # of H's (histidines) in each peptide\n",
    "peaksdb378['H'] = peaksdb378['Peptide'].str.count(\"H\")\n",
    "\n",
    "# use a count function to enumerate the # of I's (isoleucines) in each peptide\n",
    "# in peaksdb378 output, there will be no isoleucines (they're lumped in with leucines)\n",
    "peaksdb378['I'] = peaksdb378['Peptide'].str.count(\"I\")\n",
    "\n",
    "# use a count function to enumerate the # of K's (lysines) in each peptide\n",
    "peaksdb378['K'] = peaksdb378['Peptide'].str.count(\"K\")\n",
    "\n",
    "# use a count function to enumerate the # of L's (leucines) in each peptide\n",
    "# also these include the isoleucines\n",
    "peaksdb378['L'] = peaksdb378['Peptide'].str.count(\"L\")\n",
    "\n",
    "# use a count function to enumerate the # of M's (methionines) in each peptide\n",
    "peaksdb378['M'] = peaksdb378['Peptide'].str.count(\"M\")\n",
    "\n",
    "# use a count function to enumerate the # of N's (asparagines) in each peptide\n",
    "peaksdb378['N'] = peaksdb378['Peptide'].str.count(\"N\")\n",
    "\n",
    "# use a count function to enumerate the # of P's ([prolines]) in each peptide\n",
    "peaksdb378['P'] = peaksdb378['Peptide'].str.count(\"P\")\n",
    "\n",
    "# use a count function to enumerate the # of Q's (glutamines) in each peptide\n",
    "peaksdb378['Q'] = peaksdb378['Peptide'].str.count(\"Q\")\n",
    "\n",
    "# use a count function to enumerate the # of R's (arginines) in each peptide\n",
    "peaksdb378['R'] = peaksdb378['Peptide'].str.count(\"R\")\n",
    "\n",
    "# use a count function to enumerate the # of S's (serines) in each peptide\n",
    "peaksdb378['S'] = peaksdb378['Peptide'].str.count(\"S\")\n",
    "\n",
    "# use a count function to enumerate the # of T's (threonines) in each peptide\n",
    "peaksdb378['T'] = peaksdb378['Peptide'].str.count(\"T\")\n",
    "\n",
    "# use a count function to enumerate the # of V's (valines) in each peptide\n",
    "peaksdb378['V'] = peaksdb378['Peptide'].str.count(\"V\")\n",
    "\n",
    "# use a count function to enumerate the # of W's (tryptophans) in each peptide\n",
    "peaksdb378['W'] = peaksdb378['Peptide'].str.count(\"W\")\n",
    "\n",
    "# use a count function to enumerate the # of Y's (tyrosines) in each peptide\n",
    "peaksdb378['Y'] = peaksdb378['Peptide'].str.count(\"Y\")\n",
    "\n",
    "# use a count function to enumerate the # of carbamidomethylated C's in each peptide\n",
    "peaksdb378['c-carb'] = peaksdb378['Peptide'].str.count(\"57.02\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "peaksdb378['m-oxid'] = peaksdb378['Peptide'].apply(lambda x: x.count('M(+15.99)'))\n",
    "\n",
    "# use a lamba function to enumerate the # of deamidated N's in each peptide\n",
    "peaksdb378['n-deam'] = peaksdb378['Peptide'].apply(lambda x: x.count('N(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "peaksdb378['q-deam'] = peaksdb378['Peptide'].apply(lambda x: x.count('Q(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of hydroxylated K's in each peptide\n",
    "peaksdb378['k-hydr'] = peaksdb378['Peptide'].apply(lambda x: x.count('K(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of hydroxylated P's in each peptide\n",
    "#peaksdb378['p-hydr'] = peaksdb378['Peptide'].apply(lambda x: x.count('P(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "peaksdb378['r-meth'] = peaksdb378['Peptide'].apply(lambda x: x.count('R(+14.02)'))\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "peaksdb378['stripped peptide'] = peaksdb378['Peptide'].str.replace(r\"\\(.*\\)\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "peaksdb378['stripped length'] = peaksdb378['stripped peptide'].apply(len)\n",
    "\n",
    "peaksdb378['NAAF num.'] = peaksdb378['Area'] / peaksdb378['stripped length']\n",
    "\n",
    "# total the number of modifications in sequence\n",
    "peaksdb378['ptm-total'] = peaksdb378['c-carb'] + peaksdb378['m-oxid'] + peaksdb378['n-deam'] + peaksdb378['q-deam'] + peaksdb378['k-hydr'] + peaksdb378['r-meth']\n",
    "\n",
    "# turn all isoleucines into leucines\n",
    "# this helps later in comparing Unipept peptides to PeaksDB and Comet ones\n",
    "peaksdb378['stripped I-L']= peaksdb378['stripped peptide'].str.replace('I','L')\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "peaksdb378.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDB-PTMopt/378/ETNP_SKQ17_PEAKSDB20_378-100m-trap_15ppm.csv\")\n",
    "\n",
    "# check out the results\n",
    "peaksdb378.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exporting txt files of stripped peptides at confidence cutoffs:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total modified PeaksDB peptides in 378: 120\n",
      "Deduplicated modified PeaksDB peptides in 378: 120\n",
      "Total stripped PeaksDB peptides in 378: 120\n",
      "Deduplicated stripped PeaksDB peptides in 378: 120\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LPQVEGTGGDVQPSQDLVR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>VIGQNEAVDAVSNAIR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AIQQQIENPLAQQILSGELVPGK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AIDLIDEAASSIR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AIDLIDEAASSLR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          stripped peptide\n",
       "0      LPQVEGTGGDVQPSQDLVR\n",
       "1         VIGQNEAVDAVSNAIR\n",
       "2  AIQQQIENPLAQQILSGELVPGK\n",
       "3            AIDLIDEAASSIR\n",
       "4            AIDLIDEAASSLR"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# keep only peptide column peptides >20 -10lgP\n",
    "pep378moddup = peaksdb378[[\"Peptide\"]]\n",
    "\n",
    "# keep only the stripped peptide column >20 -10lgP\n",
    "# this is what we'll use for UniPept input, etc\n",
    "pep378dup = peaksdb378[[\"stripped peptide\"]]\n",
    "\n",
    "# deduplicate both of these lists\n",
    "pep378mod = pep378moddup.drop_duplicates()\n",
    "pep378 = pep378dup.drop_duplicates()\n",
    "\n",
    "# print out the #s of modified and stripped peptides, deduplicated and not\n",
    "\n",
    "print(\"Total modified PeaksDB peptides in 378:\", len(pep378moddup))\n",
    "print(\"Deduplicated modified PeaksDB peptides in 378:\", len(pep378mod))\n",
    "print(\"Total stripped PeaksDB peptides in 378:\", len(pep378dup))\n",
    "print(\"Deduplicated stripped PeaksDB peptides in 378:\", len(pep378))\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "pep378.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDB-PTMopt/378/ETNP_SKQ17_PEAKSDB20_378-100m-trap_15ppm_stripped_peptides.txt\", header=False, index=False)\n",
    "\n",
    "# made the text file into a FASTA \n",
    "!awk '{print \">\"NR\"\\n\"$0}' /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDB-PTMopt/378/ETNP_SKQ17_PEAKSDB20_378-100m-trap_15ppm_stripped_peptides.txt > \\\n",
    "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDB-PTMopt/378/ETNP_SKQ17_PEAKSDB20_378-100m-trap_15ppm_stripped_peptides.fas\n",
    "\n",
    "# look at the stripped peptides\n",
    "pep378.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## NAAF correction and exporting files with AA and PTM totals:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>K</th>\n",
       "      <th>L</th>\n",
       "      <th>...</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>Total area</th>\n",
       "      <th>Total length</th>\n",
       "      <th>% C w/ carb</th>\n",
       "      <th>% M w/ oxid</th>\n",
       "      <th>% N w/ deam</th>\n",
       "      <th>% Q w/ deam</th>\n",
       "      <th>% K w/ hydr</th>\n",
       "      <th>% R w/ meth</th>\n",
       "      <th>NAAF denom.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>133</td>\n",
       "      <td>3</td>\n",
       "      <td>84</td>\n",
       "      <td>140</td>\n",
       "      <td>14</td>\n",
       "      <td>91</td>\n",
       "      <td>7</td>\n",
       "      <td>65</td>\n",
       "      <td>81</td>\n",
       "      <td>96</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>693658100.0</td>\n",
       "      <td>1118</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.057143</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.139535</td>\n",
       "      <td>620445.527728</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                A  C   D    E   F   G  H   I   K   L  ...  r-meth  \\\n",
       "sample total  133  3  84  140  14  91  7  65  81  96  ...       6   \n",
       "\n",
       "               Total area  Total length  % C w/ carb  % M w/ oxid  \\\n",
       "sample total  693658100.0          1118          1.0         0.25   \n",
       "\n",
       "              % N w/ deam  % Q w/ deam  % K w/ hydr  % R w/ meth  \\\n",
       "sample total          0.0     0.057143          0.0     0.139535   \n",
       "\n",
       "                NAAF denom.  \n",
       "sample total  620445.527728  \n",
       "\n",
       "[1 rows x 35 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of certain columns in the modified\n",
    "# peptide dataframe above \n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'A': peaksdb378['A'].sum(),\n",
    "        'C': peaksdb378['C'].sum(),\n",
    "        'D': peaksdb378['D'].sum(),\n",
    "        'E': peaksdb378['E'].sum(),\n",
    "        'F': peaksdb378['F'].sum(),\n",
    "        'G': peaksdb378['G'].sum(),\n",
    "        'H': peaksdb378['H'].sum(),\n",
    "        'I': peaksdb378['I'].sum(),\n",
    "        'K': peaksdb378['K'].sum(),\n",
    "        'L': peaksdb378['L'].sum(),\n",
    "        'M': peaksdb378['M'].sum(),\n",
    "        'N': peaksdb378['N'].sum(),\n",
    "        'P': peaksdb378['P'].sum(),\n",
    "        'Q': peaksdb378['Q'].sum(),\n",
    "        'R': peaksdb378['R'].sum(),\n",
    "        'S': peaksdb378['S'].sum(),\n",
    "        'T': peaksdb378['T'].sum(),\n",
    "        'V': peaksdb378['V'].sum(),\n",
    "        'W': peaksdb378['W'].sum(),\n",
    "        'Y': peaksdb378['Y'].sum(),\n",
    "        'c-carb': peaksdb378['c-carb'].sum(),\n",
    "        'm-oxid': peaksdb378['m-oxid'].sum(),\n",
    "        'n-deam': peaksdb378['n-deam'].sum(),\n",
    "        'q-deam': peaksdb378['q-deam'].sum(),\n",
    "        'k-hydr': peaksdb378['k-hydr'].sum(),\n",
    "        'r-meth': peaksdb378['r-meth'].sum(),\n",
    "        'Total area': peaksdb378['Area'].sum(),\n",
    "        'Total length': peaksdb378['stripped length'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaksdb378 = pd.DataFrame(data, columns=['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', 'm-oxid', 'n-deam', 'q-deam', 'k-hydr', 'r-meth', 'Total area', 'Total length'], index=index)\n",
    "\n",
    "# calculate percentage of C's with carb (should be 1.0)\n",
    "totalpeaksdb378['% C w/ carb'] = totalpeaksdb378['c-carb'] / totalpeaksdb378['C'] \n",
    "\n",
    "# calculate percentage of M's that are oxidized\n",
    "totalpeaksdb378['% M w/ oxid'] = totalpeaksdb378['m-oxid'] / totalpeaksdb378['M'] \n",
    "\n",
    "# calculate percentage of N's that are deamidated\n",
    "totalpeaksdb378['% N w/ deam'] = totalpeaksdb378['n-deam'] / totalpeaksdb378['N'] \n",
    "\n",
    "# calculate percentage of Q's that are deamidated\n",
    "totalpeaksdb378['% Q w/ deam'] = totalpeaksdb378['q-deam'] / totalpeaksdb378['Q'] \n",
    "\n",
    "# calculate percentage of K's that are hydroxylated\n",
    "totalpeaksdb378['% K w/ hydr'] = totalpeaksdb378['k-hydr'] / totalpeaksdb378['K'] \n",
    "\n",
    "# calculate percentage of P's that are hydroxylated\n",
    "#totalpeaksdb378['% P w/ hydr'] = totalpeaksdb378['p-hydr'] / totalpeaksdb378['P'] \n",
    "\n",
    "# calculate percentage of R's that are methylated\n",
    "totalpeaksdb378['% R w/ meth'] = totalpeaksdb378['r-meth'] / totalpeaksdb378['R'] \n",
    "\n",
    "# calculate NAAF denominator for all peptides in dataset i\n",
    "totalpeaksdb378['NAAF denom.'] = totalpeaksdb378['Total area'] / totalpeaksdb378['Total length']\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "totalpeaksdb378.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDB-PTMopt/378/ETNP_SKQ17_PEAKSDB20_378-100m-trap_15ppm_totals.csv\")\n",
    "\n",
    "totalpeaksdb378.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>NAAF factor</th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>...</th>\n",
       "      <th>T-NAAF20</th>\n",
       "      <th>V-NAAF20</th>\n",
       "      <th>W-NAAF20</th>\n",
       "      <th>Y-NAAF20</th>\n",
       "      <th>ccarb-NAAF20</th>\n",
       "      <th>moxid-NAAF20</th>\n",
       "      <th>ndeam-NAAF20</th>\n",
       "      <th>qdeam-NAAF20</th>\n",
       "      <th>khydr-NAAF20</th>\n",
       "      <th>rmeth-NAAF20</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>LPQVEGTGGDVQPSQDLVR</td>\n",
       "      <td>5.412070</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>5.41207</td>\n",
       "      <td>16.236211</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>VIGQNEAVDAVSNAIR</td>\n",
       "      <td>4.200611</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>12.601832</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AIQQQIENPLAQQILSGELVPGK</td>\n",
       "      <td>0.054239</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.054239</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AIDLIDEAASSIR</td>\n",
       "      <td>3.223490</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AIDLIDEAASSLR</td>\n",
       "      <td>3.223490</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          stripped peptide  NAAF factor  A  C  D  E  F  G  H  I  ...  \\\n",
       "0      LPQVEGTGGDVQPSQDLVR     5.412070  0  0  2  1  0  3  0  0  ...   \n",
       "1         VIGQNEAVDAVSNAIR     4.200611  3  0  1  1  0  1  0  2  ...   \n",
       "2  AIQQQIENPLAQQILSGELVPGK     0.054239  2  0  0  2  0  2  0  3  ...   \n",
       "3            AIDLIDEAASSIR     3.223490  3  0  2  1  0  0  0  3  ...   \n",
       "4            AIDLIDEAASSLR     3.223490  3  0  2  1  0  0  0  2  ...   \n",
       "\n",
       "   T-NAAF20   V-NAAF20  W-NAAF20  Y-NAAF20  ccarb-NAAF20  moxid-NAAF20  \\\n",
       "0   5.41207  16.236211       0.0       0.0           0.0           0.0   \n",
       "1   0.00000  12.601832       0.0       0.0           0.0           0.0   \n",
       "2   0.00000   0.054239       0.0       0.0           0.0           0.0   \n",
       "3   0.00000   0.000000       0.0       0.0           0.0           0.0   \n",
       "4   0.00000   0.000000       0.0       0.0           0.0           0.0   \n",
       "\n",
       "   ndeam-NAAF20  qdeam-NAAF20  khydr-NAAF20  rmeth-NAAF20  \n",
       "0           0.0           0.0           0.0           0.0  \n",
       "1           0.0           0.0           0.0           0.0  \n",
       "2           0.0           0.0           0.0           0.0  \n",
       "3           0.0           0.0           0.0           0.0  \n",
       "4           0.0           0.0           0.0           0.0  \n",
       "\n",
       "[5 rows x 54 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use the calculated NAAF factor (in totalpeaksdb378 dataframe, above) to caluclate the NAAF \n",
    "# NAAF: normalized normalized area abundance factor\n",
    "\n",
    "NAAF20 = 620445.527728\n",
    "\n",
    "# use NAAF >20 -10lgP to get NAAF\n",
    "peaksdb378['NAAF factor'] = (peaksdb378['NAAF num.'])/NAAF20\n",
    "\n",
    "# make a dataframe that contains only what we need: sequences, AAs, PTMs\n",
    "peaksdb378_NAAF = peaksdb378[['stripped peptide', 'NAAF factor', 'A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', \\\n",
    "                                'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', 'm-oxid', 'n-deam', \\\n",
    "                                'q-deam', 'k-hydr', 'r-meth']].copy()\n",
    "\n",
    "# multiply the NAAF20 factor by the AAs to normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaksdb378_NAAF['A-NAAF20'] = peaksdb378_NAAF['A'] * peaksdb378['NAAF factor']\n",
    "peaksdb378_NAAF['C-NAAF20'] = peaksdb378_NAAF['C'] * peaksdb378['NAAF factor']\n",
    "peaksdb378_NAAF['D-NAAF20'] = peaksdb378_NAAF['D'] * peaksdb378['NAAF factor']\n",
    "peaksdb378_NAAF['E-NAAF20'] = peaksdb378_NAAF['E'] * peaksdb378['NAAF factor']\n",
    "peaksdb378_NAAF['F-NAAF20'] = peaksdb378_NAAF['F'] * peaksdb378['NAAF factor']\n",
    "peaksdb378_NAAF['G-NAAF20'] = peaksdb378_NAAF['G'] * peaksdb378['NAAF factor']\n",
    "peaksdb378_NAAF['H-NAAF20'] = peaksdb378_NAAF['H'] * peaksdb378['NAAF factor']\n",
    "peaksdb378_NAAF['I-NAAF20'] = peaksdb378_NAAF['I'] * peaksdb378['NAAF factor']\n",
    "peaksdb378_NAAF['K-NAAF20'] = peaksdb378_NAAF['K'] * peaksdb378['NAAF factor']\n",
    "peaksdb378_NAAF['L-NAAF20'] = peaksdb378_NAAF['L'] * peaksdb378['NAAF factor']\n",
    "peaksdb378_NAAF['M-NAAF20'] = peaksdb378_NAAF['M'] * peaksdb378['NAAF factor']\n",
    "peaksdb378_NAAF['N-NAAF20'] = peaksdb378_NAAF['N'] * peaksdb378['NAAF factor']\n",
    "peaksdb378_NAAF['P-NAAF20'] = peaksdb378_NAAF['P'] * peaksdb378['NAAF factor']\n",
    "peaksdb378_NAAF['Q-NAAF20'] = peaksdb378_NAAF['Q'] * peaksdb378['NAAF factor']\n",
    "peaksdb378_NAAF['R-NAAF20'] = peaksdb378_NAAF['R'] * peaksdb378['NAAF factor']\n",
    "peaksdb378_NAAF['S-NAAF20'] = peaksdb378_NAAF['S'] * peaksdb378['NAAF factor']\n",
    "peaksdb378_NAAF['T-NAAF20'] = peaksdb378_NAAF['T'] * peaksdb378['NAAF factor']\n",
    "peaksdb378_NAAF['V-NAAF20'] = peaksdb378_NAAF['V'] * peaksdb378['NAAF factor']\n",
    "peaksdb378_NAAF['W-NAAF20'] = peaksdb378_NAAF['W'] * peaksdb378['NAAF factor']\n",
    "peaksdb378_NAAF['Y-NAAF20'] = peaksdb378_NAAF['Y'] * peaksdb378['NAAF factor']\n",
    "\n",
    "# multiply the NAAF20 factor by the PTMs normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaksdb378_NAAF['ccarb-NAAF20'] = peaksdb378_NAAF['c-carb'] * peaksdb378_NAAF['NAAF factor']\n",
    "peaksdb378_NAAF['moxid-NAAF20'] = peaksdb378_NAAF['m-oxid'] * peaksdb378_NAAF['NAAF factor']\n",
    "peaksdb378_NAAF['ndeam-NAAF20'] = peaksdb378_NAAF['n-deam'] * peaksdb378_NAAF['NAAF factor']\n",
    "peaksdb378_NAAF['qdeam-NAAF20'] = peaksdb378_NAAF['q-deam'] * peaksdb378_NAAF['NAAF factor']\n",
    "peaksdb378_NAAF['khydr-NAAF20'] = peaksdb378_NAAF['k-hydr'] * peaksdb378_NAAF['NAAF factor']\n",
    "peaksdb378_NAAF['rmeth-NAAF20'] = peaksdb378_NAAF['r-meth'] * peaksdb378_NAAF['NAAF factor']\n",
    "\n",
    "# write the dataframe to a new csv\n",
    "peaksdb378_NAAF.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDB-PTMopt/378/ETNP_SKQ17_PEAKSDB20_378-100m-trap_15ppm_NAAF.csv\")\n",
    "\n",
    "peaksdb378_NAAF.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NAAF</th>\n",
       "      <th>A-NAAF</th>\n",
       "      <th>C-NAAF</th>\n",
       "      <th>D-NAAF</th>\n",
       "      <th>E-NAAF</th>\n",
       "      <th>F-NAAF</th>\n",
       "      <th>G-NAAF</th>\n",
       "      <th>H-NAAF</th>\n",
       "      <th>I-NAAF</th>\n",
       "      <th>K-NAAF</th>\n",
       "      <th>...</th>\n",
       "      <th>Q-deam-NAAF</th>\n",
       "      <th>K-hydr-NAAF</th>\n",
       "      <th>R-meth-NAAF</th>\n",
       "      <th>% C w/ carb. NAAF</th>\n",
       "      <th>% M w/ oxid. NAAF</th>\n",
       "      <th>% N w/ deam. NAAF</th>\n",
       "      <th>% Q w/ deam. NAAF</th>\n",
       "      <th>% K w/ hydr. NAAF</th>\n",
       "      <th>% R w/ meth. NAAF</th>\n",
       "      <th>NAAF check</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>110.106445</td>\n",
       "      <td>124.773942</td>\n",
       "      <td>0.248526</td>\n",
       "      <td>49.975971</td>\n",
       "      <td>60.189831</td>\n",
       "      <td>4.934141</td>\n",
       "      <td>80.032139</td>\n",
       "      <td>0.659536</td>\n",
       "      <td>52.289862</td>\n",
       "      <td>29.423688</td>\n",
       "      <td>...</td>\n",
       "      <td>0.231285</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.025997</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.393678</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.009469</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.012751</td>\n",
       "      <td>0.000177</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    NAAF      A-NAAF    C-NAAF     D-NAAF     E-NAAF  \\\n",
       "sample total  110.106445  124.773942  0.248526  49.975971  60.189831   \n",
       "\n",
       "                F-NAAF     G-NAAF    H-NAAF     I-NAAF     K-NAAF  ...  \\\n",
       "sample total  4.934141  80.032139  0.659536  52.289862  29.423688  ...   \n",
       "\n",
       "              Q-deam-NAAF  K-hydr-NAAF  R-meth-NAAF  % C w/ carb. NAAF  \\\n",
       "sample total     0.231285          0.0     1.025997                1.0   \n",
       "\n",
       "              % M w/ oxid. NAAF  % N w/ deam. NAAF  % Q w/ deam. NAAF  \\\n",
       "sample total           0.393678                0.0           0.009469   \n",
       "\n",
       "              % K w/ hydr. NAAF  % R w/ meth. NAAF  NAAF check  \n",
       "sample total                0.0           0.012751    0.000177  \n",
       "\n",
       "[1 rows x 34 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of NAAF normalized AAs for peaksdb378 peaksdb results\n",
    "# also contains the sums of the NAAF-corrected PTMs occurances for each affected residue\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'NAAF': peaksdb378_NAAF['NAAF factor'].sum(),\n",
    "        'A-NAAF': peaksdb378_NAAF['A-NAAF20'].sum(),\n",
    "        'C-NAAF': peaksdb378_NAAF['C-NAAF20'].sum(),\n",
    "        'D-NAAF': peaksdb378_NAAF['D-NAAF20'].sum(),\n",
    "        'E-NAAF': peaksdb378_NAAF['E-NAAF20'].sum(),\n",
    "        'F-NAAF': peaksdb378_NAAF['F-NAAF20'].sum(),\n",
    "        'G-NAAF': peaksdb378_NAAF['G-NAAF20'].sum(),\n",
    "        'H-NAAF': peaksdb378_NAAF['H-NAAF20'].sum(),\n",
    "        'I-NAAF': peaksdb378_NAAF['I-NAAF20'].sum(),\n",
    "        'K-NAAF': peaksdb378_NAAF['K-NAAF20'].sum(),\n",
    "        'L-NAAF': peaksdb378_NAAF['L-NAAF20'].sum(),\n",
    "        'M-NAAF': peaksdb378_NAAF['M-NAAF20'].sum(),\n",
    "        'N-NAAF': peaksdb378_NAAF['N-NAAF20'].sum(),\n",
    "        'P-NAAF': peaksdb378_NAAF['P-NAAF20'].sum(),\n",
    "        'Q-NAAF': peaksdb378_NAAF['Q-NAAF20'].sum(),\n",
    "        'R-NAAF': peaksdb378_NAAF['R-NAAF20'].sum(),\n",
    "        'S-NAAF': peaksdb378_NAAF['S-NAAF20'].sum(),\n",
    "        'T-NAAF': peaksdb378_NAAF['T-NAAF20'].sum(),\n",
    "        'V-NAAF': peaksdb378_NAAF['V-NAAF20'].sum(),\n",
    "        'W-NAAF': peaksdb378_NAAF['W-NAAF20'].sum(),\n",
    "        'Y-NAAF': peaksdb378_NAAF['Y-NAAF20'].sum(),\n",
    "        'C-carb-NAAF': peaksdb378_NAAF['ccarb-NAAF20'].sum(),\n",
    "        'M-oxid-NAAF': peaksdb378_NAAF['moxid-NAAF20'].sum(),\n",
    "        'N-deam-NAAF': peaksdb378_NAAF['ndeam-NAAF20'].sum(),\n",
    "        'Q-deam-NAAF': peaksdb378_NAAF['qdeam-NAAF20'].sum(),\n",
    "        'K-hydr-NAAF': peaksdb378_NAAF['khydr-NAAF20'].sum(),\n",
    "        'R-meth-NAAF': peaksdb378_NAAF['rmeth-NAAF20'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaksdb378_NAAF = pd.DataFrame(data, columns=['NAAF', 'A-NAAF', 'C-NAAF', 'D-NAAF', 'E-NAAF', 'F-NAAF', \\\n",
    "                                                   'G-NAAF', 'H-NAAF', 'I-NAAF','K-NAAF', 'L-NAAF', 'M-NAAF', \\\n",
    "                                                   'N-NAAF', 'P-NAAF', 'Q-NAAF', 'R-NAAF', 'S-NAAF', \\\n",
    "                                                   'T-NAAF', 'V-NAAF', 'W-NAAF', 'Y-NAAF', 'C-carb-NAAF', \\\n",
    "                                                   'M-oxid-NAAF', 'N-deam-NAAF', 'Q-deam-NAAF', 'K-hydr-NAAF',\\\n",
    "                                                   'R-meth-NAAF'], index=index)\n",
    "\n",
    "# calculate the NAAF-corrected % modified C, M, N, Q, K, and Rs\n",
    "\n",
    "\n",
    "totalpeaksdb378_NAAF['% C w/ carb. NAAF'] = totalpeaksdb378_NAAF['C-carb-NAAF'] / totalpeaksdb378_NAAF['C-NAAF']\n",
    "totalpeaksdb378_NAAF['% M w/ oxid. NAAF'] = totalpeaksdb378_NAAF['M-oxid-NAAF'] / totalpeaksdb378_NAAF['M-NAAF']\n",
    "totalpeaksdb378_NAAF['% N w/ deam. NAAF'] = totalpeaksdb378_NAAF['N-deam-NAAF'] / totalpeaksdb378_NAAF['N-NAAF']\n",
    "totalpeaksdb378_NAAF['% Q w/ deam. NAAF'] = totalpeaksdb378_NAAF['Q-deam-NAAF'] / totalpeaksdb378_NAAF['Q-NAAF']\n",
    "totalpeaksdb378_NAAF['% K w/ hydr. NAAF'] = totalpeaksdb378_NAAF['K-hydr-NAAF'] / totalpeaksdb378_NAAF['K-NAAF']\n",
    "totalpeaksdb378_NAAF['% R w/ meth. NAAF'] = totalpeaksdb378_NAAF['R-meth-NAAF'] / totalpeaksdb378_NAAF['R-NAAF']\n",
    "\n",
    "# calculate NAAF summed numerator over denominator (in above cell) for all peptides in dataset i: a check\n",
    "totalpeaksdb378_NAAF['NAAF check'] = totalpeaksdb378_NAAF['NAAF'] / 620445.527728\n",
    "\n",
    "# write modified dataframe to new txt file, same name + totals\n",
    "totalpeaksdb378_NAAF.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDB-PTMopt/378/ETNP_SKQ17_PEAKSDB20_378-100m-trap_15ppm_NAAF_totals\")\n",
    "\n",
    "totalpeaksdb378_NAAF.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Using BioPython to query peptide sequences\n",
    "\n",
    "I installed the BioPython package using `pip install biopython`. All instructions and information [here](https://www.tutorialspoint.com/biopython/index.htm). \n",
    "\n",
    "GitHub project: https://github.com/biopython/biopython\n",
    "\n",
    "I'm relying on the ProtParam module to parse sequences for relative AA composition, instability, secondary structure, instability, and hydrophobicity. You can read more about that module and the studies the indecies are derived from here:\n",
    "\n",
    "https://biopython.org/wiki/ProtParam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# Bio.SeqIO is the standard Sequence Input/Output interface for BioPython 1.43 and later\n",
    "# Bio.SeqIO provides a simple uniform interface to input and output assorted sequence file formats.\n",
    "# (including multiple sequence alignments), but will only deal with sequences as SeqRecord objects\n",
    "\n",
    "# for accepted file formats see https://biopython.org/wiki/SeqIO\n",
    "\n",
    "from Bio import SeqIO\n",
    "#for seq_record in SeqIO.parse(\"/home/millieginty/Documents/git-repos/2017-etnp/data/MED4/MED2_tryp_1raw_db_peptides_nmod.fasta\", \"fasta\"):\n",
    "    #print(seq_record.id)\n",
    "    #print(repr(seq_record.seq))\n",
    "    #print(len(seq_record))\n",
    "    \n",
    "# I commented the print functions out so the output doesn't take up too much space. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# seeing what the ProtParam module can do with a single protein sequence:\n",
    "\n",
    "from Bio.SeqUtils.ProtParam import ProteinAnalysis\n",
    "\n",
    "test_seq = \"MAEGEITTFTALTEKFNLPPGNYKKPKLLYCSNGGHFLRILPDGTVDGTRDRSDQHIQLQLSAESVGEVYIKSTETGQYLAMDTSGLLYGSQTPSEECLFLERLEENHYNTYTSKKHAEKNWFVGLKKNGSCKRGPRTHYGQKAILFLPLPV\"\n",
    "\n",
    "analysed_seq = ProteinAnalysis(test_seq)\n",
    "print(\"molecular weight of seq =\", analysed_seq.molecular_weight())\n",
    "\n",
    "# calculates the aromaticity value of a protein according to Lobry & Gautier (1994, Nucleic Acids Res., 22, 3174-3180). \n",
    "# it's simply the relative frequency of Phe+Trp+Tyr.\n",
    "\n",
    "analysed_seq.aromaticity()\n",
    "print(\"aromaticity of seq =\", analysed_seq.aromaticity())\n",
    "\n",
    "# secondary_structure_fraction:\n",
    "# this methods returns a list of the fraction of amino acids which tend to be in helix, turn or sheet. \n",
    "# AAs in helix: V, I, Y, F, W, L\n",
    "# AAs in turn: N, P, G, S\n",
    "# AAs in sheet: E, M, A, L\n",
    "# the returned list contains 3 values: [Helix, Turn, Sheet]\n",
    "\n",
    "analysed_seq.secondary_structure_fraction()\n",
    "print(\"frac in H T S =\", analysed_seq.secondary_structure_fraction())\n",
    "\n",
    "# the instability index, an implementation of the method of Guruprasad et al. (1990, Protein Engineering, 4, 155-161).\n",
    "# this method tests a protein for stability. \n",
    "# any value above 40 means the protein is unstable (=has a short half life)\n",
    "# NOT SURE WHAT THIS MEANS FOR PEPTIDES, BUT WE COULD DO THIS FOR PROTEINS\n",
    "\n",
    "analysed_seq.instability_index()\n",
    "print(\"instability =\", analysed_seq.instability_index())\n",
    "\n",
    "# count_amino_acids will do just that, and get_amino_acids_percent will return %'s for each AA across the sequence. \n",
    "analysed_seq.get_amino_acids_percent()\n",
    "\n",
    "# taking the returned dictionary and converting to a dataframe\n",
    "\n",
    "aadict = analysed_seq.get_amino_acids_percent()\n",
    "aadf = pd.DataFrame(list(aadict.items()),columns = ['residue','% occurance']) \n",
    "\n",
    "aadf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# use SeqIO and a loop to apply count_amino_acids to each sequence in the file\n",
    "# aatot will give us the total number of each residue in the entire sample output\n",
    "\n",
    "import collections\n",
    "from Bio import SeqIO\n",
    "from Bio.SeqUtils.ProtParam import ProteinAnalysis\n",
    "\n",
    "all_aas = collections.defaultdict(int)\n",
    "for record in SeqIO.parse(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas\", \"fasta\"):\n",
    "    x = ProteinAnalysis(str(record.seq))\n",
    "    #print(record.id, x.count_amino_acids())\n",
    "    for aa, count in x.count_amino_acids().items():\n",
    "        all_aas[aa] += count        \n",
    "        \n",
    "# made a dataframe for amino acid total counts        \n",
    "data = (all_aas)\n",
    "aatot = pd.DataFrame(data, index = ['sample sequence total'])\n",
    "aatot.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pandas import Series, DataFrame\n",
    "\n",
    "with open('/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas') as fasta_file:  # Will close handle cleanly\n",
    "    identifiers = []\n",
    "    lengths = []\n",
    "    for seq_record in SeqIO.parse(fasta_file, 'fasta'):  # (generator)\n",
    "        identifiers.append(seq_record.id)\n",
    "        lengths.append(len(seq_record.seq))\n",
    "        \n",
    "        \n",
    "#converting lists to pandas Series    \n",
    "s1 = Series(identifiers, name='ID')\n",
    "s2 = Series(lengths, name='length')\n",
    "\n",
    "#Gathering Series into a pandas DataFrame and rename index as ID column\n",
    "idseq = DataFrame(dict(ID=s1, length=s2)).set_index(['ID'])\n",
    "\n",
    "idseq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pandas import Series, DataFrame\n",
    "\n",
    "\n",
    "\n",
    "with open('/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas) as fasta_file:  # Will close handle cleanly\n",
    "    identifiers = []\n",
    "    lengths = []\n",
    "    aa = []\n",
    "    for seq_record in SeqIO.parse(fasta_file, 'fasta'):  # (generator)\n",
    "        identifiers.append(seq_record.id)\n",
    "        lengths.append(len(seq_record.seq))\n",
    "        aa.count_amino_acids(seq_record.seq)\n",
    "        \n",
    "        \n",
    "#converting lists to pandas Series    \n",
    "s1 = Series(identifiers, name='ID')\n",
    "s2 = Series(lengths, name='length')\n",
    "s3 = Series(aa, name='AAs')\n",
    "\n",
    "#Gathering Series into a pandas DataFrame and rename index as ID column\n",
    "idseq = DataFrame(dict(ID=s1, length=s2, AAs=s3)).set_index(['ID'])\n",
    "\n",
    "idseq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio.SeqUtils import ProtParam\n",
    "\n",
    "handle = open(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas\") \n",
    "for record in SeqIO.parse(handle, \"fasta\"): \n",
    "    seq = str(record.seq)\n",
    "    X = ProtParam.ProteinAnalysis(seq)\n",
    "    print(X.count_amino_acids()) \n",
    "    #print X.get_amino_acids_percent() \n",
    "    #print X.molecular_weight() \n",
    "    #print X.aromaticity() \n",
    "    #print X.instability_index() \n",
    "    #print X.flexibility() \n",
    "    #print X.isoelectric_point() \n",
    "    #print X.secondary_structure_fraction()\n",
    "    \n",
    "# made a data series from the count_amino_acids function\n",
    "# aacount = {X.count_amino_acids()}\n",
    "\n",
    "# made a pandas dataframe from the series generated above\n",
    "# aacount = pd.DataFrame(list(data.items()),columns = ['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y']) \n",
    "\n",
    "aacount = pd.DataFrame(X.count_amino_acids(), index=[0])\n",
    "\n",
    "# look at new dataframe\n",
    "\n",
    "# aacount.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
