{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Manipulation of PeaskDB de novo-assisted database search results of ETNP 2017 P2 samples LC-MS/MS data using python.\n",
    "\n",
    "Starting with:\n",
    "\n",
    "    PeaksDB search results (.csv) of PTM-optimized database searches\n",
    "    These were all searched with 15 ppm precursor tolerance and 0.5 ppm fragement ion tolerance\n",
    "    Search database included marine fungi and labrinthulomyces discovered using _de novo_ peptide sequencing\n",
    "    XInteract file includes precursor intensities and protein descriptions\n",
    "\n",
    "Goal:\n",
    "\n",
    "    Files with stripped (no PTMs) peptide lists and\n",
    "    Columns with #'s of each modification in every sequence\n",
    "    Column with stripped peptide lengths (# amino acids)\n",
    "    \n",
    "### To use for a different file:\n",
    "\n",
    "#### 1. Change the input file name in *IN 4*\n",
    "#### 2. Use 'find + replace' (Esc + F) to replace the running # (e.g., 233) for another\n",
    "#### 3. Update the NAAF factor calculated in *IN 6* into *IN 7*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We don't have technical duplicates here, sadly, unlike the MED4 Pro samples. I exported PeaksDB search results as `csv` files into my ETNP 2017 git repo:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDB-PTMopt\n"
     ]
    }
   ],
   "source": [
    "cd /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDB-PTMopt/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.0.5'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# LIBRARIES\n",
    "#import pandas library for working with tabular data\n",
    "import os\n",
    "os.getcwd()\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "from scipy.stats import kde\n",
    "#import regular expresson (regex)\n",
    "import re\n",
    "#check pandas version\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "'ETNP_SKQ17_PEAKSDB_161_231-100m-0.3-JA2_15ppm_DB search psm.csv'\r\n",
      "'ETNP_SKQ17_PEAKSDB_161_231-100m-0.3-JA2_15ppm_de novo only peptides.csv'\r\n",
      " ETNP_SKQ17_PEAKSDB_161_231-100m-0.3-JA2_15ppm_peptides.csv\r\n",
      " ETNP_SKQ17_PEAKSDB_161_231-100m-0.3-JA2_15ppm_peptides.pep.xml\r\n",
      " ETNP_SKQ17_PEAKSDB_161_231-100m-0.3-JA2_15ppm_protein-peptides.csv\r\n",
      " ETNP_SKQ17_PEAKSDB_161_231-100m-0.3-JA2_15ppm_proteins.csv\r\n",
      " ETNP_SKQ17_PEAKSDB_161_231-100m-0.3-JA2_15ppm_proteins.fasta\r\n",
      " ETNP_SKQ17_PEAKSDB20_231-100m-0.3-JA2_15ppm.csv\r\n",
      " ETNP_SKQ17_PEAKSDB20_231-100m-0.3-JA2_15ppm_NAAF.csv\r\n",
      " ETNP_SKQ17_PEAKSDB20_231-100m-0.3-JA2_15ppm_NAAF_totals\r\n",
      " ETNP_SKQ17_PEAKSDB20_231-100m-0.3-JA2_15ppm_NAAF_totals.csv\r\n",
      " ETNP_SKQ17_PEAKSDB20_231-100m-0.3-JA2_15ppm_stripped_peptides.fas\r\n",
      " ETNP_SKQ17_PEAKSDB20_231-100m-0.3-JA2_15ppm_stripped_peptides.txt\r\n",
      " ETNP_SKQ17_PEAKSDB20_231-100m-0.3-JA2_15ppm_totals.csv\r\n"
     ]
    }
   ],
   "source": [
    "ls 231/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 231: 100 m McLane pump filtered on 0.3 um GF-75"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Peptide', '-10lgP', 'Mass', 'Length', 'ppm', 'm/z', 'RT',\n",
      "       'Area 231_JA2_100m_0.3um_SKQ17_P2', 'Fraction', 'Scan', 'Source File',\n",
      "       '#Spec', '#Spec 231_JA2_100m_0.3um_SKQ17_P2', 'Accession', 'PTM',\n",
      "       'AScore'],\n",
      "      dtype='object')\n",
      "# redundant peaksdb peptides in combined dataframe 1299\n",
      "# nonredundant peaksdb peptides in combined dataframe 1299\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide</th>\n",
       "      <th>-10lgP</th>\n",
       "      <th>Mass</th>\n",
       "      <th>Length</th>\n",
       "      <th>ppm</th>\n",
       "      <th>m/z</th>\n",
       "      <th>RT</th>\n",
       "      <th>Area</th>\n",
       "      <th>Fraction</th>\n",
       "      <th>Scan</th>\n",
       "      <th>Source File</th>\n",
       "      <th>#Spec</th>\n",
       "      <th>#Spec</th>\n",
       "      <th>Accession</th>\n",
       "      <th>PTM</th>\n",
       "      <th>AScore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IVVGGPYSSVSDAASSLDSSQK</td>\n",
       "      <td>88.11</td>\n",
       "      <td>2153.0488</td>\n",
       "      <td>22</td>\n",
       "      <td>2.7</td>\n",
       "      <td>1077.5345</td>\n",
       "      <td>71.78</td>\n",
       "      <td>25300000.0</td>\n",
       "      <td>1</td>\n",
       "      <td>23179</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>ETNP_160m_PROKKA_83311:ETNP_110m_PROKKA_45826</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IVVGGPYSSVSDAASVLDGSQK</td>\n",
       "      <td>88.10</td>\n",
       "      <td>2135.0745</td>\n",
       "      <td>22</td>\n",
       "      <td>1.6</td>\n",
       "      <td>1068.5463</td>\n",
       "      <td>82.14</td>\n",
       "      <td>20600000.0</td>\n",
       "      <td>1</td>\n",
       "      <td>28172</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>ETNP_120m_PROKKA_126325:ETNP_90m_PROKKA_27133:...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AIQQQIENPLAQQILSGELVPGK</td>\n",
       "      <td>87.31</td>\n",
       "      <td>2473.3540</td>\n",
       "      <td>23</td>\n",
       "      <td>3.6</td>\n",
       "      <td>1237.6887</td>\n",
       "      <td>98.28</td>\n",
       "      <td>25900000.0</td>\n",
       "      <td>1</td>\n",
       "      <td>34897</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>gi|54036848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>QAVSADSSGSFIGGAELASLK</td>\n",
       "      <td>83.59</td>\n",
       "      <td>1993.9956</td>\n",
       "      <td>21</td>\n",
       "      <td>1.0</td>\n",
       "      <td>998.0060</td>\n",
       "      <td>81.22</td>\n",
       "      <td>6280000.0</td>\n",
       "      <td>1</td>\n",
       "      <td>27723</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>ETNP_120m_PROKKA_239025:ETNP_120m_free_PROKKA_...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LPQVEGTGGDVQPSQDLVR</td>\n",
       "      <td>83.40</td>\n",
       "      <td>1994.0068</td>\n",
       "      <td>19</td>\n",
       "      <td>1.9</td>\n",
       "      <td>998.0126</td>\n",
       "      <td>62.23</td>\n",
       "      <td>237000000.0</td>\n",
       "      <td>1</td>\n",
       "      <td>19110</td>\n",
       "      <td>20170410_ETNP-231-100m-0.3um-JA2_01.raw</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>gi|54036848</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Peptide  -10lgP       Mass  Length  ppm        m/z     RT  \\\n",
       "0   IVVGGPYSSVSDAASSLDSSQK   88.11  2153.0488      22  2.7  1077.5345  71.78   \n",
       "1   IVVGGPYSSVSDAASVLDGSQK   88.10  2135.0745      22  1.6  1068.5463  82.14   \n",
       "2  AIQQQIENPLAQQILSGELVPGK   87.31  2473.3540      23  3.6  1237.6887  98.28   \n",
       "3    QAVSADSSGSFIGGAELASLK   83.59  1993.9956      21  1.0   998.0060  81.22   \n",
       "4      LPQVEGTGGDVQPSQDLVR   83.40  1994.0068      19  1.9   998.0126  62.23   \n",
       "\n",
       "          Area  Fraction   Scan                              Source File  \\\n",
       "0   25300000.0         1  23179  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "1   20600000.0         1  28172  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "2   25900000.0         1  34897  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "3    6280000.0         1  27723  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "4  237000000.0         1  19110  20170410_ETNP-231-100m-0.3um-JA2_01.raw   \n",
       "\n",
       "   #Spec  #Spec                                          Accession  PTM AScore  \n",
       "0      6      6      ETNP_160m_PROKKA_83311:ETNP_110m_PROKKA_45826  NaN    NaN  \n",
       "1      7      7  ETNP_120m_PROKKA_126325:ETNP_90m_PROKKA_27133:...  NaN    NaN  \n",
       "2      6      6                                        gi|54036848  NaN    NaN  \n",
       "3      7      7  ETNP_120m_PROKKA_239025:ETNP_120m_free_PROKKA_...  NaN    NaN  \n",
       "4     13     13                                        gi|54036848  NaN    NaN  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read the CSV into a dataframe using the pandas read_csv function\n",
    "peaksdbdup231 = pd.read_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDB-PTMopt/231/ETNP_SKQ17_PEAKSDB_161_231-100m-0.3-JA2_15ppm_peptides.csv\")\n",
    "\n",
    "# remove redundant rows\n",
    "peaksdb231 = pd.DataFrame.drop_duplicates(peaksdbdup231)\n",
    "\n",
    "print(peaksdb231.columns)\n",
    "\n",
    "columns = ['Peptide', '-10lgP', 'Mass', 'Length', 'ppm', 'm/z', 'RT',\n",
    "       'Area', 'Fraction', 'Scan', 'Source File',\n",
    "       '#Spec', '#Spec', 'Accession', 'PTM',\n",
    "       'AScore']\n",
    "\n",
    "peaksdb231.columns = columns\n",
    "\n",
    "print(\"# redundant peaksdb peptides in combined dataframe\", len(peaksdbdup231))\n",
    "print(\"# nonredundant peaksdb peptides in combined dataframe\", len(peaksdb231))\n",
    "\n",
    "#look at the dataframe\n",
    "peaksdb231.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide</th>\n",
       "      <th>-10lgP</th>\n",
       "      <th>Mass</th>\n",
       "      <th>Length</th>\n",
       "      <th>ppm</th>\n",
       "      <th>m/z</th>\n",
       "      <th>RT</th>\n",
       "      <th>Area</th>\n",
       "      <th>Fraction</th>\n",
       "      <th>Scan</th>\n",
       "      <th>...</th>\n",
       "      <th>m-oxid</th>\n",
       "      <th>n-deam</th>\n",
       "      <th>q-deam</th>\n",
       "      <th>k-hydr</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>stripped length</th>\n",
       "      <th>NAAF num.</th>\n",
       "      <th>ptm-total</th>\n",
       "      <th>stripped I-L</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IVVGGPYSSVSDAASSLDSSQK</td>\n",
       "      <td>88.11</td>\n",
       "      <td>2153.0488</td>\n",
       "      <td>22</td>\n",
       "      <td>2.7</td>\n",
       "      <td>1077.5345</td>\n",
       "      <td>71.78</td>\n",
       "      <td>25300000.0</td>\n",
       "      <td>1</td>\n",
       "      <td>23179</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>IVVGGPYSSVSDAASSLDSSQK</td>\n",
       "      <td>22</td>\n",
       "      <td>1.150000e+06</td>\n",
       "      <td>0</td>\n",
       "      <td>LVVGGPYSSVSDAASSLDSSQK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IVVGGPYSSVSDAASVLDGSQK</td>\n",
       "      <td>88.10</td>\n",
       "      <td>2135.0745</td>\n",
       "      <td>22</td>\n",
       "      <td>1.6</td>\n",
       "      <td>1068.5463</td>\n",
       "      <td>82.14</td>\n",
       "      <td>20600000.0</td>\n",
       "      <td>1</td>\n",
       "      <td>28172</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>IVVGGPYSSVSDAASVLDGSQK</td>\n",
       "      <td>22</td>\n",
       "      <td>9.363636e+05</td>\n",
       "      <td>0</td>\n",
       "      <td>LVVGGPYSSVSDAASVLDGSQK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AIQQQIENPLAQQILSGELVPGK</td>\n",
       "      <td>87.31</td>\n",
       "      <td>2473.3540</td>\n",
       "      <td>23</td>\n",
       "      <td>3.6</td>\n",
       "      <td>1237.6887</td>\n",
       "      <td>98.28</td>\n",
       "      <td>25900000.0</td>\n",
       "      <td>1</td>\n",
       "      <td>34897</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>AIQQQIENPLAQQILSGELVPGK</td>\n",
       "      <td>23</td>\n",
       "      <td>1.126087e+06</td>\n",
       "      <td>0</td>\n",
       "      <td>ALQQQLENPLAQQLLSGELVPGK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>QAVSADSSGSFIGGAELASLK</td>\n",
       "      <td>83.59</td>\n",
       "      <td>1993.9956</td>\n",
       "      <td>21</td>\n",
       "      <td>1.0</td>\n",
       "      <td>998.0060</td>\n",
       "      <td>81.22</td>\n",
       "      <td>6280000.0</td>\n",
       "      <td>1</td>\n",
       "      <td>27723</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>QAVSADSSGSFIGGAELASLK</td>\n",
       "      <td>21</td>\n",
       "      <td>2.990476e+05</td>\n",
       "      <td>0</td>\n",
       "      <td>QAVSADSSGSFLGGAELASLK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LPQVEGTGGDVQPSQDLVR</td>\n",
       "      <td>83.40</td>\n",
       "      <td>1994.0068</td>\n",
       "      <td>19</td>\n",
       "      <td>1.9</td>\n",
       "      <td>998.0126</td>\n",
       "      <td>62.23</td>\n",
       "      <td>237000000.0</td>\n",
       "      <td>1</td>\n",
       "      <td>19110</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>LPQVEGTGGDVQPSQDLVR</td>\n",
       "      <td>19</td>\n",
       "      <td>1.247368e+07</td>\n",
       "      <td>0</td>\n",
       "      <td>LPQVEGTGGDVQPSQDLVR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Peptide  -10lgP       Mass  Length  ppm        m/z     RT  \\\n",
       "0   IVVGGPYSSVSDAASSLDSSQK   88.11  2153.0488      22  2.7  1077.5345  71.78   \n",
       "1   IVVGGPYSSVSDAASVLDGSQK   88.10  2135.0745      22  1.6  1068.5463  82.14   \n",
       "2  AIQQQIENPLAQQILSGELVPGK   87.31  2473.3540      23  3.6  1237.6887  98.28   \n",
       "3    QAVSADSSGSFIGGAELASLK   83.59  1993.9956      21  1.0   998.0060  81.22   \n",
       "4      LPQVEGTGGDVQPSQDLVR   83.40  1994.0068      19  1.9   998.0126  62.23   \n",
       "\n",
       "          Area  Fraction   Scan  ... m-oxid  n-deam  q-deam k-hydr r-meth  \\\n",
       "0   25300000.0         1  23179  ...      0       0       0      0      0   \n",
       "1   20600000.0         1  28172  ...      0       0       0      0      0   \n",
       "2   25900000.0         1  34897  ...      0       0       0      0      0   \n",
       "3    6280000.0         1  27723  ...      0       0       0      0      0   \n",
       "4  237000000.0         1  19110  ...      0       0       0      0      0   \n",
       "\n",
       "          stripped peptide  stripped length     NAAF num.  ptm-total  \\\n",
       "0   IVVGGPYSSVSDAASSLDSSQK               22  1.150000e+06          0   \n",
       "1   IVVGGPYSSVSDAASVLDGSQK               22  9.363636e+05          0   \n",
       "2  AIQQQIENPLAQQILSGELVPGK               23  1.126087e+06          0   \n",
       "3    QAVSADSSGSFIGGAELASLK               21  2.990476e+05          0   \n",
       "4      LPQVEGTGGDVQPSQDLVR               19  1.247368e+07          0   \n",
       "\n",
       "              stripped I-L  \n",
       "0   LVVGGPYSSVSDAASSLDSSQK  \n",
       "1   LVVGGPYSSVSDAASVLDGSQK  \n",
       "2  ALQQQLENPLAQQLLSGELVPGK  \n",
       "3    QAVSADSSGSFLGGAELASLK  \n",
       "4      LPQVEGTGGDVQPSQDLVR  \n",
       "\n",
       "[5 rows x 47 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use a count function to enumerate the # of A's (alanines) in each peptide\n",
    "peaksdb231['A'] = peaksdb231['Peptide'].str.count(\"A\")\n",
    "\n",
    "# use a count function to enumerate the # of C's (cysteines) in each peptide\n",
    "peaksdb231['C'] = peaksdb231['Peptide'].str.count(\"C\")\n",
    "\n",
    "# use a count function to enumerate the # of D's (aspartic acids) in each peptide\n",
    "peaksdb231['D'] = peaksdb231['Peptide'].str.count(\"D\")\n",
    "\n",
    "# use a count function to enumerate the # of E's (glutamic acids) in each peptide\n",
    "peaksdb231['E'] = peaksdb231['Peptide'].str.count(\"E\")\n",
    "\n",
    "# use a count function to enumerate the # of F's (phenylalanines) in each peptide\n",
    "peaksdb231['F'] = peaksdb231['Peptide'].str.count(\"F\")\n",
    "\n",
    "# use a count function to enumerate the # of G's (glycines) in each peptide\n",
    "peaksdb231['G'] = peaksdb231['Peptide'].str.count(\"G\")\n",
    "\n",
    "# use a count function to enumerate the # of H's (histidines) in each peptide\n",
    "peaksdb231['H'] = peaksdb231['Peptide'].str.count(\"H\")\n",
    "\n",
    "# use a count function to enumerate the # of I's (isoleucines) in each peptide\n",
    "# in peaksdb231 output, there will be no isoleucines (they're lumped in with leucines)\n",
    "peaksdb231['I'] = peaksdb231['Peptide'].str.count(\"I\")\n",
    "\n",
    "# use a count function to enumerate the # of K's (lysines) in each peptide\n",
    "peaksdb231['K'] = peaksdb231['Peptide'].str.count(\"K\")\n",
    "\n",
    "# use a count function to enumerate the # of L's (leucines) in each peptide\n",
    "# also these include the isoleucines\n",
    "peaksdb231['L'] = peaksdb231['Peptide'].str.count(\"L\")\n",
    "\n",
    "# use a count function to enumerate the # of M's (methionines) in each peptide\n",
    "peaksdb231['M'] = peaksdb231['Peptide'].str.count(\"M\")\n",
    "\n",
    "# use a count function to enumerate the # of N's (asparagines) in each peptide\n",
    "peaksdb231['N'] = peaksdb231['Peptide'].str.count(\"N\")\n",
    "\n",
    "# use a count function to enumerate the # of P's ([prolines]) in each peptide\n",
    "peaksdb231['P'] = peaksdb231['Peptide'].str.count(\"P\")\n",
    "\n",
    "# use a count function to enumerate the # of Q's (glutamines) in each peptide\n",
    "peaksdb231['Q'] = peaksdb231['Peptide'].str.count(\"Q\")\n",
    "\n",
    "# use a count function to enumerate the # of R's (arginines) in each peptide\n",
    "peaksdb231['R'] = peaksdb231['Peptide'].str.count(\"R\")\n",
    "\n",
    "# use a count function to enumerate the # of S's (serines) in each peptide\n",
    "peaksdb231['S'] = peaksdb231['Peptide'].str.count(\"S\")\n",
    "\n",
    "# use a count function to enumerate the # of T's (threonines) in each peptide\n",
    "peaksdb231['T'] = peaksdb231['Peptide'].str.count(\"T\")\n",
    "\n",
    "# use a count function to enumerate the # of V's (valines) in each peptide\n",
    "peaksdb231['V'] = peaksdb231['Peptide'].str.count(\"V\")\n",
    "\n",
    "# use a count function to enumerate the # of W's (tryptophans) in each peptide\n",
    "peaksdb231['W'] = peaksdb231['Peptide'].str.count(\"W\")\n",
    "\n",
    "# use a count function to enumerate the # of Y's (tyrosines) in each peptide\n",
    "peaksdb231['Y'] = peaksdb231['Peptide'].str.count(\"Y\")\n",
    "\n",
    "# use a count function to enumerate the # of carbamidomethylated C's in each peptide\n",
    "peaksdb231['c-carb'] = peaksdb231['Peptide'].str.count(\"57.02\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "peaksdb231['m-oxid'] = peaksdb231['Peptide'].apply(lambda x: x.count('M(+15.99)'))\n",
    "\n",
    "# use a lamba function to enumerate the # of deamidated N's in each peptide\n",
    "peaksdb231['n-deam'] = peaksdb231['Peptide'].apply(lambda x: x.count('N(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "peaksdb231['q-deam'] = peaksdb231['Peptide'].apply(lambda x: x.count('Q(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of hydroxylated K's in each peptide\n",
    "peaksdb231['k-hydr'] = peaksdb231['Peptide'].apply(lambda x: x.count('K(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of hydroxylated P's in each peptide\n",
    "#peaksdb231['p-hydr'] = peaksdb231['Peptide'].apply(lambda x: x.count('P(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "peaksdb231['r-meth'] = peaksdb231['Peptide'].apply(lambda x: x.count('R(+14.02)'))\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "peaksdb231['stripped peptide'] = peaksdb231['Peptide'].str.replace(r\"\\(.*\\)\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "peaksdb231['stripped length'] = peaksdb231['stripped peptide'].apply(len)\n",
    "\n",
    "peaksdb231['NAAF num.'] = peaksdb231['Area'] / peaksdb231['stripped length']\n",
    "\n",
    "# total the number of modifications in sequence\n",
    "peaksdb231['ptm-total'] = peaksdb231['c-carb'] + peaksdb231['m-oxid'] + peaksdb231['n-deam'] + peaksdb231['q-deam'] + peaksdb231['k-hydr'] + peaksdb231['r-meth']\n",
    "\n",
    "# turn all isoleucines into leucines\n",
    "# this helps later in comparing Unipept peptides to PeaksDB and Comet ones\n",
    "peaksdb231['stripped I-L']= peaksdb231['stripped peptide'].str.replace('I','L')\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "peaksdb231.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDB-PTMopt/231/ETNP_SKQ17_PEAKSDB20_231-100m-0.3-JA2_15ppm.csv\")\n",
    "\n",
    "# check out the results\n",
    "peaksdb231.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exporting txt files of stripped peptides at confidence cutoffs:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total modified PeaksDB peptides in 231: 1299\n",
      "Deduplicated modified PeaksDB peptides in 231: 1299\n",
      "Total stripped PeaksDB peptides in 231: 1299\n",
      "Deduplicated stripped PeaksDB peptides in 231: 1278\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IVVGGPYSSVSDAASSLDSSQK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IVVGGPYSSVSDAASVLDGSQK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AIQQQIENPLAQQILSGELVPGK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>QAVSADSSGSFIGGAELASLK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LPQVEGTGGDVQPSQDLVR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          stripped peptide\n",
       "0   IVVGGPYSSVSDAASSLDSSQK\n",
       "1   IVVGGPYSSVSDAASVLDGSQK\n",
       "2  AIQQQIENPLAQQILSGELVPGK\n",
       "3    QAVSADSSGSFIGGAELASLK\n",
       "4      LPQVEGTGGDVQPSQDLVR"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# keep only peptide column peptides >20 -10lgP\n",
    "pep231moddup = peaksdb231[[\"Peptide\"]]\n",
    "\n",
    "# keep only the stripped peptide column >20 -10lgP\n",
    "# this is what we'll use for UniPept input, etc\n",
    "pep231dup = peaksdb231[[\"stripped peptide\"]]\n",
    "\n",
    "# deduplicate both of these lists\n",
    "pep231mod = pep231moddup.drop_duplicates()\n",
    "pep231 = pep231dup.drop_duplicates()\n",
    "\n",
    "# print out the #s of modified and stripped peptides, deduplicated and not\n",
    "\n",
    "print(\"Total modified PeaksDB peptides in 231:\", len(pep231moddup))\n",
    "print(\"Deduplicated modified PeaksDB peptides in 231:\", len(pep231mod))\n",
    "print(\"Total stripped PeaksDB peptides in 231:\", len(pep231dup))\n",
    "print(\"Deduplicated stripped PeaksDB peptides in 231:\", len(pep231))\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "pep231.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDB-PTMopt/231/ETNP_SKQ17_PEAKSDB20_231-100m-0.3-JA2_15ppm_stripped_peptides.txt\", header=False, index=False)\n",
    "\n",
    "# made the text file into a FASTA \n",
    "!awk '{print \">\"NR\"\\n\"$0}' /home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDB-PTMopt/231/ETNP_SKQ17_PEAKSDB20_231-100m-0.3-JA2_15ppm_stripped_peptides.txt > \\\n",
    "/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDB-PTMopt/231/ETNP_SKQ17_PEAKSDB20_231-100m-0.3-JA2_15ppm_stripped_peptides.fas\n",
    "\n",
    "# look at the stripped peptides\n",
    "pep231.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## NAAF correction and exporting files with AA and PTM totals:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>K</th>\n",
       "      <th>L</th>\n",
       "      <th>...</th>\n",
       "      <th>r-meth</th>\n",
       "      <th>Total area</th>\n",
       "      <th>Total length</th>\n",
       "      <th>% C w/ carb</th>\n",
       "      <th>% M w/ oxid</th>\n",
       "      <th>% N w/ deam</th>\n",
       "      <th>% Q w/ deam</th>\n",
       "      <th>% K w/ hydr</th>\n",
       "      <th>% R w/ meth</th>\n",
       "      <th>NAAF denom.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>1786</td>\n",
       "      <td>77</td>\n",
       "      <td>1133</td>\n",
       "      <td>1308</td>\n",
       "      <td>590</td>\n",
       "      <td>1835</td>\n",
       "      <td>70</td>\n",
       "      <td>894</td>\n",
       "      <td>789</td>\n",
       "      <td>1443</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>3.867077e+09</td>\n",
       "      <td>16576</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.858974</td>\n",
       "      <td>0.017632</td>\n",
       "      <td>0.019332</td>\n",
       "      <td>0.010139</td>\n",
       "      <td>0.009077</td>\n",
       "      <td>233293.736728</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 A   C     D     E    F     G   H    I    K     L  ...  \\\n",
       "sample total  1786  77  1133  1308  590  1835  70  894  789  1443  ...   \n",
       "\n",
       "              r-meth    Total area  Total length  % C w/ carb  % M w/ oxid  \\\n",
       "sample total       6  3.867077e+09         16576          1.0     0.858974   \n",
       "\n",
       "              % N w/ deam  % Q w/ deam  % K w/ hydr  % R w/ meth  \\\n",
       "sample total     0.017632     0.019332     0.010139     0.009077   \n",
       "\n",
       "                NAAF denom.  \n",
       "sample total  233293.736728  \n",
       "\n",
       "[1 rows x 35 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of certain columns in the modified\n",
    "# peptide dataframe above \n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'A': peaksdb231['A'].sum(),\n",
    "        'C': peaksdb231['C'].sum(),\n",
    "        'D': peaksdb231['D'].sum(),\n",
    "        'E': peaksdb231['E'].sum(),\n",
    "        'F': peaksdb231['F'].sum(),\n",
    "        'G': peaksdb231['G'].sum(),\n",
    "        'H': peaksdb231['H'].sum(),\n",
    "        'I': peaksdb231['I'].sum(),\n",
    "        'K': peaksdb231['K'].sum(),\n",
    "        'L': peaksdb231['L'].sum(),\n",
    "        'M': peaksdb231['M'].sum(),\n",
    "        'N': peaksdb231['N'].sum(),\n",
    "        'P': peaksdb231['P'].sum(),\n",
    "        'Q': peaksdb231['Q'].sum(),\n",
    "        'R': peaksdb231['R'].sum(),\n",
    "        'S': peaksdb231['S'].sum(),\n",
    "        'T': peaksdb231['T'].sum(),\n",
    "        'V': peaksdb231['V'].sum(),\n",
    "        'W': peaksdb231['W'].sum(),\n",
    "        'Y': peaksdb231['Y'].sum(),\n",
    "        'c-carb': peaksdb231['c-carb'].sum(),\n",
    "        'm-oxid': peaksdb231['m-oxid'].sum(),\n",
    "        'n-deam': peaksdb231['n-deam'].sum(),\n",
    "        'q-deam': peaksdb231['q-deam'].sum(),\n",
    "        'k-hydr': peaksdb231['k-hydr'].sum(),\n",
    "        'r-meth': peaksdb231['r-meth'].sum(),\n",
    "        'Total area': peaksdb231['Area'].sum(),\n",
    "        'Total length': peaksdb231['stripped length'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaksdb231 = pd.DataFrame(data, columns=['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', 'm-oxid', 'n-deam', 'q-deam', 'k-hydr', 'r-meth', 'Total area', 'Total length'], index=index)\n",
    "\n",
    "# calculate percentage of C's with carb (should be 1.0)\n",
    "totalpeaksdb231['% C w/ carb'] = totalpeaksdb231['c-carb'] / totalpeaksdb231['C'] \n",
    "\n",
    "# calculate percentage of M's that are oxidized\n",
    "totalpeaksdb231['% M w/ oxid'] = totalpeaksdb231['m-oxid'] / totalpeaksdb231['M'] \n",
    "\n",
    "# calculate percentage of N's that are deamidated\n",
    "totalpeaksdb231['% N w/ deam'] = totalpeaksdb231['n-deam'] / totalpeaksdb231['N'] \n",
    "\n",
    "# calculate percentage of Q's that are deamidated\n",
    "totalpeaksdb231['% Q w/ deam'] = totalpeaksdb231['q-deam'] / totalpeaksdb231['Q'] \n",
    "\n",
    "# calculate percentage of K's that are hydroxylated\n",
    "totalpeaksdb231['% K w/ hydr'] = totalpeaksdb231['k-hydr'] / totalpeaksdb231['K'] \n",
    "\n",
    "# calculate percentage of P's that are hydroxylated\n",
    "#totalpeaksdb231['% P w/ hydr'] = totalpeaksdb231['p-hydr'] / totalpeaksdb231['P'] \n",
    "\n",
    "# calculate percentage of R's that are methylated\n",
    "totalpeaksdb231['% R w/ meth'] = totalpeaksdb231['r-meth'] / totalpeaksdb231['R'] \n",
    "\n",
    "# calculate NAAF denominator for all peptides in dataset i\n",
    "totalpeaksdb231['NAAF denom.'] = totalpeaksdb231['Total area'] / totalpeaksdb231['Total length']\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "totalpeaksdb231.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDB-PTMopt/231/ETNP_SKQ17_PEAKSDB20_231-100m-0.3-JA2_15ppm_totals.csv\")\n",
    "\n",
    "totalpeaksdb231.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>stripped peptide</th>\n",
       "      <th>NAAF factor</th>\n",
       "      <th>A</th>\n",
       "      <th>C</th>\n",
       "      <th>D</th>\n",
       "      <th>E</th>\n",
       "      <th>F</th>\n",
       "      <th>G</th>\n",
       "      <th>H</th>\n",
       "      <th>I</th>\n",
       "      <th>...</th>\n",
       "      <th>T-NAAF20</th>\n",
       "      <th>V-NAAF20</th>\n",
       "      <th>W-NAAF20</th>\n",
       "      <th>Y-NAAF20</th>\n",
       "      <th>ccarb-NAAF20</th>\n",
       "      <th>moxid-NAAF20</th>\n",
       "      <th>ndeam-NAAF20</th>\n",
       "      <th>qdeam-NAAF20</th>\n",
       "      <th>khydr-NAAF20</th>\n",
       "      <th>rmeth-NAAF20</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IVVGGPYSSVSDAASSLDSSQK</td>\n",
       "      <td>4.929408</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>14.788224</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.929408</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IVVGGPYSSVSDAASVLDGSQK</td>\n",
       "      <td>4.013668</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>16.054673</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.013668</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AIQQQIENPLAQQILSGELVPGK</td>\n",
       "      <td>4.826906</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>4.826906</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>QAVSADSSGSFIGGAELASLK</td>\n",
       "      <td>1.281850</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1.281850</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LPQVEGTGGDVQPSQDLVR</td>\n",
       "      <td>53.467720</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>53.46772</td>\n",
       "      <td>160.403160</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          stripped peptide  NAAF factor  A  C  D  E  F  G  H  I  ...  \\\n",
       "0   IVVGGPYSSVSDAASSLDSSQK     4.929408  2  0  2  0  0  2  0  1  ...   \n",
       "1   IVVGGPYSSVSDAASVLDGSQK     4.013668  2  0  2  0  0  3  0  1  ...   \n",
       "2  AIQQQIENPLAQQILSGELVPGK     4.826906  2  0  0  2  0  2  0  3  ...   \n",
       "3    QAVSADSSGSFIGGAELASLK     1.281850  4  0  1  1  1  3  0  1  ...   \n",
       "4      LPQVEGTGGDVQPSQDLVR    53.467720  0  0  2  1  0  3  0  0  ...   \n",
       "\n",
       "   T-NAAF20    V-NAAF20  W-NAAF20  Y-NAAF20  ccarb-NAAF20  moxid-NAAF20  \\\n",
       "0   0.00000   14.788224       0.0  4.929408           0.0           0.0   \n",
       "1   0.00000   16.054673       0.0  4.013668           0.0           0.0   \n",
       "2   0.00000    4.826906       0.0  0.000000           0.0           0.0   \n",
       "3   0.00000    1.281850       0.0  0.000000           0.0           0.0   \n",
       "4  53.46772  160.403160       0.0  0.000000           0.0           0.0   \n",
       "\n",
       "   ndeam-NAAF20  qdeam-NAAF20  khydr-NAAF20  rmeth-NAAF20  \n",
       "0           0.0           0.0           0.0           0.0  \n",
       "1           0.0           0.0           0.0           0.0  \n",
       "2           0.0           0.0           0.0           0.0  \n",
       "3           0.0           0.0           0.0           0.0  \n",
       "4           0.0           0.0           0.0           0.0  \n",
       "\n",
       "[5 rows x 54 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# use the calculated NAAF factor (in totalpeaksdb231 dataframe, above) to caluclate the NAAF \n",
    "# NAAF: normalized normalized area abundance factor\n",
    "\n",
    "NAAF20 = 233293.736728\n",
    "\n",
    "# use NAAF >XCorr 3 to get NAAF\n",
    "peaksdb231['NAAF factor'] = (peaksdb231['NAAF num.'])/NAAF20\n",
    "\n",
    "# make a dataframe that contains only what we need: sequences, AAs, PTMs\n",
    "peaksdb231_NAAF = peaksdb231[['stripped peptide', 'NAAF factor', 'A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', \\\n",
    "                                'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', 'm-oxid', 'n-deam', \\\n",
    "                                'q-deam', 'k-hydr', 'r-meth']].copy()\n",
    "\n",
    "# multiply the NAAF20 factor by the AAs to normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaksdb231_NAAF['A-NAAF20'] = peaksdb231_NAAF['A'] * peaksdb231['NAAF factor']\n",
    "peaksdb231_NAAF['C-NAAF20'] = peaksdb231_NAAF['C'] * peaksdb231['NAAF factor']\n",
    "peaksdb231_NAAF['D-NAAF20'] = peaksdb231_NAAF['D'] * peaksdb231['NAAF factor']\n",
    "peaksdb231_NAAF['E-NAAF20'] = peaksdb231_NAAF['E'] * peaksdb231['NAAF factor']\n",
    "peaksdb231_NAAF['F-NAAF20'] = peaksdb231_NAAF['F'] * peaksdb231['NAAF factor']\n",
    "peaksdb231_NAAF['G-NAAF20'] = peaksdb231_NAAF['G'] * peaksdb231['NAAF factor']\n",
    "peaksdb231_NAAF['H-NAAF20'] = peaksdb231_NAAF['H'] * peaksdb231['NAAF factor']\n",
    "peaksdb231_NAAF['I-NAAF20'] = peaksdb231_NAAF['I'] * peaksdb231['NAAF factor']\n",
    "peaksdb231_NAAF['K-NAAF20'] = peaksdb231_NAAF['K'] * peaksdb231['NAAF factor']\n",
    "peaksdb231_NAAF['L-NAAF20'] = peaksdb231_NAAF['L'] * peaksdb231['NAAF factor']\n",
    "peaksdb231_NAAF['M-NAAF20'] = peaksdb231_NAAF['M'] * peaksdb231['NAAF factor']\n",
    "peaksdb231_NAAF['N-NAAF20'] = peaksdb231_NAAF['N'] * peaksdb231['NAAF factor']\n",
    "peaksdb231_NAAF['P-NAAF20'] = peaksdb231_NAAF['P'] * peaksdb231['NAAF factor']\n",
    "peaksdb231_NAAF['Q-NAAF20'] = peaksdb231_NAAF['Q'] * peaksdb231['NAAF factor']\n",
    "peaksdb231_NAAF['R-NAAF20'] = peaksdb231_NAAF['R'] * peaksdb231['NAAF factor']\n",
    "peaksdb231_NAAF['S-NAAF20'] = peaksdb231_NAAF['S'] * peaksdb231['NAAF factor']\n",
    "peaksdb231_NAAF['T-NAAF20'] = peaksdb231_NAAF['T'] * peaksdb231['NAAF factor']\n",
    "peaksdb231_NAAF['V-NAAF20'] = peaksdb231_NAAF['V'] * peaksdb231['NAAF factor']\n",
    "peaksdb231_NAAF['W-NAAF20'] = peaksdb231_NAAF['W'] * peaksdb231['NAAF factor']\n",
    "peaksdb231_NAAF['Y-NAAF20'] = peaksdb231_NAAF['Y'] * peaksdb231['NAAF factor']\n",
    "\n",
    "# multiply the NAAF20 factor by the PTMs normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaksdb231_NAAF['ccarb-NAAF20'] = peaksdb231_NAAF['c-carb'] * peaksdb231_NAAF['NAAF factor']\n",
    "peaksdb231_NAAF['moxid-NAAF20'] = peaksdb231_NAAF['m-oxid'] * peaksdb231_NAAF['NAAF factor']\n",
    "peaksdb231_NAAF['ndeam-NAAF20'] = peaksdb231_NAAF['n-deam'] * peaksdb231_NAAF['NAAF factor']\n",
    "peaksdb231_NAAF['qdeam-NAAF20'] = peaksdb231_NAAF['q-deam'] * peaksdb231_NAAF['NAAF factor']\n",
    "peaksdb231_NAAF['khydr-NAAF20'] = peaksdb231_NAAF['k-hydr'] * peaksdb231_NAAF['NAAF factor']\n",
    "peaksdb231_NAAF['rmeth-NAAF20'] = peaksdb231_NAAF['r-meth'] * peaksdb231_NAAF['NAAF factor']\n",
    "\n",
    "# write the dataframe to a new csv\n",
    "peaksdb231_NAAF.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDB-PTMopt/231/ETNP_SKQ17_PEAKSDB20_231-100m-0.3-JA2_15ppm_NAAF.csv\")\n",
    "\n",
    "peaksdb231_NAAF.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NAAF</th>\n",
       "      <th>A-NAAF</th>\n",
       "      <th>C-NAAF</th>\n",
       "      <th>D-NAAF</th>\n",
       "      <th>E-NAAF</th>\n",
       "      <th>F-NAAF</th>\n",
       "      <th>G-NAAF</th>\n",
       "      <th>H-NAAF</th>\n",
       "      <th>I-NAAF</th>\n",
       "      <th>K-NAAF</th>\n",
       "      <th>...</th>\n",
       "      <th>Q-deam-NAAF</th>\n",
       "      <th>K-hydr-NAAF</th>\n",
       "      <th>R-meth-NAAF</th>\n",
       "      <th>% C w/ carb. NAAF</th>\n",
       "      <th>% M w/ oxid. NAAF</th>\n",
       "      <th>% N w/ deam. NAAF</th>\n",
       "      <th>% Q w/ deam. NAAF</th>\n",
       "      <th>% K w/ hydr. NAAF</th>\n",
       "      <th>% R w/ meth. NAAF</th>\n",
       "      <th>NAAF check</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>sample total</th>\n",
       "      <td>1502.41746</td>\n",
       "      <td>2078.244989</td>\n",
       "      <td>35.328251</td>\n",
       "      <td>1117.903003</td>\n",
       "      <td>1140.395347</td>\n",
       "      <td>301.460433</td>\n",
       "      <td>1303.225805</td>\n",
       "      <td>39.415329</td>\n",
       "      <td>1076.560933</td>\n",
       "      <td>556.854934</td>\n",
       "      <td>...</td>\n",
       "      <td>1.506736</td>\n",
       "      <td>9.37155</td>\n",
       "      <td>11.166295</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.854814</td>\n",
       "      <td>0.001211</td>\n",
       "      <td>0.002994</td>\n",
       "      <td>0.016829</td>\n",
       "      <td>0.010988</td>\n",
       "      <td>0.00644</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    NAAF       A-NAAF     C-NAAF       D-NAAF       E-NAAF  \\\n",
       "sample total  1502.41746  2078.244989  35.328251  1117.903003  1140.395347   \n",
       "\n",
       "                  F-NAAF       G-NAAF     H-NAAF       I-NAAF      K-NAAF  \\\n",
       "sample total  301.460433  1303.225805  39.415329  1076.560933  556.854934   \n",
       "\n",
       "              ...  Q-deam-NAAF  K-hydr-NAAF  R-meth-NAAF  % C w/ carb. NAAF  \\\n",
       "sample total  ...     1.506736      9.37155    11.166295                1.0   \n",
       "\n",
       "              % M w/ oxid. NAAF  % N w/ deam. NAAF  % Q w/ deam. NAAF  \\\n",
       "sample total           0.854814           0.001211           0.002994   \n",
       "\n",
       "              % K w/ hydr. NAAF  % R w/ meth. NAAF  NAAF check  \n",
       "sample total           0.016829           0.010988     0.00644  \n",
       "\n",
       "[1 rows x 34 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# made a new dataframe that contains the sums of NAAF normalized AAs for peaksdb231 peaksdb results\n",
    "# also contains the sums of the NAAF-corrected PTMs occurances for each affected residue\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'NAAF': peaksdb231_NAAF['NAAF factor'].sum(),\n",
    "        'A-NAAF': peaksdb231_NAAF['A-NAAF20'].sum(),\n",
    "        'C-NAAF': peaksdb231_NAAF['C-NAAF20'].sum(),\n",
    "        'D-NAAF': peaksdb231_NAAF['D-NAAF20'].sum(),\n",
    "        'E-NAAF': peaksdb231_NAAF['E-NAAF20'].sum(),\n",
    "        'F-NAAF': peaksdb231_NAAF['F-NAAF20'].sum(),\n",
    "        'G-NAAF': peaksdb231_NAAF['G-NAAF20'].sum(),\n",
    "        'H-NAAF': peaksdb231_NAAF['H-NAAF20'].sum(),\n",
    "        'I-NAAF': peaksdb231_NAAF['I-NAAF20'].sum(),\n",
    "        'K-NAAF': peaksdb231_NAAF['K-NAAF20'].sum(),\n",
    "        'L-NAAF': peaksdb231_NAAF['L-NAAF20'].sum(),\n",
    "        'M-NAAF': peaksdb231_NAAF['M-NAAF20'].sum(),\n",
    "        'N-NAAF': peaksdb231_NAAF['N-NAAF20'].sum(),\n",
    "        'P-NAAF': peaksdb231_NAAF['P-NAAF20'].sum(),\n",
    "        'Q-NAAF': peaksdb231_NAAF['Q-NAAF20'].sum(),\n",
    "        'R-NAAF': peaksdb231_NAAF['R-NAAF20'].sum(),\n",
    "        'S-NAAF': peaksdb231_NAAF['S-NAAF20'].sum(),\n",
    "        'T-NAAF': peaksdb231_NAAF['T-NAAF20'].sum(),\n",
    "        'V-NAAF': peaksdb231_NAAF['V-NAAF20'].sum(),\n",
    "        'W-NAAF': peaksdb231_NAAF['W-NAAF20'].sum(),\n",
    "        'Y-NAAF': peaksdb231_NAAF['Y-NAAF20'].sum(),\n",
    "        'C-carb-NAAF': peaksdb231_NAAF['ccarb-NAAF20'].sum(),\n",
    "        'M-oxid-NAAF': peaksdb231_NAAF['moxid-NAAF20'].sum(),\n",
    "        'N-deam-NAAF': peaksdb231_NAAF['ndeam-NAAF20'].sum(),\n",
    "        'Q-deam-NAAF': peaksdb231_NAAF['qdeam-NAAF20'].sum(),\n",
    "        'K-hydr-NAAF': peaksdb231_NAAF['khydr-NAAF20'].sum(),\n",
    "        'R-meth-NAAF': peaksdb231_NAAF['rmeth-NAAF20'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaksdb231_NAAF = pd.DataFrame(data, columns=['NAAF', 'A-NAAF', 'C-NAAF', 'D-NAAF', 'E-NAAF', 'F-NAAF', \\\n",
    "                                                   'G-NAAF', 'H-NAAF', 'I-NAAF','K-NAAF', 'L-NAAF', 'M-NAAF', \\\n",
    "                                                   'N-NAAF', 'P-NAAF', 'Q-NAAF', 'R-NAAF', 'S-NAAF', \\\n",
    "                                                   'T-NAAF', 'V-NAAF', 'W-NAAF', 'Y-NAAF', 'C-carb-NAAF', \\\n",
    "                                                   'M-oxid-NAAF', 'N-deam-NAAF', 'Q-deam-NAAF', 'K-hydr-NAAF',\\\n",
    "                                                   'R-meth-NAAF'], index=index)\n",
    "\n",
    "# calculate the NAAF-corrected % modified C, M, N, Q, K, and Rs\n",
    "\n",
    "\n",
    "totalpeaksdb231_NAAF['% C w/ carb. NAAF'] = totalpeaksdb231_NAAF['C-carb-NAAF'] / totalpeaksdb231_NAAF['C-NAAF']\n",
    "totalpeaksdb231_NAAF['% M w/ oxid. NAAF'] = totalpeaksdb231_NAAF['M-oxid-NAAF'] / totalpeaksdb231_NAAF['M-NAAF']\n",
    "totalpeaksdb231_NAAF['% N w/ deam. NAAF'] = totalpeaksdb231_NAAF['N-deam-NAAF'] / totalpeaksdb231_NAAF['N-NAAF']\n",
    "totalpeaksdb231_NAAF['% Q w/ deam. NAAF'] = totalpeaksdb231_NAAF['Q-deam-NAAF'] / totalpeaksdb231_NAAF['Q-NAAF']\n",
    "totalpeaksdb231_NAAF['% K w/ hydr. NAAF'] = totalpeaksdb231_NAAF['K-hydr-NAAF'] / totalpeaksdb231_NAAF['K-NAAF']\n",
    "totalpeaksdb231_NAAF['% R w/ meth. NAAF'] = totalpeaksdb231_NAAF['R-meth-NAAF'] / totalpeaksdb231_NAAF['R-NAAF']\n",
    "\n",
    "# calculate NAAF summed numerator over denominator (in above cell) for all peptides in dataset i: a check\n",
    "totalpeaksdb231_NAAF['NAAF check'] = totalpeaksdb231_NAAF['NAAF'] / 233293.736728\n",
    "\n",
    "# write modified dataframe to new txt file, same name + totals\n",
    "totalpeaksdb231_NAAF.to_csv(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PeaksDB-PTMopt/231/ETNP_SKQ17_PEAKSDB20_231-100m-0.3-JA2_15ppm_NAAF_totals\")\n",
    "\n",
    "totalpeaksdb231_NAAF.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Using BioPython to query peptide sequences\n",
    "\n",
    "I installed the BioPython package using `pip install biopython`. All instructions and information [here](https://www.tutorialspoint.com/biopython/index.htm). \n",
    "\n",
    "GitHub project: https://github.com/biopython/biopython\n",
    "\n",
    "I'm relying on the ProtParam module to parse sequences for relative AA composition, instability, secondary structure, instability, and hydrophobicity. You can read more about that module and the studies the indecies are derived from here:\n",
    "\n",
    "https://biopython.org/wiki/ProtParam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# Bio.SeqIO is the standard Sequence Input/Output interface for BioPython 1.43 and later\n",
    "# Bio.SeqIO provides a simple uniform interface to input and output assorted sequence file formats.\n",
    "# (including multiple sequence alignments), but will only deal with sequences as SeqRecord objects\n",
    "\n",
    "# for accepted file formats see https://biopython.org/wiki/SeqIO\n",
    "\n",
    "from Bio import SeqIO\n",
    "#for seq_record in SeqIO.parse(\"/home/millieginty/Documents/git-repos/2017-etnp/data/MED4/MED2_tryp_1raw_db_peptides_nmod.fasta\", \"fasta\"):\n",
    "    #print(seq_record.id)\n",
    "    #print(repr(seq_record.seq))\n",
    "    #print(len(seq_record))\n",
    "    \n",
    "# I commented the print functions out so the output doesn't take up too much space. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# seeing what the ProtParam module can do with a single protein sequence:\n",
    "\n",
    "from Bio.SeqUtils.ProtParam import ProteinAnalysis\n",
    "\n",
    "test_seq = \"MAEGEITTFTALTEKFNLPPGNYKKPKLLYCSNGGHFLRILPDGTVDGTRDRSDQHIQLQLSAESVGEVYIKSTETGQYLAMDTSGLLYGSQTPSEECLFLERLEENHYNTYTSKKHAEKNWFVGLKKNGSCKRGPRTHYGQKAILFLPLPV\"\n",
    "\n",
    "analysed_seq = ProteinAnalysis(test_seq)\n",
    "print(\"molecular weight of seq =\", analysed_seq.molecular_weight())\n",
    "\n",
    "# calculates the aromaticity value of a protein according to Lobry & Gautier (1994, Nucleic Acids Res., 22, 3174-3180). \n",
    "# it's simply the relative frequency of Phe+Trp+Tyr.\n",
    "\n",
    "analysed_seq.aromaticity()\n",
    "print(\"aromaticity of seq =\", analysed_seq.aromaticity())\n",
    "\n",
    "# secondary_structure_fraction:\n",
    "# this methods returns a list of the fraction of amino acids which tend to be in helix, turn or sheet. \n",
    "# AAs in helix: V, I, Y, F, W, L\n",
    "# AAs in turn: N, P, G, S\n",
    "# AAs in sheet: E, M, A, L\n",
    "# the returned list contains 3 values: [Helix, Turn, Sheet]\n",
    "\n",
    "analysed_seq.secondary_structure_fraction()\n",
    "print(\"frac in H T S =\", analysed_seq.secondary_structure_fraction())\n",
    "\n",
    "# the instability index, an implementation of the method of Guruprasad et al. (1990, Protein Engineering, 4, 155-161).\n",
    "# this method tests a protein for stability. \n",
    "# any value above 40 means the protein is unstable (=has a short half life)\n",
    "# NOT SURE WHAT THIS MEANS FOR PEPTIDES, BUT WE COULD DO THIS FOR PROTEINS\n",
    "\n",
    "analysed_seq.instability_index()\n",
    "print(\"instability =\", analysed_seq.instability_index())\n",
    "\n",
    "# count_amino_acids will do just that, and get_amino_acids_percent will return %'s for each AA across the sequence. \n",
    "analysed_seq.get_amino_acids_percent()\n",
    "\n",
    "# taking the returned dictionary and converting to a dataframe\n",
    "\n",
    "aadict = analysed_seq.get_amino_acids_percent()\n",
    "aadf = pd.DataFrame(list(aadict.items()),columns = ['residue','% occurance']) \n",
    "\n",
    "aadf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# use SeqIO and a loop to apply count_amino_acids to each sequence in the file\n",
    "# aatot will give us the total number of each residue in the entire sample output\n",
    "\n",
    "import collections\n",
    "from Bio import SeqIO\n",
    "from Bio.SeqUtils.ProtParam import ProteinAnalysis\n",
    "\n",
    "all_aas = collections.defaultdict(int)\n",
    "for record in SeqIO.parse(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas\", \"fasta\"):\n",
    "    x = ProteinAnalysis(str(record.seq))\n",
    "    #print(record.id, x.count_amino_acids())\n",
    "    for aa, count in x.count_amino_acids().items():\n",
    "        all_aas[aa] += count        \n",
    "        \n",
    "# made a dataframe for amino acid total counts        \n",
    "data = (all_aas)\n",
    "aatot = pd.DataFrame(data, index = ['sample sequence total'])\n",
    "aatot.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pandas import Series, DataFrame\n",
    "\n",
    "with open('/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas') as fasta_file:  # Will close handle cleanly\n",
    "    identifiers = []\n",
    "    lengths = []\n",
    "    for seq_record in SeqIO.parse(fasta_file, 'fasta'):  # (generator)\n",
    "        identifiers.append(seq_record.id)\n",
    "        lengths.append(len(seq_record.seq))\n",
    "        \n",
    "        \n",
    "#converting lists to pandas Series    \n",
    "s1 = Series(identifiers, name='ID')\n",
    "s2 = Series(lengths, name='length')\n",
    "\n",
    "#Gathering Series into a pandas DataFrame and rename index as ID column\n",
    "idseq = DataFrame(dict(ID=s1, length=s2)).set_index(['ID'])\n",
    "\n",
    "idseq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pandas import Series, DataFrame\n",
    "\n",
    "\n",
    "\n",
    "with open('/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas) as fasta_file:  # Will close handle cleanly\n",
    "    identifiers = []\n",
    "    lengths = []\n",
    "    aa = []\n",
    "    for seq_record in SeqIO.parse(fasta_file, 'fasta'):  # (generator)\n",
    "        identifiers.append(seq_record.id)\n",
    "        lengths.append(len(seq_record.seq))\n",
    "        aa.count_amino_acids(seq_record.seq)\n",
    "        \n",
    "        \n",
    "#converting lists to pandas Series    \n",
    "s1 = Series(identifiers, name='ID')\n",
    "s2 = Series(lengths, name='length')\n",
    "s3 = Series(aa, name='AAs')\n",
    "\n",
    "#Gathering Series into a pandas DataFrame and rename index as ID column\n",
    "idseq = DataFrame(dict(ID=s1, length=s2, AAs=s3)).set_index(['ID'])\n",
    "\n",
    "idseq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio.SeqUtils import ProtParam\n",
    "\n",
    "handle = open(\"/home/millieginty/Documents/git-repos/2017-etnp/data/pro2020/ETNP-SKQ17/PEAKS-PTMopt/ETNP-SKQ17-231-100m-0.3-JA2_DN50_stripped_peptides.fas\") \n",
    "for record in SeqIO.parse(handle, \"fasta\"): \n",
    "    seq = str(record.seq)\n",
    "    X = ProtParam.ProteinAnalysis(seq)\n",
    "    print(X.count_amino_acids()) \n",
    "    #print X.get_amino_acids_percent() \n",
    "    #print X.molecular_weight() \n",
    "    #print X.aromaticity() \n",
    "    #print X.instability_index() \n",
    "    #print X.flexibility() \n",
    "    #print X.isoelectric_point() \n",
    "    #print X.secondary_structure_fraction()\n",
    "    \n",
    "# made a data series from the count_amino_acids function\n",
    "# aacount = {X.count_amino_acids()}\n",
    "\n",
    "# made a pandas dataframe from the series generated above\n",
    "# aacount = pd.DataFrame(list(data.items()),columns = ['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y']) \n",
    "\n",
    "aacount = pd.DataFrame(X.count_amino_acids(), index=[0])\n",
    "\n",
    "# look at new dataframe\n",
    "\n",
    "# aacount.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
